diff --git a/include/inputstr.h b/include/inputstr.h
index c5b9ff7..a83e863 100644
--- a/include/inputstr.h
+++ b/include/inputstr.h
@@ -49,6 +49,7 @@ SOFTWARE.
 #ifndef INPUTSTRUCT_H
 #define INPUTSTRUCT_H
 
+#include <pixman.h>
 #include "input.h"
 #include "window.h"
 #include "dixstruct.h"
@@ -550,6 +551,9 @@ typedef struct _DeviceIntRec {
         XIPropertyPtr   properties;
         XIPropertyHandlerPtr handlers; /* NULL-terminated */
     } properties;
+
+    /* coordinate transformation matrix for absolute input devices */
+    struct pixman_f_transform transform;
 } DeviceIntRec;
 
 typedef struct {
