diff --git a/mi/mifillarc.c b/mi/mifillarc.c
index c8b901c..d49ae1d 100644
--- a/mi/mifillarc.c
+++ b/mi/mifillarc.c
@@ -30,8 +30,8 @@ Author:  Bob Scheifler, MIT X Consortium
 /* $Xorg: mifillarc.c,v 1.4 2001/02/09 02:05:20 xorgcvs Exp $ */
 
 #include <math.h>
-#include "X.h"
-#include "Xprotostr.h"
+#include <X11/X.h>
+#include <X11/Xprotostr.h>
 #include "regionstr.h"
 #include "gcstruct.h"
 #include "pixmapstr.h"
