diff --git a/Xi/xiquerypointer.c b/Xi/xiquerypointer.c
index da1d101..d51d383 100644
--- a/Xi/xiquerypointer.c
+++ b/Xi/xiquerypointer.c
@@ -83,7 +83,10 @@ ProcXIQueryPointer(ClientPtr client)
 
     rc = dixLookupDevice(&pDev, stuff->deviceid, client, DixReadAccess);
     if (rc != Success)
+    {
+        client->errorValue = stuff->deviceid;
         return rc;
+    }
 
     if (pDev->valuator == NULL)
     {
