diff --git a/dix/main.c b/dix/main.c
index c5d3783..9fe822b 100644
--- a/dix/main.c
+++ b/dix/main.c
@@ -411,22 +411,10 @@ main(int argc, char *argv[], char *envp[])
 	    FatalError("could not open default font '%s'", defaultTextFont);
 	}
 
-#ifdef NULL_ROOT_CURSOR
-        cm.width = 0;
-        cm.height = 0;
-        cm.xhot = 0;
-        cm.yhot = 0;
-
-        if (!(rootCursor = AllocCursor(NULL, NULL, &cm, 0, 0, 0, 0, 0, 0))) {
-            FatalError("could not create empty root cursor");
-	}
-        AddResource(FakeClientID(0), RT_CURSOR, (pointer)rootCursor);
-#else
-	if (!(rootCursor = CreateRootCursor(defaultCursorFont, 0))) {
+	if (!(rootCursor = CreateRootCursor(NULL, 0))) {
 	    FatalError("could not open default cursor font '%s'",
 		       defaultCursorFont);
 	}
-#endif
 
 #ifdef DPMSExtension
  	/* check all screens, looking for DPMS Capabilities */
