diff --git a/miext/shadow/shadow.c b/miext/shadow/shadow.c
index 2c2a262..46ddfbb 100644
--- a/miext/shadow/shadow.c
+++ b/miext/shadow/shadow.c
@@ -23,12 +23,16 @@
  */
 
 
+#ifdef HAVE_DIX_CONFIG_H
+#include <dix-config.h>
+#endif
+
 #include    <X11/X.h>
 #include    "scrnintstr.h"
 #include    "windowstr.h"
-#include    "font.h"
+#include    <X11/fonts/font.h>
 #include    "dixfontstr.h"
-#include    "fontstruct.h"
+#include    <X11/fonts/fontstruct.h>
 #include    "mi.h"
 #include    "regionstr.h"
 #include    "globals.h"
