diff --git a/xfixes/saveset.c b/xfixes/saveset.c
index ba959c2..eb3f658 100644
--- a/xfixes/saveset.c
+++ b/xfixes/saveset.c
@@ -32,6 +32,7 @@ ProcXFixesChangeSaveSet(ClientPtr client)
     Bool	toRoot, map;
     int		result;
     WindowPtr	pWin;
+
     REQUEST(xXFixesChangeSaveSetReq);
 		  
     REQUEST_SIZE_MATCH(xXFixesChangeSaveSetReq);
@@ -40,18 +41,15 @@ ProcXFixesChangeSaveSet(ClientPtr client)
         return result;
     if (client->clientAsMask == (CLIENT_BITS(pWin->drawable.id)))
         return BadMatch;
-    if ((stuff->mode != SetModeInsert) && (stuff->mode != SetModeDelete))
-    {
+    if ((stuff->mode != SetModeInsert) && (stuff->mode != SetModeDelete)) {
 	client->errorValue = stuff->mode;
 	return BadValue;
     }
-    if ((stuff->target != SaveSetNearest) && (stuff->target != SaveSetRoot))
-    {
+    if ((stuff->target != SaveSetNearest) && (stuff->target != SaveSetRoot)) {
 	client->errorValue = stuff->target;
 	return BadValue;
     }
-    if ((stuff->map != SaveSetMap) && (stuff->map != SaveSetUnmap))
-    {
+    if ((stuff->map != SaveSetMap) && (stuff->map != SaveSetUnmap)) {
 	client->errorValue = stuff->map;
 	return BadValue;
     }
