diff --git a/Xi/stubs.c b/Xi/stubs.c
index 547a4cc..f7b0fe9 100644
--- a/Xi/stubs.c
+++ b/Xi/stubs.c
@@ -1,5 +1,3 @@
-/* $Xorg: stubs.c,v 1.4 2001/02/09 02:04:35 xorgcvs Exp $ */
-
 /************************************************************
 
 Copyright 1989, 1998  The Open Group
@@ -45,7 +43,6 @@ ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
 SOFTWARE.
 
 ********************************************************/
-/* $XFree86: xc/programs/Xserver/Xi/stubs.c,v 3.4 2001/12/14 19:58:59 dawes Exp $ */
 
 /*
  * stubs.c -- stub routines for the X server side of the XINPUT
@@ -102,7 +99,6 @@ ChangeKeyboardDevice (DeviceIntPtr old_dev, DeviceIntPtr new_dev)
     return BadMatch;
     }
 
-
 /***********************************************************************
  *
  * Caller:	ProcXChangePointerDevice
@@ -134,11 +130,8 @@ ChangeKeyboardDevice (DeviceIntPtr old_dev, DeviceIntPtr new_dev)
  */
 
 int
-ChangePointerDevice (
-    DeviceIntPtr	old_dev,
-    DeviceIntPtr	new_dev,
-    unsigned char	x,
-    unsigned char	y)
+ChangePointerDevice(DeviceIntPtr old_dev,
+		    DeviceIntPtr new_dev, unsigned char x, unsigned char y)
     {
     /***********************************************************************
     InitFocusClassDeviceStruct(old_dev);	* allow focusing old ptr*
@@ -273,7 +266,8 @@ SetDeviceMode (register ClientPtr client, DeviceIntPtr dev, int mode)
  */
 
 int
-SetDeviceValuators (register ClientPtr client, DeviceIntPtr dev, int *valuators, int first_valuator, int num_valuators)
+SetDeviceValuators(register ClientPtr client, DeviceIntPtr dev, int *valuators,
+		   int first_valuator, int num_valuators)
 {
     return BadMatch;
     }
@@ -287,10 +281,10 @@ SetDeviceValuators (register ClientPtr client, DeviceIntPtr dev, int *valuators,
  */
 
 int
-ChangeDeviceControl (register ClientPtr client, DeviceIntPtr dev, xDeviceCtl *control)
-{
-    switch (control->control)
+ChangeDeviceControl(register ClientPtr client, DeviceIntPtr dev,
+		    xDeviceCtl * control)
 	{
+    switch (control->control) {
 	case DEVICE_RESOLUTION:
 	    return (BadMatch);
 	default:
