diff --git a/Xext/syncsrv.h b/Xext/syncsrv.h
index 2b70773..e7ef1f4 100644
--- a/Xext/syncsrv.h
+++ b/Xext/syncsrv.h
@@ -71,19 +71,15 @@ typedef struct _SysCounterInfo {
     CARD64	bracket_greater;
     CARD64	bracket_less;
     SyncCounterType counterType;  /* how can this counter change */
-    void        (*QueryValue)(
-			      pointer /*pCounter*/,
+    void (*QueryValue) (pointer /*pCounter */ ,
 			      CARD64 * /*freshvalue*/
 );
-    void	(*BracketValues)(
-				 pointer /*pCounter*/,
+    void (*BracketValues) (pointer /*pCounter */ ,
 				 CARD64 * /*lessthan*/,
 				 CARD64 * /*greaterthan*/
 );
 } SysCounterInfo;
 
-
-
 typedef struct _SyncAlarmClientList {
     ClientPtr	client;
     XID		delete_id;
@@ -117,28 +113,33 @@ typedef union {
     SyncAwait	    await;
 } SyncAwaitUnion;
 
-extern pointer SyncCreateSystemCounter(
-    const char */* name */,
+extern pointer SyncCreateSystemCounter(const char * /* name */ ,
     CARD64  	/* inital_value */,
     CARD64  	/* resolution */,
-    SyncCounterType /* change characterization */,
+                                       SyncCounterType
+                                       /* change characterization */ ,
     void        (* /*QueryValue*/ ) (
-        pointer /* pCounter */,
+                                                                    pointer
+                                                                    /* pCounter */
+                                                                    ,
         CARD64 * /* pValue_return */), /* XXX prototype */
     void        (* /*BracketValues*/) (
-        pointer /* pCounter */, 
-        CARD64 * /* pbracket_less */,
-        CARD64 * /* pbracket_greater */)
+                                                                       pointer
+                                                                       /* pCounter */
+                                                                       ,
+                                                                       CARD64 *
+                                                                       /* pbracket_less */
+                                                                       ,
+                                                                       CARD64 *
+                                                                       /* pbracket_greater */
+                                                                       )
 );
 
-extern void SyncChangeCounter(
-    SyncCounter *	/* pCounter*/,
+extern void SyncChangeCounter(SyncCounter * /* pCounter */ ,
     CARD64  		/* new_value */
 );
 
-extern void SyncDestroySystemCounter(
-    pointer pCounter
-);
+extern void SyncDestroySystemCounter(pointer pCounter);
 
 extern void InitServertime(void);
 
