diff --git a/exa/exa_accel.c b/exa/exa_accel.c
index 028d936..90595d4 100644
--- a/exa/exa_accel.c
+++ b/exa/exa_accel.c
@@ -1288,7 +1288,8 @@ exaFillRegionTiled (DrawablePtr	pDrawable,
     }
 
 fallback:
-    if (alu != GXcopy || !EXA_PM_IS_SOLID(pDrawable, planemask)) {
+    if (alu != GXcopy || pPatOrg->x != 0 || pPatOrg->y != 0 ||
+	!EXA_PM_IS_SOLID(pDrawable, planemask)) {
 	REGION_TRANSLATE(pScreen, pRegion, -xoff, -yoff);
 	return FALSE;
     }
