diff --git a/config/dbus.c b/config/dbus.c
index cef8ed5..37462ac 100644
--- a/config/dbus.c
+++ b/config/dbus.c
@@ -175,7 +175,7 @@ add_device(DBusMessage *message, DBusMessage *reply, DBusError *error)
 unwind:
     if (ret != Success) {
         if (dev)
-            RemoveDevice(dev);
+            RemoveDevice(dev, TRUE);
 
         err = -ret;
         dbus_message_iter_append_basic(&reply_iter, DBUS_TYPE_INT32, &err);
