diff --git a/hw/xfree86/xaa/xaaTEGlyph.c b/hw/xfree86/xaa/xaaTEGlyph.c
index cb60266..d1d12c7 100644
--- a/hw/xfree86/xaa/xaaTEGlyph.c
+++ b/hw/xfree86/xaa/xaaTEGlyph.c
@@ -126,7 +126,7 @@ GlyphScanlineFuncPtr glyph_scanline_func[32] = {
    DrawTETextScanlineGeneric, DrawTETextScanlineGeneric
 };
 
-GlyphScanlineFuncPtr *glyph_get_scanline_func(void) {
+_X_EXPORT GlyphScanlineFuncPtr *glyph_get_scanline_func(void) {
    return glyph_scanline_func;
 }
 
