diff --git a/hw/xwin/glx/wgl_ext_api.h b/hw/xwin/glx/wgl_ext_api.h
index 77efd70..b7231eb 100644
--- a/hw/xwin/glx/wgl_ext_api.h
+++ b/hw/xwin/glx/wgl_ext_api.h
@@ -60,25 +60,21 @@ BOOL wglChoosePixelFormatARBWrapper(HDC hdc,
                                      const int *piAttribIList,
                                      const FLOAT *pfAttribFList,
                                      UINT nMaxFormats,
-                                     int *piFormats,
-                                     UINT *nNumFormats);
+                                    int *piFormats, UINT * nNumFormats);
 
 HPBUFFERARB wglCreatePbufferARBWrapper(HDC hDC,
                                        int iPixelFormat,
                                        int iWidth,
-                                       int iHeight,
-                                       const int *piAttribList);
+                                       int iHeight, const int *piAttribList);
 
 HDC wglGetPbufferDCARBWrapper(HPBUFFERARB hPbuffer);
 
-int wglReleasePbufferDCARBWrapper(HPBUFFERARB hPbuffer,
-                                  HDC hDC);
+int wglReleasePbufferDCARBWrapper(HPBUFFERARB hPbuffer, HDC hDC);
 
 BOOL wglDestroyPbufferARBWrapper(HPBUFFERARB hPbuffer);
 
 BOOL wglQueryPbufferARBWrapper(HPBUFFERARB hPbuffer,
-                                int iAttribute,
-                                int *piValue);
+                               int iAttribute, int *piValue);
 
 BOOL wglSwapIntervalEXTWrapper(int interval);
 
