diff --git a/mi/mieq.c b/mi/mieq.c
index 971edf9..aef6fae 100644
--- a/mi/mieq.c
+++ b/mi/mieq.c
@@ -386,7 +386,7 @@ mieqProcessInputEvents(void)
         }
 
         /* Update the sprite now. Next event may be from different device. */
-        if (type == DeviceMotionNotify && master)
+        if (type == DeviceMotionNotify && (master || dev->isMaster))
             miPointerUpdateSprite(dev);
     }
 }
