diff --git a/Xext/sleepuntil.c b/Xext/sleepuntil.c
index 30be8c0..f8cedbe 100644
--- a/Xext/sleepuntil.c
+++ b/Xext/sleepuntil.c
@@ -1,5 +1,4 @@
 /*
- * $Xorg: sleepuntil.c,v 1.4 2001/02/09 02:04:33 xorgcvs Exp $
  *
 Copyright 1992, 1998  The Open Group
 
@@ -25,7 +24,6 @@ in this Software without prior written authorization from The Open Group.
  *
  * Author:  Keith Packard, MIT X Consortium
  */
-/* $XFree86: xc/programs/Xserver/Xext/sleepuntil.c,v 3.6 2003/07/16 01:38:30 dawes Exp $ */
 
 /* dixsleep.c - implement millisecond timeouts for X clients */
 
