diff --git a/dix/main.c b/dix/main.c
index 5c46dc1..47a932f 100644
--- a/dix/main.c
+++ b/dix/main.c
@@ -184,6 +184,9 @@ int main(int argc, char *argv[], char *envp[])
 	clients[0] = serverClient;
 	currentMaxClients = 1;
 
+	/* Initialize privates before first allocation */
+	dixResetPrivates();
+
 	/* Initialize server client devPrivates, to be reallocated as
 	 * more client privates are registered
 	 */
@@ -200,7 +203,6 @@ int main(int argc, char *argv[], char *envp[])
 	InitEvents();
 	InitSelections();
 	InitGlyphCaching();
-	dixResetPrivates();
 	dixResetRegistry();
 	ResetFontPrivateIndex();
 	InitCallbackManager();
