diff --git a/hw/xfree86/shadowfb/sfbmodule.c b/hw/xfree86/shadowfb/sfbmodule.c
index 6793c8f..663f587 100644
--- a/hw/xfree86/shadowfb/sfbmodule.c
+++ b/hw/xfree86/shadowfb/sfbmodule.c
@@ -1,6 +1,10 @@
 /* $XFree86: xc/programs/Xserver/hw/xfree86/shadowfb/sfbmodule.c,v 1.1 1999/01/31 12:38:06 dawes Exp $ */
 
 
+#ifdef HAVE_XORG_CONFIG_H
+#include <xorg-config.h>
+#endif
+
 #ifdef XFree86LOADER
 
 #include "xf86Module.h"
