diff --git a/xfixes/xfixesint.h b/xfixes/xfixesint.h
index 5f08807..7f3b119 100755
--- a/xfixes/xfixesint.h
+++ b/xfixes/xfixesint.h
@@ -61,9 +61,7 @@
 #include "selection.h"
 #include "xfixes.h"
 
-extern unsigned char	XFixesReqCode;
 extern int		XFixesEventBase;
-extern int		XFixesClientPrivateIndex;
 
 typedef struct _XFixesClient {
     CARD32	major_version;
@@ -73,7 +71,6 @@ typedef struct _XFixesClient {
 #define GetXFixesClient(pClient)    ((XFixesClientPtr) (pClient)->devPrivates[XFixesClientPrivateIndex].ptr)
 
 extern int	(*ProcXFixesVector[XFixesNumberRequests])(ClientPtr);
-extern int	(*SProcXFixesVector[XFixesNumberRequests])(ClientPtr);
 
 /* Initialize extension at server startup time */
 
