diff --git a/include/privates.h b/include/privates.h
index e377b30..9539a29 100644
--- a/include/privates.h
+++ b/include/privates.h
@@ -28,13 +28,6 @@ typedef struct _Private {
 } PrivateRec;
 
 /*
- * Backwards compatibility macro.  Use to get the proper PrivateRec
- * reference from any of the structure types that supported the old
- * devPrivates mechanism.
- */
-#define DEVPRIV_PTR(foo) ((PrivateRec **)(&(foo)->devPrivates[0].ptr))
-
-/*
  * Request pre-allocated private space for your driver/module.
  * Calling this is not necessary if only a pointer by itself is needed.
  */
@@ -154,7 +147,4 @@ dixLookupPrivateOffset(RESTYPE type);
 extern int
 dixRegisterPrivateOffset(RESTYPE type, unsigned offset);
 
-/* Used by the legacy support, don't rely on this being here */
-#define PadToLong(w) ((((w) + sizeof(long)-1) / sizeof(long)) * sizeof(long))
-
 #endif /* PRIVATES_H */
