diff --git a/hw/xfree86/ddc/ddcProperty.c b/hw/xfree86/ddc/ddcProperty.c
index a4384f1..66d734a 100644
--- a/hw/xfree86/ddc/ddcProperty.c
+++ b/hw/xfree86/ddc/ddcProperty.c
@@ -45,7 +45,10 @@ addRootWindowProperties(ScrnInfoPtr pScrn, xf86MonPtr DDC)
     Bool makeEDID1prop = FALSE;
     Bool makeEDID2prop = FALSE;
 
-    if (DDC->ver.version == 1) {
+    if (DDC->flags & MONITOR_DISPLAYID) {
+	/* Don't bother, use RANDR already */
+	return;
+    } else if (DDC->ver.version == 1) {
 	makeEDID1prop = TRUE;
     } else if (DDC->ver.version == 2) {
 	int checksum1;
