diff --git a/hw/kdrive/neomagic/neomagic.c b/hw/kdrive/neomagic/neomagic.c
index 68dbb91..b9c6f6f 100644
--- a/hw/kdrive/neomagic/neomagic.c
+++ b/hw/kdrive/neomagic/neomagic.c
@@ -328,7 +328,6 @@ KdCardFuncs    neoFuncs = {
 
     neoDrawInit,              // initAccel
     neoDrawEnable,            // enableAccel
-    neoDrawSync,              // syncAccel
     neoDrawDisable,           // disableAccel
     neoDrawFini,              // finiAccel
 
