diff --git a/xkb/XKBGAlloc.c b/xkb/XKBGAlloc.c
index 796931c..d1adea3 100644
--- a/xkb/XKBGAlloc.c
+++ b/xkb/XKBGAlloc.c
@@ -114,11 +114,9 @@ register char *ptr;
     }
     if (freeAll) {
 	(*num_inout)= (*sz_inout)= 0;
-	if (*elems) {
 	    free(*elems);
 	    *elems= NULL;
 	}
-    }
     else if (first+count>=(*num_inout))
 	*num_inout= first;
     else {
@@ -137,14 +135,10 @@ _XkbClearProperty(char *prop_in)
 {
 XkbPropertyPtr	prop= (XkbPropertyPtr)prop_in;
 
-    if (prop->name) {
 	free(prop->name);
 	prop->name= NULL;
-    }
-    if (prop->value) {
 	free(prop->value);
 	prop->value= NULL;
-    }
     return;
 }
 
