diff --git a/hw/kdrive/ephyr/ephyr_draw.c b/hw/kdrive/ephyr/ephyr_draw.c
index 0c1a5d0..84faecc 100644
--- a/hw/kdrive/ephyr/ephyr_draw.c
+++ b/hw/kdrive/ephyr/ephyr_draw.c
@@ -226,6 +226,12 @@ static Bool
 ephyrCheckComposite(int op, PicturePtr pSrcPicture, PicturePtr pMaskPicture,
 		    PicturePtr pDstPicture)
 {
+    /* Exercise the component alpha helper, so fail on this case like a normal
+     * driver
+     */
+    if (pMaskPicture && pMaskPicture->componentAlpha && op == PictOpOver)
+	return FALSE;
+
     return TRUE;
 }
 
