diff --git a/hw/xfree86/common/xf86xv.c b/hw/xfree86/common/xf86xv.c
index 85b13d3..abbe033 100644
--- a/hw/xfree86/common/xf86xv.c
+++ b/hw/xfree86/common/xf86xv.c
@@ -1123,6 +1123,10 @@ xf86XVClipNotify(WindowPtr pWin, int dx, int dy)
 
      pPriv->pCompositeClip = NULL;
 
+     if (pPriv->AdaptorRec->ClipNotify)
+        (*pPriv->AdaptorRec->ClipNotify)(pPriv->pScrn, pPriv->DevPriv.ptr,
+                                         pWin, dx, dy);
+
      /* Stop everything except images, but stop them too if the
 	window isn't visible.  But we only remove the images. */
 
