diff --git a/fb/fbpoint.c b/fb/fbpoint.c
index 5ba24e0..5229d69 100644
--- a/fb/fbpoint.c
+++ b/fb/fbpoint.c
@@ -35,6 +35,8 @@ typedef void	(*FbDots)  (FbBits	*dst,
 			    BoxPtr	pBox,
 			    xPoint	*pts,
 			    int		npt,
+			    int		xorg,
+			    int		yorg,
 			    int		xoff,
 			    int		yoff,
 			    FbBits	and,
@@ -47,6 +49,8 @@ fbDots (FbBits	    *dstOrig,
 	BoxPtr	    pBox,
 	xPoint	    *pts,
 	int	    npt,
+	int	    xorg,
+	int	    yorg,
 	int	    xoff,
 	int	    yoff,
 	FbBits	    andOrig,
@@ -66,13 +70,13 @@ fbDots (FbBits	    *dstOrig,
     y2 = pBox->y2;
     while (npt--)
     {
-	x = pts->x + xoff;
-	y = pts->y + yoff;
+	x = pts->x + xorg;
+	y = pts->y + yorg;
 	pts++;
 	if (x1 <= x && x < x2 && y1 <= y && y < y2)
 	{
-	    x *= dstBpp;
-	    d = dst + (y * dstStride) + (x >> FB_STIP_SHIFT);
+	    x = (x + xoff) * dstBpp;
+	    d = dst + ((y + yoff) * dstStride) + (x >> FB_STIP_SHIFT);
 	    x &= FB_STIP_MASK;
 #ifdef FB_24BIT
 	    if (dstBpp == 24)
@@ -156,5 +160,5 @@ fbPolyPoint (DrawablePtr    pDrawable,
     for (nBox = REGION_NUM_RECTS (pClip), pBox = REGION_RECTS (pClip);
 	 nBox--; pBox++)
 	(*dots) (dst, dstStride, dstBpp, pBox, pptInit, nptInit, 
-		 pDrawable->x + dstXoff, pDrawable->y + dstYoff, and, xor);
+		 pDrawable->x, pDrawable->y, dstXoff, dstYoff, and, xor);
 }
