diff --git a/xkb/ddxLoad.c b/xkb/ddxLoad.c
index 25a07f0..fc49d99 100644
--- a/xkb/ddxLoad.c
+++ b/xkb/ddxLoad.c
@@ -48,7 +48,7 @@ THE USE OR PERFORMANCE OF THIS SOFTWARE.
 #include <X11/extensions/XI.h>
 #include "xkb.h"
 
-#if defined(CSRG_BASED) || defined(linux) || defined(__osf__) || defined(__GNU__)
+#if defined(CSRG_BASED) || defined(linux) || defined(__GNU__)
 #include <paths.h>
 #endif
 
