diff --git a/hw/xfree86/parser/scan.c b/hw/xfree86/parser/scan.c
index 03cbc8a..cdca9ca 100644
--- a/hw/xfree86/parser/scan.c
+++ b/hw/xfree86/parser/scan.c
@@ -1207,21 +1207,21 @@ xf86getBoolValue(Bool *val, const char *str)
 	if (*str == '\0') {
 		*val = TRUE;
 	} else {
-		if (strcmp(str, "1") == 0)
+		if (xf86nameCompare(str, "1") == 0)
 			*val = TRUE;
-		else if (strcmp(str, "on") == 0)
+		else if (xf86nameCompare(str, "on") == 0)
 			*val = TRUE;
-		else if (strcmp(str, "true") == 0)
+		else if (xf86nameCompare(str, "true") == 0)
 			*val = TRUE;
-		else if (strcmp(str, "yes") == 0)
+		else if (xf86nameCompare(str, "yes") == 0)
 			*val = TRUE;
-		else if (strcmp(str, "0") == 0)
+		else if (xf86nameCompare(str, "0") == 0)
 			*val = FALSE;
-		else if (strcmp(str, "off") == 0)
+		else if (xf86nameCompare(str, "off") == 0)
 			*val = FALSE;
-		else if (strcmp(str, "false") == 0)
+		else if (xf86nameCompare(str, "false") == 0)
 			*val = FALSE;
-		else if (strcmp(str, "no") == 0)
+		else if (xf86nameCompare(str, "no") == 0)
 			*val = FALSE;
 		else
 			return FALSE;
