diff --git a/hw/darwin/darwin.c b/hw/darwin/darwin.c
index 7b48368..4e34954 100644
--- a/hw/darwin/darwin.c
+++ b/hw/darwin/darwin.c
@@ -156,7 +156,7 @@ DarwinPrintBanner()
 #endif
   ErrorF("\nRelease Date: %s\n", XF86_DATE);
   ErrorF("X Protocol Version %d, Revision %d, %s\n",
-         X_PROTOCOL, X_PROTOCOL_REVISION, XORG_RELEASE );
+         X_PROTOCOL, X_PROTOCOL_REVISION, XORG_VERSION_CURRENT );
   ErrorF("Build Operating System: %s %s\n", OSNAME, OSVENDOR);
 #ifdef HAS_UTSNAME
   {
