diff --git a/hw/xfree86/utils/xorgcfg/interface.c b/hw/xfree86/utils/xorgcfg/interface.c
index b6ac82a..9c280e0 100644
--- a/hw/xfree86/utils/xorgcfg/interface.c
+++ b/hw/xfree86/utils/xorgcfg/interface.c
@@ -68,8 +68,7 @@
 #define DefaultXFree86Dir	"/usr"
 #endif
 
-#define IS_KBDDRIV(S) ((strcasecmp((S),"kbd") == 0) || \
-	(strcasecmp((S), "keyboard") == 0))
+#define IS_KBDDRIV(S) ((strcasecmp((S),"kbd") == 0)
 
 /*
  * Prototypes
