diff --git a/GL/glx/indirect_size_get.c b/GL/glx/indirect_size_get.c
index 9285714..f64fb7e 100644
--- a/GL/glx/indirect_size_get.c
+++ b/GL/glx/indirect_size_get.c
@@ -652,6 +652,10 @@ __glGetBooleanv_size(GLenum e)
     case GL_WEIGHT_ARRAY_SIZE_ARB:
     case GL_WEIGHT_ARRAY_ARB:
     case GL_PACK_INVERT_MESA:
+    case GL_STENCIL_BACK_FUNC_ATI:
+    case GL_STENCIL_BACK_FAIL_ATI:
+    case GL_STENCIL_BACK_PASS_DEPTH_FAIL_ATI:
+    case GL_STENCIL_BACK_PASS_DEPTH_PASS_ATI:
     case GL_FRAGMENT_PROGRAM_ARB:
     case GL_MAX_DRAW_BUFFERS_ARB:
 /*      case GL_MAX_DRAW_BUFFERS_ATI:*/
