diff --git a/Xi/allowev.h b/Xi/allowev.h
index 2d1132d..199744f 100644
--- a/Xi/allowev.h
+++ b/Xi/allowev.h
@@ -1,4 +1,3 @@
-/* $XFree86: xc/programs/Xserver/Xi/allowev.h,v 3.1 1996/04/15 11:18:23 dawes Exp $ */
 /************************************************************
 
 Copyright 1996 by Thomas E. Dickey <dickey@clark.net>
@@ -31,14 +30,10 @@ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 #ifndef ALLOWEV_H
 #define ALLOWEV_H 1
 
-int
-SProcXAllowDeviceEvents(
-	ClientPtr              /* client */
+int SProcXAllowDeviceEvents(ClientPtr	/* client */
 	);
 
-int
-ProcXAllowDeviceEvents(
-	ClientPtr              /* client */
+int ProcXAllowDeviceEvents(ClientPtr	/* client */
 	);
 
 #endif /* ALLOWEV_H */
