diff --git a/Xext/xres.c b/Xext/xres.c
index 6263066..9f3624b 100644
--- a/Xext/xres.c
+++ b/Xext/xres.c
@@ -2,7 +2,7 @@
    Copyright (c) 2002  XFree86 Inc
 */
 /* $XFree86: xc/programs/Xserver/Xext/xres.c,v 1.7tsi Exp $ */
-/* $XdotOrg: xserver/xorg/Xext/xres.c,v 1.7 2005/07/03 08:53:36 daniels Exp $ */
+/* $XdotOrg: xserver/xorg/Xext/xres.c,v 1.8 2006/02/11 19:16:50 alanc Exp $ */
 
 #define NEED_EVENTS
 #define NEED_REPLIES
@@ -224,7 +224,7 @@ ProcXResQueryClientPixmapBytes (ClientPtr client)
     rep.sequenceNumber = client->sequence;
     rep.length = 0;
     rep.bytes = bytes;
-#ifdef XSERVER64
+#ifdef _XSERVER64
     rep.bytes_overflow = bytes >> 32;
 #else
     rep.bytes_overflow = 0;
