diff --git a/hw/xnest/Screen.c b/hw/xnest/Screen.c
index 1ce79cc..660e54f 100644
--- a/hw/xnest/Screen.c
+++ b/hw/xnest/Screen.c
@@ -14,6 +14,10 @@ is" without express or implied warranty.
 */
 /* $XFree86: xc/programs/Xserver/hw/xnest/Screen.c,v 3.12 2003/11/14 22:25:59 dawes Exp $ */
 
+#ifdef HAVE_XNEST_CONFIG_H
+#include <xnest-config.h>
+#endif
+
 #include <X11/X.h>
 #include <X11/Xproto.h>
 #include "scrnintstr.h"
