diff --git a/xkb/xkbInit.c b/xkb/xkbInit.c
index 5ac06fe..7c611bc 100644
--- a/xkb/xkbInit.c
+++ b/xkb/xkbInit.c
@@ -594,6 +594,7 @@ InitKeyboardDeviceStruct(DeviceIntPtr dev, XkbRMLVOSet *rmlvo,
     InitFocusClassDeviceStruct(dev);
 
     xkbi->kbdProc = ctrl_func;
+    dev->kbdfeed->BellProc = bell_func;
     dev->kbdfeed->CtrlProc = XkbDDXKeybdCtrlProc;
 
     dev->kbdfeed->ctrl = defaultKeyboardControl;
