diff --git a/dix/getevents.c b/dix/getevents.c
index b2be9bd..b735318 100644
--- a/dix/getevents.c
+++ b/dix/getevents.c
@@ -400,7 +400,7 @@ getValuatorEvents(EventList *events, DeviceIntPtr pDev, int first_valuator,
         xv = (deviceValuator*)events->event;
         xv->type = DeviceValuator;
         xv->first_valuator = i;
-        xv->num_valuators = ((num_valuators - i) > 6) ? 6 : (num_valuators - i);
+        xv->num_valuators = ((final_valuator - i) > 6) ? 6 : (final_valuator - i);
         xv->deviceid = pDev->id;
         switch (final_valuator - i) {
         case 6:
