diff --git a/hw/kdrive/src/kkeymap.h b/hw/kdrive/src/kkeymap.h
index ccc5e86..f66cfa4 100644
--- a/hw/kdrive/src/kkeymap.h
+++ b/hw/kdrive/src/kkeymap.h
@@ -1,6 +1,4 @@
 /*
- * Id: kkeymap.h,v 1.1 1999/11/02 03:54:46 keithp Exp $
- *
  * Copyright © 1999 Keith Packard
  *
  * Permission to use, copy, modify, distribute, and sell this software and its
@@ -21,7 +19,6 @@
  * TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
  * PERFORMANCE OF THIS SOFTWARE.
  */
-/* $RCSId: xc/programs/Xserver/hw/kdrive/kkeymap.h,v 1.1 1999/11/19 13:53:50 hohndel Exp $ */
 /*
  *  All global variables and functions pertaining to keyboard key mapping
  *  live in this header file.
