diff --git a/hw/xwin/winclipboardunicode.c b/hw/xwin/winclipboardunicode.c
index ba86915..a297bf2 100644
--- a/hw/xwin/winclipboardunicode.c
+++ b/hw/xwin/winclipboardunicode.c
@@ -54,13 +54,11 @@ winClipboardDetectUnicodeSupport (void)
     {
     case VER_PLATFORM_WIN32_NT:
       /* Unicode supported on NT only */
-      ErrorF ("DetectUnicodeSupport - Windows NT/2000/XP\n");
       fReturn = TRUE;
       break;
 
     case VER_PLATFORM_WIN32_WINDOWS:
       /* Unicode is not supported on non-NT */
-      ErrorF ("DetectUnicodeSupport - Windows 95/98/Me\n");
       fReturn = FALSE;
       break;
     }
