diff --git a/hw/xnest/Pointer.h b/hw/xnest/Pointer.h
index 27a2011..872dfb8 100644
--- a/hw/xnest/Pointer.h
+++ b/hw/xnest/Pointer.h
@@ -23,6 +23,8 @@ is" without express or implied warranty.
         (ButtonPressMask | ButtonReleaseMask | PointerMotionMask | \
 	 EnterWindowMask | LeaveWindowMask)
 
+extern DeviceIntPtr xnestPointerDevice;
+
 void xnestChangePointerControl(DeviceIntPtr pDev, PtrCtrl *ctrl);
 int xnestPointerProc(DeviceIntPtr pDev, int onoff);
 
