diff --git a/hw/xgl/xglpict.c b/hw/xgl/xglpict.c
index 90f1c90..dcb1f48 100644
--- a/hw/xgl/xglpict.c
+++ b/hw/xgl/xglpict.c
@@ -164,6 +164,9 @@ xglChangePicture (PicturePtr pPicture,
     if (pPicture->stateChanges & CPComponentAlpha)
 	pPixmapPriv->pictureMask |= xglPCComponentAlphaMask;
 
+    if (pPicture->stateChanges & CPDither)
+	pPixmapPriv->pictureMask |= xglPCDitherMask;
+
     XGL_PICTURE_SCREEN_UNWRAP (ChangePicture);
     (*pPictureScreen->ChangePicture) (pPicture, mask);
     XGL_PICTURE_SCREEN_WRAP (ChangePicture, xglChangePicture);
@@ -269,6 +272,14 @@ xglUpdatePicture (PicturePtr pPicture)
 	    glitz_surface_set_component_alpha (surface, 0);
     }
 
+    if (pPixmapPriv->pictureMask & xglPCDitherMask)
+    {
+	if (pPicture->dither)
+	    glitz_surface_set_dither (surface, 1);
+	else
+	    glitz_surface_set_dither (surface, 0);
+    }
+
     pPixmapPriv->pictureMask &= ~XGL_PICTURE_CHANGES (~0);
 }
 
