diff --git a/Xi/listdev.h b/Xi/listdev.h
index 125ea35..bdd67ce 100644
--- a/Xi/listdev.h
+++ b/Xi/listdev.h
@@ -1,4 +1,3 @@
-/* $XFree86: xc/programs/Xserver/Xi/listdev.h,v 3.1 1996/04/15 11:18:57 dawes Exp $ */
 /************************************************************
 
 Copyright 1996 by Thomas E. Dickey <dickey@clark.net>
@@ -31,26 +30,18 @@ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 #ifndef LISTDEV_H
 #define LISTDEV_H 1
 
-int
-SProcXListInputDevices(
-	ClientPtr              /* client */
+int SProcXListInputDevices(ClientPtr	/* client */
 	);
 
-int
-ProcXListInputDevices (
-	ClientPtr              /* client */
+int ProcXListInputDevices(ClientPtr	/* client */
 	);
 
-void
-SizeDeviceInfo (
-	DeviceIntPtr           /* d */,
+void SizeDeviceInfo(DeviceIntPtr /* d */ ,
 	int *                  /* namesize */,
 	int *                  /* size */
 	);
 
-void
-ListDeviceInfo (
-	ClientPtr              /* client */,
+void ListDeviceInfo(ClientPtr /* client */ ,
 	DeviceIntPtr           /* d */,
 	xDeviceInfoPtr         /* dev */,
 	char **                /* devbuf */,
@@ -58,44 +49,32 @@ ListDeviceInfo (
 	char **                /* namebuf */
 	);
 
-void
-CopyDeviceName (
-	char **                /* namebuf */,
+void CopyDeviceName(char ** /* namebuf */ ,
 	char *                 /* name */
 	);
 
-void
-CopySwapDevice (
-	ClientPtr              /* client */,
+void CopySwapDevice(ClientPtr /* client */ ,
 	DeviceIntPtr           /* d */,
 	int                    /* num_classes */,
 	char **                /* buf */
 	);
 
-void
-CopySwapKeyClass (
-	ClientPtr              /* client */,
+void CopySwapKeyClass(ClientPtr /* client */ ,
 	KeyClassPtr            /* k */,
 	char **                /* buf */
 	);
 
-void
-CopySwapButtonClass (
-	ClientPtr              /* client */,
+void CopySwapButtonClass(ClientPtr /* client */ ,
 	ButtonClassPtr         /* b */,
 	char **                /* buf */
 	);
 
-int
-CopySwapValuatorClass (
-	ClientPtr              /* client */,
+int CopySwapValuatorClass(ClientPtr /* client */ ,
 	ValuatorClassPtr       /* v */,
 	char **                /* buf */
 	);
 
-void
-SRepXListInputDevices (
-	ClientPtr              /* client */,
+void SRepXListInputDevices(ClientPtr /* client */ ,
 	int                    /* size */,
 	xListInputDevicesReply * /* rep */
 	);
