diff --git a/hw/xfree86/dri/dri.c b/hw/xfree86/dri/dri.c
index 72139de..dcd455c 100644
--- a/hw/xfree86/dri/dri.c
+++ b/hw/xfree86/dri/dri.c
@@ -1654,16 +1654,12 @@ DRICopyWindow(WindowPtr pWin, DDXPointRec ptOldOrg, RegionPtr prgnSrc)
 static void
 DRIGetSecs(long *secs, long *usecs)
 {
-#ifdef XFree86LOADER
-    getsecs(secs,usecs);
-#else
     struct timeval tv;
 
     gettimeofday(&tv, NULL);
 
     *secs  = tv.tv_sec;
     *usecs = tv.tv_usec;
-#endif
 }
 
 static unsigned long
