diff --git a/hw/kdrive/ephyr/hostx.c b/hw/kdrive/ephyr/hostx.c
index d50c6b1..dd988dc 100644
--- a/hw/kdrive/ephyr/hostx.c
+++ b/hw/kdrive/ephyr/hostx.c
@@ -1211,7 +1211,6 @@ hostx_set_window_geometry (int a_win, EphyrBox *a_geo)
 
     XMoveWindow (dpy, a_win, a_geo->x, a_geo->y) ;
     XResizeWindow (dpy, a_win, a_geo->width, a_geo->height) ;
-    XFlush (dpy) ;
     EPHYR_LOG ("leave\n") ;
     return TRUE;
 }
