diff --git a/randr/randrstr.h b/randr/randrstr.h
index 9e7bbf3..93960e5 100644
--- a/randr/randrstr.h
+++ b/randr/randrstr.h
@@ -825,10 +825,10 @@ RROutputDestroy (RROutputPtr	output);
 extern _X_EXPORT int
 ProcRRGetOutputInfo (ClientPtr client);
 
-extern int
+extern _X_EXPORT int
 ProcRRSetOutputPrimary (ClientPtr client);
 
-extern int
+extern _X_EXPORT int
 ProcRRGetOutputPrimary (ClientPtr client);
 
 /*
