diff --git a/exa/exa_migration.c b/exa/exa_migration.c
index f31cce3..571650c 100644
--- a/exa/exa_migration.c
+++ b/exa/exa_migration.c
@@ -348,7 +348,7 @@ exaDoMoveInPixmap (ExaMigrationPtr migrate)
     pPixmap->drawable.serialNumber = NEXT_SERIAL_NUMBER;
 }
 
-_X_EXPORT void
+void
 exaMoveInPixmap (PixmapPtr pPixmap)
 {
     static ExaMigrationRec migrate = { .as_dst = FALSE, .as_src = TRUE,
@@ -389,7 +389,7 @@ exaDoMoveOutPixmap (ExaMigrationPtr migrate)
     }
 }
 
-_X_EXPORT void
+void
 exaMoveOutPixmap (PixmapPtr pPixmap)
 {
     static ExaMigrationRec migrate = { .as_dst = FALSE, .as_src = TRUE,
