diff --git a/render/mipict.h b/render/mipict.h
index 60baf7f..b3d4bae 100644
--- a/render/mipict.h
+++ b/render/mipict.h
@@ -80,14 +80,6 @@ miChangePictureFilter (PicturePtr pPicture,
 		       xFixed     *params,
 		       int	  nparams);
 
-Bool
-miClipPicture (RegionPtr    pRegion,
-	       PicturePtr   pPicture,
-	       INT16	    xReg,
-	       INT16	    yReg,
-	       INT16	    xPict,
-	       INT16	    yPict);
-
 void
 miCompositeSourceValidate (PicturePtr	pPicture,
 			   INT16	x,
@@ -170,12 +162,6 @@ void
 miTriangleBounds (int ntri, xTriangle *tris, BoxPtr bounds);
 
 void
-miRasterizeTriangle (PicturePtr	pMask,
-		     xTriangle	*tri,
-		     int	x_off,
-		     int	y_off);
-
-void
 miTriangles (CARD8	    op,
 	     PicturePtr	    pSrc,
 	     PicturePtr	    pDst,
