diff --git a/dix/glyphcurs.c b/dix/glyphcurs.c
index f74b137..f10400f 100644
--- a/dix/glyphcurs.c
+++ b/dix/glyphcurs.c
@@ -91,7 +91,7 @@ ServerBitsFromGlyph(FontPtr pfont, unsigned ch, CursorMetricPtr cm, unsigned cha
 
     pScreen = screenInfo.screens[0];
     nby = BitmapBytePad(cm->width) * (long)cm->height;
-    pbits = xcalloc(1, nby);
+    pbits = calloc(1, nby);
     if (!pbits)
 	return BadAlloc;
 
@@ -105,7 +105,7 @@ ServerBitsFromGlyph(FontPtr pfont, unsigned ch, CursorMetricPtr cm, unsigned cha
 	    (*pScreen->DestroyPixmap)(ppix);
 	if (pGC)
 	    FreeScratchGC(pGC);
-	xfree(pbits);
+	free(pbits);
 	return BadAlloc;
     }
 
