diff --git a/cfb/cfbwindow.c b/cfb/cfbwindow.c
index 2c5eada..7160f54 100644
--- a/cfb/cfbwindow.c
+++ b/cfb/cfbwindow.c
@@ -47,6 +47,10 @@ SOFTWARE.
 ******************************************************************/
 /* $XFree86: xc/programs/Xserver/cfb/cfbwindow.c,v 1.5tsi Exp $ */
 
+#ifdef HAVE_DIX_CONFIG_H
+#include <dix-config.h>
+#endif
+
 #include <X11/X.h>
 #include "scrnintstr.h"
 #include "windowstr.h"
