diff --git a/hw/xfree86/common/xf86Priv.h b/hw/xfree86/common/xf86Priv.h
index 0794de7..f40a217 100644
--- a/hw/xfree86/common/xf86Priv.h
+++ b/hw/xfree86/common/xf86Priv.h
@@ -37,6 +37,8 @@
 #ifndef _XF86PRIV_H
 #define _XF86PRIV_H
 
+#include <pciaccess.h>
+
 #include "xf86Privstr.h"
 #include "propertyst.h"
 
@@ -107,7 +109,7 @@ extern Bool xf86Resetting;
 extern Bool xf86Initialising;
 extern Bool xf86ProbeFailed;
 extern int xf86NumScreens;
-extern pciVideoPtr *xf86PciVideoInfo;
+extern struct pci_device ** xf86PciVideoInfo;
 extern xf86CurrentAccessRec xf86CurrentAccess;
 extern const char *xf86VisualNames[];
 extern int xf86Verbose;                 /* verbosity level */
@@ -154,9 +156,6 @@ void xf86ClearEntityListForScreen(int scrnIndex);
 void xf86AddDevToEntity(int entityIndex, GDevPtr dev);
 extern void xf86PostPreInit(void);
 extern void xf86PostScreenInit(void);
-extern memType getValidBIOSBase(PCITAG tag, int num);
-extern memType getEmptyPciRange(PCITAG tag, int base_reg);
-extern int pciTestMultiDeviceCard(int bus, int dev, int func, PCITAG** pTag);
 
 /* xf86Config.c */
 
@@ -197,6 +196,7 @@ void xf86CloseLog(void);
 Bool xf86LoadModules(char **list, pointer *optlist);
 int xf86SetVerbosity(int verb);
 int xf86SetLogVerbosity(int verb);
+Bool xf86CallDriverProbe( struct _DriverRec * drv, Bool detect_only );
 
 /* xf86Io.c */
 
