diff --git a/Xi/getprop.c b/Xi/getprop.c
index bbca245..5308410 100644
--- a/Xi/getprop.c
+++ b/Xi/getprop.c
@@ -1,5 +1,3 @@
-/* $Xorg: getprop.c,v 1.4 2001/02/09 02:04:34 xorgcvs Exp $ */
-
 /************************************************************
 
 Copyright 1989, 1998  The Open Group
@@ -45,7 +43,6 @@ ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
 SOFTWARE.
 
 ********************************************************/
-/* $XFree86: xc/programs/Xserver/Xi/getprop.c,v 3.5 2001/08/23 14:56:19 alanh Exp $ */
 
 /***********************************************************************
  *
@@ -119,21 +116,18 @@ ProcXGetDeviceDontPropagateList (register ClientPtr client)
     rep.count = 0;
 
     pWin = (WindowPtr) LookupWindow (stuff->window, client);
-    if (!pWin)
-        {
+    if (!pWin) {
 	client->errorValue = stuff->window;
 	SendErrorToClient(client, IReqCode, X_GetDeviceDontPropagateList, 0, 
 		BadWindow);
 	return Success;
         }
 
-    if ((others = wOtherInputMasks(pWin)) != 0)
-	{
+    if ((others = wOtherInputMasks(pWin)) != 0) {
 	for (i=0; i<EMASKSIZE; i++)
 	    tbuf = ClassFromMask (NULL, others->dontPropagateMask[i], i, 
 		&count, COUNT);
-	if (count)
-	    {
+	if (count) {
 	    rep.count = count;
 	    buf = (XEventClass *) xalloc (rep.count * sizeof(XEventClass));
 	    rep.length = (rep.count * sizeof (XEventClass) + 3) >> 2;
@@ -145,11 +139,9 @@ ProcXGetDeviceDontPropagateList (register ClientPtr client)
 	    }
 	}
 
-    WriteReplyToClient (client, sizeof (xGetDeviceDontPropagateListReply), 
-	&rep);
+    WriteReplyToClient(client, sizeof(xGetDeviceDontPropagateListReply), &rep);
 
-    if (count)
-	{
+    if (count) {
 	client->pSwapReplyFunc = (ReplySwapPtr)Swap32Write;
 	WriteSwappedDataToClient( client, count * sizeof(XEventClass), buf);
 	xfree (buf);
@@ -165,18 +157,17 @@ ProcXGetDeviceDontPropagateList (register ClientPtr client)
  */
 
 XEventClass
-*ClassFromMask (XEventClass *buf, Mask mask, int maskndx, CARD16 *count, int mode)
+    * ClassFromMask(XEventClass * buf, Mask mask, int maskndx, CARD16 * count,
+		    int mode)
 {
     int		i,j;
     int		id = maskndx;
     Mask	tmask = 0x80000000;
 
     for (i=0; i<32; i++,tmask>>=1)
-	if (tmask & mask)
-	    {
+	if (tmask & mask) {
 	    for (j=0; j<ExtEventIndex; j++)
-		if (EventInfo[j].mask == tmask)
-		    {
+		if (EventInfo[j].mask == tmask) {
 		    if (mode == COUNT)
 			(*count)++;
 		    else
@@ -194,7 +185,8 @@ XEventClass
  */
 
 void
-SRepXGetDeviceDontPropagateList (ClientPtr client, int size, xGetDeviceDontPropagateListReply *rep)
+SRepXGetDeviceDontPropagateList(ClientPtr client, int size,
+				xGetDeviceDontPropagateListReply * rep)
 {
     register char n;
 
