diff --git a/include/dix.h b/include/dix.h
index ebd8f2f..a19b901 100644
--- a/include/dix.h
+++ b/include/dix.h
@@ -365,8 +365,7 @@ extern _X_EXPORT int DeliverEventsToWindow(
     xEventPtr /* pEvents */,
     int /* count */,
     Mask /* filter */,
-    GrabPtr /* grab */,
-    int /* mskidx */);
+    GrabPtr /* grab */);
 
 extern _X_EXPORT int DeliverDeviceEvents(
     WindowPtr /* pWin */,
