diff --git a/hw/xfree86/parser/Pointer.c b/hw/xfree86/parser/Pointer.c
index 65397cd..e8ad290 100644
--- a/hw/xfree86/parser/Pointer.c
+++ b/hw/xfree86/parser/Pointer.c
@@ -189,7 +189,7 @@ xf86parsePointerSection (void)
 				if (xf86getSubToken (&(ptr->inp_comment)) != NUMBER || val.num < 0) {
 					Error (ZAXISMAPPING_MSG, NULL);
 				}
-				if (asprintf(&s, "%ul %ul", val1, val.num) == -1)
+				if (asprintf(&s, "%lu %u", val1, val.num) == -1)
 				    s = NULL;
 				break;
 			case XAXIS:
