diff --git a/Xi/setmmap.c b/Xi/setmmap.c
index 7af5e10..6ed18e0 100644
--- a/Xi/setmmap.c
+++ b/Xi/setmmap.c
@@ -55,11 +55,11 @@ SOFTWARE.
 
 #define	 NEED_EVENTS			/* for inputstr.h    */
 #define	 NEED_REPLIES
-#include "X.h"				/* for inputstr.h    */
-#include "Xproto.h"			/* Request macro     */
+#include <X11/X.h>				/* for inputstr.h    */
+#include <X11/Xproto.h>			/* Request macro     */
 #include "inputstr.h"			/* DeviceIntPtr	     */
-#include "XI.h"
-#include "XIproto.h"
+#include <X11/extensions/XI.h>
+#include <X11/extensions/XIproto.h>
 #include "exevents.h"
 #include "extnsionst.h"
 #include "extinit.h"			/* LookupDeviceIntRec */
