diff --git a/hw/xfree86/common/xf86Bus.h b/hw/xfree86/common/xf86Bus.h
index ea86c1c..b7d1608 100644
--- a/hw/xfree86/common/xf86Bus.h
+++ b/hw/xfree86/common/xf86Bus.h
@@ -74,15 +74,6 @@ typedef struct {
     IOADDRESS                   domainIO;
 } EntityRec, *EntityPtr;
 
-/* asynchronous event handling */
-#ifdef async
-typedef struct _AsyncQRec {
-    void (*func)(pointer);
-    pointer arg;
-    struct _AsyncQRec *next;
-} AsyncQRec, *AsyncQPtr;
-#endif
-
 #define NO_SEPARATE_IO_FROM_MEM 0x0001
 #define NO_SEPARATE_MEM_FROM_IO 0x0002
 #define NEED_VGA_ROUTED 0x0004
