diff --git a/randr/rrdispatch.c b/randr/rrdispatch.c
index ffb46a4..ad1439d 100644
--- a/randr/rrdispatch.c
+++ b/randr/rrdispatch.c
@@ -108,7 +108,7 @@ ProcRRSelectInput (ClientPtr client)
 	if (!pRREvent)
 	{
 	    /* build the entry */
-	    pRREvent = (RREventPtr) xalloc (sizeof (RREventRec));
+	    pRREvent = (RREventPtr) malloc(sizeof (RREventRec));
 	    if (!pRREvent)
 		return BadAlloc;
 	    pRREvent->next = 0;
@@ -131,7 +131,7 @@ ProcRRSelectInput (ClientPtr client)
 	     */
 	    if (!pHead)
 	    {
-		pHead = (RREventPtr *) xalloc (sizeof (RREventPtr));
+		pHead = (RREventPtr *) malloc(sizeof (RREventPtr));
 		if (!pHead ||
 		    !AddResource (pWin->drawable.id, RREventType, (pointer)pHead))
 		{
@@ -174,7 +174,7 @@ ProcRRSelectInput (ClientPtr client)
 		    pNewRREvent->next = pRREvent->next;
 		else
 		    *pHead = pRREvent->next;
-		xfree (pRREvent);
+		free(pRREvent);
 	    }
 	}
     }
