diff --git a/hw/xfree86/dri/xf86dri.c b/hw/xfree86/dri/xf86dri.c
index 9690e88..fdf0e99 100644
--- a/hw/xfree86/dri/xf86dri.c
+++ b/hw/xfree86/dri/xf86dri.c
@@ -404,10 +404,8 @@ ProcXF86DRICreateDrawable(
     if (rc != Success)
 	return rc;
 
-    if (!DRICreateDrawable( screenInfo.screens[stuff->screen],
-			    (Drawable)stuff->drawable,
-			    pDrawable,
-			    (drm_drawable_t *)&rep.hHWDrawable)) {
+    if (!DRICreateDrawable(screenInfo.screens[stuff->screen], client,
+			   pDrawable, (drm_drawable_t *)&rep.hHWDrawable)) {
 	return BadValue;
     }
 
@@ -435,8 +433,7 @@ ProcXF86DRIDestroyDrawable(
     if (rc != Success)
 	return rc;
 
-    if (!DRIDestroyDrawable( screenInfo.screens[stuff->screen], 
-			     (Drawable)stuff->drawable,
+    if (!DRIDestroyDrawable(screenInfo.screens[stuff->screen], client,
 			     pDrawable)) {
 	return BadValue;
     }
