diff --git a/hw/xfree86/common/xf86Module.h b/hw/xfree86/common/xf86Module.h
index f0ac3bd..c668c3b 100644
--- a/hw/xfree86/common/xf86Module.h
+++ b/hw/xfree86/common/xf86Module.h
@@ -86,7 +86,7 @@ typedef enum {
  */
 #define ABI_ANSIC_VERSION	SET_ABI_VERSION(0, 2)
 #define ABI_VIDEODRV_VERSION	SET_ABI_VERSION(0, 7)
-#define ABI_XINPUT_VERSION	SET_ABI_VERSION(0, 4)
+#define ABI_XINPUT_VERSION	SET_ABI_VERSION(0, 5)
 #define ABI_EXTENSION_VERSION	SET_ABI_VERSION(0, 2)
 #define ABI_FONT_VERSION	SET_ABI_VERSION(0, 4)
 
