diff --git a/hw/xfree86/common/xf86xvmc.c b/hw/xfree86/common/xf86xvmc.c
index 40c82c7..7dc4e9d 100644
--- a/hw/xfree86/common/xf86xvmc.c
+++ b/hw/xfree86/common/xf86xvmc.c
@@ -41,19 +41,13 @@
 #include "resource.h"
 #include "dixstruct.h"
 
-#ifdef XFree86LOADER
 #include "xvmodproc.h"
-#endif
 
 #include "xf86xvpriv.h"
 #include "xf86xvmc.h"
 
 typedef int (*XvMCScreenInitProcPtr)(ScreenPtr, int, XvMCAdaptorPtr);
-#ifdef XFree86LOADER
 _X_EXPORT XvMCScreenInitProcPtr XvMCScreenInitProc = NULL;
-#else
-_X_EXPORT XvMCScreenInitProcPtr XvMCScreenInitProc = XvMCScreenInit;
-#endif
 
 
 typedef struct {
