diff --git a/Xi/querydev.h b/Xi/querydev.h
index 92ab2bd..34e87bd 100644
--- a/Xi/querydev.h
+++ b/Xi/querydev.h
@@ -39,4 +39,7 @@ void SRepXIQueryDevice(ClientPtr client, int size, xXIQueryDeviceReply *rep);
 int SizeDeviceClasses(DeviceIntPtr dev);
 int ListDeviceClasses(DeviceIntPtr dev, char* any, uint16_t* nclasses);
 int GetDeviceUse(DeviceIntPtr dev, uint16_t *attachment);
+int ListButtonInfo(DeviceIntPtr dev, xXIButtonInfo* info);
+int ListKeyInfo(DeviceIntPtr dev, xXIKeyInfo* info);
+int ListValuatorInfo(DeviceIntPtr dev, xXIValuatorInfo* info, int axisnumber);
 #endif /* QUERYDEV_H */
