diff --git a/hw/xnest/Init.c b/hw/xnest/Init.c
index 5bf0300..4699111 100644
--- a/hw/xnest/Init.c
+++ b/hw/xnest/Init.c
@@ -129,22 +129,6 @@ void
 DarwinHandleGUI(int argc, char *argv[])
 {
 }
-
-void GlxExtensionInit();
-void GlxWrapInitVisuals(void *procPtr);
-
-void
-DarwinGlxExtensionInit()
-{
-    GlxExtensionInit();
-}
-
-void
-DarwinGlxWrapInitVisuals(
-    void *procPtr)
-{
-    GlxWrapInitVisuals(procPtr);
-}
 #endif
 
 void OsVendorInit()
