diff --git a/Xext/xtest.c b/Xext/xtest.c
index 0a6b1c5..6ee8430 100644
--- a/Xext/xtest.c
+++ b/Xext/xtest.c
@@ -99,10 +99,16 @@ static DISPATCH_PROC(SProcXTestFakeInput);
 static DISPATCH_PROC(SProcXTestGetVersion);
 static DISPATCH_PROC(SProcXTestGrabControl);
 
+Bool
+XTestInitPrivates(void)
+{
+    return dixRegisterPrivateKey(&XTestDevicePrivateKeyRec, PRIVATE_DEVICE, 0);
+}
+
 void
 XTestExtensionInit(INITARGS)
 {
-    if (!dixRegisterPrivateKey(&XTestDevicePrivateKeyRec, PRIVATE_DEVICE, 0))
+    if (!XTestInitPrivates())
 	return;
 
     AddExtension(XTestExtensionName, 0, 0,
