diff --git a/include/dix.h b/include/dix.h
index 57ffee9..970b1d8 100644
--- a/include/dix.h
+++ b/include/dix.h
@@ -460,10 +460,6 @@ extern int GrabDevice(
     CARD8 * /* status */,
     Bool /* coreGrab */);
 
-extern void RemoveOtherCoreGrabs(
-    ClientPtr /* client */,
-    DeviceIntPtr /* dev */);
-
 extern void InitEvents(void);
 extern void InitSprite(
         DeviceIntPtr /* pDev */, 
