diff --git a/hw/xquartz/mach-startup/stub.c b/hw/xquartz/mach-startup/stub.c
index 8bbd4d9..a011c4e 100644
--- a/hw/xquartz/mach-startup/stub.c
+++ b/hw/xquartz/mach-startup/stub.c
@@ -178,17 +178,15 @@ static void send_fd_handoff(int connected_fd, int launchd_fd) {
     
     *((int*)CMSG_DATA(cmsg)) = launchd_fd;
     
-#ifdef DEBUG
-    fprintf(stderr, "Xquartz: Handoff connection established.  Sending message.\n");
-#endif
     if(sendmsg(connected_fd, &msg, 0) < 0) {
-        fprintf(stderr, "Xquartz: Error sending $DISPLAY file descriptor: %s\n", strerror(errno));
+        fprintf(stderr, "Xquartz: Error sending $DISPLAY file descriptor over fd %d: %d -- %s\n", connected_fd, errno, strerror(errno));
         return;
     }
 
 #ifdef DEBUG
-    fprintf(stderr, "Xquartz: Message sent.  Closing.\n");
+    fprintf(stderr, "Xquartz: Message sent.  Closing handoff fd.\n");
 #endif
+
     close(connected_fd);
 }
 
@@ -286,6 +284,10 @@ int main(int argc, char **argv, char **envp) {
                 continue;
             }
 
+#ifdef DEBUG
+            fprintf(stderr, "Xquartz: Handoff connection established (try %d of %d) on fd %d, \"%s\".  Sending message.\n", (int)try+1, (int)try_max, handoff_fd, handoff_socket_filename);
+#endif
+
             send_fd_handoff(handoff_fd, launchd_fd);            
             close(handoff_fd);
             break;
