diff --git a/mi/miinitext.c b/mi/miinitext.c
index 3d9522d..cb34473 100644
--- a/mi/miinitext.c
+++ b/mi/miinitext.c
@@ -384,9 +384,6 @@ extern void DamageExtensionInit(INITARGS);
 #ifdef COMPOSITE
 extern void CompositeExtensionInit(INITARGS);
 #endif
-#ifdef MPX
-extern void MPXExtensionInit(INITARGS);
-#endif
 
 /* The following is only a small first step towards run-time
  * configurable extensions.
@@ -666,9 +663,6 @@ InitExtensions(argc, argv)
 #ifdef DAMAGE
     if (!noDamageExtension) DamageExtensionInit();
 #endif
-#ifdef MPX
-    if (!noMPXExtension) MPXExtensionInit();
-#endif
 }
 
 void
@@ -739,9 +733,6 @@ static ExtensionModule staticExtensions[] = {
 #ifdef XEVIE
     { XevieExtensionInit, "XEVIE", &noXevieExtension, NULL },
 #endif 
-#ifdef MPX
-    { MPXExtensionInit, "MPX", &noMPXExtension, NULL },
-#endif
     { NULL, NULL, NULL, NULL, NULL }
 };
     
