diff --git a/exa/exa_priv.h b/exa/exa_priv.h
index bab8aa2..9bed5c8 100644
--- a/exa/exa_priv.h
+++ b/exa/exa_priv.h
@@ -35,6 +35,9 @@
 #include <X11/X.h>
 #define NEED_EVENTS
 #include <X11/Xproto.h>
+#ifdef MITSHM
+#include "shmint.h"
+#endif
 #include "scrnintstr.h"
 #include "pixmapstr.h"
 #include "windowstr.h"
@@ -306,6 +309,10 @@ exaGetSpans (DrawablePtr pDrawable, int wMax, DDXPointPtr ppt, int *pwidth,
 
 extern const GCOps exaOps;
 
+#ifdef MITSHM
+extern ShmFuncs exaShmFuncs;
+#endif
+
 #ifdef RENDER
 void
 ExaCheckComposite (CARD8      op,
