diff --git a/hw/vfb/InitOutput.c b/hw/vfb/InitOutput.c
index 04887ce..4de0878 100644
--- a/hw/vfb/InitOutput.c
+++ b/hw/vfb/InitOutput.c
@@ -939,7 +939,7 @@ vfbScreenInit(int index, ScreenPtr pScreen, int argc, char **argv)
 
     if (!ret) return FALSE;
 
-    /* miInitializeBackingStore(pScreen); */
+    miInitializeBackingStore(pScreen);
 
     /*
      * Circumvent the backing store that was just initialised.  This amounts
