diff --git a/GL/mesa/X/xf86glx.c b/GL/mesa/X/xf86glx.c
index b1125a7..5085eb9 100644
--- a/GL/mesa/X/xf86glx.c
+++ b/GL/mesa/X/xf86glx.c
@@ -140,12 +140,12 @@ __glXMesaDrawableSwapBuffers(__GLXdrawable *base)
 
 
 static __GLXdrawable *
-__glXMesaContextCreateDrawable(__GLXcontext *context,
+__glXMesaScreenCreateDrawable(__GLXscreen *screen,
 			       DrawablePtr pDraw,
-			       XID drawId)
+			      XID drawId,
+			      __GLcontextModes *modes)
 {
     __GLXMESAdrawable *glxPriv;
-    __GLXscreen *pGlxScreen;
     XMesaVisual xm_vis;
 
     glxPriv = xalloc(sizeof *glxPriv);
@@ -154,7 +154,7 @@ __glXMesaContextCreateDrawable(__GLXcontext *context,
 
     memset(glxPriv, 0, sizeof *glxPriv);
 
-    if (!__glXDrawableInit(&glxPriv->base, context, pDraw, drawId)) {
+    if (!__glXDrawableInit(&glxPriv->base, screen, pDraw, drawId, modes)) {
         xfree(glxPriv);
 	return NULL;
     }
@@ -163,21 +163,10 @@ __glXMesaContextCreateDrawable(__GLXcontext *context,
     glxPriv->base.resize        = __glXMesaDrawableResize;
     glxPriv->base.swapBuffers   = __glXMesaDrawableSwapBuffers;
 
-    pGlxScreen = __glXActiveScreens[pDraw->pScreen->myNum];
-
-    if (glxPriv->base.type == DRAWABLE_WINDOW) {
-	VisualID vid = wVisual((WindowPtr)pDraw);
-
-	glxPriv->base.modes = _gl_context_modes_find_visual(pGlxScreen->modes,
-							    vid);
-    } else {
-	glxPriv->base.modes = glxPriv->base.pGlxPixmap->modes;
-    }
-
-    xm_vis = find_mesa_visual(pGlxScreen, glxPriv->base.modes->visualID);
+    xm_vis = find_mesa_visual(screen, modes->visualID);
     if (xm_vis == NULL) {
 	ErrorF("find_mesa_visual returned NULL for visualID = 0x%04x\n",
-	       glxPriv->base.modes->visualID);
+	       modes->visualID);
 	xfree(glxPriv);
 	return NULL;
     }
@@ -197,7 +186,7 @@ __glXMesaContextDestroy(__GLXcontext *baseContext)
     __GLXMESAcontext *context = (__GLXMESAcontext *) baseContext;
 
     XMesaDestroyContext(context->xmesa);
-    __glXContextDestroy(context);
+    __glXContextDestroy(&context->base);
     xfree(context);
 }
 
@@ -268,7 +257,6 @@ __glXMesaScreenCreateContext(__GLXscreen *screen,
     context->base.loseCurrent    = __glXMesaContextLoseCurrent;
     context->base.copy           = __glXMesaContextCopy;
     context->base.forceCurrent   = __glXMesaContextForceCurrent;
-    context->base.createDrawable = __glXMesaContextCreateDrawable;
 
     xm_vis = find_mesa_visual(screen, modes->visualID);
     if (!xm_vis) {
@@ -419,6 +407,7 @@ __glXMesaScreenProbe(ScreenPtr pScreen)
 
     screen->base.destroy       = __glXMesaScreenDestroy;
     screen->base.createContext = __glXMesaScreenCreateContext;
+    screen->base.createDrawable = __glXMesaScreenCreateDrawable;
     screen->base.pScreen       = pScreen;
 
     /*
