diff --git a/hw/xfree86/loader/xf86sym.c b/hw/xfree86/loader/xf86sym.c
index 8a45fe7..0d1b38c 100644
--- a/hw/xfree86/loader/xf86sym.c
+++ b/hw/xfree86/loader/xf86sym.c
@@ -187,7 +187,7 @@ extern long __umodsi3(long, long);
 #include <sys/io.h>
 #endif
 
-#if defined(__powerpc__) && (defined(Lynx) || defined(linux))
+#if defined(__powerpc__) && defined(linux)
 void _restf14();
 void _restf17();
 void _restf18();
@@ -754,7 +754,7 @@ _X_HIDDEN void *xfree86LookupTab[] = {
     SYMFUNC(outb)
     SYMFUNC(outw)
     SYMFUNC(outl)
-# if defined(NO_INLINE) || defined(Lynx)
+# if defined(NO_INLINE)
     SYMFUNC(mem_barrier)
     SYMFUNC(ldl_u)
     SYMFUNC(eieio)
@@ -769,34 +769,6 @@ _X_HIDDEN void *xfree86LookupTab[] = {
     SYMFUNC(stw_u)
     SYMFUNC(write_mem_barrier)
 # endif
-# if defined(Lynx)
-    SYMFUNC(_restf14)
-    SYMFUNC(_restf17)
-    SYMFUNC(_restf18)
-    SYMFUNC(_restf19)
-    SYMFUNC(_restf20)
-    SYMFUNC(_restf22)
-    SYMFUNC(_restf23)
-    SYMFUNC(_restf24)
-    SYMFUNC(_restf25)
-    SYMFUNC(_restf26)
-    SYMFUNC(_restf27)
-    SYMFUNC(_restf28)
-    SYMFUNC(_restf29)
-    SYMFUNC(_savef14)
-    SYMFUNC(_savef17)
-    SYMFUNC(_savef18)
-    SYMFUNC(_savef19)
-    SYMFUNC(_savef20)
-    SYMFUNC(_savef22)
-    SYMFUNC(_savef23)
-    SYMFUNC(_savef24)
-    SYMFUNC(_savef25)
-    SYMFUNC(_savef26)
-    SYMFUNC(_savef27)
-    SYMFUNC(_savef28)
-    SYMFUNC(_savef29)
-# endif
 # if PPCIO_DEBUG
     SYMFUNC(debug_inb)
     SYMFUNC(debug_inw)
@@ -807,39 +779,23 @@ _X_HIDDEN void *xfree86LookupTab[] = {
 # endif
 #endif
 #if defined(__GNUC__)
-#if !defined(Lynx)
     SYMFUNC(__div64)
-#endif
-#if !defined(Lynx)	/* FIXME: test on others than x86 and !3.1.0a/x86 */
     SYMFUNC(__divdf3)
-#endif
     SYMFUNC(__divdi3)
-#if !defined(Lynx)
     SYMFUNC(__divsf3)
     SYMFUNC(__divsi3)
-#endif
     SYMFUNC(__moddi3)
-#if !defined(Lynx)
     SYMFUNC(__modsi3)
-#endif
-#if !defined(Lynx)
     SYMFUNC(__mul64)
-#endif
-#if !defined(Lynx)
     SYMFUNC(__muldf3)
-#endif
     SYMFUNC(__muldi3)
-#if !defined(Lynx)
     SYMFUNC(__mulsf3)
     SYMFUNC(__mulsi3)
     SYMFUNC(__udivdi3)
     SYMFUNC(__udivsi3)
-#endif
     SYMFUNC(__umoddi3)
-#if !defined(Lynx)
     SYMFUNC(__umodsi3)
 #endif
-#endif
 #if defined(__ia64__)
     SYMFUNC(outw)
     SYMFUNC(outb)
@@ -886,7 +842,7 @@ _X_HIDDEN void *xfree86LookupTab[] = {
     SYMVAR(res8514Exclusive)
     SYMVAR(res8514Shared)
 
-#if defined(__powerpc__) && (!defined(NO_INLINE) || defined(Lynx))
+#if defined(__powerpc__) && !defined(NO_INLINE)
     SYMVAR(ioBase)
 #endif
 
