diff --git a/hw/kdrive/linux/bus.c b/hw/kdrive/linux/bus.c
index a0d9b12..ba8a6cd 100644
--- a/hw/kdrive/linux/bus.c
+++ b/hw/kdrive/linux/bus.c
@@ -77,7 +77,7 @@ BusInit (KdPointerInfo *pi)
                 close(fd);
                 if (pi->path)
                     xfree(pi->path);
-                pi->path = KdSaveString(BusNames[i]);
+                pi->path = strdup(BusNames[i]);
                 return Success;
             }
         }
