diff --git a/hw/xfree86/exa/exaPriv.h b/hw/xfree86/exa/exaPriv.h
index a53f370..41938a7 100644
--- a/hw/xfree86/exa/exaPriv.h
+++ b/hw/xfree86/exa/exaPriv.h
@@ -70,6 +70,7 @@ typedef struct {
     CopyWindowProcPtr 		 SavedCopyWindow;
 #ifdef RENDER
     CompositeProcPtr             SavedComposite;
+    GlyphsProcPtr                SavedGlyphs;
 #endif
     EnableDisableFBAccessProcPtr SavedEnableDisableFBAccess;
     Bool			 wrappedEnableDisableFB;
@@ -242,6 +243,12 @@ exaEnableDisableFBAccess (int index, Bool enable);
 
 /* exa.c */
 void
+exaDrawableUseScreen(DrawablePtr pDrawable);
+
+void
+exaDrawableUseMemory(DrawablePtr pDrawable);
+
+void
 exaPixmapUseScreen (PixmapPtr pPixmap);
 
 void
@@ -299,4 +306,15 @@ exaComposite(CARD8	op,
 	     CARD16	width,
 	     CARD16	height);
 
+void
+exaGlyphs (CARD8	op,
+	  PicturePtr	pSrc,
+	  PicturePtr	pDst,
+	  PictFormatPtr	maskFormat,
+	  INT16		xSrc,
+	  INT16		ySrc,
+	  int		nlist,
+	  GlyphListPtr	list,
+	  GlyphPtr	*glyphs);
+
 #endif /* EXAPRIV_H */
