diff --git a/hw/xfree86/ramdac/TI.c b/hw/xfree86/ramdac/TI.c
index f421fc3..1c541d7 100644
--- a/hw/xfree86/ramdac/TI.c
+++ b/hw/xfree86/ramdac/TI.c
@@ -115,10 +115,8 @@ TIramdacCalculateMNPForClock(
     VCO = 8.0 * IntRef * best_m / best_n;
     ActualClock = VCO / ( 1 << p );
 
-#ifdef DEBUG
-    ErrorF( "f_out=%ld f_vco=%.1f n=%d m=%d p=%d\n",
+    DebugF( "f_out=%ld f_vco=%.1f n=%d m=%d p=%d\n",
 	    ActualClock, VCO, *rN, *rM, *rP);
-#endif
 
     return (ActualClock);
 }
