diff --git a/xkb/xkbActions.c b/xkb/xkbActions.c
index 3f0b766..afc44ac 100644
--- a/xkb/xkbActions.c
+++ b/xkb/xkbActions.c
@@ -1032,7 +1032,7 @@ register int	i;
     xkbi->filters= realloc(xkbi->filters,
                             xkbi->szFilters * sizeof(XkbFilterRec));
     /* 6/21/93 (ef) -- XXX! deal with allocation failure */
-    bzero(&xkbi->filters[xkbi->szFilters/2],
+    memset(&xkbi->filters[xkbi->szFilters/2], 0,
             (xkbi->szFilters/2)*sizeof(XkbFilterRec));
     return &xkbi->filters[xkbi->szFilters/2];
 }
