diff --git a/GL/apple/indirect.c b/GL/apple/indirect.c
index ba54186..2729ab0 100644
--- a/GL/apple/indirect.c
+++ b/GL/apple/indirect.c
@@ -36,15 +36,30 @@
 #endif
 
 #include "dri.h"
-#include "quartz.h"
 
-//#include <CoreGraphics/CoreGraphics.h>
+#define GL_EXT_histogram 1
+#define GL_EXT_polygon_offset 1
+#define GL_SGIS_pixel_texture 1
+#define GL_SGIX_pixel_texture 1
+#define GL_EXT_multisample 1
+#define GL_SGIS_multisample 1
+#define GL_EXT_vertex_array 1
+#define GL_ARB_point_parameters 1
+#define GL_NV_vertex_array_range 1
+#define GL_MESA_resize_buffers 1
+#define GL_ARB_window_pos 1
+#define GL_EXT_cull_vertex 1
+#define GL_NV_vertex_program 1
+#define GL_APPLE_fence 1
+#define GL_IBM_multimode_draw_arrays 1
+#define GL_EXT_fragment_shader 1
+
 #include <OpenGL/OpenGL.h>
 #include <OpenGL/CGLContext.h>
 
 // X11 and X11's glx
 #include <GL/gl.h>
-#include <miscstruct.h>
+#include <GL/glxproto.h>
 #include <windowstr.h>
 #include <resource.h>
 #include <GL/glxint.h>
@@ -62,15 +77,17 @@
 #include "x-list.h"
 
 #include <dispatch.h>
-
+#define GLAPIENTRYP *
+typedef unsigned long long GLuint64EXT;
+typedef long long GLint64EXT;
+#include <Xplugin.h>
 #include "glcontextmodes.h"
 #include <glapi.h>
 #include <glapitable.h>
 
 // ggs: needed to call back to glx with visual configs
 extern void GlxSetVisualConfigs(int nconfigs, __GLXvisualConfig *configs, void **configprivs);
-
-#define GLAQUA_DEBUG 1
+__GLXprovider * GlxGetMesaProvider (void);
 
 // Write debugging output, or not
 #ifdef GLAQUA_DEBUG
@@ -80,9 +97,13 @@ extern void GlxSetVisualConfigs(int nconfigs, __GLXvisualConfig *configs, void *
 #endif
 
 static void setup_dispatch_table(void);
+GLuint __glFloorLog2(GLuint val);
+void warn_func(void * p1, char *format, ...);
 
 // some prototypes
 static __GLXscreen * __glXAquaScreenProbe(ScreenPtr pScreen);
+static __GLXdrawable * __glXAquaScreenCreateDrawable(__GLXscreen *screen, DrawablePtr pDraw, XID drawId, __GLcontextModes *modes);
+
 static Bool glAquaInitVisuals(VisualPtr *visualp, DepthPtr *depthp,
                               int *nvisualp, int *ndepthp,
                               int *rootDepthp, VisualID *defaultVisp,
@@ -90,16 +111,12 @@ static Bool glAquaInitVisuals(VisualPtr *visualp, DepthPtr *depthp,
 static void glAquaSetVisualConfigs(int nconfigs, __GLXvisualConfig *configs,
                                    void **privates);
 
-static __GLinterface *glAquaCreateContext(__GLimports *imports,
-                                          __GLcontextModes *mode,
-                                          __GLinterface *shareGC);
 static void glAquaResetExtension(void);
 static void __glXAquaContextDestroy(__GLXcontext *baseContext);
 static int __glXAquaContextMakeCurrent(__GLXcontext *baseContext);
 static int __glXAquaContextLoseCurrent(__GLXcontext *baseContext);
 static int __glXAquaContextForceCurrent(__GLXcontext *baseContext);
 static int __glXAquaContextCopy(__GLXcontext *baseDst, __GLXcontext *baseSrc, unsigned long mask);
-static __GLXdrawable * __glXAquaContextCreateDrawable(__GLXcontext *context, DrawablePtr pDraw, XID drawId);
 
 static CGLPixelFormatObj makeFormat(__GLcontextModes *mode);
 
@@ -112,7 +129,7 @@ __GLXprovider __glXMesaProvider = {
 __GLXprovider *
 GlxGetMesaProvider (void)
 {
-  ErrorF("GlxGetMesaProvider\n");
+  GLAQUA_DEBUG_MSG("GlxGetMesaProvider\n");
   return &__glXMesaProvider;
 }
 
@@ -167,7 +184,7 @@ __glXAquaScreenCreateContext(__GLXscreen *screen,
   context->base.loseCurrent    = __glXAquaContextLoseCurrent;
   context->base.copy           = __glXAquaContextCopy;
   context->base.forceCurrent   = __glXAquaContextForceCurrent;
-  context->base.createDrawable = __glXAquaContextCreateDrawable;
+  //  context->base.createDrawable = __glXAquaContextCreateDrawable;
 
   context->pixelFormat = makeFormat(modes);
   if (!context->pixelFormat) {
@@ -259,7 +276,7 @@ static void surface_notify(void *_arg, void *data) {
     case AppleDRISurfaceNotifyDestroyed:
         if (surface_hash != NULL)
             x_hash_table_remove(surface_hash, (void *) arg->id);
-	        draw->base->pDraw = NULL;
+	        draw->base.pDraw = NULL;
 			draw->sid = 0;
         break;
 
@@ -324,6 +341,7 @@ static void attach(__GLXAquaContext *context, __GLXAquaDrawable *draw) {
     } 
 }
 
+#if 0     // unused
 static void unattach(__GLXAquaContext *context) {
 	x_list *lst;
 	GLAQUA_DEBUG_MSG("unattach\n");
@@ -345,12 +363,12 @@ static void unattach(__GLXAquaContext *context) {
         context->sid = 0;
     }
 }
+#endif
 
 static int __glXAquaContextMakeCurrent(__GLXcontext *baseContext) {
     CGLError gl_err;
     __GLXAquaContext *context = (__GLXAquaContext *) baseContext;
 	__GLXAquaDrawable *drawPriv = (__GLXAquaDrawable *) context->base.drawPriv;
-    __GLXAquaDrawable *readPriv = (__GLXAquaDrawable *) context->base.readPriv;
 
     GLAQUA_DEBUG_MSG("glAquaMakeCurrent (ctx 0x%p)\n", baseContext);
     
@@ -363,12 +381,6 @@ static int __glXAquaContextMakeCurrent(__GLXcontext *baseContext) {
     return gl_err == 0;
 }
 
-static GLboolean glAquaShareContext(__GLcontext *gc, __GLcontext *gcShare)
-{
-  GLAQUA_DEBUG_MSG("glAquaShareContext unimplemented\n");
-  return GL_TRUE;
-}
-
 static int __glXAquaContextCopy(__GLXcontext *baseDst, __GLXcontext *baseSrc, unsigned long mask)
 {
     CGLError gl_err;
@@ -405,12 +417,7 @@ static GLboolean __glXAquaDrawableResize(__GLXdrawable *base)  {
     return GL_TRUE;
 }
 
-static void glAquaNotifyDestroy(__GLcontext *gc) {
-    GLAQUA_DEBUG_MSG("unimplemented glAquaNotifyDestroy");
-}
-
 static GLboolean __glXAquaDrawableSwapBuffers(__GLXdrawable *base) {
-    __GLXAquaDrawable *glxPriv = (__GLXAquaDrawable *) base;
     CGLError gl_err;
 	__GLXAquaContext * drawableCtx;
 //    GLAQUA_DEBUG_MSG("glAquaDrawableSwapBuffers(%p)\n",base);
@@ -867,7 +874,7 @@ static Bool init_visuals(int *nvisualp, VisualPtr *visualp,
     glAquaScreens[screenInfo.numScreens-1].num_vis = numNewVisuals;
     //    glAquaScreens[screenInfo.numScreens-1].priv = glXVisualPriv;
 
-    /* Set up depth's VisualIDs */
+    /* set up depth's VisualIDs */
     for (i = 0; i < ndepth; i++) {
         int numVids = 0;
         VisualID *pVids = NULL;
@@ -1027,7 +1034,7 @@ static Bool glAquaInitVisuals(VisualPtr *visualp, DepthPtr *depthp,
         glAquaInitVisualConfigs(); /* ensure the visual configs are setup */
 
     /*
-     * Setup the visuals supported by this particular screen.
+     * setup the visuals supported by this particular screen.
      */
     return init_visuals(nvisualp, visualp, defaultVisp,
                         *ndepthp, *depthp, *rootDepthp);
@@ -1109,7 +1116,7 @@ static void init_screen_visuals(__GLXAquaScreen *screen) {
                 pVis[j].blueMask  == modes->blueMask &&
                 !used[j]) {
 
-                    /* Set the VisualID */
+                    /* set the VisualID */
                     modes->visualID = pVis[j].vid;
 
                     /* Mark this visual used */
@@ -1145,6 +1152,7 @@ static __GLXscreen * __glXAquaScreenProbe(ScreenPtr pScreen) {
 
   screen->base.destroy       = __glXAquaScreenDestroy;
   screen->base.createContext = __glXAquaScreenCreateContext;
+  screen->base.createDrawable = __glXAquaScreenCreateDrawable;
   screen->base.pScreen       = pScreen;
 
   init_screen_visuals(screen);
@@ -1163,20 +1171,20 @@ static void __glXAquaDrawableDestroy(__GLXdrawable *base) {
 }
 
 static __GLXdrawable *
-__glXAquaContextCreateDrawable(__GLXcontext *context,
+__glXAquaScreenCreateDrawable(__GLXscreen *screen,
 			       DrawablePtr pDraw,
-			       XID drawId)
-{
+			      XID drawId,
+			      __GLcontextModes *modes) {
   __GLXAquaDrawable *glxPriv;
-  __GLXscreen *pGlxScreen;
 
-  GLAQUA_DEBUG_MSG("glAquaContextCreateDrawable(%p,%p,%d)\n", context, pDraw, drawId);
-  if (glxPriv == NULL) return NULL;
+  GLAQUA_DEBUG_MSG("glAquaScreenCreateDrawable(%p,%p,%d,%p)\n", context, pDraw, drawId, modes);
+
   glxPriv = xalloc(sizeof *glxPriv);
+  if (glxPriv == NULL) return NULL;
 
   memset(glxPriv, 0, sizeof *glxPriv);
 
-  if (!__glXDrawableInit(&glxPriv->base, context, pDraw, drawId)) {
+  if (!__glXDrawableInit(&glxPriv->base, screen, pDraw, drawId, modes)) {
     xfree(glxPriv);
     return NULL;
   }
@@ -1184,15 +1192,7 @@ __glXAquaContextCreateDrawable(__GLXcontext *context,
   glxPriv->base.destroy       = __glXAquaDrawableDestroy;
   glxPriv->base.resize        = __glXAquaDrawableResize;
   glxPriv->base.swapBuffers   = __glXAquaDrawableSwapBuffers;
-
-  pGlxScreen = __glXActiveScreens[pDraw->pScreen->myNum];
-
-  if (glxPriv->base.type == DRAWABLE_WINDOW) {
-    VisualID vid = wVisual((WindowPtr)pDraw);
-
-    glxPriv->base.modes = _gl_context_modes_find_visual(pGlxScreen->modes, vid);
-  } else 
-    glxPriv->base.modes = glxPriv->base.pGlxPixmap->modes;
+  //  glxPriv->base.copySubBuffer = __glXAquaDrawableCopySubBuffer;
 
     return &glxPriv->base;
 }
@@ -1216,18 +1216,59 @@ GLuint __glFloorLog2(GLuint val)
     return c;
 }
 
+void warn_func(void * p1, char *format, ...) {
+  va_list v;
+  va_start(v, format);
+  vfprintf(stderr, format, v);
+  va_end(v);
+}
+
 static void setup_dispatch_table(void) {
 	struct _glapi_table *disp=_glapi_get_dispatch();
+  _glapi_set_warning_func((_glapi_warning_func)warn_func);
+  _glapi_noop_enable_warnings(TRUE);
 
- SET_NewList(disp, glNewList);
- SET_EndList(disp, glEndList);
- SET_CallList(disp, glCallList);
- SET_CallLists(disp, glCallLists);
- SET_DeleteLists(disp, glDeleteLists);
- SET_GenLists(disp, glGenLists);
- SET_ListBase(disp, glListBase);
+  SET_Accum(disp, glAccum);
+  SET_ActiveStencilFaceEXT(disp, glActiveStencilFaceEXT);
+  SET_ActiveTextureARB(disp, glActiveTextureARB);
+//SET_AlphaFragmentOp1ATI(disp, glAlphaFragmentOp1EXT);   // <-- EXT -> ATI
+//SET_AlphaFragmentOp2ATI(disp, glAlphaFragmentOp2EXT);
+//SET_AlphaFragmentOp3ATI(disp, glAlphaFragmentOp3EXT);
+  SET_AlphaFunc(disp, glAlphaFunc);
+//SET_AreProgramsResidentNV(disp, glAreProgramsResidentNV);
+  SET_AreTexturesResident(disp, glAreTexturesResident); 
+  SET_ArrayElement(disp, glArrayElement);
+  SET_AttachObjectARB(disp, glAttachObjectARB);
  SET_Begin(disp, glBegin);
+//SET_BeginFragmentShaderATI(disp, glBeginFragmentShaderEXT);   // <-- EXT -> ATI
+  SET_BeginQueryARB(disp, glBeginQueryARB);
+  SET_BindAttribLocationARB(disp, glBindAttribLocationARB);
+  SET_BindBufferARB(disp, glBindBufferARB);
+//SET_BindFragmentShaderATI(disp, glBindFragmentShaderEXT);     // <-- EXT -> ATI
+  SET_BindFramebufferEXT(disp, glBindFramebufferEXT);
+//SET_BindProgramNV(disp, glBindProgramNV);
+  SET_BindRenderbufferEXT(disp, glBindRenderbufferEXT);
+  SET_BindTexture(disp, glBindTexture);
  SET_Bitmap(disp, glBitmap);
+  SET_BlendColor(disp, glBlendColor);
+  SET_BlendEquation(disp, glBlendEquation);
+  SET_BlendEquationSeparateEXT(disp, glBlendEquationSeparateEXT);
+  SET_BlendFunc(disp, glBlendFunc);
+  SET_BlendFuncSeparateEXT(disp, glBlendFuncSeparateEXT);
+  SET_BlitFramebufferEXT(disp, glBlitFramebufferEXT);
+  SET_BufferDataARB(disp, glBufferDataARB);
+  SET_BufferSubDataARB(disp, glBufferSubDataARB);
+  SET_CallList(disp, glCallList);
+  SET_CallLists(disp, glCallLists);
+  SET_CheckFramebufferStatusEXT(disp, glCheckFramebufferStatusEXT);
+  SET_Clear(disp, glClear);
+  SET_ClearAccum(disp, glClearAccum);
+  SET_ClearColor(disp, glClearColor);
+  SET_ClearDepth(disp, glClearDepth);
+  SET_ClearIndex(disp, glClearIndex);
+  SET_ClearStencil(disp, glClearStencil);
+  SET_ClientActiveTextureARB(disp, glClientActiveTextureARB);
+  SET_ClipPlane(disp, glClipPlane);
  SET_Color3b(disp, glColor3b);
  SET_Color3bv(disp, glColor3bv);
  SET_Color3d(disp, glColor3d);
@@ -1260,9 +1301,229 @@ static void setup_dispatch_table(void) {
  SET_Color4uiv(disp, glColor4uiv);
  SET_Color4us(disp, glColor4us);
  SET_Color4usv(disp, glColor4usv);
+//SET_ColorFragmentOp1ATI(disp, glColorFragmentOp1EXT);    // <-- EXT -> ATI
+//SET_ColorFragmentOp2ATI(disp, glColorFragmentOp2EXT);
+//SET_ColorFragmentOp3ATI(disp, glColorFragmentOp3EXT);
+  SET_ColorMask(disp, glColorMask);
+  SET_ColorMaterial(disp, glColorMaterial);
+  SET_ColorPointer(disp, glColorPointer);
+//SET_ColorPointerEXT(disp, glColorPointerEXT);
+  SET_ColorSubTable(disp, glColorSubTable);
+  SET_ColorTable(disp, glColorTable);
+  SET_ColorTableParameterfv(disp, glColorTableParameterfv);
+  SET_ColorTableParameteriv(disp, glColorTableParameteriv);
+  SET_CombinerInputNV(disp, glCombinerInputNV);
+  SET_CombinerOutputNV(disp, glCombinerOutputNV);
+  SET_CombinerParameterfNV(disp, glCombinerParameterfNV);
+  SET_CombinerParameterfvNV(disp, glCombinerParameterfvNV);
+  SET_CombinerParameteriNV(disp, glCombinerParameteriNV);
+  SET_CombinerParameterivNV(disp, glCombinerParameterivNV);
+  SET_CompileShaderARB(disp, glCompileShaderARB);
+  SET_CompressedTexImage1DARB(disp, glCompressedTexImage1DARB);
+  SET_CompressedTexImage2DARB(disp, glCompressedTexImage2DARB);
+  SET_CompressedTexImage3DARB(disp, glCompressedTexImage3DARB);
+  SET_CompressedTexSubImage1DARB(disp, glCompressedTexSubImage1DARB);
+  SET_CompressedTexSubImage2DARB(disp, glCompressedTexSubImage2DARB);
+  SET_CompressedTexSubImage3DARB(disp, glCompressedTexSubImage3DARB);
+  SET_ConvolutionFilter1D(disp, glConvolutionFilter1D);
+  SET_ConvolutionFilter2D(disp, glConvolutionFilter2D);
+  SET_ConvolutionParameterf(disp, glConvolutionParameterf);
+  SET_ConvolutionParameterfv(disp, glConvolutionParameterfv);
+  SET_ConvolutionParameteri(disp, glConvolutionParameteri);
+  SET_ConvolutionParameteriv(disp, glConvolutionParameteriv);
+  SET_CopyColorSubTable(disp, glCopyColorSubTable);
+  SET_CopyColorTable(disp, glCopyColorTable);
+  SET_CopyConvolutionFilter1D(disp, glCopyConvolutionFilter1D);
+  SET_CopyConvolutionFilter2D(disp, glCopyConvolutionFilter2D);
+  SET_CopyPixels(disp, glCopyPixels);
+  SET_CopyTexImage1D(disp, glCopyTexImage1D);
+  SET_CopyTexImage2D(disp, glCopyTexImage2D);
+  SET_CopyTexSubImage1D(disp, glCopyTexSubImage1D);
+  SET_CopyTexSubImage2D(disp, glCopyTexSubImage2D);
+  SET_CopyTexSubImage3D(disp, glCopyTexSubImage3D);
+  SET_CreateProgramObjectARB(disp, glCreateProgramObjectARB);
+  SET_CreateShaderObjectARB(disp, glCreateShaderObjectARB);
+  SET_CullFace(disp, glCullFace);
+//SET_CullParameterdvEXT(disp, glCullParameterdvEXT);
+//SET_CullParameterfvEXT(disp, glCullParameterfvEXT);
+  SET_DeleteBuffersARB(disp, glDeleteBuffersARB);
+  SET_DeleteFencesNV(disp, glDeleteFencesAPPLE);
+//SET_DeleteFragmentShaderATI(disp, glDeleteFragmentShaderEXT);      // <-- EXT -> ATI
+  SET_DeleteFramebuffersEXT(disp, glDeleteFramebuffersEXT);
+  SET_DeleteLists(disp, glDeleteLists);
+  SET_DeleteObjectARB(disp, glDeleteObjectARB);
+//SET_DeleteProgramsNV(disp, glDeleteProgramsNV);
+  SET_DeleteQueriesARB(disp, glDeleteQueriesARB);
+  SET_DeleteRenderbuffersEXT(disp, glDeleteRenderbuffersEXT);
+  SET_DeleteTextures(disp, glDeleteTextures);
+  SET_DepthBoundsEXT(disp, glDepthBoundsEXT);
+  SET_DepthFunc(disp, glDepthFunc);
+  SET_DepthMask(disp, glDepthMask);
+  SET_DepthRange(disp, glDepthRange);
+  SET_DetachObjectARB(disp, glDetachObjectARB);
+  SET_Disable(disp, glDisable);
+  SET_DisableClientState(disp, glDisableClientState);
+  SET_DisableVertexAttribArrayARB(disp, glDisableVertexAttribArrayARB);
+  SET_DrawArrays(disp, glDrawArrays);
+  SET_DrawBuffer(disp, glDrawBuffer);
+  SET_DrawBuffersARB(disp, glDrawBuffersARB);
+  SET_DrawElements(disp, glDrawElements);
+  SET_DrawPixels(disp, glDrawPixels);
+  SET_DrawRangeElements(disp, glDrawRangeElements);
  SET_EdgeFlag(disp, glEdgeFlag);
+  SET_EdgeFlagPointer(disp, glEdgeFlagPointer);
+//SET_EdgeFlagPointerEXT(disp, glEdgeFlagPointerEXT);
  SET_EdgeFlagv(disp, glEdgeFlagv);
+  SET_Enable(disp, glEnable);
+  SET_EnableClientState(disp, glEnableClientState);
+  SET_EnableVertexAttribArrayARB(disp, glEnableVertexAttribArrayARB);
  SET_End(disp, glEnd);
+//SET_EndFragmentShaderATI(disp, glEndFragmentShaderEXT);        // <-- EXT -> ATI
+  SET_EndList(disp, glEndList);
+  SET_EndQueryARB(disp, glEndQueryARB);
+  SET_EvalCoord1d(disp, glEvalCoord1d);
+  SET_EvalCoord1dv(disp, glEvalCoord1dv);
+  SET_EvalCoord1f(disp, glEvalCoord1f);
+  SET_EvalCoord1fv(disp, glEvalCoord1fv);
+  SET_EvalCoord2d(disp, glEvalCoord2d);
+  SET_EvalCoord2dv(disp, glEvalCoord2dv);
+  SET_EvalCoord2f(disp, glEvalCoord2f);
+  SET_EvalCoord2fv(disp, glEvalCoord2fv);
+  SET_EvalMesh1(disp, glEvalMesh1);
+  SET_EvalMesh2(disp, glEvalMesh2);
+  SET_EvalPoint1(disp, glEvalPoint1);
+  SET_EvalPoint2(disp, glEvalPoint2);
+//SET_ExecuteProgramNV(disp, glExecuteProgramNV);
+  SET_FeedbackBuffer(disp, glFeedbackBuffer);
+  SET_FinalCombinerInputNV(disp, glFinalCombinerInputNV);
+  SET_Finish(disp, glFinish);
+  SET_FinishFenceNV(disp, glFinishFenceAPPLE);       // <-- APPLE -> NV
+  SET_Flush(disp, glFlush);
+//SET_FlushVertexArrayRangeNV(disp, glFlushVertexArrayRangeNV);
+  SET_FogCoordPointerEXT(disp, glFogCoordPointerEXT);
+  SET_FogCoorddEXT(disp, glFogCoorddEXT);
+  SET_FogCoorddvEXT(disp, glFogCoorddvEXT);
+  SET_FogCoordfEXT(disp, glFogCoordfEXT);
+  SET_FogCoordfvEXT(disp, glFogCoordfvEXT);
+  SET_Fogf(disp, glFogf);
+  SET_Fogfv(disp, glFogfv);
+  SET_Fogi(disp, glFogi);
+  SET_Fogiv(disp, glFogiv);
+  SET_FramebufferRenderbufferEXT(disp, glFramebufferRenderbufferEXT);
+  SET_FramebufferTexture1DEXT(disp, glFramebufferTexture1DEXT);
+  SET_FramebufferTexture2DEXT(disp, glFramebufferTexture2DEXT);
+  SET_FramebufferTexture3DEXT(disp, glFramebufferTexture3DEXT);
+  SET_FrontFace(disp, glFrontFace);
+  SET_Frustum(disp, glFrustum);
+  SET_GenBuffersARB(disp, glGenBuffersARB);
+  SET_GenFencesNV(disp, glGenFencesAPPLE);            // <-- APPLE -> NV
+//SET_GenFragmentShadersATI(disp, glGenFragmentShadersEXT);         // <-- EXT -> ATI
+  SET_GenFramebuffersEXT(disp, glGenFramebuffersEXT);
+  SET_GenLists(disp, glGenLists);
+//SET_GenProgramsNV(disp, glGenProgramsNV);
+  SET_GenQueriesARB(disp, glGenQueriesARB);
+  SET_GenRenderbuffersEXT(disp, glGenRenderbuffersEXT);
+  SET_GenTextures(disp, glGenTextures);
+  SET_GenerateMipmapEXT(disp, glGenerateMipmapEXT);
+  SET_GetActiveAttribARB(disp, glGetActiveAttribARB);
+  SET_GetActiveUniformARB(disp, glGetActiveUniformARB);
+  SET_GetAttachedObjectsARB(disp, glGetAttachedObjectsARB);
+  SET_GetAttribLocationARB(disp, glGetAttribLocationARB);
+  SET_GetBooleanv(disp, glGetBooleanv);
+  SET_GetBufferParameterivARB(disp, glGetBufferParameterivARB);
+  SET_GetBufferPointervARB(disp, glGetBufferPointervARB);
+  SET_GetBufferSubDataARB(disp, glGetBufferSubDataARB);
+  SET_GetClipPlane(disp, glGetClipPlane);
+  SET_GetColorTable(disp, glGetColorTable);
+  SET_GetColorTableParameterfv(disp, glGetColorTableParameterfv);
+  SET_GetColorTableParameteriv(disp, glGetColorTableParameteriv);
+  SET_GetCombinerInputParameterfvNV(disp, glGetCombinerInputParameterfvNV);
+  SET_GetCombinerInputParameterivNV(disp, glGetCombinerInputParameterivNV);
+  SET_GetCombinerOutputParameterfvNV(disp, glGetCombinerOutputParameterfvNV);
+  SET_GetCombinerOutputParameterivNV(disp, glGetCombinerOutputParameterivNV);
+  SET_GetCompressedTexImageARB(disp, glGetCompressedTexImageARB);
+  SET_GetConvolutionFilter(disp, glGetConvolutionFilter);
+  SET_GetConvolutionParameterfv(disp, glGetConvolutionParameterfv);
+  SET_GetConvolutionParameteriv(disp, glGetConvolutionParameteriv);
+  SET_GetDoublev(disp, glGetDoublev);
+  SET_GetError(disp, glGetError);
+//SET_GetFenceivNV(disp, glGetFenceivNV);
+  SET_GetFinalCombinerInputParameterfvNV(disp, glGetFinalCombinerInputParameterfvNV);
+  SET_GetFinalCombinerInputParameterivNV(disp, glGetFinalCombinerInputParameterivNV);
+  SET_GetFloatv(disp, glGetFloatv);
+  SET_GetFramebufferAttachmentParameterivEXT(disp, glGetFramebufferAttachmentParameterivEXT);
+  SET_GetHandleARB(disp, glGetHandleARB);
+  SET_GetHistogram(disp, glGetHistogram);
+  SET_GetHistogramParameterfv(disp, glGetHistogramParameterfv);
+  SET_GetHistogramParameteriv(disp, glGetHistogramParameteriv);
+  SET_GetInfoLogARB(disp, glGetInfoLogARB);
+  SET_GetIntegerv(disp, glGetIntegerv);
+  SET_GetLightfv(disp, glGetLightfv);
+  SET_GetLightiv(disp, glGetLightiv);
+  SET_GetMapdv(disp, glGetMapdv);
+  SET_GetMapfv(disp, glGetMapfv);
+  SET_GetMapiv(disp, glGetMapiv);
+  SET_GetMaterialfv(disp, glGetMaterialfv);
+  SET_GetMaterialiv(disp, glGetMaterialiv);
+  SET_GetMinmax(disp, glGetMinmax);
+  SET_GetMinmaxParameterfv(disp, glGetMinmaxParameterfv);
+  SET_GetMinmaxParameteriv(disp, glGetMinmaxParameteriv);
+  SET_GetObjectParameterfvARB(disp, glGetObjectParameterfvARB);
+  SET_GetObjectParameterivARB(disp, glGetObjectParameterivARB);
+  SET_GetPixelMapfv(disp, glGetPixelMapfv);
+  SET_GetPixelMapuiv(disp, glGetPixelMapuiv);
+  SET_GetPixelMapusv(disp, glGetPixelMapusv);
+//SET_GetPixelTexGenParameterfvSGIS(disp, glGetPixelTexGenParameterfvSGIS);
+//SET_GetPixelTexGenParameterivSGIS(disp, glGetPixelTexGenParameterivSGIS);
+  SET_GetPointerv(disp, glGetPointerv);
+  SET_GetPolygonStipple(disp, glGetPolygonStipple);
+  SET_GetProgramEnvParameterdvARB(disp, glGetProgramEnvParameterdvARB);
+  SET_GetProgramEnvParameterfvARB(disp, glGetProgramEnvParameterfvARB);
+  SET_GetProgramLocalParameterdvARB(disp, glGetProgramLocalParameterdvARB);
+  SET_GetProgramLocalParameterfvARB(disp, glGetProgramLocalParameterfvARB);
+//SET_GetProgramNamedParameterdvNV(disp, glGetProgramNamedParameterdvNV);
+//SET_GetProgramNamedParameterfvNV(disp, glGetProgramNamedParameterfvNV);
+//SET_GetProgramParameterdvNV(disp, glGetProgramParameterdvNV);
+//SET_GetProgramParameterfvNV(disp, glGetProgramParameterfvNV);
+  SET_GetProgramStringARB(disp, glGetProgramStringARB);
+//SET_GetProgramStringNV(disp, glGetProgramStringNV);
+  SET_GetProgramivARB(disp, glGetProgramivARB);
+//SET_GetProgramivNV(disp, glGetProgramivNV);
+//SET_GetQueryObjecti64vEXT(disp, glGetQueryObjecti64vEXT);
+  SET_GetQueryObjectivARB(disp, glGetQueryObjectivARB);
+//SET_GetQueryObjectui64vEXT(disp, glGetQueryObjectui64vEXT);
+  SET_GetQueryObjectuivARB(disp, glGetQueryObjectuivARB);
+  SET_GetQueryivARB(disp, glGetQueryivARB);
+  SET_GetRenderbufferParameterivEXT(disp, glGetRenderbufferParameterivEXT);
+  SET_GetSeparableFilter(disp, glGetSeparableFilter);
+  SET_GetShaderSourceARB(disp, glGetShaderSourceARB);
+  SET_GetString(disp, glGetString);
+  SET_GetTexEnvfv(disp, glGetTexEnvfv);
+  SET_GetTexEnviv(disp, glGetTexEnviv);
+  SET_GetTexGendv(disp, glGetTexGendv);
+  SET_GetTexGenfv(disp, glGetTexGenfv);
+  SET_GetTexGeniv(disp, glGetTexGeniv);
+  SET_GetTexImage(disp, glGetTexImage);
+  SET_GetTexLevelParameterfv(disp, glGetTexLevelParameterfv);
+  SET_GetTexLevelParameteriv(disp, glGetTexLevelParameteriv);
+  SET_GetTexParameterfv(disp, glGetTexParameterfv);
+  SET_GetTexParameteriv(disp, glGetTexParameteriv);
+//SET_GetTrackMatrixivNV(disp, glGetTrackMatrixivNV);
+  SET_GetUniformLocationARB(disp, glGetUniformLocationARB);
+  SET_GetUniformfvARB(disp, glGetUniformfvARB);
+  SET_GetUniformivARB(disp, glGetUniformivARB);
+//SET_GetVertexAttribPointervNV(disp, glGetVertexAttribPointervNV);
+  SET_GetVertexAttribdvARB(disp, glGetVertexAttribdvARB);
+//SET_GetVertexAttribdvNV(disp, glGetVertexAttribdvNV);
+  SET_GetVertexAttribfvARB(disp, glGetVertexAttribfvARB);
+//SET_GetVertexAttribfvNV(disp, glGetVertexAttribfvNV);
+  SET_GetVertexAttribivARB(disp, glGetVertexAttribivARB);
+//SET_GetVertexAttribivNV(disp, glGetVertexAttribivNV);
+  SET_Hint(disp, glHint);
+  SET_Histogram(disp, glHistogram);
+  SET_IndexMask(disp, glIndexMask);
+  SET_IndexPointer(disp, glIndexPointer);
+//SET_IndexPointerEXT(disp, glIndexPointerEXT);
  SET_Indexd(disp, glIndexd);
  SET_Indexdv(disp, glIndexdv);
  SET_Indexf(disp, glIndexf);
@@ -1271,6 +1532,96 @@ static void setup_dispatch_table(void) {
  SET_Indexiv(disp, glIndexiv);
  SET_Indexs(disp, glIndexs);
  SET_Indexsv(disp, glIndexsv);
+  SET_Indexub(disp, glIndexub);
+  SET_Indexubv(disp, glIndexubv);
+  SET_InitNames(disp, glInitNames);
+  SET_InterleavedArrays(disp, glInterleavedArrays);
+  SET_IsBufferARB(disp, glIsBufferARB);
+  SET_IsEnabled(disp, glIsEnabled);
+  SET_IsFenceNV(disp, glIsFenceAPPLE);           // <-- APPLE -> NV
+  SET_IsFramebufferEXT(disp, glIsFramebufferEXT);
+  SET_IsList(disp, glIsList);
+//SET_IsProgramNV(disp, glIsProgramNV);
+  SET_IsQueryARB(disp, glIsQueryARB);
+  SET_IsRenderbufferEXT(disp, glIsRenderbufferEXT);
+  SET_IsTexture(disp, glIsTexture);
+  SET_LightModelf(disp, glLightModelf);
+  SET_LightModelfv(disp, glLightModelfv);
+  SET_LightModeli(disp, glLightModeli);
+  SET_LightModeliv(disp, glLightModeliv);
+  SET_Lightf(disp, glLightf);
+  SET_Lightfv(disp, glLightfv);
+  SET_Lighti(disp, glLighti);
+  SET_Lightiv(disp, glLightiv);
+  SET_LineStipple(disp, glLineStipple);
+  SET_LineWidth(disp, glLineWidth);
+  SET_LinkProgramARB(disp, glLinkProgramARB);
+  SET_ListBase(disp, glListBase);
+  SET_LoadIdentity(disp, glLoadIdentity);
+  SET_LoadMatrixd(disp, glLoadMatrixd);
+  SET_LoadMatrixf(disp, glLoadMatrixf);
+  SET_LoadName(disp, glLoadName);
+//SET_LoadProgramNV(disp, glLoadProgramNV);
+  SET_LoadTransposeMatrixdARB(disp, glLoadTransposeMatrixdARB);
+  SET_LoadTransposeMatrixfARB(disp, glLoadTransposeMatrixfARB);
+  SET_LockArraysEXT(disp, glLockArraysEXT);
+  SET_LogicOp(disp, glLogicOp);
+  SET_Map1d(disp, glMap1d);
+  SET_Map1f(disp, glMap1f);
+  SET_Map2d(disp, glMap2d);
+  SET_Map2f(disp, glMap2f);
+  SET_MapBufferARB(disp, glMapBufferARB);
+  SET_MapGrid1d(disp, glMapGrid1d);
+  SET_MapGrid1f(disp, glMapGrid1f);
+  SET_MapGrid2d(disp, glMapGrid2d);
+  SET_MapGrid2f(disp, glMapGrid2f);
+  SET_Materialf(disp, glMaterialf);
+  SET_Materialfv(disp, glMaterialfv);
+  SET_Materiali(disp, glMateriali);
+  SET_Materialiv(disp, glMaterialiv);
+  SET_MatrixMode(disp, glMatrixMode);
+  SET_Minmax(disp, glMinmax);
+  SET_MultMatrixd(disp, glMultMatrixd);
+  SET_MultMatrixf(disp, glMultMatrixf);
+  SET_MultTransposeMatrixdARB(disp, glMultTransposeMatrixdARB);
+  SET_MultTransposeMatrixfARB(disp, glMultTransposeMatrixfARB);
+  SET_MultiDrawArraysEXT(disp, glMultiDrawArraysEXT);
+  SET_MultiDrawElementsEXT(disp, glMultiDrawElementsEXT);
+//SET_MultiModeDrawArraysIBM(disp, glMultiModeDrawArraysIBM);
+//SET_MultiModeDrawElementsIBM(disp, glMultiModeDrawElementsIBM);
+  SET_MultiTexCoord1dARB(disp, glMultiTexCoord1dARB);
+  SET_MultiTexCoord1dvARB(disp, glMultiTexCoord1dvARB);
+  SET_MultiTexCoord1fARB(disp, glMultiTexCoord1fARB);
+  SET_MultiTexCoord1fvARB(disp, glMultiTexCoord1fvARB);
+  SET_MultiTexCoord1iARB(disp, glMultiTexCoord1iARB);
+  SET_MultiTexCoord1ivARB(disp, glMultiTexCoord1ivARB);
+  SET_MultiTexCoord1sARB(disp, glMultiTexCoord1sARB);
+  SET_MultiTexCoord1svARB(disp, glMultiTexCoord1svARB);
+  SET_MultiTexCoord2dARB(disp, glMultiTexCoord2dARB);
+  SET_MultiTexCoord2dvARB(disp, glMultiTexCoord2dvARB);
+  SET_MultiTexCoord2fARB(disp, glMultiTexCoord2fARB);
+  SET_MultiTexCoord2fvARB(disp, glMultiTexCoord2fvARB);
+  SET_MultiTexCoord2iARB(disp, glMultiTexCoord2iARB);
+  SET_MultiTexCoord2ivARB(disp, glMultiTexCoord2ivARB);
+  SET_MultiTexCoord2sARB(disp, glMultiTexCoord2sARB);
+  SET_MultiTexCoord2svARB(disp, glMultiTexCoord2svARB);
+  SET_MultiTexCoord3dARB(disp, glMultiTexCoord3dARB);
+  SET_MultiTexCoord3dvARB(disp, glMultiTexCoord3dvARB);
+  SET_MultiTexCoord3fARB(disp, glMultiTexCoord3fARB);
+  SET_MultiTexCoord3fvARB(disp, glMultiTexCoord3fvARB);
+  SET_MultiTexCoord3iARB(disp, glMultiTexCoord3iARB);
+  SET_MultiTexCoord3ivARB(disp, glMultiTexCoord3ivARB);
+  SET_MultiTexCoord3sARB(disp, glMultiTexCoord3sARB);
+  SET_MultiTexCoord3svARB(disp, glMultiTexCoord3svARB);
+  SET_MultiTexCoord4dARB(disp, glMultiTexCoord4dARB);
+  SET_MultiTexCoord4dvARB(disp, glMultiTexCoord4dvARB);
+  SET_MultiTexCoord4fARB(disp, glMultiTexCoord4fARB);
+  SET_MultiTexCoord4fvARB(disp, glMultiTexCoord4fvARB);
+  SET_MultiTexCoord4iARB(disp, glMultiTexCoord4iARB);
+  SET_MultiTexCoord4ivARB(disp, glMultiTexCoord4ivARB);
+  SET_MultiTexCoord4sARB(disp, glMultiTexCoord4sARB);
+  SET_MultiTexCoord4svARB(disp, glMultiTexCoord4svARB);
+  SET_NewList(disp, glNewList);
  SET_Normal3b(disp, glNormal3b);
  SET_Normal3bv(disp, glNormal3bv);
  SET_Normal3d(disp, glNormal3d);
@@ -1281,6 +1632,61 @@ static void setup_dispatch_table(void) {
  SET_Normal3iv(disp, glNormal3iv);
  SET_Normal3s(disp, glNormal3s);
  SET_Normal3sv(disp, glNormal3sv);
+  SET_NormalPointer(disp, glNormalPointer);
+//SET_NormalPointerEXT(disp, glNormalPointerEXT);
+  SET_Ortho(disp, glOrtho);
+//SET_PassTexCoordATI(disp, glPassTexCoordEXT);         // <-- EXT -> ATI
+  SET_PassThrough(disp, glPassThrough);
+  SET_PixelMapfv(disp, glPixelMapfv);
+  SET_PixelMapuiv(disp, glPixelMapuiv);
+  SET_PixelMapusv(disp, glPixelMapusv);
+  SET_PixelStoref(disp, glPixelStoref);
+  SET_PixelStorei(disp, glPixelStorei);
+//SET_PixelTexGenParameterfSGIS(disp, glPixelTexGenParameterfSGIS);
+//SET_PixelTexGenParameterfvSGIS(disp, glPixelTexGenParameterfvSGIS);
+//SET_PixelTexGenParameteriSGIS(disp, glPixelTexGenParameteriSGIS);
+//SET_PixelTexGenParameterivSGIS(disp, glPixelTexGenParameterivSGIS);
+//  SET_PixelTexGenSGIX(disp, glPixelTexGenSGIX);
+  SET_PixelTransferf(disp, glPixelTransferf);
+  SET_PixelTransferi(disp, glPixelTransferi);
+  SET_PixelZoom(disp, glPixelZoom);
+  SET_PointParameterfEXT(disp, glPointParameterfARB);      // <-- ARB -> EXT
+  SET_PointParameterfvEXT(disp, glPointParameterfvARB);    // <-- ARB -> EXT
+  SET_PointParameteriNV(disp, glPointParameteriNV);
+  SET_PointParameterivNV(disp, glPointParameterivNV);
+  SET_PointSize(disp, glPointSize);
+  SET_PolygonMode(disp, glPolygonMode);
+  SET_PolygonOffset(disp, glPolygonOffset);
+//SET_PolygonOffsetEXT(disp, glPolygonOffsetEXT);
+  SET_PolygonStipple(disp, glPolygonStipple);
+  SET_PopAttrib(disp, glPopAttrib);
+  SET_PopClientAttrib(disp, glPopClientAttrib);
+  SET_PopMatrix(disp, glPopMatrix);
+  SET_PopName(disp, glPopName);
+  SET_PrioritizeTextures(disp, glPrioritizeTextures);
+  SET_ProgramEnvParameter4dARB(disp, glProgramEnvParameter4dARB);
+  SET_ProgramEnvParameter4dvARB(disp, glProgramEnvParameter4dvARB);
+  SET_ProgramEnvParameter4fARB(disp, glProgramEnvParameter4fARB);
+  SET_ProgramEnvParameter4fvARB(disp, glProgramEnvParameter4fvARB);
+  SET_ProgramLocalParameter4dARB(disp, glProgramLocalParameter4dARB);
+  SET_ProgramLocalParameter4dvARB(disp, glProgramLocalParameter4dvARB);
+  SET_ProgramLocalParameter4fARB(disp, glProgramLocalParameter4fARB);
+  SET_ProgramLocalParameter4fvARB(disp, glProgramLocalParameter4fvARB);
+//SET_ProgramNamedParameter4dNV(disp, glProgramNamedParameter4dNV);
+//SET_ProgramNamedParameter4dvNV(disp, glProgramNamedParameter4dvNV);
+//SET_ProgramNamedParameter4fNV(disp, glProgramNamedParameter4fNV);
+//SET_ProgramNamedParameter4fvNV(disp, glProgramNamedParameter4fvNV);
+//SET_ProgramParameter4dNV(disp, glProgramParameter4dNV);
+//SET_ProgramParameter4dvNV(disp, glProgramParameter4dvNV);
+//SET_ProgramParameter4fNV(disp, glProgramParameter4fNV);
+//SET_ProgramParameter4fvNV(disp, glProgramParameter4fvNV);
+//SET_ProgramParameters4dvNV(disp, glProgramParameters4dvNV);
+//SET_ProgramParameters4fvNV(disp, glProgramParameters4fvNV);
+  SET_ProgramStringARB(disp, glProgramStringARB);
+  SET_PushAttrib(disp, glPushAttrib);
+  SET_PushClientAttrib(disp, glPushClientAttrib);
+  SET_PushMatrix(disp, glPushMatrix);
+  SET_PushName(disp, glPushName);
  SET_RasterPos2d(disp, glRasterPos2d);
  SET_RasterPos2dv(disp, glRasterPos2dv);
  SET_RasterPos2f(disp, glRasterPos2f);
@@ -1305,6 +1711,8 @@ static void setup_dispatch_table(void) {
  SET_RasterPos4iv(disp, glRasterPos4iv);
  SET_RasterPos4s(disp, glRasterPos4s);
  SET_RasterPos4sv(disp, glRasterPos4sv);
+  SET_ReadBuffer(disp, glReadBuffer);
+  SET_ReadPixels(disp, glReadPixels);
  SET_Rectd(disp, glRectd);
  SET_Rectdv(disp, glRectdv);
  SET_Rectf(disp, glRectf);
@@ -1313,6 +1721,51 @@ static void setup_dispatch_table(void) {
  SET_Rectiv(disp, glRectiv);
  SET_Rects(disp, glRects);
  SET_Rectsv(disp, glRectsv);
+  SET_RenderMode(disp, glRenderMode);
+  SET_RenderbufferStorageEXT(disp, glRenderbufferStorageEXT);
+//SET_RequestResidentProgramsNV(disp, glRequestResidentProgramsNV);
+  SET_ResetHistogram(disp, glResetHistogram);
+  SET_ResetMinmax(disp, glResetMinmax);
+//SET_ResizeBuffersMESA(disp, glResizeBuffersMESA);
+  SET_Rotated(disp, glRotated);
+  SET_Rotatef(disp, glRotatef);
+  SET_SampleCoverageARB(disp, glSampleCoverageARB);
+//SET_SampleMapATI(disp, glSampleMapEXT);       // <-- EXT -> ATI
+//SET_SampleMaskSGIS(disp, glSampleMaskSGIS);
+//SET_SamplePatternSGIS(disp, glSamplePatternSGIS);
+  SET_Scaled(disp, glScaled);
+  SET_Scalef(disp, glScalef);
+  SET_Scissor(disp, glScissor);
+  SET_SecondaryColor3bEXT(disp, glSecondaryColor3bEXT);
+  SET_SecondaryColor3bvEXT(disp, glSecondaryColor3bvEXT);
+  SET_SecondaryColor3dEXT(disp, glSecondaryColor3dEXT);
+  SET_SecondaryColor3dvEXT(disp, glSecondaryColor3dvEXT);
+  SET_SecondaryColor3fEXT(disp, glSecondaryColor3fEXT);
+  SET_SecondaryColor3fvEXT(disp, glSecondaryColor3fvEXT);
+  SET_SecondaryColor3iEXT(disp, glSecondaryColor3iEXT);
+  SET_SecondaryColor3ivEXT(disp, glSecondaryColor3ivEXT);
+  SET_SecondaryColor3sEXT(disp, glSecondaryColor3sEXT);
+  SET_SecondaryColor3svEXT(disp, glSecondaryColor3svEXT);
+  SET_SecondaryColor3ubEXT(disp, glSecondaryColor3ubEXT);
+  SET_SecondaryColor3ubvEXT(disp, glSecondaryColor3ubvEXT);
+  SET_SecondaryColor3uiEXT(disp, glSecondaryColor3uiEXT);
+  SET_SecondaryColor3uivEXT(disp, glSecondaryColor3uivEXT);
+  SET_SecondaryColor3usEXT(disp, glSecondaryColor3usEXT);
+  SET_SecondaryColor3usvEXT(disp, glSecondaryColor3usvEXT);
+  SET_SecondaryColorPointerEXT(disp, glSecondaryColorPointerEXT);
+  SET_SelectBuffer(disp, glSelectBuffer);
+  SET_SeparableFilter2D(disp, glSeparableFilter2D);
+  SET_SetFenceNV(disp, glSetFenceAPPLE);  // <-- APPLE -> NV
+//SET_SetFragmentShaderConstantATI(disp, glSetFragmentShaderConstantEXT);   // <-- EXT -> ATI
+  SET_ShadeModel(disp, glShadeModel);
+  SET_ShaderSourceARB(disp, glShaderSourceARB);
+  SET_StencilFunc(disp, glStencilFunc);
+  SET_StencilFuncSeparate(disp, glStencilFuncSeparate);
+  SET_StencilMask(disp, glStencilMask);
+  SET_StencilMaskSeparate(disp, glStencilMaskSeparate);
+  SET_StencilOp(disp, glStencilOp);
+  SET_StencilOpSeparate(disp, glStencilOpSeparate);
+  SET_TestFenceNV(disp, glTestFenceAPPLE); // <-- APPLE -> NV
  SET_TexCoord1d(disp, glTexCoord1d);
  SET_TexCoord1dv(disp, glTexCoord1dv);
  SET_TexCoord1f(disp, glTexCoord1f);
@@ -1345,6 +1798,54 @@ static void setup_dispatch_table(void) {
  SET_TexCoord4iv(disp, glTexCoord4iv);
  SET_TexCoord4s(disp, glTexCoord4s);
  SET_TexCoord4sv(disp, glTexCoord4sv);
+  SET_TexCoordPointer(disp, glTexCoordPointer);
+//SET_TexCoordPointerEXT(disp, glTexCoordPointerEXT);
+  SET_TexEnvf(disp, glTexEnvf);
+  SET_TexEnvfv(disp, glTexEnvfv);
+  SET_TexEnvi(disp, glTexEnvi);
+  SET_TexEnviv(disp, glTexEnviv);
+  SET_TexGend(disp, glTexGend);
+  SET_TexGendv(disp, glTexGendv);
+  SET_TexGenf(disp, glTexGenf);
+  SET_TexGenfv(disp, glTexGenfv);
+  SET_TexGeni(disp, glTexGeni);
+  SET_TexGeniv(disp, glTexGeniv);
+  SET_TexImage1D(disp, glTexImage1D);
+  SET_TexImage2D(disp, glTexImage2D);
+  SET_TexImage3D(disp, glTexImage3D);
+  SET_TexParameterf(disp, glTexParameterf);
+  SET_TexParameterfv(disp, glTexParameterfv);
+  SET_TexParameteri(disp, glTexParameteri);
+  SET_TexParameteriv(disp, glTexParameteriv);
+  SET_TexSubImage1D(disp, glTexSubImage1D);
+  SET_TexSubImage2D(disp, glTexSubImage2D);
+  SET_TexSubImage3D(disp, glTexSubImage3D);
+//SET_TrackMatrixNV(disp, glTrackMatrixNV);
+  SET_Translated(disp, glTranslated);
+  SET_Translatef(disp, glTranslatef);
+  SET_Uniform1fARB(disp, glUniform1fARB);
+  SET_Uniform1fvARB(disp, glUniform1fvARB);
+  SET_Uniform1iARB(disp, glUniform1iARB);
+  SET_Uniform1ivARB(disp, glUniform1ivARB);
+  SET_Uniform2fARB(disp, glUniform2fARB);
+  SET_Uniform2fvARB(disp, glUniform2fvARB);
+  SET_Uniform2iARB(disp, glUniform2iARB);
+  SET_Uniform2ivARB(disp, glUniform2ivARB);
+  SET_Uniform3fARB(disp, glUniform3fARB);
+  SET_Uniform3fvARB(disp, glUniform3fvARB);
+  SET_Uniform3iARB(disp, glUniform3iARB);
+  SET_Uniform3ivARB(disp, glUniform3ivARB);
+  SET_Uniform4fARB(disp, glUniform4fARB);
+  SET_Uniform4fvARB(disp, glUniform4fvARB);
+  SET_Uniform4iARB(disp, glUniform4iARB);
+  SET_Uniform4ivARB(disp, glUniform4ivARB);
+  SET_UniformMatrix2fvARB(disp, glUniformMatrix2fvARB);
+  SET_UniformMatrix3fvARB(disp, glUniformMatrix3fvARB);
+  SET_UniformMatrix4fvARB(disp, glUniformMatrix4fvARB);
+  SET_UnlockArraysEXT(disp, glUnlockArraysEXT);
+  SET_UnmapBufferARB(disp, glUnmapBufferARB);
+  SET_UseProgramObjectARB(disp, glUseProgramObjectARB);
+  SET_ValidateProgramARB(disp, glValidateProgramARB);
  SET_Vertex2d(disp, glVertex2d);
  SET_Vertex2dv(disp, glVertex2dv);
  SET_Vertex2f(disp, glVertex2f);
@@ -1369,459 +1870,7 @@ static void setup_dispatch_table(void) {
  SET_Vertex4iv(disp, glVertex4iv);
  SET_Vertex4s(disp, glVertex4s);
  SET_Vertex4sv(disp, glVertex4sv);
- SET_ClipPlane(disp, glClipPlane);
- SET_ColorMaterial(disp, glColorMaterial);
- SET_CullFace(disp, glCullFace);
- SET_Fogf(disp, glFogf);
- SET_Fogfv(disp, glFogfv);
- SET_Fogi(disp, glFogi);
- SET_Fogiv(disp, glFogiv);
- SET_FrontFace(disp, glFrontFace);
- SET_Hint(disp, glHint);
- SET_Lightf(disp, glLightf);
- SET_Lightfv(disp, glLightfv);
- SET_Lighti(disp, glLighti);
- SET_Lightiv(disp, glLightiv);
- SET_LightModelf(disp, glLightModelf);
- SET_LightModelfv(disp, glLightModelfv);
- SET_LightModeli(disp, glLightModeli);
- SET_LightModeliv(disp, glLightModeliv);
- SET_LineStipple(disp, glLineStipple);
- SET_LineWidth(disp, glLineWidth);
- SET_Materialf(disp, glMaterialf);
- SET_Materialfv(disp, glMaterialfv);
- SET_Materiali(disp, glMateriali);
- SET_Materialiv(disp, glMaterialiv);
- SET_PointSize(disp, glPointSize);
- SET_PolygonMode(disp, glPolygonMode);
- SET_PolygonStipple(disp, glPolygonStipple);
- SET_Scissor(disp, glScissor);
- SET_ShadeModel(disp, glShadeModel);
- SET_TexParameterf(disp, glTexParameterf);
- SET_TexParameterfv(disp, glTexParameterfv);
- SET_TexParameteri(disp, glTexParameteri);
- SET_TexParameteriv(disp, glTexParameteriv);
- SET_TexImage1D(disp, glTexImage1D);
- SET_TexImage2D(disp, glTexImage2D);
- SET_TexEnvf(disp, glTexEnvf);
- SET_TexEnvfv(disp, glTexEnvfv);
- SET_TexEnvi(disp, glTexEnvi);
- SET_TexEnviv(disp, glTexEnviv);
- SET_TexGend(disp, glTexGend);
- SET_TexGendv(disp, glTexGendv);
- SET_TexGenf(disp, glTexGenf);
- SET_TexGenfv(disp, glTexGenfv);
- SET_TexGeni(disp, glTexGeni);
- SET_TexGeniv(disp, glTexGeniv);
- SET_FeedbackBuffer(disp, glFeedbackBuffer);
- SET_SelectBuffer(disp, glSelectBuffer);
- SET_RenderMode(disp, glRenderMode);
- SET_InitNames(disp, glInitNames);
- SET_LoadName(disp, glLoadName);
- SET_PassThrough(disp, glPassThrough);
- SET_PopName(disp, glPopName);
- SET_PushName(disp, glPushName);
- SET_DrawBuffer(disp, glDrawBuffer);
- SET_Clear(disp, glClear);
- SET_ClearAccum(disp, glClearAccum);
- SET_ClearIndex(disp, glClearIndex);
- SET_ClearColor(disp, glClearColor);
- SET_ClearStencil(disp, glClearStencil);
- SET_ClearDepth(disp, glClearDepth);
- SET_StencilMask(disp, glStencilMask);
- SET_ColorMask(disp, glColorMask);
- SET_DepthMask(disp, glDepthMask);
- SET_IndexMask(disp, glIndexMask);
- SET_Accum(disp, glAccum);
- SET_Disable(disp, glDisable);
- SET_Enable(disp, glEnable);
- SET_Finish(disp, glFinish);
- SET_Flush(disp, glFlush);
- SET_PopAttrib(disp, glPopAttrib);
- SET_PushAttrib(disp, glPushAttrib);
- SET_Map1d(disp, glMap1d);
- SET_Map1f(disp, glMap1f);
- SET_Map2d(disp, glMap2d);
- SET_Map2f(disp, glMap2f);
- SET_MapGrid1d(disp, glMapGrid1d);
- SET_MapGrid1f(disp, glMapGrid1f);
- SET_MapGrid2d(disp, glMapGrid2d);
- SET_MapGrid2f(disp, glMapGrid2f);
- SET_EvalCoord1d(disp, glEvalCoord1d);
- SET_EvalCoord1dv(disp, glEvalCoord1dv);
- SET_EvalCoord1f(disp, glEvalCoord1f);
- SET_EvalCoord1fv(disp, glEvalCoord1fv);
- SET_EvalCoord2d(disp, glEvalCoord2d);
- SET_EvalCoord2dv(disp, glEvalCoord2dv);
- SET_EvalCoord2f(disp, glEvalCoord2f);
- SET_EvalCoord2fv(disp, glEvalCoord2fv);
- SET_EvalMesh1(disp, glEvalMesh1);
- SET_EvalPoint1(disp, glEvalPoint1);
- SET_EvalMesh2(disp, glEvalMesh2);
- SET_EvalPoint2(disp, glEvalPoint2);
- SET_AlphaFunc(disp, glAlphaFunc);
- SET_BlendFunc(disp, glBlendFunc);
- SET_LogicOp(disp, glLogicOp);
- SET_StencilFunc(disp, glStencilFunc);
- SET_StencilOp(disp, glStencilOp);
- SET_DepthFunc(disp, glDepthFunc);
- SET_PixelZoom(disp, glPixelZoom);
- SET_PixelTransferf(disp, glPixelTransferf);
- SET_PixelTransferi(disp, glPixelTransferi);
- SET_PixelStoref(disp, glPixelStoref);
- SET_PixelStorei(disp, glPixelStorei);
- SET_PixelMapfv(disp, glPixelMapfv);
- SET_PixelMapuiv(disp, glPixelMapuiv);
- SET_PixelMapusv(disp, glPixelMapusv);
- SET_ReadBuffer(disp, glReadBuffer);
- SET_CopyPixels(disp, glCopyPixels);
- SET_ReadPixels(disp, glReadPixels);
- SET_DrawPixels(disp, glDrawPixels);
- SET_GetBooleanv(disp, glGetBooleanv);
- SET_GetClipPlane(disp, glGetClipPlane);
- SET_GetDoublev(disp, glGetDoublev);
- SET_GetError(disp, glGetError);
- SET_GetFloatv(disp, glGetFloatv);
- SET_GetIntegerv(disp, glGetIntegerv);
- SET_GetLightfv(disp, glGetLightfv);
- SET_GetLightiv(disp, glGetLightiv);
- SET_GetMapdv(disp, glGetMapdv);
- SET_GetMapfv(disp, glGetMapfv);
- SET_GetMapiv(disp, glGetMapiv);
- SET_GetMaterialfv(disp, glGetMaterialfv);
- SET_GetMaterialiv(disp, glGetMaterialiv);
- SET_GetPixelMapfv(disp, glGetPixelMapfv);
- SET_GetPixelMapuiv(disp, glGetPixelMapuiv);
- SET_GetPixelMapusv(disp, glGetPixelMapusv);
- SET_GetPolygonStipple(disp, glGetPolygonStipple);
- SET_GetString(disp, glGetString);
- SET_GetTexEnvfv(disp, glGetTexEnvfv);
- SET_GetTexEnviv(disp, glGetTexEnviv);
- SET_GetTexGendv(disp, glGetTexGendv);
- SET_GetTexGenfv(disp, glGetTexGenfv);
- SET_GetTexGeniv(disp, glGetTexGeniv);
- SET_GetTexImage(disp, glGetTexImage);
- SET_GetTexParameterfv(disp, glGetTexParameterfv);
- SET_GetTexParameteriv(disp, glGetTexParameteriv);
- SET_GetTexLevelParameterfv(disp, glGetTexLevelParameterfv);
- SET_GetTexLevelParameteriv(disp, glGetTexLevelParameteriv);
- SET_IsEnabled(disp, glIsEnabled);
- SET_IsList(disp, glIsList);
- SET_DepthRange(disp, glDepthRange);
- SET_Frustum(disp, glFrustum);
- SET_LoadIdentity(disp, glLoadIdentity);
- SET_LoadMatrixf(disp, glLoadMatrixf);
- SET_LoadMatrixd(disp, glLoadMatrixd);
- SET_MatrixMode(disp, glMatrixMode);
- SET_MultMatrixf(disp, glMultMatrixf);
- SET_MultMatrixd(disp, glMultMatrixd);
- SET_Ortho(disp, glOrtho);
- SET_PopMatrix(disp, glPopMatrix);
- SET_PushMatrix(disp, glPushMatrix);
- SET_Rotated(disp, glRotated);
- SET_Rotatef(disp, glRotatef);
- SET_Scaled(disp, glScaled);
- SET_Scalef(disp, glScalef);
- SET_Translated(disp, glTranslated);
- SET_Translatef(disp, glTranslatef);
- SET_Viewport(disp, glViewport);
- SET_ArrayElement(disp, glArrayElement);
- SET_BindTexture(disp, glBindTexture);
- SET_ColorPointer(disp, glColorPointer);
- SET_DisableClientState(disp, glDisableClientState);
- SET_DrawArrays(disp, glDrawArrays);
- SET_DrawElements(disp, glDrawElements);
- SET_EdgeFlagPointer(disp, glEdgeFlagPointer);
- SET_EnableClientState(disp, glEnableClientState);
- SET_IndexPointer(disp, glIndexPointer);
- SET_Indexub(disp, glIndexub);
- SET_Indexubv(disp, glIndexubv);
- SET_InterleavedArrays(disp, glInterleavedArrays);
- SET_NormalPointer(disp, glNormalPointer);
- SET_PolygonOffset(disp, glPolygonOffset);
- SET_TexCoordPointer(disp, glTexCoordPointer);
- SET_VertexPointer(disp, glVertexPointer);
- SET_AreTexturesResident(disp, glAreTexturesResident); 
- SET_CopyTexImage1D(disp, glCopyTexImage1D);
- SET_CopyTexImage2D(disp, glCopyTexImage2D);
- SET_CopyTexSubImage1D(disp, glCopyTexSubImage1D);
- SET_CopyTexSubImage2D(disp, glCopyTexSubImage2D);
- SET_DeleteTextures(disp, glDeleteTextures);
- SET_GenTextures(disp, glGenTextures);
- SET_GetPointerv(disp, glGetPointerv);
- SET_IsTexture(disp, glIsTexture);
- SET_PrioritizeTextures(disp, glPrioritizeTextures);
- SET_TexSubImage1D(disp, glTexSubImage1D);
- SET_TexSubImage2D(disp, glTexSubImage2D);
- SET_PopClientAttrib(disp, glPopClientAttrib);
- SET_PushClientAttrib(disp, glPushClientAttrib);
- SET_BlendColor(disp, glBlendColor);
- SET_BlendEquation(disp, glBlendEquation);
- SET_DrawRangeElements(disp, glDrawRangeElements);
- SET_ColorTable(disp, glColorTable);
- SET_ColorTableParameterfv(disp, glColorTableParameterfv);
- SET_ColorTableParameteriv(disp, glColorTableParameteriv);
- SET_CopyColorTable(disp, glCopyColorTable);
- SET_GetColorTable(disp, glGetColorTable);
- SET_GetColorTableParameterfv(disp, glGetColorTableParameterfv);
- SET_GetColorTableParameteriv(disp, glGetColorTableParameteriv);
- SET_ColorSubTable(disp, glColorSubTable);
- SET_CopyColorSubTable(disp, glCopyColorSubTable);
- SET_ConvolutionFilter1D(disp, glConvolutionFilter1D);
- SET_ConvolutionFilter2D(disp, glConvolutionFilter2D);
- SET_ConvolutionParameterf(disp, glConvolutionParameterf);
- SET_ConvolutionParameterfv(disp, glConvolutionParameterfv);
- SET_ConvolutionParameteri(disp, glConvolutionParameteri);
- SET_ConvolutionParameteriv(disp, glConvolutionParameteriv);
- SET_CopyConvolutionFilter1D(disp, glCopyConvolutionFilter1D);
- SET_CopyConvolutionFilter2D(disp, glCopyConvolutionFilter2D);
- SET_GetConvolutionFilter(disp, glGetConvolutionFilter);
- SET_GetConvolutionParameterfv(disp, glGetConvolutionParameterfv);
- SET_GetConvolutionParameteriv(disp, glGetConvolutionParameteriv);
- SET_GetSeparableFilter(disp, glGetSeparableFilter);
- SET_SeparableFilter2D(disp, glSeparableFilter2D);
- SET_GetHistogram(disp, glGetHistogram);
- SET_GetHistogramParameterfv(disp, glGetHistogramParameterfv);
- SET_GetHistogramParameteriv(disp, glGetHistogramParameteriv);
- SET_GetMinmax(disp, glGetMinmax);
- SET_GetMinmaxParameterfv(disp, glGetMinmaxParameterfv);
- SET_GetMinmaxParameteriv(disp, glGetMinmaxParameteriv);
- SET_Histogram(disp, glHistogram);
- SET_Minmax(disp, glMinmax);
- SET_ResetHistogram(disp, glResetHistogram);
- SET_ResetMinmax(disp, glResetMinmax);
- SET_TexImage3D(disp, glTexImage3D);
- SET_TexSubImage3D(disp, glTexSubImage3D);
- SET_CopyTexSubImage3D(disp, glCopyTexSubImage3D);
- SET_ActiveTextureARB(disp, glActiveTextureARB);
- SET_ClientActiveTextureARB(disp, glClientActiveTextureARB);
- SET_MultiTexCoord1dARB(disp, glMultiTexCoord1dARB);
- SET_MultiTexCoord1dvARB(disp, glMultiTexCoord1dvARB);
- SET_MultiTexCoord1fARB(disp, glMultiTexCoord1fARB);
- SET_MultiTexCoord1fvARB(disp, glMultiTexCoord1fvARB);
- SET_MultiTexCoord1iARB(disp, glMultiTexCoord1iARB);
- SET_MultiTexCoord1ivARB(disp, glMultiTexCoord1ivARB);
- SET_MultiTexCoord1sARB(disp, glMultiTexCoord1sARB);
- SET_MultiTexCoord1svARB(disp, glMultiTexCoord1svARB);
- SET_MultiTexCoord2dARB(disp, glMultiTexCoord2dARB);
- SET_MultiTexCoord2dvARB(disp, glMultiTexCoord2dvARB);
- SET_MultiTexCoord2fARB(disp, glMultiTexCoord2fARB);
- SET_MultiTexCoord2fvARB(disp, glMultiTexCoord2fvARB);
- SET_MultiTexCoord2iARB(disp, glMultiTexCoord2iARB);
- SET_MultiTexCoord2ivARB(disp, glMultiTexCoord2ivARB);
- SET_MultiTexCoord2sARB(disp, glMultiTexCoord2sARB);
- SET_MultiTexCoord2svARB(disp, glMultiTexCoord2svARB);
- SET_MultiTexCoord3dARB(disp, glMultiTexCoord3dARB);
- SET_MultiTexCoord3dvARB(disp, glMultiTexCoord3dvARB);
- SET_MultiTexCoord3fARB(disp, glMultiTexCoord3fARB);
- SET_MultiTexCoord3fvARB(disp, glMultiTexCoord3fvARB);
- SET_MultiTexCoord3iARB(disp, glMultiTexCoord3iARB);
- SET_MultiTexCoord3ivARB(disp, glMultiTexCoord3ivARB);
- SET_MultiTexCoord3sARB(disp, glMultiTexCoord3sARB);
- SET_MultiTexCoord3svARB(disp, glMultiTexCoord3svARB);
- SET_MultiTexCoord4dARB(disp, glMultiTexCoord4dARB);
- SET_MultiTexCoord4dvARB(disp, glMultiTexCoord4dvARB);
- SET_MultiTexCoord4fARB(disp, glMultiTexCoord4fARB);
- SET_MultiTexCoord4fvARB(disp, glMultiTexCoord4fvARB);
- SET_MultiTexCoord4iARB(disp, glMultiTexCoord4iARB);
- SET_MultiTexCoord4ivARB(disp, glMultiTexCoord4ivARB);
- SET_MultiTexCoord4sARB(disp, glMultiTexCoord4sARB);
- SET_MultiTexCoord4svARB(disp, glMultiTexCoord4svARB);
- SET_LoadTransposeMatrixfARB(disp, glLoadTransposeMatrixfARB);
- SET_LoadTransposeMatrixdARB(disp, glLoadTransposeMatrixdARB);
- SET_MultTransposeMatrixfARB(disp, glMultTransposeMatrixfARB);
- SET_MultTransposeMatrixdARB(disp, glMultTransposeMatrixdARB);
- SET_SampleCoverageARB(disp, glSampleCoverageARB);
- SET_DrawBuffersARB(disp, glDrawBuffersARB);
-/* SET_PolygonOffsetEXT(disp, glPolygonOffsetEXT);
- SET_GetTexFilterFuncSGIS(disp, glGetTexFilterFuncSGIS);
- SET_TexFilterFuncSGIS(disp, glTexFilterFuncSGIS);
- SET_GetHistogramEXT(disp, glGetHistogramEXT);
- SET_GetHistogramParameterfvEXT(disp, glGetHistogramParameterfvEXT);
- SET_GetHistogramParameterivEXT(disp, glGetHistogramParameterivEXT);
- SET_GetMinmaxEXT(disp, glGetMinmaxEXT);
- SET_GetMinmaxParameterfvEXT(disp, glGetMinmaxParameterfvEXT);
- SET_GetMinmaxParameterivEXT(disp, glGetMinmaxParameterivEXT);
- SET_GetConvolutionFilterEXT(disp, glGetConvolutionFilterEXT);
- SET_GetConvolutionParameterfvEXT(disp, glGetConvolutionParameterfvEXT);
- SET_GetConvolutionParameterivEXT(disp, glGetConvolutionParameterivEXT);
- SET_GetSeparableFilterEXT(disp, glGetSeparableFilterEXT);
- SET_GetColorTableSGI(disp, glGetColorTableSGI);
- SET_GetColorTableParameterfvSGI(disp, glGetColorTableParameterfvSGI);
- SET_GetColorTableParameterivSGI(disp, glGetColorTableParameterivSGI);
- SET_PixelTexGenSGIX(disp, glPixelTexGenSGIX);
- SET_PixelTexGenParameteriSGIS(disp, glPixelTexGenParameteriSGIS);
- SET_PixelTexGenParameterivSGIS(disp, glPixelTexGenParameterivSGIS);
- SET_PixelTexGenParameterfSGIS(disp, glPixelTexGenParameterfSGIS);
- SET_PixelTexGenParameterfvSGIS(disp, glPixelTexGenParameterfvSGIS);
- SET_GetPixelTexGenParameterivSGIS(disp, glGetPixelTexGenParameterivSGIS);
- SET_GetPixelTexGenParameterfvSGIS(disp, glGetPixelTexGenParameterfvSGIS);
- SET_TexImage4DSGIS(disp, glTexImage4DSGIS);
- SET_TexSubImage4DSGIS(disp, glTexSubImage4DSGIS); */
- SET_AreTexturesResidentEXT(disp, glAreTexturesResident);
- SET_GenTexturesEXT(disp, glGenTextures);
- SET_IsTextureEXT(disp, glIsTexture);
-/* SET_DetailTexFuncSGIS(disp, glDetailTexFuncSGIS);
- SET_GetDetailTexFuncSGIS(disp, glGetDetailTexFuncSGIS);
- SET_SharpenTexFuncSGIS(disp, glSharpenTexFuncSGIS);
- SET_GetSharpenTexFuncSGIS(disp, glGetSharpenTexFuncSGIS);
- SET_SampleMaskSGIS(disp, glSampleMaskSGIS);
- SET_SamplePatternSGIS(disp, glSamplePatternSGIS);
- SET_ColorPointerEXT(disp, glColorPointerEXT);
- SET_EdgeFlagPointerEXT(disp, glEdgeFlagPointerEXT);
- SET_IndexPointerEXT(disp, glIndexPointerEXT);
- SET_NormalPointerEXT(disp, glNormalPointerEXT);
- SET_TexCoordPointerEXT(disp, glTexCoordPointerEXT);
- SET_VertexPointerEXT(disp, glVertexPointerEXT);
- SET_SpriteParameterfSGIX(disp, glSpriteParameterfSGIX);
- SET_SpriteParameterfvSGIX(disp, glSpriteParameterfvSGIX);
- SET_SpriteParameteriSGIX(disp, glSpriteParameteriSGIX);
- SET_SpriteParameterivSGIX(disp, glSpriteParameterivSGIX);
- SET_PointParameterfEXT(disp, glPointParameterfEXT);
- SET_PointParameterfvEXT(disp, glPointParameterfvEXT);
- SET_GetInstrumentsSGIX(disp, glGetInstrumentsSGIX);
- SET_InstrumentsBufferSGIX(disp, glInstrumentsBufferSGIX);
- SET_PollInstrumentsSGIX(disp, glPollInstrumentsSGIX);
- SET_ReadInstrumentsSGIX(disp, glReadInstrumentsSGIX);
- SET_StartInstrumentsSGIX(disp, glStartInstrumentsSGIX);
- SET_StopInstrumentsSGIX(disp, glStopInstrumentsSGIX);
- SET_FrameZoomSGIX(disp, glFrameZoomSGIX);
- SET_TagSampleBufferSGIX(disp, glTagSampleBufferSGIX);
- SET_ReferencePlaneSGIX(disp, glReferencePlaneSGIX);
- SET_FlushRasterSGIX(disp, glFlushRasterSGIX);
- SET_GetListParameterfvSGIX(disp, glGetListParameterfvSGIX);
- SET_GetListParameterivSGIX(disp, glGetListParameterivSGIX);
- SET_ListParameterfSGIX(disp, glListParameterfSGIX);
- SET_ListParameterfvSGIX(disp, glListParameterfvSGIX);
- SET_ListParameteriSGIX(disp, glListParameteriSGIX);
- SET_ListParameterivSGIX(disp, glListParameterivSGIX);
- SET_FragmentColorMaterialSGIX(disp, glFragmentColorMaterialSGIX);
- SET_FragmentLightfSGIX(disp, glFragmentLightfSGIX);
- SET_FragmentLightfvSGIX(disp, glFragmentLightfvSGIX);
- SET_FragmentLightiSGIX(disp, glFragmentLightiSGIX);
- SET_FragmentLightivSGIX(disp, glFragmentLightivSGIX);
- SET_FragmentLightModelfSGIX(disp, glFragmentLightModelfSGIX);
- SET_FragmentLightModelfvSGIX(disp, glFragmentLightModelfvSGIX);
- SET_FragmentLightModeliSGIX(disp, glFragmentLightModeliSGIX);
- SET_FragmentLightModelivSGIX(disp, glFragmentLightModelivSGIX);
- SET_FragmentMaterialfSGIX(disp, glFragmentMaterialfSGIX);
- SET_FragmentMaterialfvSGIX(disp, glFragmentMaterialfvSGIX);
- SET_FragmentMaterialiSGIX(disp, glFragmentMaterialiSGIX);
- SET_FragmentMaterialivSGIX(disp, glFragmentMaterialivSGIX);
- SET_GetFragmentLightfvSGIX(disp, glGetFragmentLightfvSGIX);
- SET_GetFragmentLightivSGIX(disp, glGetFragmentLightivSGIX);
- SET_GetFragmentMaterialfvSGIX(disp, glGetFragmentMaterialfvSGIX);
- SET_GetFragmentMaterialivSGIX(disp, glGetFragmentMaterialivSGIX);
- SET_LightEnviSGIX(disp, glLightEnviSGIX); 
- SET_VertexWeightfEXT(disp, glVertexWeightfEXT);
- SET_VertexWeightfvEXT(disp, glVertexWeightfvEXT);
- SET_VertexWeightPointerEXT(disp, glVertexWeightPointerEXT);
- SET_FlushVertexArrayRangeNV(disp, glFlushVertexArrayRangeNV);
- SET_VertexArrayRangeNV(disp, glVertexArrayRangeNV);
- SET_CombinerParameterfvNV(disp, glCombinerParameterfvNV);
- SET_CombinerParameterfNV(disp, glCombinerParameterfNV);
- SET_CombinerParameterivNV(disp, glCombinerParameterivNV);
- SET_CombinerParameteriNV(disp, glCombinerParameteriNV);
- SET_CombinerInputNV(disp, glCombinerInputNV);
- SET_CombinerOutputNV(disp, glCombinerOutputNV);
- SET_FinalCombinerInputNV(disp, glFinalCombinerInputNV);
- SET_GetCombinerInputParameterfvNV(disp, glGetCombinerInputParameterfvNV);
- SET_GetCombinerInputParameterivNV(disp, glGetCombinerInputParameterivNV);
- SET_GetCombinerOutputParameterfvNV(disp, glGetCombinerOutputParameterfvNV);
- SET_GetCombinerOutputParameterivNV(disp, glGetCombinerOutputParameterivNV);
- SET_GetFinalCombinerInputParameterfvNV(disp, glGetFinalCombinerInputParameterfvNV);
- SET_GetFinalCombinerInputParameterivNV(disp, glGetFinalCombinerInputParameterivNV);
- SET_ResizeBuffersMESA(disp, glResizeBuffersMESA);
- SET_WindowPos2dMESA(disp, glWindowPos2dMESA);
- SET_WindowPos2dvMESA(disp, glWindowPos2dvMESA);
- SET_WindowPos2fMESA(disp, glWindowPos2fMESA);
- SET_WindowPos2fvMESA(disp, glWindowPos2fvMESA);
- SET_WindowPos2iMESA(disp, glWindowPos2iMESA);
- SET_WindowPos2ivMESA(disp, glWindowPos2ivMESA);
- SET_WindowPos2sMESA(disp, glWindowPos2sMESA);
- SET_WindowPos2svMESA(disp, glWindowPos2svMESA);
- SET_WindowPos3dMESA(disp, glWindowPos3dMESA);
- SET_WindowPos3dvMESA(disp, glWindowPos3dvMESA);
- SET_WindowPos3fMESA(disp, glWindowPos3fMESA);
- SET_WindowPos3fvMESA(disp, glWindowPos3fvMESA);
- SET_WindowPos3iMESA(disp, glWindowPos3iMESA);
- SET_WindowPos3ivMESA(disp, glWindowPos3ivMESA);
- SET_WindowPos3sMESA(disp, glWindowPos3sMESA);
- SET_WindowPos3svMESA(disp, glWindowPos3svMESA);
- SET_WindowPos4dMESA(disp, glWindowPos4dMESA);
- SET_WindowPos4dvMESA(disp, glWindowPos4dvMESA);
- SET_WindowPos4fMESA(disp, glWindowPos4fMESA);
- SET_WindowPos4fvMESA(disp, glWindowPos4fvMESA);
- SET_WindowPos4iMESA(disp, glWindowPos4iMESA);
- SET_WindowPos4ivMESA(disp, glWindowPos4ivMESA);
- SET_WindowPos4sMESA(disp, glWindowPos4sMESA);
- SET_WindowPos4svMESA(disp, glWindowPos4svMESA);
- SET_BlendFuncSeparateEXT(disp, glBlendFuncSeparateEXT);
- SET_IndexMaterialEXT(disp, glIndexMaterialEXT);
- SET_IndexFuncEXT(disp, glIndexFuncEXT);
- SET_LockArraysEXT(disp, glLockArraysEXT);
- SET_UnlockArraysEXT(disp, glUnlockArraysEXT);
- SET_CullParameterdvEXT(disp, glCullParameterdvEXT);
- SET_CullParameterfvEXT(disp, glCullParameterfvEXT);
- SET_HintPGI(disp, glHintPGI);
- SET_FogCoordfEXT(disp, glFogCoordfEXT);
- SET_FogCoordfvEXT(disp, glFogCoordfvEXT);
- SET_FogCoorddEXT(disp, glFogCoorddEXT);
- SET_FogCoorddvEXT(disp, glFogCoorddvEXT);
- SET_FogCoordPointerEXT(disp, glFogCoordPointerEXT);
- SET_GetColorTableEXT(disp, glGetColorTableEXT);
- SET_GetColorTableParameterivEXT(disp, glGetColorTableParameterivEXT);
- SET_GetColorTableParameterfvEXT(disp, glGetColorTableParameterfvEXT);
- SET_TbufferMask3DFX(disp, glTbufferMask3DFX);
- SET_CompressedTexImage3DARB(disp, glCompressedTexImage3DARB);
- SET_CompressedTexImage2DARB(disp, glCompressedTexImage2DARB);
- SET_CompressedTexImage1DARB(disp, glCompressedTexImage1DARB);
- SET_CompressedTexSubImage3DARB(disp, glCompressedTexSubImage3DARB);
- SET_CompressedTexSubImage2DARB(disp, glCompressedTexSubImage2DARB);
- SET_CompressedTexSubImage1DARB(disp, glCompressedTexSubImage1DARB);
- SET_GetCompressedTexImageARB(disp, glGetCompressedTexImageARB);
- SET_SecondaryColor3bEXT(disp, glSecondaryColor3bEXT);
- SET_SecondaryColor3bvEXT(disp, glSecondaryColor3bvEXT);
- SET_SecondaryColor3dEXT(disp, glSecondaryColor3dEXT);
- SET_SecondaryColor3dvEXT(disp, glSecondaryColor3dvEXT);
- SET_SecondaryColor3fEXT(disp, glSecondaryColor3fEXT);
- SET_SecondaryColor3fvEXT(disp, glSecondaryColor3fvEXT);
- SET_SecondaryColor3iEXT(disp, glSecondaryColor3iEXT);
- SET_SecondaryColor3ivEXT(disp, glSecondaryColor3ivEXT);
- SET_SecondaryColor3sEXT(disp, glSecondaryColor3sEXT);
- SET_SecondaryColor3svEXT(disp, glSecondaryColor3svEXT);
- SET_SecondaryColor3ubEXT(disp, glSecondaryColor3ubEXT);
- SET_SecondaryColor3ubvEXT(disp, glSecondaryColor3ubvEXT);
- SET_SecondaryColor3uiEXT(disp, glSecondaryColor3uiEXT);
- SET_SecondaryColor3uivEXT(disp, glSecondaryColor3uivEXT);
- SET_SecondaryColor3usEXT(disp, glSecondaryColor3usEXT);
- SET_SecondaryColor3usvEXT(disp, glSecondaryColor3usvEXT);
- SET_SecondaryColorPointerEXT(disp, glSecondaryColorPointerEXT);
- SET_AreProgramsResidentNV(disp, glAreProgramsResidentNV);
- SET_BindProgramNV(disp, glBindProgramNV);
- SET_DeleteProgramsNV(disp, glDeleteProgramsNV);
- SET_ExecuteProgramNV(disp, glExecuteProgramNV);
- SET_GenProgramsNV(disp, glGenProgramsNV);
- SET_GetProgramParameterdvNV(disp, glGetProgramParameterdvNV);
- SET_GetProgramParameterfvNV(disp, glGetProgramParameterfvNV);
- SET_GetProgramivNV(disp, glGetProgramivNV);
- SET_GetProgramStringNV(disp, glGetProgramStringNV);
- SET_GetTrackMatrixivNV(disp, glGetTrackMatrixivNV);
- SET_GetVertexAttribdvARB(disp, glGetVertexAttribdvARB);
- SET_GetVertexAttribfvARB(disp, glGetVertexAttribfvARB);
- SET_GetVertexAttribivARB(disp, glGetVertexAttribivARB);
- SET_GetVertexAttribPointervNV(disp, glGetVertexAttribPointervNV);
- SET_IsProgramNV(disp, glIsProgramNV);
- SET_LoadProgramNV(disp, glLoadProgramNV);
- SET_ProgramParameter4dNV(disp, glProgramParameter4dNV);
- SET_ProgramParameter4dvNV(disp, glProgramParameter4dvNV);
- SET_ProgramParameter4fNV(disp, glProgramParameter4fNV);
- SET_ProgramParameter4fvNV(disp, glProgramParameter4fvNV);
- SET_ProgramParameters4dvNV(disp, glProgramParameters4dvNV);
- SET_ProgramParameters4fvNV(disp, glProgramParameters4fvNV);
- SET_RequestResidentProgramsNV(disp, glRequestResidentProgramsNV);
- SET_TrackMatrixNV(disp, glTrackMatrixNV);
- SET_VertexAttribPointerNV(disp, glVertexAttribPointerNV);
+//SET_VertexArrayRangeNV(disp, glVertexArrayRangeNV);
  SET_VertexAttrib1dARB(disp, glVertexAttrib1dARB);
  SET_VertexAttrib1dvARB(disp, glVertexAttrib1dvARB);
  SET_VertexAttrib1fARB(disp, glVertexAttrib1fARB);
@@ -1840,614 +1889,50 @@ static void setup_dispatch_table(void) {
  SET_VertexAttrib3fvARB(disp, glVertexAttrib3fvARB);
  SET_VertexAttrib3sARB(disp, glVertexAttrib3sARB);
  SET_VertexAttrib3svARB(disp, glVertexAttrib3svARB);
- SET_VertexAttrib4dARB(disp, glVertexAttrib4dARB);
- SET_VertexAttrib4dvARB(disp, glVertexAttrib4dvARB);
- SET_VertexAttrib4fARB(disp, glVertexAttrib4fARB);
- SET_VertexAttrib4fvARB(disp, glVertexAttrib4fvARB);
- SET_VertexAttrib4sARB(disp, glVertexAttrib4sARB);
- SET_VertexAttrib4svARB(disp, glVertexAttrib4svARB);
- SET_VertexAttrib4NubARB(disp, glVertexAttrib4NubARB);
- SET_VertexAttrib4NubvARB(disp, glVertexAttrib4NubvARB);
- SET_VertexAttribs1dvNV(disp, glVertexAttribs1dvNV);
- SET_VertexAttribs1fvNV(disp, glVertexAttribs1fvNV);
- SET_VertexAttribs1svNV(disp, glVertexAttribs1svNV);
- SET_VertexAttribs2dvNV(disp, glVertexAttribs2dvNV);
- SET_VertexAttribs2fvNV(disp, glVertexAttribs2fvNV);
- SET_VertexAttribs2svNV(disp, glVertexAttribs2svNV);
- SET_VertexAttribs3dvNV(disp, glVertexAttribs3dvNV);
- SET_VertexAttribs3fvNV(disp, glVertexAttribs3fvNV);
- SET_VertexAttribs3svNV(disp, glVertexAttribs3svNV);
- SET_VertexAttribs4dvNV(disp, glVertexAttribs4dvNV);
- SET_VertexAttribs4fvNV(disp, glVertexAttribs4fvNV);
- SET_VertexAttribs4svNV(disp, glVertexAttribs4svNV);
- SET_VertexAttribs4ubvNV(disp, glVertexAttribs4ubvNV);
- SET_PointParameteriNV(disp, glPointParameteriNV);
- SET_PointParameterivNV(disp, glPointParameterivNV);
- SET_MultiDrawArraysEXT(disp, glMultiDrawArraysEXT);
- SET_MultiDrawElementsEXT(disp, glMultiDrawElementsEXT);
- SET_ActiveStencilFaceEXT(disp, glActiveStencilFaceEXT);
- SET_DeleteFencesNV(disp, glDeleteFencesNV);
- SET_GenFencesNV(disp, glGenFencesNV);
- SET_IsFenceNV(disp, glIsFenceNV);
- SET_TestFenceNV(disp, glTestFenceNV);
- SET_GetFenceivNV(disp, glGetFenceivNV);
- SET_FinishFenceNV(disp, glFinishFenceNV);
- SET_SetFenceNV(disp, glSetFenceNV);
- SET_VertexAttrib4bvARB(disp, glVertexAttrib4bvARB);
- SET_VertexAttrib4ivARB(disp, glVertexAttrib4ivARB);
- SET_VertexAttrib4ubvARB(disp, glVertexAttrib4ubvARB);
- SET_VertexAttrib4usvARB(disp, glVertexAttrib4usvARB);
- SET_VertexAttrib4uivARB(disp, glVertexAttrib4uivARB);
  SET_VertexAttrib4NbvARB(disp, glVertexAttrib4NbvARB);
- SET_VertexAttrib4NsvARB(disp, glVertexAttrib4NsvARB);
  SET_VertexAttrib4NivARB(disp, glVertexAttrib4NivARB);
- SET_VertexAttrib4NusvARB(disp, glVertexAttrib4NusvARB);
+  SET_VertexAttrib4NsvARB(disp, glVertexAttrib4NsvARB);
+  SET_VertexAttrib4NubARB(disp, glVertexAttrib4NubARB);
+  SET_VertexAttrib4NubvARB(disp, glVertexAttrib4NubvARB);
  SET_VertexAttrib4NuivARB(disp, glVertexAttrib4NuivARB);
- SET_VertexAttribPointerARB(disp, glVertexAttribPointerARB);
- SET_EnableVertexAttribArrayARB(disp, glEnableVertexAttribArrayARB);
- SET_DisableVertexAttribArrayARB(disp, glDisableVertexAttribArrayARB);
- SET_ProgramStringARB(disp, glProgramStringARB);
- SET_ProgramEnvParameter4dARB(disp, glProgramEnvParameter4dARB);
- SET_ProgramEnvParameter4dvARB(disp, glProgramEnvParameter4dvARB);
- SET_ProgramEnvParameter4fARB(disp, glProgramEnvParameter4fARB);
- SET_ProgramEnvParameter4fvARB(disp, glProgramEnvParameter4fvARB);
- SET_ProgramLocalParameter4dARB(disp, glProgramLocalParameter4dARB);
- SET_ProgramLocalParameter4dvARB(disp, glProgramLocalParameter4dvARB);
- SET_ProgramLocalParameter4fARB(disp, glProgramLocalParameter4fARB);
- SET_ProgramLocalParameter4fvARB(disp, glProgramLocalParameter4fvARB);
- SET_GetProgramEnvParameterdvARB(disp, glGetProgramEnvParameterdvARB);
- SET_GetProgramEnvParameterfvARB(disp, glGetProgramEnvParameterfvARB);
- SET_GetProgramLocalParameterdvARB(disp, glGetProgramLocalParameterdvARB);
- SET_GetProgramLocalParameterfvARB(disp, glGetProgramLocalParameterfvARB);
- SET_GetProgramivARB(disp, glGetProgramivARB);
- SET_GetProgramStringARB(disp, glGetProgramStringARB);
- SET_ProgramNamedParameter4fNV(disp, glProgramNamedParameter4fNV);
- SET_ProgramNamedParameter4dNV(disp, glProgramNamedParameter4dNV);
- SET_ProgramNamedParameter4fvNV(disp, glProgramNamedParameter4fvNV);
- SET_ProgramNamedParameter4dvNV(disp, glProgramNamedParameter4dvNV);
- SET_GetProgramNamedParameterfvNV(disp, glGetProgramNamedParameterfvNV);
- SET_GetProgramNamedParameterdvNV(disp, glGetProgramNamedParameterdvNV);
- SET_BindBufferARB(disp, glBindBufferARB);
- SET_BufferDataARB(disp, glBufferDataARB);
- SET_BufferSubDataARB(disp, glBufferSubDataARB);
- SET_DeleteBuffersARB(disp, glDeleteBuffersARB);
- SET_GenBuffersARB(disp, glGenBuffersARB);
- SET_GetBufferParameterivARB(disp, glGetBufferParameterivARB);
- SET_GetBufferPointervARB(disp, glGetBufferPointervARB);
- SET_GetBufferSubDataARB(disp, glGetBufferSubDataARB);
- SET_IsBufferARB(disp, glIsBufferARB);
- SET_MapBufferARB(disp, glMapBufferARB);
- SET_UnmapBufferARB(disp, glUnmapBufferARB);
- SET_DepthBoundsEXT(disp, glDepthBoundsEXT);
- SET_GenQueriesARB(disp, glGenQueriesARB);
- SET_DeleteQueriesARB(disp, glDeleteQueriesARB);
- SET_IsQueryARB(disp, glIsQueryARB);
- SET_BeginQueryARB(disp, glBeginQueryARB);
- SET_EndQueryARB(disp, glEndQueryARB);
- SET_GetQueryivARB(disp, glGetQueryivARB);
- SET_GetQueryObjectivARB(disp, glGetQueryObjectivARB);
- SET_GetQueryObjectuivARB(disp, glGetQueryObjectuivARB);
- SET_MultiModeDrawArraysIBM(disp, glMultiModeDrawArraysIBM);
- SET_MultiModeDrawElementsIBM(disp, glMultiModeDrawElementsIBM);
- SET_BlendEquationSeparateEXT(disp, glBlendEquationSeparateEXT);
- SET_DeleteObjectARB(disp, glDeleteObjectARB);
- SET_GetHandleARB(disp, glGetHandleARB);
- SET_DetachObjectARB(disp, glDetachObjectARB);
- SET_CreateShaderObjectARB(disp, glCreateShaderObjectARB);
- SET_ShaderSourceARB(disp, glShaderSourceARB);
- SET_CompileShaderARB(disp, glCompileShaderARB);
- SET_CreateProgramObjectARB(disp, glCreateProgramObjectARB);
- SET_AttachObjectARB(disp, glAttachObjectARB);
- SET_LinkProgramARB(disp, glLinkProgramARB);
- SET_UseProgramObjectARB(disp, glUseProgramObjectARB);
- SET_ValidateProgramARB(disp, glValidateProgramARB);
- SET_Uniform1fARB(disp, glUniform1fARB);
- SET_Uniform2fARB(disp, glUniform2fARB);
- SET_Uniform3fARB(disp, glUniform3fARB);
- SET_Uniform4fARB(disp, glUniform4fARB);
- SET_Uniform1iARB(disp, glUniform1iARB);
- SET_Uniform2iARB(disp, glUniform2iARB);
- SET_Uniform3iARB(disp, glUniform3iARB);
- SET_Uniform4iARB(disp, glUniform4iARB);
- SET_Uniform1fvARB(disp, glUniform1fvARB);
- SET_Uniform2fvARB(disp, glUniform2fvARB);
- SET_Uniform3fvARB(disp, glUniform3fvARB);
- SET_Uniform4fvARB(disp, glUniform4fvARB);
- SET_Uniform1ivARB(disp, glUniform1ivARB);
- SET_Uniform2ivARB(disp, glUniform2ivARB);
- SET_Uniform3ivARB(disp, glUniform3ivARB);
- SET_Uniform4ivARB(disp, glUniform4ivARB);
- SET_UniformMatrix2fvARB(disp, glUniformMatrix2fvARB);
- SET_UniformMatrix3fvARB(disp, glUniformMatrix3fvARB);
- SET_UniformMatrix4fvARB(disp, glUniformMatrix4fvARB);
- SET_GetObjectParameterfvARB(disp, glGetObjectParameterfvARB);
- SET_GetObjectParameterivARB(disp, glGetObjectParameterivARB);
- SET_GetInfoLogARB(disp, glGetInfoLogARB);
- SET_GetAttachedObjectsARB(disp, glGetAttachedObjectsARB);
- SET_GetUniformLocationARB(disp, glGetUniformLocationARB);
- SET_GetActiveUniformARB(disp, glGetActiveUniformARB);
- SET_GetUniformfvARB(disp, glGetUniformfvARB);
- SET_GetUniformivARB(disp, glGetUniformivARB);
- SET_GetShaderSourceARB(disp, glGetShaderSourceARB);
- SET_BindAttribLocationARB(disp, glBindAttribLocationARB);
- SET_GetActiveAttribARB(disp, glGetActiveAttribARB);
- SET_GetAttribLocationARB(disp, glGetAttribLocationARB);
- SET_GetVertexAttribdvNV(disp, glGetVertexAttribdvNV);
- SET_GetVertexAttribfvNV(disp, glGetVertexAttribfvNV);
- SET_GetVertexAttribivNV(disp, glGetVertexAttribivNV);
- SET_VertexAttrib1dNV(disp, glVertexAttrib1dNV);
- SET_VertexAttrib1dvNV(disp, glVertexAttrib1dvNV);
- SET_VertexAttrib1fNV(disp, glVertexAttrib1fNV);
- SET_VertexAttrib1fvNV(disp, glVertexAttrib1fvNV);
- SET_VertexAttrib1sNV(disp, glVertexAttrib1sNV);
- SET_VertexAttrib1svNV(disp, glVertexAttrib1svNV);
- SET_VertexAttrib2dNV(disp, glVertexAttrib2dNV);
- SET_VertexAttrib2dvNV(disp, glVertexAttrib2dvNV);
- SET_VertexAttrib2fNV(disp, glVertexAttrib2fNV);
- SET_VertexAttrib2fvNV(disp, glVertexAttrib2fvNV);
- SET_VertexAttrib2sNV(disp, glVertexAttrib2sNV);
- SET_VertexAttrib2svNV(disp, glVertexAttrib2svNV);
- SET_VertexAttrib3dNV(disp, glVertexAttrib3dNV);
- SET_VertexAttrib3dvNV(disp, glVertexAttrib3dvNV);
- SET_VertexAttrib3fNV(disp, glVertexAttrib3fNV);
- SET_VertexAttrib3fvNV(disp, glVertexAttrib3fvNV);
- SET_VertexAttrib3sNV(disp, glVertexAttrib3sNV);
- SET_VertexAttrib3svNV(disp, glVertexAttrib3svNV);
- SET_VertexAttrib4dNV(disp, glVertexAttrib4dNV);
- SET_VertexAttrib4dvNV(disp, glVertexAttrib4dvNV);
- SET_VertexAttrib4fNV(disp, glVertexAttrib4fNV);
- SET_VertexAttrib4fvNV(disp, glVertexAttrib4fvNV);
- SET_VertexAttrib4sNV(disp, glVertexAttrib4sNV);
- SET_VertexAttrib4svNV(disp, glVertexAttrib4svNV);
- SET_VertexAttrib4ubNV(disp, glVertexAttrib4ubNV);
- SET_VertexAttrib4ubvNV(disp, glVertexAttrib4ubvNV);
- SET_GenFragmentShadersATI(disp, glGenFragmentShadersATI);
- SET_BindFragmentShaderATI(disp, glBindFragmentShaderATI);
- SET_DeleteFragmentShaderATI(disp, glDeleteFragmentShaderATI);
- SET_BeginFragmentShaderATI(disp, glBeginFragmentShaderATI);
- SET_EndFragmentShaderATI(disp, glEndFragmentShaderATI);
- SET_PassTexCoordATI(disp, glPassTexCoordATI);
- SET_SampleMapATI(disp, glSampleMapATI);
- SET_ColorFragmentOp1ATI(disp, glColorFragmentOp1ATI);
- SET_ColorFragmentOp2ATI(disp, glColorFragmentOp2ATI);
- SET_ColorFragmentOp3ATI(disp, glColorFragmentOp3ATI);
- SET_AlphaFragmentOp1ATI(disp, glAlphaFragmentOp1ATI);
- SET_AlphaFragmentOp2ATI(disp, glAlphaFragmentOp2ATI);
- SET_AlphaFragmentOp3ATI(disp, glAlphaFragmentOp3ATI);
- SET_SetFragmentShaderConstantATI(disp, glSetFragmentShaderConstantATI);
- SET_IsRenderbufferEXT(disp, glIsRenderbufferEXT);
- SET_BindRenderbufferEXT(disp, glBindRenderbufferEXT);
- SET_DeleteRenderbuffersEXT(disp, glDeleteRenderbuffersEXT);
- SET_GenRenderbuffersEXT(disp, glGenRenderbuffersEXT);
- SET_RenderbufferStorageEXT(disp, glRenderbufferStorageEXT);
- SET_GetRenderbufferParameterivEXT(disp, glGetRenderbufferParameterivEXT);
- SET_IsFramebufferEXT(disp, glIsFramebufferEXT);
- SET_BindFramebufferEXT(disp, glBindFramebufferEXT);
- SET_DeleteFramebuffersEXT(disp, glDeleteFramebuffersEXT);
- SET_GenFramebuffersEXT(disp, glGenFramebuffersEXT);
- SET_CheckFramebufferStatusEXT(disp, glCheckFramebufferStatusEXT);
- SET_FramebufferTexture1DEXT(disp, glFramebufferTexture1DEXT);
- SET_FramebufferTexture2DEXT(disp, glFramebufferTexture2DEXT);
- SET_FramebufferTexture3DEXT(disp, glFramebufferTexture3DEXT);
- SET_FramebufferRenderbufferEXT(disp, glFramebufferRenderbufferEXT);
- SET_GetFramebufferAttachmentParameterivEXT(disp, glGetFramebufferAttachmentParameterivEXT);
- SET_GenerateMipmapEXT(disp, glGenerateMipmapEXT);
- SET_StencilFuncSeparate(disp, glStencilFuncSeparate);
- SET_StencilOpSeparate(disp, glStencilOpSeparate);
- SET_StencilMaskSeparate(disp, glStencilMaskSeparate);
- SET_GetQueryObjecti64vEXT(disp, glGetQueryObjecti64vEXT);
- SET_GetQueryObjectui64vEXT(disp, glGetQueryObjectui64vEXT);
- SET_BlitFramebufferEXT(disp, glBlitFramebufferEXT);
- SET_LoadTransposeMatrixfARB(disp, glLoadTransposeMatrixfARB);
- SET_LoadTransposeMatrixdARB(disp, glLoadTransposeMatrixdARB);
- SET_MultTransposeMatrixfARB(disp, glMultTransposeMatrixfARB);
- SET_MultTransposeMatrixdARB(disp, glMultTransposeMatrixdARB);
- SET_SampleCoverageARB(disp, glSampleCoverageARB);
- SET_DrawBuffersARB(disp, glDrawBuffersARB);
- SET_PolygonOffsetEXT(disp, glPolygonOffsetEXT);
- SET_GetTexFilterFuncSGIS(disp, glGetTexFilterFuncSGIS);
- SET_TexFilterFuncSGIS(disp, glTexFilterFuncSGIS);
- SET_GetHistogramEXT(disp, glGetHistogramEXT);
- SET_GetHistogramParameterfvEXT(disp, glGetHistogramParameterfvEXT);
- SET_GetHistogramParameterivEXT(disp, glGetHistogramParameterivEXT);
- SET_GetMinmaxEXT(disp, glGetMinmaxEXT);
- SET_GetMinmaxParameterfvEXT(disp, glGetMinmaxParameterfvEXT);
- SET_GetMinmaxParameterivEXT(disp, glGetMinmaxParameterivEXT);
- SET_GetConvolutionFilterEXT(disp, glGetConvolutionFilterEXT);
- SET_GetConvolutionParameterfvEXT(disp, glGetConvolutionParameterfvEXT);
- SET_GetConvolutionParameterivEXT(disp, glGetConvolutionParameterivEXT);
- SET_GetSeparableFilterEXT(disp, glGetSeparableFilterEXT);
- SET_GetColorTableSGI(disp, glGetColorTableSGI);
- SET_GetColorTableParameterfvSGI(disp, glGetColorTableParameterfvSGI);
- SET_GetColorTableParameterivSGI(disp, glGetColorTableParameterivSGI);
- SET_PixelTexGenSGIX(disp, glPixelTexGenSGIX);
- SET_PixelTexGenParameteriSGIS(disp, glPixelTexGenParameteriSGIS);
- SET_PixelTexGenParameterivSGIS(disp, glPixelTexGenParameterivSGIS);
- SET_PixelTexGenParameterfSGIS(disp, glPixelTexGenParameterfSGIS);
- SET_PixelTexGenParameterfvSGIS(disp, glPixelTexGenParameterfvSGIS);
- SET_GetPixelTexGenParameterivSGIS(disp, glGetPixelTexGenParameterivSGIS);
- SET_GetPixelTexGenParameterfvSGIS(disp, glGetPixelTexGenParameterfvSGIS);
- SET_TexImage4DSGIS(disp, glTexImage4DSGIS);
- SET_TexSubImage4DSGIS(disp, glTexSubImage4DSGIS);
- SET_AreTexturesResidentEXT(disp, glAreTexturesResidentEXT);
- SET_GenTexturesEXT(disp, glGenTexturesEXT);
- SET_IsTextureEXT(disp, glIsTextureEXT);
- SET_DetailTexFuncSGIS(disp, glDetailTexFuncSGIS);
- SET_GetDetailTexFuncSGIS(disp, glGetDetailTexFuncSGIS);
- SET_SharpenTexFuncSGIS(disp, glSharpenTexFuncSGIS);
- SET_GetSharpenTexFuncSGIS(disp, glGetSharpenTexFuncSGIS);
- SET_SampleMaskSGIS(disp, glSampleMaskSGIS);
- SET_SamplePatternSGIS(disp, glSamplePatternSGIS);
- SET_ColorPointerEXT(disp, glColorPointerEXT);
- SET_EdgeFlagPointerEXT(disp, glEdgeFlagPointerEXT);
- SET_IndexPointerEXT(disp, glIndexPointerEXT);
- SET_NormalPointerEXT(disp, glNormalPointerEXT);
- SET_TexCoordPointerEXT(disp, glTexCoordPointerEXT);
- SET_VertexPointerEXT(disp, glVertexPointerEXT);
- SET_SpriteParameterfSGIX(disp, glSpriteParameterfSGIX);
- SET_SpriteParameterfvSGIX(disp, glSpriteParameterfvSGIX);
- SET_SpriteParameteriSGIX(disp, glSpriteParameteriSGIX);
- SET_SpriteParameterivSGIX(disp, glSpriteParameterivSGIX);
- SET_PointParameterfEXT(disp, glPointParameterfEXT);
- SET_PointParameterfvEXT(disp, glPointParameterfvEXT);
- SET_GetInstrumentsSGIX(disp, glGetInstrumentsSGIX);
- SET_InstrumentsBufferSGIX(disp, glInstrumentsBufferSGIX);
- SET_PollInstrumentsSGIX(disp, glPollInstrumentsSGIX);
- SET_ReadInstrumentsSGIX(disp, glReadInstrumentsSGIX);
- SET_StartInstrumentsSGIX(disp, glStartInstrumentsSGIX);
- SET_StopInstrumentsSGIX(disp, glStopInstrumentsSGIX);
- SET_FrameZoomSGIX(disp, glFrameZoomSGIX);
- SET_TagSampleBufferSGIX(disp, glTagSampleBufferSGIX);
- SET_ReferencePlaneSGIX(disp, glReferencePlaneSGIX);
- SET_FlushRasterSGIX(disp, glFlushRasterSGIX);
- SET_GetListParameterfvSGIX(disp, glGetListParameterfvSGIX);
- SET_GetListParameterivSGIX(disp, glGetListParameterivSGIX);
- SET_ListParameterfSGIX(disp, glListParameterfSGIX);
- SET_ListParameterfvSGIX(disp, glListParameterfvSGIX);
- SET_ListParameteriSGIX(disp, glListParameteriSGIX);
- SET_ListParameterivSGIX(disp, glListParameterivSGIX);
- SET_FragmentColorMaterialSGIX(disp, glFragmentColorMaterialSGIX);
- SET_FragmentLightfSGIX(disp, glFragmentLightfSGIX);
- SET_FragmentLightfvSGIX(disp, glFragmentLightfvSGIX);
- SET_FragmentLightiSGIX(disp, glFragmentLightiSGIX);
- SET_FragmentLightivSGIX(disp, glFragmentLightivSGIX);
- SET_FragmentLightModelfSGIX(disp, glFragmentLightModelfSGIX);
- SET_FragmentLightModelfvSGIX(disp, glFragmentLightModelfvSGIX);
- SET_FragmentLightModeliSGIX(disp, glFragmentLightModeliSGIX);
- SET_FragmentLightModelivSGIX(disp, glFragmentLightModelivSGIX);
- SET_FragmentMaterialfSGIX(disp, glFragmentMaterialfSGIX);
- SET_FragmentMaterialfvSGIX(disp, glFragmentMaterialfvSGIX);
- SET_FragmentMaterialiSGIX(disp, glFragmentMaterialiSGIX);
- SET_FragmentMaterialivSGIX(disp, glFragmentMaterialivSGIX);
- SET_GetFragmentLightfvSGIX(disp, glGetFragmentLightfvSGIX);
- SET_GetFragmentLightivSGIX(disp, glGetFragmentLightivSGIX);
- SET_GetFragmentMaterialfvSGIX(disp, glGetFragmentMaterialfvSGIX);
- SET_GetFragmentMaterialivSGIX(disp, glGetFragmentMaterialivSGIX);
- SET_LightEnviSGIX(disp, glLightEnviSGIX);
- SET_VertexWeightfEXT(disp, glVertexWeightfEXT);
- SET_VertexWeightfvEXT(disp, glVertexWeightfvEXT);
- SET_VertexWeightPointerEXT(disp, glVertexWeightPointerEXT);
- SET_FlushVertexArrayRangeNV(disp, glFlushVertexArrayRangeNV);
- SET_VertexArrayRangeNV(disp, glVertexArrayRangeNV);
- SET_CombinerParameterfvNV(disp, glCombinerParameterfvNV);
- SET_CombinerParameterfNV(disp, glCombinerParameterfNV);
- SET_CombinerParameterivNV(disp, glCombinerParameterivNV);
- SET_CombinerParameteriNV(disp, glCombinerParameteriNV);
- SET_CombinerInputNV(disp, glCombinerInputNV);
- SET_CombinerOutputNV(disp, glCombinerOutputNV);
- SET_FinalCombinerInputNV(disp, glFinalCombinerInputNV);
- SET_GetCombinerInputParameterfvNV(disp, glGetCombinerInputParameterfvNV);
- SET_GetCombinerInputParameterivNV(disp, glGetCombinerInputParameterivNV);
- SET_GetCombinerOutputParameterfvNV(disp, glGetCombinerOutputParameterfvNV);
- SET_GetCombinerOutputParameterivNV(disp, glGetCombinerOutputParameterivNV);
- SET_GetFinalCombinerInputParameterfvNV(disp, glGetFinalCombinerInputParameterfvNV);
- SET_GetFinalCombinerInputParameterivNV(disp, glGetFinalCombinerInputParameterivNV);
- SET_ResizeBuffersMESA(disp, glResizeBuffersMESA);
- SET_WindowPos2dMESA(disp, glWindowPos2dMESA);
- SET_WindowPos2dvMESA(disp, glWindowPos2dvMESA);
- SET_WindowPos2fMESA(disp, glWindowPos2fMESA);
- SET_WindowPos2fvMESA(disp, glWindowPos2fvMESA);
- SET_WindowPos2iMESA(disp, glWindowPos2iMESA);
- SET_WindowPos2ivMESA(disp, glWindowPos2ivMESA);
- SET_WindowPos2sMESA(disp, glWindowPos2sMESA);
- SET_WindowPos2svMESA(disp, glWindowPos2svMESA);
- SET_WindowPos3dMESA(disp, glWindowPos3dMESA);
- SET_WindowPos3dvMESA(disp, glWindowPos3dvMESA);
- SET_WindowPos3fMESA(disp, glWindowPos3fMESA);
- SET_WindowPos3fvMESA(disp, glWindowPos3fvMESA);
- SET_WindowPos3iMESA(disp, glWindowPos3iMESA);
- SET_WindowPos3ivMESA(disp, glWindowPos3ivMESA);
- SET_WindowPos3sMESA(disp, glWindowPos3sMESA);
- SET_WindowPos3svMESA(disp, glWindowPos3svMESA);
- SET_WindowPos4dMESA(disp, glWindowPos4dMESA);
- SET_WindowPos4dvMESA(disp, glWindowPos4dvMESA);
- SET_WindowPos4fMESA(disp, glWindowPos4fMESA);
- SET_WindowPos4fvMESA(disp, glWindowPos4fvMESA);
- SET_WindowPos4iMESA(disp, glWindowPos4iMESA);
- SET_WindowPos4ivMESA(disp, glWindowPos4ivMESA);
- SET_WindowPos4sMESA(disp, glWindowPos4sMESA);
- SET_WindowPos4svMESA(disp, glWindowPos4svMESA);
- SET_BlendFuncSeparateEXT(disp, glBlendFuncSeparateEXT);
- SET_IndexMaterialEXT(disp, glIndexMaterialEXT);
- SET_IndexFuncEXT(disp, glIndexFuncEXT);
- SET_LockArraysEXT(disp, glLockArraysEXT);
- SET_UnlockArraysEXT(disp, glUnlockArraysEXT);
- SET_CullParameterdvEXT(disp, glCullParameterdvEXT);
- SET_CullParameterfvEXT(disp, glCullParameterfvEXT);
- SET_HintPGI(disp, glHintPGI);
- SET_FogCoordfEXT(disp, glFogCoordfEXT);
- SET_FogCoordfvEXT(disp, glFogCoordfvEXT);
- SET_FogCoorddEXT(disp, glFogCoorddEXT);
- SET_FogCoorddvEXT(disp, glFogCoorddvEXT);
- SET_FogCoordPointerEXT(disp, glFogCoordPointerEXT);
- SET_GetColorTableEXT(disp, glGetColorTableEXT);
- SET_GetColorTableParameterivEXT(disp, glGetColorTableParameterivEXT);
- SET_GetColorTableParameterfvEXT(disp, glGetColorTableParameterfvEXT);
- SET_TbufferMask3DFX(disp, glTbufferMask3DFX);
- SET_CompressedTexImage3DARB(disp, glCompressedTexImage3DARB);
- SET_CompressedTexImage2DARB(disp, glCompressedTexImage2DARB);
- SET_CompressedTexImage1DARB(disp, glCompressedTexImage1DARB);
- SET_CompressedTexSubImage3DARB(disp, glCompressedTexSubImage3DARB);
- SET_CompressedTexSubImage2DARB(disp, glCompressedTexSubImage2DARB);
- SET_CompressedTexSubImage1DARB(disp, glCompressedTexSubImage1DARB);
- SET_GetCompressedTexImageARB(disp, glGetCompressedTexImageARB);
- SET_SecondaryColor3bEXT(disp, glSecondaryColor3bEXT);
- SET_SecondaryColor3bvEXT(disp, glSecondaryColor3bvEXT);
- SET_SecondaryColor3dEXT(disp, glSecondaryColor3dEXT);
- SET_SecondaryColor3dvEXT(disp, glSecondaryColor3dvEXT);
- SET_SecondaryColor3fEXT(disp, glSecondaryColor3fEXT);
- SET_SecondaryColor3fvEXT(disp, glSecondaryColor3fvEXT);
- SET_SecondaryColor3iEXT(disp, glSecondaryColor3iEXT);
- SET_SecondaryColor3ivEXT(disp, glSecondaryColor3ivEXT);
- SET_SecondaryColor3sEXT(disp, glSecondaryColor3sEXT);
- SET_SecondaryColor3svEXT(disp, glSecondaryColor3svEXT);
- SET_SecondaryColor3ubEXT(disp, glSecondaryColor3ubEXT);
- SET_SecondaryColor3ubvEXT(disp, glSecondaryColor3ubvEXT);
- SET_SecondaryColor3uiEXT(disp, glSecondaryColor3uiEXT);
- SET_SecondaryColor3uivEXT(disp, glSecondaryColor3uivEXT);
- SET_SecondaryColor3usEXT(disp, glSecondaryColor3usEXT);
- SET_SecondaryColor3usvEXT(disp, glSecondaryColor3usvEXT);
- SET_SecondaryColorPointerEXT(disp, glSecondaryColorPointerEXT);
- SET_AreProgramsResidentNV(disp, glAreProgramsResidentNV);
- SET_BindProgramNV(disp, glBindProgramNV);
- SET_DeleteProgramsNV(disp, glDeleteProgramsNV);
- SET_ExecuteProgramNV(disp, glExecuteProgramNV);
- SET_GenProgramsNV(disp, glGenProgramsNV);
- SET_GetProgramParameterdvNV(disp, glGetProgramParameterdvNV);
- SET_GetProgramParameterfvNV(disp, glGetProgramParameterfvNV);
- SET_GetProgramivNV(disp, glGetProgramivNV);
- SET_GetProgramStringNV(disp, glGetProgramStringNV);
- SET_GetTrackMatrixivNV(disp, glGetTrackMatrixivNV);
- SET_GetVertexAttribdvARB(disp, glGetVertexAttribdvARB);
- SET_GetVertexAttribfvARB(disp, glGetVertexAttribfvARB);
- SET_GetVertexAttribivARB(disp, glGetVertexAttribivARB);
- SET_GetVertexAttribPointervNV(disp, glGetVertexAttribPointervNV);
- SET_IsProgramNV(disp, glIsProgramNV);
- SET_LoadProgramNV(disp, glLoadProgramNV);
- SET_ProgramParameter4dNV(disp, glProgramParameter4dNV);
- SET_ProgramParameter4dvNV(disp, glProgramParameter4dvNV);
- SET_ProgramParameter4fNV(disp, glProgramParameter4fNV);
- SET_ProgramParameter4fvNV(disp, glProgramParameter4fvNV);
- SET_ProgramParameters4dvNV(disp, glProgramParameters4dvNV);
- SET_ProgramParameters4fvNV(disp, glProgramParameters4fvNV);
- SET_RequestResidentProgramsNV(disp, glRequestResidentProgramsNV);
- SET_TrackMatrixNV(disp, glTrackMatrixNV);
- SET_VertexAttribPointerNV(disp, glVertexAttribPointerNV);
- SET_VertexAttrib1dARB(disp, glVertexAttrib1dARB);
- SET_VertexAttrib1dvARB(disp, glVertexAttrib1dvARB);
- SET_VertexAttrib1fARB(disp, glVertexAttrib1fARB);
- SET_VertexAttrib1fvARB(disp, glVertexAttrib1fvARB);
- SET_VertexAttrib1sARB(disp, glVertexAttrib1sARB);
- SET_VertexAttrib1svARB(disp, glVertexAttrib1svARB);
- SET_VertexAttrib2dARB(disp, glVertexAttrib2dARB);
- SET_VertexAttrib2dvARB(disp, glVertexAttrib2dvARB);
- SET_VertexAttrib2fARB(disp, glVertexAttrib2fARB);
- SET_VertexAttrib2fvARB(disp, glVertexAttrib2fvARB);
- SET_VertexAttrib2sARB(disp, glVertexAttrib2sARB);
- SET_VertexAttrib2svARB(disp, glVertexAttrib2svARB);
- SET_VertexAttrib3dARB(disp, glVertexAttrib3dARB);
- SET_VertexAttrib3dvARB(disp, glVertexAttrib3dvARB);
- SET_VertexAttrib3fARB(disp, glVertexAttrib3fARB);
- SET_VertexAttrib3fvARB(disp, glVertexAttrib3fvARB);
- SET_VertexAttrib3sARB(disp, glVertexAttrib3sARB);
- SET_VertexAttrib3svARB(disp, glVertexAttrib3svARB);
+  SET_VertexAttrib4NusvARB(disp, glVertexAttrib4NusvARB);
+  SET_VertexAttrib4bvARB(disp, glVertexAttrib4bvARB);
  SET_VertexAttrib4dARB(disp, glVertexAttrib4dARB);
  SET_VertexAttrib4dvARB(disp, glVertexAttrib4dvARB);
  SET_VertexAttrib4fARB(disp, glVertexAttrib4fARB);
  SET_VertexAttrib4fvARB(disp, glVertexAttrib4fvARB);
+  SET_VertexAttrib4ivARB(disp, glVertexAttrib4ivARB);
  SET_VertexAttrib4sARB(disp, glVertexAttrib4sARB);
  SET_VertexAttrib4svARB(disp, glVertexAttrib4svARB);
- SET_VertexAttrib4NubARB(disp, glVertexAttrib4NubARB);
- SET_VertexAttrib4NubvARB(disp, glVertexAttrib4NubvARB);
- SET_VertexAttribs1dvNV(disp, glVertexAttribs1dvNV);
- SET_VertexAttribs1fvNV(disp, glVertexAttribs1fvNV);
- SET_VertexAttribs1svNV(disp, glVertexAttribs1svNV);
- SET_VertexAttribs2dvNV(disp, glVertexAttribs2dvNV);
- SET_VertexAttribs2fvNV(disp, glVertexAttribs2fvNV);
- SET_VertexAttribs2svNV(disp, glVertexAttribs2svNV);
- SET_VertexAttribs3dvNV(disp, glVertexAttribs3dvNV);
- SET_VertexAttribs3fvNV(disp, glVertexAttribs3fvNV);
- SET_VertexAttribs3svNV(disp, glVertexAttribs3svNV);
- SET_VertexAttribs4dvNV(disp, glVertexAttribs4dvNV);
- SET_VertexAttribs4fvNV(disp, glVertexAttribs4fvNV);
- SET_VertexAttribs4svNV(disp, glVertexAttribs4svNV);
- SET_VertexAttribs4ubvNV(disp, glVertexAttribs4ubvNV);
- SET_PointParameteriNV(disp, glPointParameteriNV);
- SET_PointParameterivNV(disp, glPointParameterivNV);
- SET_MultiDrawArraysEXT(disp, glMultiDrawArraysEXT);
- SET_MultiDrawElementsEXT(disp, glMultiDrawElementsEXT);
- SET_ActiveStencilFaceEXT(disp, glActiveStencilFaceEXT);
- SET_DeleteFencesNV(disp, glDeleteFencesNV);
- SET_GenFencesNV(disp, glGenFencesNV);
- SET_IsFenceNV(disp, glIsFenceNV);
- SET_TestFenceNV(disp, glTestFenceNV);
- SET_GetFenceivNV(disp, glGetFenceivNV);
- SET_FinishFenceNV(disp, glFinishFenceNV);
- SET_SetFenceNV(disp, glSetFenceNV);
- SET_VertexAttrib4bvARB(disp, glVertexAttrib4bvARB);
- SET_VertexAttrib4ivARB(disp, glVertexAttrib4ivARB);
  SET_VertexAttrib4ubvARB(disp, glVertexAttrib4ubvARB);
- SET_VertexAttrib4usvARB(disp, glVertexAttrib4usvARB);
  SET_VertexAttrib4uivARB(disp, glVertexAttrib4uivARB);
- SET_VertexAttrib4NbvARB(disp, glVertexAttrib4NbvARB);
- SET_VertexAttrib4NsvARB(disp, glVertexAttrib4NsvARB);
- SET_VertexAttrib4NivARB(disp, glVertexAttrib4NivARB);
- SET_VertexAttrib4NusvARB(disp, glVertexAttrib4NusvARB);
- SET_VertexAttrib4NuivARB(disp, glVertexAttrib4NuivARB);
+  SET_VertexAttrib4usvARB(disp, glVertexAttrib4usvARB);
  SET_VertexAttribPointerARB(disp, glVertexAttribPointerARB);
- SET_EnableVertexAttribArrayARB(disp, glEnableVertexAttribArrayARB);
- SET_DisableVertexAttribArrayARB(disp, glDisableVertexAttribArrayARB);
- SET_ProgramStringARB(disp, glProgramStringARB);
- SET_ProgramEnvParameter4dARB(disp, glProgramEnvParameter4dARB);
- SET_ProgramEnvParameter4dvARB(disp, glProgramEnvParameter4dvARB);
- SET_ProgramEnvParameter4fARB(disp, glProgramEnvParameter4fARB);
- SET_ProgramEnvParameter4fvARB(disp, glProgramEnvParameter4fvARB);
- SET_ProgramLocalParameter4dARB(disp, glProgramLocalParameter4dARB);
- SET_ProgramLocalParameter4dvARB(disp, glProgramLocalParameter4dvARB);
- SET_ProgramLocalParameter4fARB(disp, glProgramLocalParameter4fARB);
- SET_ProgramLocalParameter4fvARB(disp, glProgramLocalParameter4fvARB);
- SET_GetProgramEnvParameterdvARB(disp, glGetProgramEnvParameterdvARB);
- SET_GetProgramEnvParameterfvARB(disp, glGetProgramEnvParameterfvARB);
- SET_GetProgramLocalParameterdvARB(disp, glGetProgramLocalParameterdvARB);
- SET_GetProgramLocalParameterfvARB(disp, glGetProgramLocalParameterfvARB);
- SET_GetProgramivARB(disp, glGetProgramivARB);
- SET_GetProgramStringARB(disp, glGetProgramStringARB);
- SET_ProgramNamedParameter4fNV(disp, glProgramNamedParameter4fNV);
- SET_ProgramNamedParameter4dNV(disp, glProgramNamedParameter4dNV);
- SET_ProgramNamedParameter4fvNV(disp, glProgramNamedParameter4fvNV);
- SET_ProgramNamedParameter4dvNV(disp, glProgramNamedParameter4dvNV);
- SET_GetProgramNamedParameterfvNV(disp, glGetProgramNamedParameterfvNV);
- SET_GetProgramNamedParameterdvNV(disp, glGetProgramNamedParameterdvNV);
- SET_BindBufferARB(disp, glBindBufferARB);
- SET_BufferDataARB(disp, glBufferDataARB);
- SET_BufferSubDataARB(disp, glBufferSubDataARB);
- SET_DeleteBuffersARB(disp, glDeleteBuffersARB);
- SET_GenBuffersARB(disp, glGenBuffersARB);
- SET_GetBufferParameterivARB(disp, glGetBufferParameterivARB);
- SET_GetBufferPointervARB(disp, glGetBufferPointervARB);
- SET_GetBufferSubDataARB(disp, glGetBufferSubDataARB);
- SET_IsBufferARB(disp, glIsBufferARB);
- SET_MapBufferARB(disp, glMapBufferARB);
- SET_UnmapBufferARB(disp, glUnmapBufferARB);
- SET_DepthBoundsEXT(disp, glDepthBoundsEXT);
- SET_GenQueriesARB(disp, glGenQueriesARB);
- SET_DeleteQueriesARB(disp, glDeleteQueriesARB);
- SET_IsQueryARB(disp, glIsQueryARB);
- SET_BeginQueryARB(disp, glBeginQueryARB);
- SET_EndQueryARB(disp, glEndQueryARB);
- SET_GetQueryivARB(disp, glGetQueryivARB);
- SET_GetQueryObjectivARB(disp, glGetQueryObjectivARB);
- SET_GetQueryObjectuivARB(disp, glGetQueryObjectuivARB);
- SET_MultiModeDrawArraysIBM(disp, glMultiModeDrawArraysIBM);
- SET_MultiModeDrawElementsIBM(disp, glMultiModeDrawElementsIBM);
- SET_BlendEquationSeparateEXT(disp, glBlendEquationSeparateEXT);
- SET_DeleteObjectARB(disp, glDeleteObjectARB);
- SET_GetHandleARB(disp, glGetHandleARB);
- SET_DetachObjectARB(disp, glDetachObjectARB);
- SET_CreateShaderObjectARB(disp, glCreateShaderObjectARB);
- SET_ShaderSourceARB(disp, glShaderSourceARB);
- SET_CompileShaderARB(disp, glCompileShaderARB);
- SET_CreateProgramObjectARB(disp, glCreateProgramObjectARB);
- SET_AttachObjectARB(disp, glAttachObjectARB);
- SET_LinkProgramARB(disp, glLinkProgramARB);
- SET_UseProgramObjectARB(disp, glUseProgramObjectARB);
- SET_ValidateProgramARB(disp, glValidateProgramARB);
- SET_Uniform1fARB(disp, glUniform1fARB);
- SET_Uniform2fARB(disp, glUniform2fARB);
- SET_Uniform3fARB(disp, glUniform3fARB);
- SET_Uniform4fARB(disp, glUniform4fARB);
- SET_Uniform1iARB(disp, glUniform1iARB);
- SET_Uniform2iARB(disp, glUniform2iARB);
- SET_Uniform3iARB(disp, glUniform3iARB);
- SET_Uniform4iARB(disp, glUniform4iARB);
- SET_Uniform1fvARB(disp, glUniform1fvARB);
- SET_Uniform2fvARB(disp, glUniform2fvARB);
- SET_Uniform3fvARB(disp, glUniform3fvARB);
- SET_Uniform4fvARB(disp, glUniform4fvARB);
- SET_Uniform1ivARB(disp, glUniform1ivARB);
- SET_Uniform2ivARB(disp, glUniform2ivARB);
- SET_Uniform3ivARB(disp, glUniform3ivARB);
- SET_Uniform4ivARB(disp, glUniform4ivARB);
- SET_UniformMatrix2fvARB(disp, glUniformMatrix2fvARB);
- SET_UniformMatrix3fvARB(disp, glUniformMatrix3fvARB);
- SET_UniformMatrix4fvARB(disp, glUniformMatrix4fvARB);
- SET_GetObjectParameterfvARB(disp, glGetObjectParameterfvARB);
- SET_GetObjectParameterivARB(disp, glGetObjectParameterivARB);
- SET_GetInfoLogARB(disp, glGetInfoLogARB);
- SET_GetAttachedObjectsARB(disp, glGetAttachedObjectsARB);
- SET_GetUniformLocationARB(disp, glGetUniformLocationARB);
- SET_GetActiveUniformARB(disp, glGetActiveUniformARB);
- SET_GetUniformfvARB(disp, glGetUniformfvARB);
- SET_GetUniformivARB(disp, glGetUniformivARB);
- SET_GetShaderSourceARB(disp, glGetShaderSourceARB);
- SET_BindAttribLocationARB(disp, glBindAttribLocationARB);
- SET_GetActiveAttribARB(disp, glGetActiveAttribARB);
- SET_GetAttribLocationARB(disp, glGetAttribLocationARB);
- SET_GetVertexAttribdvNV(disp, glGetVertexAttribdvNV);
- SET_GetVertexAttribfvNV(disp, glGetVertexAttribfvNV);
- SET_GetVertexAttribivNV(disp, glGetVertexAttribivNV);
- SET_VertexAttrib1dNV(disp, glVertexAttrib1dNV);
- SET_VertexAttrib1dvNV(disp, glVertexAttrib1dvNV);
- SET_VertexAttrib1fNV(disp, glVertexAttrib1fNV);
- SET_VertexAttrib1fvNV(disp, glVertexAttrib1fvNV);
- SET_VertexAttrib1sNV(disp, glVertexAttrib1sNV);
- SET_VertexAttrib1svNV(disp, glVertexAttrib1svNV);
- SET_VertexAttrib2dNV(disp, glVertexAttrib2dNV);
- SET_VertexAttrib2dvNV(disp, glVertexAttrib2dvNV);
- SET_VertexAttrib2fNV(disp, glVertexAttrib2fNV);
- SET_VertexAttrib2fvNV(disp, glVertexAttrib2fvNV);
- SET_VertexAttrib2sNV(disp, glVertexAttrib2sNV);
- SET_VertexAttrib2svNV(disp, glVertexAttrib2svNV);
- SET_VertexAttrib3dNV(disp, glVertexAttrib3dNV);
- SET_VertexAttrib3dvNV(disp, glVertexAttrib3dvNV);
- SET_VertexAttrib3fNV(disp, glVertexAttrib3fNV);
- SET_VertexAttrib3fvNV(disp, glVertexAttrib3fvNV);
- SET_VertexAttrib3sNV(disp, glVertexAttrib3sNV);
- SET_VertexAttrib3svNV(disp, glVertexAttrib3svNV);
- SET_VertexAttrib4dNV(disp, glVertexAttrib4dNV);
- SET_VertexAttrib4dvNV(disp, glVertexAttrib4dvNV);
- SET_VertexAttrib4fNV(disp, glVertexAttrib4fNV);
- SET_VertexAttrib4fvNV(disp, glVertexAttrib4fvNV);
- SET_VertexAttrib4sNV(disp, glVertexAttrib4sNV);
- SET_VertexAttrib4svNV(disp, glVertexAttrib4svNV);
- SET_VertexAttrib4ubNV(disp, glVertexAttrib4ubNV);
- SET_VertexAttrib4ubvNV(disp, glVertexAttrib4ubvNV);
- SET_GenFragmentShadersATI(disp, glGenFragmentShadersATI);
- SET_BindFragmentShaderATI(disp, glBindFragmentShaderATI);
- SET_DeleteFragmentShaderATI(disp, glDeleteFragmentShaderATI);
- SET_BeginFragmentShaderATI(disp, glBeginFragmentShaderATI);
- SET_EndFragmentShaderATI(disp, glEndFragmentShaderATI);
- SET_PassTexCoordATI(disp, glPassTexCoordATI);
- SET_SampleMapATI(disp, glSampleMapATI);
- SET_ColorFragmentOp1ATI(disp, glColorFragmentOp1ATI);
- SET_ColorFragmentOp2ATI(disp, glColorFragmentOp2ATI);
- SET_ColorFragmentOp3ATI(disp, glColorFragmentOp3ATI);
- SET_AlphaFragmentOp1ATI(disp, glAlphaFragmentOp1ATI);
- SET_AlphaFragmentOp2ATI(disp, glAlphaFragmentOp2ATI);
- SET_AlphaFragmentOp3ATI(disp, glAlphaFragmentOp3ATI);
- SET_SetFragmentShaderConstantATI(disp, glSetFragmentShaderConstantATI);
- SET_IsRenderbufferEXT(disp, glIsRenderbufferEXT);
- SET_BindRenderbufferEXT(disp, glBindRenderbufferEXT);
- SET_DeleteRenderbuffersEXT(disp, glDeleteRenderbuffersEXT);
- SET_GenRenderbuffersEXT(disp, glGenRenderbuffersEXT);
- SET_RenderbufferStorageEXT(disp, glRenderbufferStorageEXT);
- SET_GetRenderbufferParameterivEXT(disp, glGetRenderbufferParameterivEXT);
- SET_IsFramebufferEXT(disp, glIsFramebufferEXT);
- SET_BindFramebufferEXT(disp, glBindFramebufferEXT);
- SET_DeleteFramebuffersEXT(disp, glDeleteFramebuffersEXT);
- SET_GenFramebuffersEXT(disp, glGenFramebuffersEXT);
- SET_CheckFramebufferStatusEXT(disp, glCheckFramebufferStatusEXT);
- SET_FramebufferTexture1DEXT(disp, glFramebufferTexture1DEXT);
- SET_FramebufferTexture2DEXT(disp, glFramebufferTexture2DEXT);
- SET_FramebufferTexture3DEXT(disp, glFramebufferTexture3DEXT);
- SET_FramebufferRenderbufferEXT(disp, glFramebufferRenderbufferEXT);
- SET_GetFramebufferAttachmentParameterivEXT(disp, glGetFramebufferAttachmentParameterivEXT);
- SET_GenerateMipmapEXT(disp, glGenerateMipmapEXT);
- SET_StencilFuncSeparate(disp, glStencilFuncSeparate);
- SET_StencilOpSeparate(disp, glStencilOpSeparate);
- SET_StencilMaskSeparate(disp, glStencilMaskSeparate);
- SET_GetQueryObjecti64vEXT(disp, glGetQueryObjecti64vEXT);
- SET_GetQueryObjectui64vEXT(disp, glGetQueryObjectui64vEXT);
- SET_BlitFramebufferEXT(disp, glBlitFramebufferEXT); */
-
+  SET_VertexPointer(disp, glVertexPointer);
+//  SET_VertexPointerEXT(disp, glVertexPointerEXT);
+  SET_Viewport(disp, glViewport);
+  SET_WindowPos2dMESA(disp, glWindowPos2dARB);
+  SET_WindowPos2dvMESA(disp, glWindowPos2dvARB);
+  SET_WindowPos2fMESA(disp, glWindowPos2fARB);
+  SET_WindowPos2fvMESA(disp, glWindowPos2fvARB);
+  SET_WindowPos2iMESA(disp, glWindowPos2iARB);
+  SET_WindowPos2ivMESA(disp, glWindowPos2ivARB);
+  SET_WindowPos2sMESA(disp, glWindowPos2sARB);
+  SET_WindowPos2svMESA(disp, glWindowPos2svARB);
+  SET_WindowPos3dMESA(disp, glWindowPos3dARB);
+  SET_WindowPos3dvMESA(disp, glWindowPos3dvARB);
+  SET_WindowPos3fMESA(disp, glWindowPos3fARB);
+  SET_WindowPos3fvMESA(disp, glWindowPos3fvARB);
+  SET_WindowPos3iMESA(disp, glWindowPos3iARB);
+  SET_WindowPos3ivMESA(disp, glWindowPos3ivARB);
+  SET_WindowPos3sMESA(disp, glWindowPos3sARB);
+  SET_WindowPos3svMESA(disp, glWindowPos3svARB);
+//SET_WindowPos4dMESA(disp, glWindowPos4dMESA);
+//SET_WindowPos4dvMESA(disp, glWindowPos4dvMESA);
+//SET_WindowPos4fMESA(disp, glWindowPos4fMESA);
+//SET_WindowPos4fvMESA(disp, glWindowPos4fvMESA);
+//SET_WindowPos4iMESA(disp, glWindowPos4iMESA);
+//SET_WindowPos4ivMESA(disp, glWindowPos4ivMESA);
+//SET_WindowPos4sMESA(disp, glWindowPos4sMESA);
+//SET_WindowPos4svMESA(disp, glWindowPos4svMESA);
 }
