diff --git a/randr/rrtransform.c b/randr/rrtransform.c
index 06f6298..7fd4fa0 100644
--- a/randr/rrtransform.c
+++ b/randr/rrtransform.c
@@ -38,7 +38,7 @@ void
 RRTransformFini (RRTransformPtr transform)
 {
     if (transform->params)
-	xfree (transform->params);
+	free(transform->params);
 }
 
 Bool
@@ -75,7 +75,7 @@ RRTransformSetFilter (RRTransformPtr	dst,
 
     if (nparams)
     {
-	new_params = xalloc (nparams * sizeof (xFixed));
+	new_params = malloc(nparams * sizeof (xFixed));
 	if (!new_params)
 	    return FALSE;
 	memcpy (new_params, params, nparams * sizeof (xFixed));
@@ -83,7 +83,7 @@ RRTransformSetFilter (RRTransformPtr	dst,
     else
 	new_params = NULL;
     if (dst->params)
-	xfree (dst->params);
+	free(dst->params);
     dst->filter = filter;
     dst->params = new_params;
     dst->nparams = nparams;
