diff --git a/mfb/mfbcmap.c b/mfb/mfbcmap.c
index 15c7324..a16964e 100644
--- a/mfb/mfbcmap.c
+++ b/mfb/mfbcmap.c
@@ -46,6 +46,10 @@ SOFTWARE.
 
 ******************************************************************/
 /* $Xorg: mfbcmap.c,v 1.4 2001/02/09 02:05:18 xorgcvs Exp $ */
+#ifdef HAVE_DIX_CONFIG_H
+#include <dix-config.h>
+#endif
+
 #include <X11/X.h>
 #include "scrnintstr.h"
 #include "colormapst.h"
