diff --git a/mi/micursor.c b/mi/micursor.c
index 8339e22..ce5778c 100644
--- a/mi/micursor.c
+++ b/mi/micursor.c
@@ -46,6 +46,10 @@ SOFTWARE.
 
 ******************************************************************/
 /* $Xorg: micursor.c,v 1.4 2001/02/09 02:05:20 xorgcvs Exp $ */
+#ifdef HAVE_DIX_CONFIG_H
+#include <dix-config.h>
+#endif
+
 #include "scrnintstr.h"
 #include "cursor.h"
 #include <X11/misc.h>
