diff --git a/dix/getevents.c b/dix/getevents.c
index 7c018c1..a08d4b2 100644
--- a/dix/getevents.c
+++ b/dix/getevents.c
@@ -175,8 +175,8 @@ set_valuators(DeviceIntPtr dev, DeviceEvent* event, int first_valuator,
         SetBit(event->valuators.mask, i);
         if (dev->valuator->mode == Absolute)
             SetBit(event->valuators.mode, i);
-        event->valuators.data_frac[first_valuator + i] =
-            dev->last.remainder[first_valuator + i] * (1 << 16) * (1 << 16);
+        event->valuators.data_frac[i] =
+            dev->last.remainder[i] * (1 << 16) * (1 << 16);
     }
 
     memcpy(&event->valuators.data[first_valuator],
