diff --git a/exa/exa_priv.h b/exa/exa_priv.h
index 90af553..03ee0ed 100644
--- a/exa/exa_priv.h
+++ b/exa/exa_priv.h
@@ -137,14 +137,6 @@ extern int exaPixmapPrivateIndex;
 /** Align an offset to a power-of-two alignment */
 #define EXA_ALIGN2(offset, align) (((offset) + (align) - 1) & ~((align) - 1))
 
-/**
- * Returns TRUE if the given planemask covers all the significant bits in the
- * pixel values for pDrawable.
- */
-#define EXA_PM_IS_SOLID(_pDrawable, _pm) \
-	(((_pm) & FbFullMask((_pDrawable)->depth)) == \
-	 FbFullMask((_pDrawable)->depth))
-
 #define EXA_PIXMAP_SCORE_MOVE_IN    10
 #define EXA_PIXMAP_SCORE_MAX	    20
 #define EXA_PIXMAP_SCORE_MOVE_OUT   -10
