diff --git a/hw/xfree86/exa/exaPriv.h b/hw/xfree86/exa/exaPriv.h
index ca1cdb1..db83d1b 100644
--- a/hw/xfree86/exa/exaPriv.h
+++ b/hw/xfree86/exa/exaPriv.h
@@ -118,6 +118,10 @@ typedef struct {
     int		    score;
     int		    devKind;
     DevUnion	    devPrivate;
+
+    /* True if the in-screen copy has been modified compared to the
+     * system-memory copy.
+     */
     Bool	    dirty;
     unsigned int    size;
 } ExaPixmapPrivRec, *ExaPixmapPrivPtr;
