diff --git a/Xext/sync.c b/Xext/sync.c
index c88d81c..e1383ce 100644
--- a/Xext/sync.c
+++ b/Xext/sync.c
@@ -302,20 +302,31 @@ SyncDeleteTriggerFromCounter(pTrigger)
     SyncTrigger *pTrigger;
 {
     SyncTriggerList *pCur;
+    SyncTriggerList *pPrev;
 
     /* pCounter needs to be stored in pTrigger before calling here. */
 
     if (!pTrigger->pCounter)
 	return;
 
-    for (pCur = pTrigger->pCounter->pTriglist; pCur; pCur = pCur->next)
+    pPrev = NULL;
+    pCur = pTrigger->pCounter->pTriglist;
+
+    while (pCur)
     {
 	if (pCur->pTrigger == pTrigger)
 	{
+	    if (pPrev)
+		pPrev->next = pCur->next;
+	    else
 	    pTrigger->pCounter->pTriglist = pCur->next;
+	    
 	    xfree(pCur);
 	    break;
 	}
+	
+	pPrev = pCur;
+	pCur = pCur->next;
     }
 
     if (IsSystemCounter(pTrigger->pCounter))
