diff --git a/GL/glx/indirect_size_get.c b/GL/glx/indirect_size_get.c
index 1a1b123..e7eaf97 100644
--- a/GL/glx/indirect_size_get.c
+++ b/GL/glx/indirect_size_get.c
@@ -613,6 +613,7 @@ __glGetBooleanv_size(GLenum e)
 /*      case GL_CLIENT_ACTIVE_TEXTURE_ARB:*/
     case GL_MAX_TEXTURE_UNITS:
 /*      case GL_MAX_TEXTURE_UNITS_ARB:*/
+    case GL_MAX_RENDERBUFFER_SIZE_EXT:
     case GL_TEXTURE_COMPRESSION_HINT:
 /*      case GL_TEXTURE_COMPRESSION_HINT_ARB:*/
     case GL_TEXTURE_RECTANGLE_ARB:
@@ -715,6 +716,10 @@ __glGetBooleanv_size(GLenum e)
     case GL_ACTIVE_STENCIL_FACE_EXT:
     case GL_TEXTURE_BINDING_1D_ARRAY_EXT:
     case GL_TEXTURE_BINDING_2D_ARRAY_EXT:
+    case GL_DRAW_FRAMEBUFFER_BINDING_EXT:
+    case GL_RENDERBUFFER_BINDING_EXT:
+    case GL_READ_FRAMEBUFFER_BINDING_EXT:
+    case GL_MAX_COLOR_ATTACHMENTS_EXT:
     case GL_RASTER_POSITION_UNCLIPPED_IBM:
         return 1;
     case GL_SMOOTH_POINT_SIZE_RANGE:
