diff --git a/hw/xgl/egl/xegl.c b/hw/xgl/egl/xegl.c
index aa2252a..d250ff2 100644
--- a/hw/xgl/egl/xegl.c
+++ b/hw/xgl/egl/xegl.c
@@ -563,15 +563,11 @@ InitOutput (ScreenInfo *pScreenInfo,
         glitz_egl_init (NULL);
     }
 
-    templ.types.window     = 1;
-    templ.types.pbuffer    = 1;
     templ.samples          = 1;
     templ.doublebuffer     = 1;
     templ.color.alpha_size = 8;
 
-    mask =
-	GLITZ_FORMAT_WINDOW_MASK |
-	GLITZ_FORMAT_SAMPLES_MASK;
+    mask = GLITZ_FORMAT_SAMPLES_MASK;
 
     for (i = 0; i < sizeof (extraMask) / sizeof (extraMask[0]); i++)
     {
