diff --git a/dix/resource.c b/dix/resource.c
index 9f3794b..62481d0 100644
--- a/dix/resource.c
+++ b/dix/resource.c
@@ -102,6 +102,10 @@ Equipment Corporation.
 /* $XFree86: xc/programs/Xserver/dix/resource.c,v 3.13 2003/09/24 02:43:13 dawes Exp $ */
 
 #define NEED_EVENTS
+#ifdef HAVE_DIX_CONFIG_H
+#include <dix-config.h>
+#endif
+
 #include <X11/X.h>
 #include <X11/misc.h>
 #include <X11/os.h>
