diff --git a/hw/xfree86/common/xf86Cursor.c b/hw/xfree86/common/xf86Cursor.c
index 44c1a77..9e6e622 100644
--- a/hw/xfree86/common/xf86Cursor.c
+++ b/hw/xfree86/common/xf86Cursor.c
@@ -406,18 +406,14 @@ xf86CursorOffScreen(ScreenPtr *pScreen, int *x, int *y)
 /*
  * xf86CrossScreen --
  *      Switch to another screen
+ *
+ *	Currently nothing special happens, but mi assumes the CrossScreen
+ *	method exists.
  */
 
-/* NEED TO CHECK THIS */
-/* ARGSUSED */
 static void
 xf86CrossScreen (ScreenPtr pScreen, Bool entering)
 {
-#if 0
-  if (xf86Info.sharedMonitor)
-    (XF86SCRNINFO(pScreen)->EnterLeaveMonitor)(entering);
-  (XF86SCRNINFO(pScreen)->EnterLeaveCursor)(entering);
-#endif
 }
 
 
