diff --git a/fb/fb.h b/fb/fb.h
index a924f49..379a00a 100644
--- a/fb/fb.h
+++ b/fb/fb.h
@@ -2094,10 +2094,6 @@ fbFillRegionTiled (DrawablePtr	pDrawable,
 		   RegionPtr	pRegion,
 		   PixmapPtr	pTile);
 
-void
-fbPaintWindow(WindowPtr pWin, RegionPtr pRegion, int what);
-
-
 pixman_image_t *image_from_pict (PicturePtr pict,
 				 Bool       has_clip);
 void free_pixman_pict (PicturePtr, pixman_image_t *);
