diff --git a/Xprint/ps/PsPolygon.c b/Xprint/ps/PsPolygon.c
index a1ae67f..c4c30bc 100644
--- a/Xprint/ps/PsPolygon.c
+++ b/Xprint/ps/PsPolygon.c
@@ -215,8 +215,6 @@ PsPolyFillRect(
      */    
     if (nRects == 1)
     {
-      extern Bool noDbeExtension;
-
       if ( (pRects[0].x==0) && (pRects[0].y==0) &&
            (pRects[0].width==pDrawable->width) && (pRects[0].height==pDrawable->height) &&
            (pGC->fillStyle == FillSolid) &&
@@ -227,7 +225,7 @@ PsPolyFillRect(
 #endif /* DEBUG_gismobile */
         /* Remove all content from the pixmap as it would be covered
          * by the whole rect anyway */
-        PsScrubPixmap(pDrawable);
+        PsScrubPixmap((PixmapPtr)pDrawable);
       }
     }
 #endif /* DBE */
