diff --git a/exa/exa.h b/exa/exa.h
index 9e2edca..43573ae 100644
--- a/exa/exa.h
+++ b/exa/exa.h
@@ -47,9 +47,9 @@ typedef enum _ExaOffscreenState {
 } ExaOffscreenState;
 
 struct _ExaOffscreenArea {
-    int                 offset;
-    int                 save_offset;
-    int                 size;
+    int                 base_offset;	/* allocation base */
+    int                 offset;         /* aligned offset */
+    int                 size;           /* total allocation size */
     int                 score;
     pointer             privData;
 
