diff --git a/hw/xfree86/dri/xf86dri.c b/hw/xfree86/dri/xf86dri.c
index fdf0e99..130afdf 100644
--- a/hw/xfree86/dri/xf86dri.c
+++ b/hw/xfree86/dri/xf86dri.c
@@ -337,19 +337,9 @@ ProcXF86DRICreateContext(
     rep.sequenceNumber = client->sequence;
 
     pScreen = screenInfo.screens[stuff->screen];
-    visual = pScreen->visuals;
-
-    /* Find the requested X visual */
-    for (i = 0; i < pScreen->numVisuals; i++, visual++)
-	if (visual->vid == stuff->visual)
-	    break;
-    if (i == pScreen->numVisuals) {
-	/* No visual found */
-	return BadValue;
-    }
 
     if (!DRICreateContext( pScreen,
-			   visual,
+			   NULL,
 			   stuff->context,
 			   (drm_context_t *)&rep.hHWContext)) {
 	return BadValue;
