diff --git a/hw/xfree86/dixmods/cfb32module.c b/hw/xfree86/dixmods/cfb32module.c
index 0ead3ef..1451594 100644
--- a/hw/xfree86/dixmods/cfb32module.c
+++ b/hw/xfree86/dixmods/cfb32module.c
@@ -1,4 +1,3 @@
-/* $XFree86: xc/programs/Xserver/cfb32/cfbmodule.c,v 1.8 1999/01/26 05:53:49 dawes Exp $ */
 /*
  * Copyright (C) 1998 The XFree86 Project, Inc.  All Rights Reserved.
  *
@@ -31,8 +30,6 @@
 
 #define PSZ 32
 
-#ifdef XFree86LOADER
-
 #include "xf86Module.h"
 #include "cfb.h"
 
@@ -52,7 +49,7 @@ static XF86ModuleVersionInfo VersRec =
 	{0,0,0,0}       /* signature, to be patched into the file by a tool */
 };
 
-XF86ModuleData cfb32ModuleData = { &VersRec, cfb32Setup, NULL };
+_X_EXPORT XF86ModuleData cfb32ModuleData = { &VersRec, cfb32Setup, NULL };
 
 static pointer
 cfb32Setup(pointer module, pointer opts, int *errmaj, int *errmin)
@@ -61,5 +58,3 @@ cfb32Setup(pointer module, pointer opts, int *errmaj, int *errmin)
     return LoadSubModule(module, "cfb", NULL, NULL, NULL, NULL,
 			 errmaj, errmin);
 }
-
-#endif
