diff --git a/dbe/dbe.c b/dbe/dbe.c
index f6c729d..855135c 100644
--- a/dbe/dbe.c
+++ b/dbe/dbe.c
@@ -1,4 +1,4 @@
-/* $XdotOrg$ */
+/* $XdotOrg: xc/programs/Xserver/dbe/dbe.c,v 1.2 2004/04/23 19:03:43 eich Exp $ */
 /* $Xorg: dbe.c,v 1.3 2000/08/17 19:48:16 cpqbld Exp $ */
 /******************************************************************************
  * 
@@ -37,8 +37,8 @@
 /* INCLUDES */
 
 #define NEED_EVENTS
-#include "X.h"
-#include "Xproto.h"
+#include <X11/X.h>
+#include <X11/Xproto.h>
 #include "scrnintstr.h"
 #include "extnsionst.h"
 #include "gcstruct.h"
