diff --git a/hw/xfree86/os-support/sco/sco_iop.c b/hw/xfree86/os-support/sco/sco_iop.c
index 86210d7..3da4fcf 100644
--- a/hw/xfree86/os-support/sco/sco_iop.c
+++ b/hw/xfree86/os-support/sco/sco_iop.c
@@ -64,7 +64,7 @@ extern long sysi86 (int cmd, ...);
 
 static Bool IOEnabled = FALSE;
 
-_X_EXPORT Bool
+Bool
 xf86EnableIO(void)
 {
 	if (IOEnabled)
@@ -79,7 +79,7 @@ xf86EnableIO(void)
 	return TRUE;
 }
 
-_X_EXPORT void
+void
 xf86DisableIO(void)
 {
 	if (!IOEnabled)
