diff --git a/config/udev.c b/config/udev.c
index 9934490..cd46eec 100644
--- a/config/udev.c
+++ b/config/udev.c
@@ -189,17 +189,14 @@ device_added(struct udev_device *udev_device)
         }
     }
 
+    add_option(&options, "config_info", config_info);
+
     LogMessage(X_INFO, "config/udev: Adding input device %s (%s)\n",
                name, path);
     rc = NewInputDeviceRequest(options, &attrs, &dev);
     if (rc != Success)
         goto unwind;
 
-    for (; dev; dev = dev->next) {
-        free(dev->config_info);
-        dev->config_info = strdup(config_info);
-    }
-
  unwind:
     free(config_info);
     while (!dev && (tmpo = options)) {
