diff --git a/mi/mispritest.h b/mi/mispritest.h
index 5075f05..e8251ea 100644
--- a/mi/mispritest.h
+++ b/mi/mispritest.h
@@ -57,9 +57,6 @@ typedef struct {
     /* window procedures */
     CopyWindowProcPtr			CopyWindow;
     
-    /* backing store procedures */
-    SaveDoomedAreasProcPtr		SaveDoomedAreas;
-    
     /* colormap procedures */
     InstallColormapProcPtr		InstallColormap;
     StoreColorsProcPtr			StoreColors;
