diff --git a/hw/xwin/winmouse.c b/hw/xwin/winmouse.c
index 3193e3e..99509f2 100644
--- a/hw/xwin/winmouse.c
+++ b/hw/xwin/winmouse.c
@@ -356,15 +356,12 @@ winMouseButtonsHandle (ScreenPtr pScreen,
 /**
  * Enqueue a motion event.
  *
- *  XXX: miPointerMove does exactly this, but is static :-( (and uses a static buffer)
- *
  */
 void winEnqueueMotion(int x, int y)
 {
   int valuators[2];
   ValuatorMask mask;
 
-  miPointerSetPosition(g_pwinPointer, POINTER_RELATIVE, &x, &y);
   valuators[0] = x;
   valuators[1] = y;
 
