diff --git a/hw/dmx/glxProxy/glxvisuals.h b/hw/dmx/glxProxy/glxvisuals.h
index 0529d94..e1e0238 100644
--- a/hw/dmx/glxProxy/glxvisuals.h
+++ b/hw/dmx/glxProxy/glxvisuals.h
@@ -33,10 +33,14 @@
 
 int glxVisualsMatch( __GLXvisualConfig *v1, __GLXvisualConfig *v2 );
 
-VisualID glxMatchGLXVisualInConfigList( __GLXvisualConfig *pGlxVisual, __GLXvisualConfig *configs, int nconfigs );
+VisualID glxMatchGLXVisualInConfigList(__GLXvisualConfig * pGlxVisual,
+                                       __GLXvisualConfig * configs,
+                                       int nconfigs);
 
-VisualID glxMatchVisualInConfigList( ScreenPtr pScreen, VisualPtr pVisual, __GLXvisualConfig *configs, int nconfigs );
+VisualID glxMatchVisualInConfigList(ScreenPtr pScreen, VisualPtr pVisual,
+                                    __GLXvisualConfig * configs, int nconfigs);
 
-VisualPtr glxMatchVisual( ScreenPtr pScreen, VisualPtr pVisual, ScreenPtr pMatchScreen );
+VisualPtr glxMatchVisual(ScreenPtr pScreen, VisualPtr pVisual,
+                         ScreenPtr pMatchScreen);
 
 #endif
