diff --git a/hw/xfree86/common/xf86Cursor.c b/hw/xfree86/common/xf86Cursor.c
index 29b690a..6740faf 100644
--- a/hw/xfree86/common/xf86Cursor.c
+++ b/hw/xfree86/common/xf86Cursor.c
@@ -227,8 +227,6 @@ xf86SwitchMode(ScreenPtr pScreen, DisplayModePtr mode)
       if (IsPointerDevice(dev) && dev->spriteInfo->spriteOwner)
           break;
   }
-  if (!dev)
-      dev = inputInfo.pointer;
 
   pCursorScreen = miPointerGetScreen(dev);
   if (pScreen == pCursorScreen)
