diff --git a/hw/xwin/winmultiwindowwm.c b/hw/xwin/winmultiwindowwm.c
index 1e21de3..c9b1584 100644
--- a/hw/xwin/winmultiwindowwm.c
+++ b/hw/xwin/winmultiwindowwm.c
@@ -634,7 +634,7 @@ winMultiWindowWMProc (void *pArg)
   ErrorF ("winMultiWindowWMProc ()\n");
 #endif
 
-  /* Loop until we explicity break out */
+  /* Loop until we explicitly break out */
   for (;;)
     {
       WMMsgNodePtr	pNode;
