diff --git a/hw/xfree86/common/xf86fbman.c b/hw/xfree86/common/xf86fbman.c
index debd828..a65b00b 100644
--- a/hw/xfree86/common/xf86fbman.c
+++ b/hw/xfree86/common/xf86fbman.c
@@ -936,11 +936,17 @@ localAllocateOffscreenLinear(
    extents = REGION_EXTENTS(pScreen, offman->InitialBoxes);
    pitch = extents->x2 - extents->x1;
 
-   if(gran && ((gran > pitch) || (pitch % gran))) {
+   if (gran && gran > pitch) {
 	/* we can't match the specified alignment with XY allocations */
 	xfree(link);
 	return NULL;
    }
+   if (gran && (pitch % gran)) {
+       /* pitch and granularity aren't a perfect match, let's allocate
+	* a bit more so we can align later on
+	*/
+       length += gran - 1;
+   }
 
    if(length < pitch) { /* special case */
 	w = length;
@@ -963,6 +969,8 @@ localAllocateOffscreenLinear(
 	linear->pScreen = pScreen;
 	linear->size = h * w;
 	linear->offset = (pitch * area->box.y1) + area->box.x1;
+	if (gran && linear->offset % gran)
+		linear->offset += gran - (linear->offset % gran);
 	linear->granularity = gran;
 	linear->MoveLinearCallback = moveCB;
 	linear->RemoveLinearCallback = removeCB;
