diff --git a/xkb/xkbLEDs.c b/xkb/xkbLEDs.c
index 26e6d87..7cdd21a 100644
--- a/xkb/xkbLEDs.c
+++ b/xkb/xkbLEDs.c
@@ -35,8 +35,8 @@ THE USE OR PERFORMANCE OF THIS SOFTWARE.
 #include "misc.h"
 #include "inputstr.h"
 
-#include "XI.h"
-#include "XKBsrv.h"
+#include <X11/extensions/XI.h>
+#include <X11/extensions/XKBsrv.h>
 #include "xkb.h"
 
 /***====================================================================***/
