diff --git a/include/dixstruct.h b/include/dixstruct.h
index f38a64d..1ac7d84 100644
--- a/include/dixstruct.h
+++ b/include/dixstruct.h
@@ -130,15 +130,8 @@ typedef struct _Client {
     int         requestLogIndex;
 #endif
     unsigned long replyBytesRemaining;
-#ifdef XCSECURITY
-    XID		authId;
-    unsigned int trustLevel;
-    pointer (* CheckAccess)(
-	    ClientPtr /*pClient*/,
-	    XID /*id*/,
-	    RESTYPE /*classes*/,
-	    Mask /*access_mode*/,
-	    pointer /*resourceval*/);
+#ifdef XACE
+    pointer securityState[4];   /* 4 slots for use */
 #endif
 #ifdef XAPPGROUP
     struct _AppGroupRec*	appgroup;
