diff --git a/include/globals.h b/include/globals.h
index 17bca82..d147df9 100644
--- a/include/globals.h
+++ b/include/globals.h
@@ -2,7 +2,7 @@
 #ifndef _XSERV_GLOBAL_H_
 #define _XSERV_GLOBAL_H_
 
-#include "window.h"	/* for WindowPtr */
+#include "window.h"             /* for WindowPtr */
 
 /* Global X server variables that are visible to mi, dix, os, and ddx */
 
@@ -122,4 +122,4 @@ extern _X_EXPORT Bool noXvExtension;
 extern _X_EXPORT Bool noDRI2Extension;
 #endif
 
-#endif /* !_XSERV_GLOBAL_H_ */
+#endif                          /* !_XSERV_GLOBAL_H_ */
