diff --git a/hw/xfree86/common/compiler.h b/hw/xfree86/common/compiler.h
index fe041c7..040d8cd 100644
--- a/hw/xfree86/common/compiler.h
+++ b/hw/xfree86/common/compiler.h
@@ -75,20 +75,6 @@
 #  endif
 # endif /* __inline */
 
-# if defined(IODEBUG) && defined(__GNUC__)
-#  define outb RealOutb
-#  define outw RealOutw
-#  define outl RealOutl
-#  define inb RealInb
-#  define inw RealInw
-#  define inl RealInl
-# endif
-
-
-# if defined(__SUNPRO_C)
-#  define DO_PROTOTYPES
-# endif
-
 # if defined(NO_INLINE) || defined(DO_PROTOTYPES)
 
 #  if !defined(__arm__)
@@ -1527,23 +1513,6 @@ inl(unsigned short port)
 #   define write_mem_barrier()   /* NOP */
 #  endif /* __GNUC__ */
 
-
-#  if defined(IODEBUG) && defined(__GNUC__)
-#   undef inb
-#   undef inw
-#   undef inl
-#   undef outb
-#   undef outw
-#   undef outl
-#   define inb(a) __extension__ ({unsigned char __c=RealInb(a); ErrorF("inb(0x%03x) = 0x%02x\t@ line %4d, file %s\n", a, __c, __LINE__, __FILE__);__c;})
-#   define inw(a) __extension__ ({unsigned short __c=RealInw(a); ErrorF("inw(0x%03x) = 0x%04x\t@ line %4d, file %s\n", a, __c, __LINE__, __FILE__);__c;})
-#   define inl(a) __extension__ ({unsigned int __c=RealInl(a); ErrorF("inl(0x%03x) = 0x%08x\t@ line %4d, file %s\n", a, __c, __LINE__, __FILE__);__c;})
-
-#   define outb(a,b) (ErrorF("outb(0x%03x, 0x%02x)\t@ line %4d, file %s\n", a, b, __LINE__, __FILE__),RealOutb(a,b))
-#   define outw(a,b) (ErrorF("outw(0x%03x, 0x%04x)\t@ line %4d, file %s\n", a, b, __LINE__, __FILE__),RealOutw(a,b))
-#   define outl(a,b) (ErrorF("outl(0x%03x, 0x%08x)\t@ line %4d, file %s\n", a, b, __LINE__, __FILE__),RealOutl(a,b))
-#  endif
-
 # endif /* NO_INLINE */
 
 # ifdef __alpha__
