diff --git a/hw/xfree86/common/xf86Config.c b/hw/xfree86/common/xf86Config.c
index 2b8c2d5..5d424af 100644
--- a/hw/xfree86/common/xf86Config.c
+++ b/hw/xfree86/common/xf86Config.c
@@ -83,7 +83,8 @@ extern DeviceAssocRec mouse_assoc;
 #endif
 
 #if (defined(i386) || defined(__i386__)) && \
-    (defined(__FreeBSD__) || defined(__NetBSD__) || defined(linux) || \
+    (defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || \
+     defined(__NetBSD__) || defined(linux) || \
      (defined(SVR4) && !defined(sun)) || defined(__GNU__))
 #define SUPPORT_PC98
 #endif
