diff --git a/hw/xfree86/common/xf86.h b/hw/xfree86/common/xf86.h
index 54332e3..e1e0cd7 100644
--- a/hw/xfree86/common/xf86.h
+++ b/hw/xfree86/common/xf86.h
@@ -263,7 +263,6 @@ extern _X_EXPORT Bool xf86GetVidModeEnabled(void);
 extern _X_EXPORT Bool xf86GetModInDevAllowNonLocal(void);
 extern _X_EXPORT Bool xf86GetModInDevEnabled(void);
 extern _X_EXPORT Bool xf86GetAllowMouseOpenFail(void);
-extern _X_EXPORT Bool xf86IsPc98(void);
 extern _X_EXPORT void xf86DisableRandR(void);
 extern _X_EXPORT CARD32 xorgGetVersion(void);
 extern _X_EXPORT CARD32 xf86GetModuleVersion(pointer module);
