diff --git a/Xi/grabdev.h b/Xi/grabdev.h
index d149da5..881982f 100644
--- a/Xi/grabdev.h
+++ b/Xi/grabdev.h
@@ -1,4 +1,3 @@
-/* $XFree86: xc/programs/Xserver/Xi/grabdev.h,v 3.1 1996/04/15 11:18:51 dawes Exp $ */
 /************************************************************
 
 Copyright 1996 by Thomas E. Dickey <dickey@clark.net>
@@ -31,19 +30,13 @@ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 #ifndef GRABDEV_H
 #define GRABDEV_H 1
 
-int
-SProcXGrabDevice(
-	ClientPtr              /* client */
+int SProcXGrabDevice(ClientPtr	/* client */
 	);
 
-int
-ProcXGrabDevice(
-	ClientPtr              /* client */
+int ProcXGrabDevice(ClientPtr	/* client */
 	);
 
-int
-CreateMaskFromList (
-	ClientPtr              /* client */,
+int CreateMaskFromList(ClientPtr /* client */ ,
 	XEventClass *          /* list */,
 	int                    /* count */,
 	struct tmask           /* mask */[],
@@ -51,9 +44,7 @@ CreateMaskFromList (
 	int                    /* req */
 	);
 
-void
-SRepXGrabDevice (
-	ClientPtr              /* client */,
+void SRepXGrabDevice(ClientPtr /* client */ ,
 	int                    /* size */,
 	xGrabDeviceReply *     /* rep */
 	);
