diff --git a/xkb/xkbLEDs.c b/xkb/xkbLEDs.c
index 59cdba4..6ca80d7 100644
--- a/xkb/xkbLEDs.c
+++ b/xkb/xkbLEDs.c
@@ -745,12 +745,14 @@ XkbFlushLedEvents(	DeviceIntPtr			dev,
 			XkbDDXAccessXBeep(dev, _BEEP_LED_OFF, XkbAccessXFeedbackMask);
 	}
     }
-    if (ed && (ed->reason)) {
+    if (ed) {
+	if (ed->reason) {
 	if ((dev!=kbd)&&(ed->reason&XkbXI_IndicatorStateMask))
 	    XkbDDXUpdateDeviceIndicators(dev,sli,sli->effectiveState);
 	XkbSendExtensionDeviceNotify(dev,cause->client,ed);
     }
     bzero((char *)ed,sizeof(XkbExtensionDeviceNotify));
+    }
     return;
 }
 
