diff --git a/hw/xfree86/common/xf86Xinput.h b/hw/xfree86/common/xf86Xinput.h
index fe65643..4c752cd 100644
--- a/hw/xfree86/common/xf86Xinput.h
+++ b/hw/xfree86/common/xf86Xinput.h
@@ -83,6 +83,10 @@
 #define XI86_POINTER_CAPABLE	0x40 /* capable of being a core pointer */
 #define XI86_KEYBOARD_CAPABLE	0x80 /* capable of being a core keyboard */
 
+#ifdef MPX
+#define XI86_MP_DEVICE 0x100 /* device is multipointer device */
+#endif
+
 #define XI_PRIVATE(dev) \
 	(((LocalDevicePtr)((dev)->public.devicePrivate))->private)
 
