diff --git a/Xi/getkmap.c b/Xi/getkmap.c
index 5de0446..041e2c6 100644
--- a/Xi/getkmap.c
+++ b/Xi/getkmap.c
@@ -1,5 +1,3 @@
-/* $Xorg: getkmap.c,v 1.4 2001/02/09 02:04:34 xorgcvs Exp $ */
-
 /************************************************************
 
 Copyright 1989, 1998  The Open Group
@@ -45,7 +43,6 @@ ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
 SOFTWARE.
 
 ********************************************************/
-/* $XFree86: xc/programs/Xserver/Xi/getkmap.c,v 3.3 2001/01/17 22:13:24 dawes Exp $ */
 
 /********************************************************************
  *
@@ -105,35 +102,28 @@ ProcXGetDeviceKeyMapping(register ClientPtr client)
     REQUEST_SIZE_MATCH(xGetDeviceKeyMappingReq);
 
     dev = LookupDeviceIntRec (stuff->deviceid);
-    if (dev == NULL)
-	{
+    if (dev == NULL) {
 	SendErrorToClient (client, IReqCode, X_GetDeviceKeyMapping, 0, 
 		BadDevice);
 	return Success;
 	}
 
-    if (dev->key == NULL)
-	{
-	SendErrorToClient (client, IReqCode, X_GetDeviceKeyMapping, 0, 
-		BadMatch);
+    if (dev->key == NULL) {
+	SendErrorToClient(client, IReqCode, X_GetDeviceKeyMapping, 0, BadMatch);
 	return Success;
 	}
     k =  &dev->key->curKeySyms;
 
     if ((stuff->firstKeyCode < k->minKeyCode) ||
-        (stuff->firstKeyCode > k->maxKeyCode))
-        {
+	(stuff->firstKeyCode > k->maxKeyCode)) {
 	client->errorValue = stuff->firstKeyCode;
-	SendErrorToClient (client, IReqCode, X_GetDeviceKeyMapping, 0, 
-		BadValue);
+	SendErrorToClient(client, IReqCode, X_GetDeviceKeyMapping, 0, BadValue);
 	return Success;
         }
 
-    if (stuff->firstKeyCode + stuff->count > k->maxKeyCode + 1)
-        {
+    if (stuff->firstKeyCode + stuff->count > k->maxKeyCode + 1) {
 	client->errorValue = stuff->count;
-	SendErrorToClient (client, IReqCode, X_GetDeviceKeyMapping, 0, 
-		BadValue);
+	SendErrorToClient(client, IReqCode, X_GetDeviceKeyMapping, 0, BadValue);
 	return Success;
         }
 
@@ -145,8 +135,7 @@ ProcXGetDeviceKeyMapping(register ClientPtr client)
     WriteReplyToClient(client, sizeof(xGetDeviceKeyMappingReply), &rep);
 
     client->pSwapReplyFunc = (ReplySwapPtr)CopySwap32Write;
-    WriteSwappedDataToClient(
-	client,
+    WriteSwappedDataToClient(client,
 	k->mapWidth * stuff->count * sizeof(KeySym),
 	&k->map[(stuff->firstKeyCode - k->minKeyCode) *
 	k->mapWidth]);
@@ -162,7 +151,8 @@ ProcXGetDeviceKeyMapping(register ClientPtr client)
  */
 
 void
-SRepXGetDeviceKeyMapping (ClientPtr client, int size, xGetDeviceKeyMappingReply *rep)
+SRepXGetDeviceKeyMapping(ClientPtr client, int size,
+			 xGetDeviceKeyMappingReply * rep)
 {
     register char n;
 
@@ -170,4 +160,3 @@ SRepXGetDeviceKeyMapping (ClientPtr client, int size, xGetDeviceKeyMappingReply
     swapl(&rep->length, n);
     WriteToClient(client, size, (char *)rep);
     }
-
