diff --git a/hw/xfree86/os-support/xf86OSmouse.h b/hw/xfree86/os-support/xf86OSmouse.h
index 1eb4fad..7cb7322 100644
--- a/hw/xfree86/os-support/xf86OSmouse.h
+++ b/hw/xfree86/os-support/xf86OSmouse.h
@@ -287,7 +287,7 @@ typedef struct _MouseDevRec {
 /* Generalize for other axes. */
 #define MSE_NOAXISMAP	MSE_NOZMAP
 
-#define MSE_MAXBUTTONS	12
+#define MSE_MAXBUTTONS	24
 #define MSE_DFLTBUTTONS	 3
 
 #endif /* _XF86OSMOUSE_H_ */
