diff --git a/hw/kdrive/ephyr/ephyrlog.h b/hw/kdrive/ephyr/ephyrlog.h
index 71f7977..a07a0a0 100644
--- a/hw/kdrive/ephyr/ephyrlog.h
+++ b/hw/kdrive/ephyr/ephyrlog.h
@@ -31,11 +31,11 @@
 #include <assert.h>
 #include "os.h"
 
-#ifdef NDEBUG
+#ifndef DEBUG
 /*we are not in debug mode*/
 #define EPHYR_LOG(...)
 #define EPHYR_LOG_ERROR(...)
-#endif /*NDEBUG*/
+#endif /*!DEBUG*/
 
 #define ERROR_LOG_LEVEL 3
 #define INFO_LOG_LEVEL 4
