diff --git a/ilbm/ilbmtegblt.c b/ilbm/ilbmtegblt.c
index 2c39ef7..0f6a8e5 100644
--- a/ilbm/ilbmtegblt.c
+++ b/ilbm/ilbmtegblt.c
@@ -52,9 +52,9 @@ SOFTWARE.
 /* Modified jun 95 by Geert Uytterhoeven (Geert.Uytterhoeven@cs.kuleuven.ac.be)
    to use interleaved bitplanes instead of normal bitplanes */
 
-#include "X.h"
-#include "Xmd.h"
-#include "Xproto.h"
+#include <X11/X.h>
+#include <X11/Xmd.h>
+#include <X11/Xproto.h>
 #include "ilbm.h"
 #include "fontstruct.h"
 #include "dixfontstr.h"
