diff --git a/hw/xfree86/common/xf86Xinput.c b/hw/xfree86/common/xf86Xinput.c
index 2436231..31c3008 100644
--- a/hw/xfree86/common/xf86Xinput.c
+++ b/hw/xfree86/common/xf86Xinput.c
@@ -179,15 +179,33 @@ xf86ActivateDevice(LocalDevicePtr local)
         dev->coreEvents = local->flags & XI86_ALWAYS_CORE; 
         dev->spriteInfo->spriteOwner = !(local->flags & XI86_SHARED_POINTER);
 
-        RegisterOtherDevice(dev);
-
+        if (DeviceIsPointerType(dev))
+        {
 #ifdef XKB
-        if (!DeviceIsPointerType(dev) && !noXkbExtension)
+            dev->public.processInputProc = CoreProcessPointerEvent;
+            dev->public.realInputProc = CoreProcessPointerEvent;
+#else
+            dev->public.processInputProc = ProcessPointerEvent;
+            dev->public.realInputProc = ProcessPointerEvent;
+#endif
+            dev->deviceGrab.ActivateGrab = ActivatePointerGrab;
+            dev->deviceGrab.DeactivateGrab = DeactivatePointerGrab;
+        } else 
         {
-            XkbSetExtension(dev, ProcessKeyboardEvent);
-        }
+#ifdef XKB
+            dev->public.processInputProc = CoreProcessKeyboardEvent;
+            dev->public.realInputProc = CoreProcessKeyboardEvent;
+#else
+            dev->public.processInputProc = ProcessKeyboardEvent;
+            dev->public.realInputProc = ProcessKeyboardEvent;
 #endif
+            dev->deviceGrab.ActivateGrab = ActivateKeyboardGrab;
+            dev->deviceGrab.DeactivateGrab = DeactivateKeyboardGrab;
+        }
 
+        RegisterOtherDevice(dev);
+        if (!noXkbExtension)
+            XkbSetExtension(dev, ProcessOtherEvent);
 
         if (serverGeneration == 1) 
             xf86Msg(X_INFO, "XINPUT: Adding extended input device \"%s\" (type: %s)\n",
