diff --git a/hw/xfree86/exa/exa_offscreen.c b/hw/xfree86/exa/exa_offscreen.c
index 9ff88b4..c961998 100644
--- a/hw/xfree86/exa/exa_offscreen.c
+++ b/hw/xfree86/exa/exa_offscreen.c
@@ -368,7 +368,8 @@ exaOffscreenInit (ScreenPtr pScreen)
 
     area->state = ExaOffscreenAvail;
     area->base_offset = pExaScr->info->card.offScreenBase;
-    area->size = pExaScr->info->card.memorySize - area->offset;
+    area->offset = area->base_offset;
+    area->size = pExaScr->info->card.memorySize - area->base_offset;
     area->save = NULL;
     area->next = NULL;
     area->score = 0;
