diff --git a/hw/xfree86/utils/xorgcfg/text-mode.c b/hw/xfree86/utils/xorgcfg/text-mode.c
index c1fa67e..0b6e654 100644
--- a/hw/xfree86/utils/xorgcfg/text-mode.c
+++ b/hw/xfree86/utils/xorgcfg/text-mode.c
@@ -39,7 +39,7 @@
 #endif
 #include <ctype.h>
 #include <X11/Xlib.h>
-#include "xkbstr.h"
+#include <X11/extensions/XKBstr.h>
 #include <X11/extensions/XKBrules.h>
 #include "cards.h"
 #include "config.h"
