diff --git a/hw/xfree86/common/xf86DoProbe.c b/hw/xfree86/common/xf86DoProbe.c
index b74f37e..e76fd91 100644
--- a/hw/xfree86/common/xf86DoProbe.c
+++ b/hw/xfree86/common/xf86DoProbe.c
@@ -97,9 +97,7 @@ DoProbe()
 
 	xf86MsgVerb(X_INFO, 3, "Probing in driver %s\n",  drv->driverName);
 
-	if (drv->Probe == NULL) continue;
-
-	probeResult = (*drv->Probe)( drv, PROBE_DETECT );
+	probeResult = xf86CallDriverProbe( drv, TRUE );
 	if (!probeResult) {
 	    xf86ErrorF("Probe in driver `%s' returns FALSE\n",
 		drv->driverName);
