diff --git a/dix/pixmap.c b/dix/pixmap.c
index 82e388c..7ab8681 100644
--- a/dix/pixmap.c
+++ b/dix/pixmap.c
@@ -88,7 +88,7 @@ FreeScratchPixmapHeader(PixmapPtr pPixmap)
 }
 
 
-Bool
+_X_EXPORT Bool
 CreateScratchPixmapsForScreen(int scrnum)
 {
     /* let it be created on first use */
@@ -97,7 +97,7 @@ CreateScratchPixmapsForScreen(int scrnum)
 }
 
 
-void
+_X_EXPORT void
 FreeScratchPixmapsForScreen(int scrnum)
 {
     FreeScratchPixmapHeader(screenInfo.screens[scrnum]->pScratchPixmap);
