diff --git a/hw/kdrive/ati/ati_dri.h b/hw/kdrive/ati/ati_dri.h
index 2e2b950..985f7eb 100644
--- a/hw/kdrive/ati/ati_dri.h
+++ b/hw/kdrive/ati/ati_dri.h
@@ -1,6 +1,4 @@
 /*
- * $Id$
- *
  * Copyright © 2003 Eric Anholt
  *
  * Permission to use, copy, modify, distribute, and sell this software and its
@@ -21,7 +19,6 @@
  * TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
  * PERFORMANCE OF THIS SOFTWARE.
  */
-/* $Header$ */
 
 #ifndef _ATI_DRI_H_
 #define _ATI_DRI_H_
