diff --git a/Xi/getdctl.c b/Xi/getdctl.c
index 4287028..6090b81 100644
--- a/Xi/getdctl.c
+++ b/Xi/getdctl.c
@@ -127,7 +127,6 @@ static void CopySwapDeviceCore (ClientPtr client, DeviceIntPtr dev, char *buf)
     if (client->swapped) {
         swaps(&c->control);
         swaps(&c->length);
-        swaps(&c->status);
     }
 }
 
@@ -142,7 +141,6 @@ static void CopySwapDeviceEnable (ClientPtr client, DeviceIntPtr dev, char *buf)
     if (client->swapped) {
         swaps(&e->control);
         swaps(&e->length);
-        swaps(&e->enable);
     }
 }
 
