diff --git a/xkb/ddxLoad.c b/xkb/ddxLoad.c
index 906715a..c2b1f18 100644
--- a/xkb/ddxLoad.c
+++ b/xkb/ddxLoad.c
@@ -38,8 +38,8 @@ THE USE OR PERFORMANCE OF THIS SOFTWARE.
 #include "scrnintstr.h"
 #include "windowstr.h"
 #define	XKBSRV_NEED_FILE_FUNCS
-#include "XKBsrv.h"
-#include "XI.h"
+#include <X11/extensions/XKBsrv.h>
+#include <X11/extensions/XI.h>
 #include "xkb.h"
 
 #if defined(CSRG_BASED) || defined(linux) || defined(__sgi) || defined(AIXV3) || defined(__osf__) || defined(__GNU__)
