diff --git a/include/os.h b/include/os.h
index 0a23a73..7c84772 100644
--- a/include/os.h
+++ b/include/os.h
@@ -230,8 +230,6 @@ extern pointer XNFalloc(unsigned long /*amount*/);
 extern pointer XNFcalloc(unsigned long /*amount*/);
 extern pointer XNFrealloc(pointer /*ptr*/, unsigned long /*amount*/);
 
-extern void OsInitAllocator(void);
-
 extern char *Xstrdup(const char *s);
 extern char *XNFstrdup(const char *s);
 extern char *Xprintf(const char *fmt, ...);
