diff --git a/hw/xfree86/loader/xf86sym.c b/hw/xfree86/loader/xf86sym.c
index c257d2d..da369df 100644
--- a/hw/xfree86/loader/xf86sym.c
+++ b/hw/xfree86/loader/xf86sym.c
@@ -316,7 +316,6 @@ _X_HIDDEN void *xfree86LookupTab[] = {
     SYMFUNC(xf86GetNumEntityInstances)
     SYMFUNC(xf86GetDevFromEntity)
     SYMFUNC(xf86GetPciInfoForEntity)
-    SYMFUNC(xf86DeallocateResourcesForEntity)
     SYMFUNC(xf86RegisterResources)
     SYMFUNC(xf86CheckPciMemBase)
     SYMFUNC(xf86IsEntityPrimary)
