diff --git a/include/inputstr.h b/include/inputstr.h
index d96e24a..b2304dd 100644
--- a/include/inputstr.h
+++ b/include/inputstr.h
@@ -400,7 +400,10 @@ typedef struct _DeviceIntRec {
     int			nPrivates;
     DeviceUnwrapProc    unwrapProc;
     SpriteInfoPtr       spriteInfo;
+    union {
     DeviceIntPtr        master;       /* master device */
+    DeviceIntPtr        lastSlave;    /* last slave device used */
+    } u;
 } DeviceIntRec;
 
 typedef struct {
