diff --git a/hw/xnest/Pointer.h b/hw/xnest/Pointer.h
index 890726f..d0047c5 100644
--- a/hw/xnest/Pointer.h
+++ b/hw/xnest/Pointer.h
@@ -23,7 +23,7 @@ is" without express or implied warranty.
 
 extern DeviceIntPtr xnestPointerDevice;
 
-void xnestChangePointerControl(DeviceIntPtr pDev, PtrCtrl *ctrl);
+void xnestChangePointerControl(DeviceIntPtr pDev, PtrCtrl * ctrl);
 int xnestPointerProc(DeviceIntPtr pDev, int onoff);
 
-#endif /* XNESTPOINTER_H */
+#endif                          /* XNESTPOINTER_H */
