diff --git a/hw/xfree86/common/xf86Bus.h b/hw/xfree86/common/xf86Bus.h
index 518294c..83ba83c 100644
--- a/hw/xfree86/common/xf86Bus.h
+++ b/hw/xfree86/common/xf86Bus.h
@@ -89,10 +89,6 @@ typedef struct {
 
 #define NEED_SHARED (NEED_MEM_SHARED | NEED_IO_SHARED)
 
-#define busType bus.type
-#define isaBusId bus.id.isa
-#define sbusBusId bus.id.sbus
-
 struct x_BusAccRec;
 typedef void (*BusAccProcPtr)(struct x_BusAccRec *ptr);
 
