diff --git a/include/swaprep.h b/include/swaprep.h
index 2c026c6..d18b39f 100644
--- a/include/swaprep.h
+++ b/include/swaprep.h
@@ -120,17 +120,6 @@ extern void SQueryKeymapReply(
     int /* size */,
     xQueryKeymapReply * /* pRep */);
 
-#ifdef LBX
-extern void SwapCharInfo(
-    xCharInfo * /* pInfo */);
-#endif
-
-#ifdef LBX
-extern void SwapFont(
-    xQueryFontReply * /* pr */,
-    Bool /* hasGlyphs */);
-#endif
-
 extern void SQueryFontReply(
     ClientPtr /* pClient */,
     int /* size */,
