diff --git a/hw/xquartz/darwin.c b/hw/xquartz/darwin.c
index fa8d4ce..29ab836 100644
--- a/hw/xquartz/darwin.c
+++ b/hw/xquartz/darwin.c
@@ -79,7 +79,7 @@
 
 aslclient aslc;
 
-void debug_asl (const char *file, const char *function, int line, const char *fmt, ...) {
+void xq_asl_log (int level, const char *subsystem, const char *file, const char *function, int line, const char *fmt, ...) {
     va_list args;
     aslmsg msg = asl_new(ASL_TYPE_MSG);
 
@@ -93,10 +93,12 @@ void debug_asl (const char *file, const char *function, int line, const char *fm
             asl_set(msg, "Line", _line);
             free(_line);
         }
+        if(subsystem)
+            asl_set(msg, "Subsystem", subsystem);
     }
 
     va_start(args, fmt);
-    asl_vlog(aslc, msg, ASL_LEVEL_DEBUG, fmt, args);
+    asl_vlog(aslc, msg, level, fmt, args);
     va_end(args);
 
     if(msg)
