diff --git a/hw/dmx/input/lnx-keyboard.c b/hw/dmx/input/lnx-keyboard.c
index c5dada9..55ff9f8 100644
--- a/hw/dmx/input/lnx-keyboard.c
+++ b/hw/dmx/input/lnx-keyboard.c
@@ -358,7 +358,7 @@ static unsigned char at2lnx[NUM_KEYCODES] =
 /** Create a private structure for use within this file. */
 pointer kbdLinuxCreatePrivate(DeviceIntPtr pKeyboard)
 {
-    myPrivate *priv = xcalloc(1, sizeof(*priv));
+    myPrivate *priv = calloc(1, sizeof(*priv));
     priv->fd        = -1;
     priv->pKeyboard = pKeyboard;
     return priv;
@@ -367,7 +367,7 @@ pointer kbdLinuxCreatePrivate(DeviceIntPtr pKeyboard)
 /** Destroy a private structure. */
 void kbdLinuxDestroyPrivate(pointer priv)
 {
-    if (priv) xfree(priv);
+    if (priv) free(priv);
 }
 
 /** Ring the bell.
