diff --git a/mi/miscrinit.c b/mi/miscrinit.c
index 2008a68..28d3316 100644
--- a/mi/miscrinit.c
+++ b/mi/miscrinit.c
@@ -189,6 +189,19 @@ miScreenDevPrivateInit(ScreenPtr pScreen, int width, pointer pbits)
     return TRUE;
 }
 
+static PixmapPtr
+miGetScreenPixmap(ScreenPtr pScreen)
+{
+    return (PixmapPtr)(pScreen->devPrivate);
+}
+
+static void
+miSetScreenPixmap(PixmapPtr pPix)
+{
+    if (pPix)
+	pPix->drawable.pScreen->devPrivate = (pointer)pPix;
+}
+
 _X_EXPORT Bool
 miScreenInit(
     ScreenPtr pScreen,
@@ -297,16 +310,3 @@ miSetZeroLineBias(ScreenPtr pScreen, unsigned int bias)
 {
     dixSetPrivate(&pScreen->devPrivates, miZeroLineScreenKey, (pointer)bias);
 }
-
-PixmapPtr
-miGetScreenPixmap(ScreenPtr pScreen)
-{
-    return (PixmapPtr)(pScreen->devPrivate);
-}
-
-void
-miSetScreenPixmap(PixmapPtr pPix)
-{
-    if (pPix)
-	pPix->drawable.pScreen->devPrivate = (pointer)pPix;
-}
