diff --git a/Xext/syncsrv.h b/Xext/syncsrv.h
index 64e42cd..aa7dfb9 100644
--- a/Xext/syncsrv.h
+++ b/Xext/syncsrv.h
@@ -51,24 +51,8 @@ PERFORMANCE OF THIS SOFTWARE.
 #ifndef _SYNCSRV_H_
 #define _SYNCSRV_H_
 
-#define CARD64 XSyncValue /* XXX temporary! need real 64 bit values for Alpha */
-
-/* Sync object types */
-#define SYNC_COUNTER		0
-
-typedef struct _SyncObject {
-    ClientPtr		client;	/* Owning client. 0 for system counters */
-    struct _SyncTriggerList *pTriglist;	/* list of triggers */
-    XID			id;		/* resource ID */
-    unsigned char	type;		/* SYNC_* */
-    Bool		beingDestroyed;	/* in process of going away */
-} SyncObject;
-
-typedef struct _SyncCounter {
-    SyncObject		sync;		/* Common sync object data */
-    CARD64		value;		/* counter value */
-    struct _SysCounterInfo *pSysCounterInfo; /* NULL if not a system counter */
-} SyncCounter;
+#include "misync.h"
+#include "misyncstr.h"
 
 /*
  * The System Counter interface
@@ -100,29 +84,6 @@ typedef struct _SysCounterInfo {
 
 
 
-typedef struct _SyncTrigger {
-    SyncObject *pSync;
-    CARD64	wait_value;	/* wait value */
-    unsigned int value_type;     /* Absolute or Relative */
-    unsigned int test_type;	/* transition or Comparision type */
-    CARD64	test_value;	/* trigger event threshold value */
-    Bool	(*CheckTrigger)(
-				struct _SyncTrigger * /*pTrigger*/,
-				CARD64 /*newval*/
-				);
-    void	(*TriggerFired)(
-				struct _SyncTrigger * /*pTrigger*/
-				);
-    void	(*CounterDestroyed)(
-				struct _SyncTrigger * /*pTrigger*/
-				    );
-} SyncTrigger;
-
-typedef struct _SyncTriggerList {
-    SyncTrigger *pTrigger;
-    struct _SyncTriggerList *next;
-} SyncTriggerList;
-
 typedef struct _SyncAlarmClientList {
     ClientPtr	client;
     XID		delete_id;
@@ -179,6 +140,7 @@ extern void SyncChangeCounter(
 extern void SyncDestroySystemCounter(
     pointer pCounter
 );
+
 extern void InitServertime(void);
 
 extern void SyncExtensionInit(void);
