diff --git a/hw/xfree86/common/xf86Priv.h b/hw/xfree86/common/xf86Priv.h
index f3dfd70..5de48ac 100644
--- a/hw/xfree86/common/xf86Priv.h
+++ b/hw/xfree86/common/xf86Priv.h
@@ -150,9 +150,10 @@ Bool xf86PathIsSafe(const char *path);
 extern const DisplayModeRec xf86DefaultModes[];
 extern const int xf86NumDefaultModes;
 
-/* xf86DoProbe.c */
-void DoProbe(void);
+/* xf86Configure.c */
 void DoConfigure(void);
+
+/* xf86ShowOpts.c */
 void DoShowOptions(void);
 
 /* xf86Events.c */
