diff --git a/dix/atom.c b/dix/atom.c
index d78fa56..53cd5c3 100644
--- a/dix/atom.c
+++ b/dix/atom.c
@@ -74,7 +74,7 @@ static NodePtr *nodeTable;
 
 void FreeAtom(NodePtr patom);
 
-Atom 
+_X_EXPORT Atom 
 MakeAtom(char *string, unsigned len, Bool makeit)
 {
     register    NodePtr * np;
@@ -151,7 +151,7 @@ MakeAtom(char *string, unsigned len, Bool makeit)
 	return None;
 }
 
-Bool
+_X_EXPORT Bool
 ValidAtom(Atom atom)
 {
     return (atom != None) && (atom <= lastAtom);
