diff --git a/hw/xwin/winwndproc.c b/hw/xwin/winwndproc.c
index 5e84510..4e4eff2 100644
--- a/hw/xwin/winwndproc.c
+++ b/hw/xwin/winwndproc.c
@@ -57,7 +57,6 @@ Bool				g_fButton[3] = { FALSE, FALSE, FALSE };
 extern Bool			g_fClipboard;
 extern HWND			g_hDlgDepthChange;
 extern Bool			g_fKeyboardHookLL;
-extern HWND			g_hwndKeyboardFocus;
 extern Bool			g_fSoftwareCursor;
 extern DWORD			g_dwCurrentThreadID;
 
@@ -964,9 +963,6 @@ winWindowProc (HWND hwnd, UINT message,
       if (s_pScreenPriv == NULL || s_pScreenInfo->fIgnoreInput)
 	break;
 
-      /* Save handle of our main window that last received focus */
-      g_hwndKeyboardFocus = hwnd;
-
       /* Restore the state of all mode keys */
       winRestoreModeKeyStates ();
 
@@ -979,9 +975,6 @@ winWindowProc (HWND hwnd, UINT message,
       if (s_pScreenPriv == NULL || s_pScreenInfo->fIgnoreInput)
 	break;
 
-      /* Clear handle of our main window that last received focus */
-      g_hwndKeyboardFocus = NULL;
-
       /* Release any pressed keys */
       winKeybdReleaseKeys ();
 
