diff --git a/ilbm/ilbmfillsp.c b/ilbm/ilbmfillsp.c
index f7fe907..95eb220 100644
--- a/ilbm/ilbmfillsp.c
+++ b/ilbm/ilbmfillsp.c
@@ -52,8 +52,8 @@ SOFTWARE.
 /* Modified jun 95 by Geert Uytterhoeven (Geert.Uytterhoeven@cs.kuleuven.ac.be)
    to use interleaved bitplanes instead of normal bitplanes */
 
-#include "X.h"
-#include "Xmd.h"
+#include <X11/X.h>
+#include <X11/Xmd.h>
 #include "gcstruct.h"
 #include "window.h"
 #include "pixmapstr.h"
