diff --git a/hw/xfree86/dummylib/xf86screens.c b/hw/xfree86/dummylib/xf86screens.c
deleted file mode 100644
index 5585533..0000000
--- a/hw/xfree86/dummylib/xf86screens.c
+++ /dev/null
@@ -1,12 +0,0 @@
-#ifdef HAVE_XORG_CONFIG_H
-#include <xorg-config.h>
-#endif
-
-#include <X11/X.h>
-#include "os.h"
-#include "xf86.h"
-#include "xf86Priv.h"
-
-/* Dummy variables */
-ScrnInfoPtr *xf86Screens = NULL;
-
