diff --git a/Xext/xcmisc.c b/Xext/xcmisc.c
index 4f92896..aed2e4e 100644
--- a/Xext/xcmisc.c
+++ b/Xext/xcmisc.c
@@ -64,8 +64,7 @@ XCMiscExtensionInit(INITARGS)
 }
 
 static int
-ProcXCMiscGetVersion(client)
-    ClientPtr client;
+ProcXCMiscGetVersion(ClientPtr client)
 {
     xXCMiscGetVersionReply rep;
     int n;
@@ -86,8 +85,7 @@ ProcXCMiscGetVersion(client)
 }
 
 static int
-ProcXCMiscGetXIDRange(client)
-    ClientPtr client;
+ProcXCMiscGetXIDRange(ClientPtr client)
 {
     xXCMiscGetXIDRangeReply rep;
     int n;
@@ -110,8 +108,7 @@ ProcXCMiscGetXIDRange(client)
 }
 
 static int
-ProcXCMiscGetXIDList(client)
-    ClientPtr client;
+ProcXCMiscGetXIDList(ClientPtr client)
 {
     REQUEST(xXCMiscGetXIDListReq);
     xXCMiscGetXIDListReply rep;
@@ -150,8 +147,7 @@ ProcXCMiscGetXIDList(client)
 }
 
 static int
-ProcXCMiscDispatch (client)
-    ClientPtr	client;
+ProcXCMiscDispatch (ClientPtr client)
 {
     REQUEST(xReq);
     switch (stuff->data)
@@ -168,8 +164,7 @@ ProcXCMiscDispatch (client)
 }
 
 static int
-SProcXCMiscGetVersion(client)
-    ClientPtr	client;
+SProcXCMiscGetVersion(ClientPtr client)
 {
     int n;
     REQUEST(xXCMiscGetVersionReq);
@@ -182,8 +177,7 @@ SProcXCMiscGetVersion(client)
 }
 
 static int
-SProcXCMiscGetXIDRange(client)
-    ClientPtr	client;
+SProcXCMiscGetXIDRange(ClientPtr client)
 {
     int n;
     REQUEST(xReq);
@@ -193,8 +187,7 @@ SProcXCMiscGetXIDRange(client)
 }
 
 static int
-SProcXCMiscGetXIDList(client)
-    ClientPtr	client;
+SProcXCMiscGetXIDList(ClientPtr client)
 {
     int n;
     REQUEST(xXCMiscGetXIDListReq);
@@ -205,8 +198,7 @@ SProcXCMiscGetXIDList(client)
 }
 
 static int
-SProcXCMiscDispatch (client)
-    ClientPtr	client;
+SProcXCMiscDispatch (ClientPtr client)
 {
     REQUEST(xReq);
     switch (stuff->data)
