diff --git a/Xi/querydp.c b/Xi/querydp.c
index 2925dde..2e2cfaa 100644
--- a/Xi/querydp.c
+++ b/Xi/querydp.c
@@ -127,6 +127,7 @@ ProcXIQueryPointer(ClientPtr client)
     {
         int i, down;
         rep.buttons_len = ((pDev->button->numButtons/8) + 3)/4;
+        rep.length += rep.buttons_len;
         buttons = xcalloc(rep.buttons_len, 4);
         if (!buttons)
             return BadAlloc;
