diff --git a/exa/exa_migration_mixed.c b/exa/exa_migration_mixed.c
index 771c2c3..915bed9 100644
--- a/exa/exa_migration_mixed.c
+++ b/exa/exa_migration_mixed.c
@@ -109,6 +109,10 @@ exaCreateDriverPixmap_mixed(PixmapPtr pPixmap)
     if (!w || !h)
 	goto finish;
 
+    /* we do not malloc memory by default. */
+    if (!sys_buffer)
+	goto finish;
+
     if (!pExaScr->info->UploadToScreen)
 	goto fallback;
 
