diff --git a/os/connection.c b/os/connection.c
index c5fc5a0..6f480de 100644
--- a/os/connection.c
+++ b/os/connection.c
@@ -114,7 +114,7 @@ SOFTWARE.
 #define Pid_t pid_t
 
 
-#ifdef HAS_GETPEERUCRED
+#ifdef HAVE_GETPEERUCRED
 # include <ucred.h>
 # include <zone.h>
 #endif
@@ -122,7 +122,7 @@ SOFTWARE.
 #ifdef XSERVER_DTRACE
 # include <sys/types.h>
 typedef const char *string;
-# ifndef HAS_GETPEERUCRED
+# ifndef HAVE_GETPEERUCRED
 #  define zoneid_t int
 # endif
 # include "../dix/Xserver-dtrace.h"
@@ -282,7 +282,7 @@ InitConnectionLimits(void)
     lastfdesc = sysconf(_SC_OPEN_MAX) - 1;
 #endif
 
-#ifdef HAS_GETDTABLESIZE
+#ifdef HAVE_GETDTABLESIZE
     if (lastfdesc < 0)
 	lastfdesc = getdtablesize() - 1;
 #endif
