diff --git a/dix/devices.c b/dix/devices.c
index 904c404..b179ce3 100644
--- a/dix/devices.c
+++ b/dix/devices.c
@@ -592,8 +592,6 @@ CorePointerProc(DeviceIntPtr pDev, int what)
  * Initialise the two core devices, VCP and VCK (see events.c).
  * Both devices are not tied to physical devices, but guarantee that there is
  * always a keyboard and a pointer present and keep the protocol semantics.
- * Both core devices are NOT part of the device list and act only as a
- * fallback if no physical device is available.
  *
  * The devices are activated but not enabled.
  *
