diff --git a/hw/xwin/windialogs.c b/hw/xwin/windialogs.c
index 3ec9b78..270ff78 100644
--- a/hw/xwin/windialogs.c
+++ b/hw/xwin/windialogs.c
@@ -324,7 +324,7 @@ winDisplayExitDialog (winPrivScreenPtr pScreenPriv)
   
   /* Set focus to the Cancel button */
   PostMessage (g_hDlgExit, WM_NEXTDLGCTL,
-	       GetDlgItem (g_hDlgExit, IDCANCEL), TRUE);
+	       (WPARAM)GetDlgItem (g_hDlgExit, IDCANCEL), TRUE);
 }
 
 #define CONNECTED_CLIENTS_FORMAT	"There %s currently %d client%s connected."
@@ -594,7 +594,7 @@ winDisplayAboutDialog (winPrivScreenPtr pScreenPriv)
   
   /* Set focus to the OK button */
   PostMessage (g_hDlgAbout, WM_NEXTDLGCTL,
-	       GetDlgItem (g_hDlgAbout, IDOK), TRUE);
+	       (WPARAM)GetDlgItem (g_hDlgAbout, IDOK), TRUE);
 }
 
 
@@ -675,7 +675,7 @@ winAboutDlgProc (HWND hwndDialog, UINT message,
 
 	case ID_ABOUT_CHANGELOG:
 	  {
-	    HINSTANCE iReturn;
+	    int iReturn;
 #ifdef __CYGWIN__
 	    const char *	pszCygPath = "/usr/X11R6/share/doc/"
 	      "xorg-x11-xwin/changelog.html";
@@ -688,7 +688,7 @@ winAboutDlgProc (HWND hwndDialog, UINT message,
 		    "devel/server/changelog.html";
 #endif
 	    
-	    iReturn = ShellExecute (NULL,
+	    iReturn = (int)ShellExecute (NULL,
                                     "open",
                                     pszWinPath,
                                     NULL,
@@ -708,7 +708,7 @@ winAboutDlgProc (HWND hwndDialog, UINT message,
 	    const char *	pszPath = __VENDORDWEBSUPPORT__;
 	    int			iReturn;
 	    
-	    iReturn = ShellExecute (NULL,
+	    iReturn = (int)ShellExecute (NULL,
                                     "open",
                                     pszPath,
                                     NULL,
@@ -728,7 +728,7 @@ winAboutDlgProc (HWND hwndDialog, UINT message,
 	    const char *	pszPath = "http://x.cygwin.com/docs/ug/";
 	    int			iReturn;
 	    
-	    iReturn = ShellExecute (NULL,
+	    iReturn = (int)ShellExecute (NULL,
                                     "open",
                                     pszPath,
                                     NULL,
@@ -748,7 +748,7 @@ winAboutDlgProc (HWND hwndDialog, UINT message,
 	    const char *	pszPath = "http://x.cygwin.com/docs/faq/";
 	    int			iReturn;
 	    
-	    iReturn = ShellExecute (NULL,
+	    iReturn = (int)ShellExecute (NULL,
                                     "open",
                                     pszPath,
                                     NULL,
