diff --git a/hw/xfree86/dixmods/extmod/modinit.h b/hw/xfree86/dixmods/extmod/modinit.h
index 6134213..1de1485 100644
--- a/hw/xfree86/dixmods/extmod/modinit.h
+++ b/hw/xfree86/dixmods/extmod/modinit.h
@@ -66,9 +66,6 @@ extern void ResExtensionInit(INITARGS);
 #ifdef SHM
 extern void ShmExtensionInit(INITARGS);
 #include <X11/extensions/shmstr.h>
-extern void ShmSetPixmapFormat(
-    ScreenPtr pScreen,
-    int format);
 extern void ShmRegisterFuncs(
     ScreenPtr pScreen,
     ShmFuncsPtr funcs);
