diff --git a/dix/devices.c b/dix/devices.c
index 97bb7df..dbd1b4e 100644
--- a/dix/devices.c
+++ b/dix/devices.c
@@ -312,8 +312,11 @@ EnableDevice(DeviceIntPtr dev, BOOL sendevent)
                 PairDevices(NULL, other, dev);
         } else
         {
+            if (dev->coreEvents)
             other = (IsPointerDevice(dev)) ? inputInfo.pointer :
                 inputInfo.keyboard;
+            else
+                other = NULL; /* auto-float non-core devices */
             AttachDevice(NULL, dev, other);
         }
     }
@@ -2335,7 +2338,7 @@ AttachDevice(ClientPtr client, DeviceIntPtr dev, DeviceIntPtr master)
         return BadDevice;
 
     /* set from floating to floating? */
-    if (!dev->u.master && !master)
+    if (!dev->u.master && !master && dev->enabled)
         return Success;
 
     /* free the existing sprite. */
@@ -2357,7 +2360,13 @@ AttachDevice(ClientPtr client, DeviceIntPtr dev, DeviceIntPtr master)
      */
     if (!master)
     {
-        WindowPtr currentRoot = dev->spriteInfo->sprite->spriteTrace[0];
+        WindowPtr currentRoot;
+
+        if (dev->spriteInfo->sprite)
+            currentRoot = dev->spriteInfo->sprite->spriteTrace[0];
+        else /* new device auto-set to floating */
+            currentRoot = WindowTable[0];
+
         /* we need to init a fake sprite */
         screen = currentRoot->drawable.pScreen;
         screen->DeviceCursorInitialize(dev, screen);
