diff --git a/hw/xfree86/common/xf86Globals.c b/hw/xfree86/common/xf86Globals.c
index 7102476..489b964 100644
--- a/hw/xfree86/common/xf86Globals.c
+++ b/hw/xfree86/common/xf86Globals.c
@@ -126,7 +126,6 @@ xf86InfoRec xf86Info = {
 #endif
 	TRUE,		/* pmFlag */
 	LogNone,	/* syncLog */
-	0,		/* estimateSizesAggressively */
 	FALSE,		/* kbdCustomKeycodes */
 	FALSE,		/* disableRandR */
 	X_DEFAULT	/* randRFrom */
