diff --git a/test/xi2/protocol-eventconvert.c b/test/xi2/protocol-eventconvert.c
index 9872b79..faa9f40 100644
--- a/test/xi2/protocol-eventconvert.c
+++ b/test/xi2/protocol-eventconvert.c
@@ -1001,10 +1001,12 @@ test_convert_XITouchOwnershipEvent(void)
         test_XITouchOwnershipEvent(&in);
     }
 
-    for (i = 1; i <= 0xFFFFFFFF; i <<= 1)
+    for (i = 1; ; i <<= 1)
     {
         in.touchid = i;
         test_XITouchOwnershipEvent(&in);
+	if (i == (1 << 31))
+	    break;
     }
 }
 
