diff --git a/include/ptrveloc.h b/include/ptrveloc.h
index c14e12d..6ca309c 100644
--- a/include/ptrveloc.h
+++ b/include/ptrveloc.h
@@ -106,7 +106,7 @@ InitVelocityData(DeviceVelocityPtr vel);
 extern _X_EXPORT void
 InitTrackers(DeviceVelocityPtr vel, int ntracker);
 
-extern _X_EXPORT short
+extern _X_EXPORT BOOL
 ProcessVelocityData2D(DeviceVelocityPtr vel, int dx, int dy, int time);
 
 extern _X_EXPORT float
