diff --git a/hw/xfree86/common/xf86Globals.c b/hw/xfree86/common/xf86Globals.c
index f72f1b6..dfc2bd7 100644
--- a/hw/xfree86/common/xf86Globals.c
+++ b/hw/xfree86/common/xf86Globals.c
@@ -98,37 +98,40 @@ InputInfoPtr xf86InputDevs = NULL;
 /* Globals that video drivers may not access */
 
 xf86InfoRec xf86Info = {
-	-1,		/* consoleFd */
-	-1,		/* vtno */
-	FALSE,		/* vtSysreq */
-	SKWhenNeeded,	/* ddxSpecialKeys */
-	-1,		/* lastEventTime */
-	FALSE,		/* vtRequestsPending */
-	FALSE,		/* dontVTSwitch */
-	FALSE,		/* dontZap */
-	FALSE,		/* dontZoom */
-	FALSE,		/* notrapSignals */
-	FALSE,		/* caughtSignal */
-	NULL,		/* currentScreen */
+    .consoleFd                  = -1,
+    .vtno                       = -1,
+    .vtSysreq                   = FALSE,
+    .ddxSpecialKeys             = SKWhenNeeded,
+    .lastEventTime              = -1,
+    .vtRequestsPending          = FALSE,
+    .dontVTSwitch               = FALSE,
+    .dontZap                    = FALSE,
+    .dontZoom                   = FALSE,
+    .notrapSignals              = FALSE,
+    .caughtSignal               = FALSE,
+    .currentScreen              = NULL,
 #ifdef CSRG_BASED
-	-1,		/* screenFd */
-	-1,		/* consType */
+    .screenFd                   = -1,
+    .consType                   = -1,
 #endif
-	FALSE,		/* allowMouseOpenFail */
-	TRUE,		/* vidModeEnabled */
-	FALSE,		/* vidModeAllowNonLocal */
-	TRUE,		/* miscModInDevEnabled */
-	FALSE,		/* miscModInDevAllowNonLocal */
-	Pix24DontCare,	/* pixmap24 */
-	X_DEFAULT,	/* pix24From */
+    .allowMouseOpenFail         = FALSE,
+    .vidModeEnabled             = TRUE,
+    .vidModeAllowNonLocal       = FALSE,
+    .miscModInDevEnabled        = TRUE,
+    .miscModInDevAllowNonLocal  = FALSE,
+    .pixmap24                   = Pix24DontCare,
+    .pix24From                  = X_DEFAULT,
 #ifdef __i386__
-	FALSE,		/* pc98 */
+    .pc98                       = FALSE,
 #endif
-	TRUE,		/* pmFlag */
-	LogNone,	/* syncLog */
-	FALSE,		/* kbdCustomKeycodes */
-	FALSE,		/* disableRandR */
-	X_DEFAULT	/* randRFrom */
+    .pmFlag                     = TRUE,
+    .log                        = LogNone,
+    .kbdCustomKeycodes          = FALSE,
+    .disableRandR               = FALSE,
+    .randRFrom                  = X_DEFAULT,
+    .allowEmptyInput            = TRUE,
+    .autoAddDevices             = TRUE,
+    .autoEnableDevices          = TRUE
 };
 const char *xf86ConfigFile = NULL;
 const char *xf86InputDeviceList = NULL;
