diff --git a/GL/glx/glxext.c b/GL/glx/glxext.c
index 4d6bfd7..55463c7 100644
--- a/GL/glx/glxext.c
+++ b/GL/glx/glxext.c
@@ -27,6 +27,7 @@
 #include "glxserver.h"
 #include <windowstr.h>
 #include <propertyst.h>
+#include "privates.h"
 #include <os.h>
 #include "g_disptab.h"
 #include "unpack.h"
@@ -53,7 +54,7 @@ RESTYPE __glXSwapBarrierRes;
 */
 xGLXSingleReply __glXReply;
 
-static int glxClientPrivateIndex;
+static DevPrivateKey glxClientPrivateKey = &glxClientPrivateKey;
 
 /*
 ** Client that called into GLX dispatch.
@@ -218,7 +219,7 @@ int __glXError(int error)
 __GLXclientState *
 glxGetClient(ClientPtr pClient)
 {
-    return (__GLXclientState *) pClient->devPrivates[glxClientPrivateIndex].ptr;
+    return dixLookupPrivate(&pClient->devPrivates, glxClientPrivateKey);
 }
 
 static void
@@ -288,9 +289,7 @@ void GlxExtensionInit(void)
     __glXDrawableRes = CreateNewResourceType((DeleteType)DrawableGone);
     __glXSwapBarrierRes = CreateNewResourceType((DeleteType)SwapBarrierGone);
 
-    glxClientPrivateIndex = AllocateClientPrivateIndex ();
-    if (!AllocateClientPrivate (glxClientPrivateIndex,
-				sizeof (__GLXclientState)))
+    if (!dixRequestPrivate(glxClientPrivateKey, sizeof (__GLXclientState)))
 	return;
     if (!AddCallback (&ClientStateCallback, glxClientCallback, 0))
 	return;
