diff --git a/cfb/cfbgc.c b/cfb/cfbgc.c
index 3937bd6..9cf53d3 100644
--- a/cfb/cfbgc.c
+++ b/cfb/cfbgc.c
@@ -48,11 +48,15 @@ SOFTWARE.
 
 /* $Xorg: cfbgc.c,v 1.4 2001/02/09 02:04:37 xorgcvs Exp $ */
 
+#ifdef HAVE_DIX_CONFIG_H
+#include <dix-config.h>
+#endif
+
 #include <X11/X.h>
 #include <X11/Xmd.h>
 #include <X11/Xproto.h>
 #include "cfb.h"
-#include "fontstruct.h"
+#include <X11/fonts/fontstruct.h>
 #include "dixfontstr.h"
 #include "gcstruct.h"
 #include "windowstr.h"
