diff --git a/hw/xfree86/common/xf86RandR.c b/hw/xfree86/common/xf86RandR.c
index 26e6297..f9ea4e6 100644
--- a/hw/xfree86/common/xf86RandR.c
+++ b/hw/xfree86/common/xf86RandR.c
@@ -262,7 +262,7 @@ xf86RandRSetConfig (ScreenPtr		pScreen,
     if (pScreen == miPointerCurrentScreen ())
     {
 	if (px < pSize->width && py < pSize->height)
-	    (*pScreen->SetCursorPosition) (pScreen, px, py, FALSE);
+	    (*scrp->PointerMoved) (pScreen->myNum, px, py);
     }
 
     return TRUE;
