diff --git a/dix/events.c b/dix/events.c
index 0157368..658dbc3 100644
--- a/dix/events.c
+++ b/dix/events.c
@@ -1922,12 +1922,6 @@ ReleaseActiveGrabs(ClientPtr client)
 		(*dev->deviceGrab.DeactivateGrab)(dev);
 		done = FALSE;
 	    }
-
-	    if (dev->deviceGrab.grab && SameClient(dev->deviceGrab.grab, client))
-	    {
-		(*dev->deviceGrab.DeactivateGrab)(dev);
-		done = FALSE;
-	    }
 	}
     } while (!done);
 }
