diff --git a/exa/exa.h b/exa/exa.h
index d7219f0..9d9fb35 100644
--- a/exa/exa.h
+++ b/exa/exa.h
@@ -705,6 +705,9 @@ typedef struct _ExaDriver {
                               pointer pPixData);
 
     /** @} */
+    void *(*CreatePixmap2)(ScreenPtr pScreen, int width, int height,
+                           int depth, int usage_hint, int bitsPerPixel);
+
 } ExaDriverRec, *ExaDriverPtr;
 
 /** @name EXA driver flags
