diff --git a/GL/glx/indirect_table.c b/GL/glx/indirect_table.c
index 8671a84..3e294ad 100644
--- a/GL/glx/indirect_table.c
+++ b/GL/glx/indirect_table.c
@@ -703,7 +703,7 @@ static const void *Render_function_table[384][2] = {
     /* [ 353] =  4225 */ {NULL, NULL},
     /* [ 354] =  4226 */ {NULL, NULL},
     /* [ 355] =  4227 */ {NULL, NULL},
-    /* [ 356] =  4228 */ {NULL, NULL},
+    /* [ 356] =  4228 */ {__glXDisp_BlendEquationSeparateEXT, __glXDispSwap_BlendEquationSeparateEXT},
     /* [ 357] =  4229 */ {NULL, NULL},
     /* [ 358] =  4230 */ {__glXDisp_VertexAttrib4bvARB, __glXDispSwap_VertexAttrib4bvARB},
     /* [ 359] =  4231 */ {__glXDisp_VertexAttrib4ivARB, __glXDispSwap_VertexAttrib4ivARB},
@@ -1090,7 +1090,7 @@ static const int_fast16_t Render_size_table[384][2] = {
     /* [353] =  4225 */ {  0, ~0},
     /* [354] =  4226 */ {  0, ~0},
     /* [355] =  4227 */ {  0, ~0},
-    /* [356] =  4228 */ {  0, ~0},
+    /* [356] =  4228 */ { 12, ~0},
     /* [357] =  4229 */ {  0, ~0},
     /* [358] =  4230 */ { 12, ~0},
     /* [359] =  4231 */ { 24, ~0},
