diff --git a/hw/dmx/dmxinit.c b/hw/dmx/dmxinit.c
index e5598e3..f481cf5 100644
--- a/hw/dmx/dmxinit.c
+++ b/hw/dmx/dmxinit.c
@@ -846,6 +846,12 @@ void AbortDDX(void)
     }
 }
 
+#ifdef DDXBEFORERESET
+void ddxBeforeReset(void)
+{
+}
+#endif
+
 /** This function is called in Xserver/dix/main.c from \a main() when
  * dispatchException & DE_TERMINATE (which is the only way to exit the
  * main loop without an interruption. */
