diff --git a/afb/afbbresd.c b/afb/afbbresd.c
index 13cbf6a..a5a9cd1 100644
--- a/afb/afbbresd.c
+++ b/afb/afbbresd.c
@@ -48,6 +48,10 @@ SOFTWARE.
 ******************************************************************/
 /* $XConsortium: afbbresd.c,v 1.10 94/04/17 20:28:18 dpw Exp $ */
 
+#ifdef HAVE_DIX_CONFIG_H
+#include <dix-config.h>
+#endif
+
 #include <X11/X.h>
 #include <X11/misc.h>
 #include "afb.h"
