diff --git a/mi/mieq.c b/mi/mieq.c
index cca565a..e7b7ff0 100644
--- a/mi/mieq.c
+++ b/mi/mieq.c
@@ -391,16 +391,13 @@ mieqProcessInputEvents(void)
 
             /* If someone's registered a custom event handler, let them
              * steal it. */
-#if 0
             if (handler)
             {
-                /* FIXME: this is broken now, InternalEvents! */
-                handler(DequeueScreen(dev)->myNum, event, dev, 1);
+                handler(DequeueScreen(dev)->myNum, event, dev);
                 if (master)
                     handler(DequeueScreen(master)->myNum,
-                            masterEvents->event, master, 1);
+                            (InternalEvent*)masterEvents->event, master);
             } else
-#endif
             {
                 /* process slave first, then master */
                 dev->public.processInputProc(event, dev);
