diff --git a/os/utils.c b/os/utils.c
index bf52ea0..709b5df 100644
--- a/os/utils.c
+++ b/os/utils.c
@@ -533,6 +533,7 @@ void UseMsg(void)
 #ifdef RENDER
     ErrorF("-render [default|mono|gray|color] set render color alloc policy\n");
 #endif
+    ErrorF("-retro                 start with classic stipple and cursor\n");
     ErrorF("-s #                   screen-saver timeout (minutes)\n");
     ErrorF("-t #                   mouse threshold (pixels)\n");
     ErrorF("-terminate             terminate at server reset\n");
@@ -641,8 +642,7 @@ ProcessCommandLine(int argc, char *argv[])
 	    else
 		UseMsg();
 	}
-	else if ( strcmp( argv[i], "-br") == 0)
-	    blackRoot = TRUE;
+	else if ( strcmp( argv[i], "-br") == 0) ; /* default */
 	else if ( strcmp( argv[i], "+bs") == 0)
 	    enableBackingStore = TRUE;
 	else if ( strcmp( argv[i], "-bs") == 0)
@@ -828,6 +828,8 @@ ProcessCommandLine(int argc, char *argv[])
 	    defaultKeyboardControl.autoRepeat = TRUE;
 	else if ( strcmp( argv[i], "-r") == 0)
 	    defaultKeyboardControl.autoRepeat = FALSE;
+	else if ( strcmp( argv[i], "-retro") == 0)
+	    party_like_its_1989 = TRUE;
 	else if ( strcmp( argv[i], "-s") == 0)
 	{
 	    if(++i < argc)
