diff --git a/mi/midispcur.c b/mi/midispcur.c
index 71b374f..597d272 100644
--- a/mi/midispcur.c
+++ b/mi/midispcur.c
@@ -184,7 +184,9 @@ miDCCloseScreen (index, pScreen)
     tossPix (pScreenPriv->pSave);
     tossPix (pScreenPriv->pTemp);
 #ifdef ARGB_CURSOR
+#if 0				/* This has been free()d before */
     tossPict (pScreenPriv->pRootPicture);
+#endif 
     tossPict (pScreenPriv->pTempPicture);
 #endif
     xfree ((pointer) pScreenPriv);
