diff --git a/hw/kdrive/ati/ati_video.c b/hw/kdrive/ati/ati_video.c
index 3960e6e..8e47d9e 100644
--- a/hw/kdrive/ati/ati_video.c
+++ b/hw/kdrive/ati/ati_video.c
@@ -217,11 +217,13 @@ R128DisplayVideo(KdScreenInfo *screen, ATIPortPrivPtr pPortPriv)
 	OUT_REG(R128_REG_SCALE_3D_DATATYPE, srcDatatype);
 
 	OUT_RING(DMA_PACKET0(R128_REG_SCALE_PITCH, 5));
-	OUT_RING(pPortPriv->src_pitch / 16);
-	OUT_RING((pPortPriv->src_w << 16) / pPortPriv->dst_w);
-	OUT_RING((pPortPriv->src_h << 16) / pPortPriv->dst_h);
-	OUT_RING(0x0);
-	OUT_RING(0x0);
+	OUT_RING_REG(R128_REG_SCALE_PITCH, pPortPriv->src_pitch / 16);
+	OUT_RING_REG(R128_REG_SCALE_X_INC,
+	    (pPortPriv->src_w << 16) / pPortPriv->dst_w);
+	OUT_RING_REG(R128_REG_SCALE_Y_INC,
+	    (pPortPriv->src_h << 16) / pPortPriv->dst_h);
+	OUT_RING_REG(R128_REG_SCALE_HACC, 0x0);
+	OUT_RING_REG(R128_REG_SCALE_VACC, 0x0);
 
 	END_DMA();
 
@@ -240,19 +242,16 @@ R128DisplayVideo(KdScreenInfo *screen, ATIPortPrivPtr pPortPriv)
 		srch = pPortPriv->src_h - srcY;
 
 		BEGIN_DMA(6);
-		/* R128_REG_SCALE_SRC_HEIGHT_WIDTH,
-		 * R128_REG_SCALE_OFFSET_0
-		 */
 		OUT_RING(DMA_PACKET0(R128_REG_SCALE_SRC_HEIGHT_WIDTH, 2));
-		OUT_RING((srch << 16) | srcw);
-		OUT_RING(pPortPriv->src_offset + srcY * pPortPriv->src_pitch +
-		    srcX * 2);
-		/* R128_REG_SCALE_DST_X_Y
-		 * R128_REG_SCALE_DST_HEIGHT_WIDTH
-		 */
+		OUT_RING_REG(R128_REG_SCALE_SRC_HEIGHT_WIDTH,
+		    (srch << 16) | srcw);
+		OUT_RING_REG(R128_REG_SCALE_OFFSET_0, pPortPriv->src_offset +
+		    srcY * pPortPriv->src_pitch + srcX * 2);
+
 		OUT_RING(DMA_PACKET0(R128_REG_SCALE_DST_X_Y, 2));
-		OUT_RING((dstX << 16) | dstY);
-		OUT_RING((dsth << 16) | dstw);
+		OUT_RING_REG(R128_REG_SCALE_DST_X_Y, (dstX << 16) | dstY);
+		OUT_RING_REG(R128_REG_SCALE_DST_HEIGHT_WIDTH,
+		    (dsth << 16) | dstw);
 		END_DMA();
 		pBox++;
 	}
@@ -340,14 +339,12 @@ RadeonDisplayVideo(KdScreenInfo *screen, ATIPortPrivPtr pPortPriv)
 
 	BEGIN_DMA(8);
 
-	/* RADEON_REG_PP_CNTL,
-	 * RADEON_REG_RB3D_CNTL, 
-	 * RADEON_REG_RB3D_COLOROFFSET
-	 */
 	OUT_RING(DMA_PACKET0(RADEON_REG_PP_CNTL, 3));
-	OUT_RING(RADEON_TEX_0_ENABLE | RADEON_TEX_BLEND_0_ENABLE);
-	OUT_RING(dst_format | RADEON_ALPHA_BLEND_ENABLE);
-	OUT_RING(dst_offset);
+	OUT_RING_REG(RADEON_REG_PP_CNTL,
+	    RADEON_TEX_0_ENABLE | RADEON_TEX_BLEND_0_ENABLE);
+	OUT_RING_REG(RADEON_REG_RB3D_CNTL,
+	    dst_format | RADEON_ALPHA_BLEND_ENABLE);
+	OUT_RING_REG(RADEON_REG_RB3D_COLOROFFSET, dst_offset);
 
 	OUT_REG(RADEON_REG_RB3D_COLORPITCH, dst_pitch >> pixel_shift);
 
@@ -363,79 +360,66 @@ RadeonDisplayVideo(KdScreenInfo *screen, ATIPortPrivPtr pPortPriv)
 		OUT_REG(R200_REG_SE_VTX_FMT_1,
 		    (2 << R200_VTX_TEX0_COMP_CNT_SHIFT));
 
-		/* R200_REG_PP_TXFILTER_0,
-		 * R200_REG_PP_TXFORMAT_0,
-		 * R200_REG_PP_TXFORMAT_X_0,
-		 * R200_REG_PP_TXSIZE_0,
-		 * R200_REG_PP_TXPITCH_0
-		 */
 		OUT_RING(DMA_PACKET0(R200_REG_PP_TXFILTER_0, 5));
-		OUT_RING(R200_MAG_FILTER_LINEAR |
+		OUT_RING_REG(R200_REG_PP_TXFILTER_0,
+		    R200_MAG_FILTER_LINEAR |
 		    R200_MIN_FILTER_LINEAR |
 		    R200_YUV_TO_RGB);
-		OUT_RING(txformat);
-		OUT_RING(0);
-		OUT_RING((pPixmap->drawable.width - 1) |
+		OUT_RING_REG(R200_REG_PP_TXFORMAT_0, txformat);
+		OUT_RING_REG(R200_REG_PP_TXFORMAT_X_0, 0);
+		OUT_RING_REG(R200_REG_PP_TXSIZE_0,
+		    (pPixmap->drawable.width - 1) |
 		    ((pPixmap->drawable.height - 1) << RADEON_TEX_VSIZE_SHIFT));
-		OUT_RING(pPortPriv->src_pitch - 32);
+		OUT_RING_REG(R200_REG_PP_TXPITCH_0, pPortPriv->src_pitch - 32);
 
 		OUT_REG(R200_PP_TXOFFSET_0, pPortPriv->src_offset);
 
-		/* R200_REG_PP_TXCBLEND_0,
-		 * R200_REG_PP_TXCBLEND2_0
-		 * R200_REG_PP_TXABLEND_0
-		 * R200_REG_PP_TXABLEND2_0
-		 */
 		OUT_RING(DMA_PACKET0(R200_REG_PP_TXCBLEND_0, 4));
-		OUT_RING(
+		OUT_RING_REG(R200_REG_PP_TXCBLEND_0,
 		    R200_TXC_ARG_A_ZERO |
 		    R200_TXC_ARG_B_ZERO |
 		    R200_TXC_ARG_C_R0_COLOR |
 		    R200_TXC_OP_MADD);
-		OUT_RING(R200_TXC_CLAMP_0_1 | R200_TXC_OUTPUT_REG_R0);
-		OUT_RING(
+		OUT_RING_REG(R200_REG_PP_TXCBLEND2_0,
+		    R200_TXC_CLAMP_0_1 | R200_TXC_OUTPUT_REG_R0);
+		OUT_RING_REG(R200_REG_PP_TXABLEND_0,
 		    R200_TXA_ARG_A_ZERO |
 		    R200_TXA_ARG_B_ZERO |
 		    R200_TXA_ARG_C_R0_ALPHA |
 		    R200_TXA_OP_MADD);
-		OUT_RING(R200_TXA_CLAMP_0_1 | R200_TXA_OUTPUT_REG_R0);
+		OUT_RING_REG(R200_REG_PP_TXABLEND2_0,
+		    R200_TXA_CLAMP_0_1 | R200_TXA_OUTPUT_REG_R0);
 
 		END_DMA();
 	} else {
-		BEGIN_DMA(11);
+		BEGIN_DMA(9);
 
-		/* RADEON_REG_PP_TXFILTER_0,
-		 * RADEON_REG_PP_TXFORMAT_0,
-		 * RADEON_REG_PP_TXOFFSET_0
-		 */
-		OUT_RING(DMA_PACKET0(RADEON_REG_PP_TXFILTER_0, 3));
-		OUT_RING(RADEON_MAG_FILTER_LINEAR |
+		OUT_RING(DMA_PACKET0(RADEON_REG_PP_TXFILTER_0, 5));
+		OUT_RING_REG(RADEON_REG_PP_TXFILTER_0, RADEON_MAG_FILTER_LINEAR |
 		    RADEON_MIN_FILTER_LINEAR |
 		    RADEON_YUV_TO_RGB);
-		OUT_RING(txformat);
-		OUT_RING(pPortPriv->src_offset);
-
-		/* RADEON_REG_PP_TEX_SIZE_0,
-		 * RADEON_REG_PP_TEX_PITCH_0
-		 */
-		OUT_RING(DMA_PACKET0(RADEON_REG_PP_TEX_SIZE_0, 2));
-		OUT_RING((pPixmap->drawable.width - 1) |
-		    ((pPixmap->drawable.height - 1) << RADEON_TEX_VSIZE_SHIFT));
-		OUT_RING(pPortPriv->src_pitch - 32);
-
-		OUT_REG(RADEON_REG_PP_TXCBLEND_0,
+		OUT_RING_REG(RADEON_REG_PP_TXFORMAT_0, txformat);
+		OUT_RING_REG(RADEON_REG_PP_TXOFFSET_0, pPortPriv->src_offset);
+		OUT_RING_REG(RADEON_REG_PP_TXCBLEND_0,
 		    RADEON_COLOR_ARG_A_ZERO |
 		    RADEON_COLOR_ARG_B_ZERO |
 		    RADEON_COLOR_ARG_C_T0_COLOR |
 		    RADEON_BLEND_CTL_ADD |
 		    RADEON_CLAMP_TX);
-		OUT_REG(RADEON_REG_PP_TXABLEND_0,
+		OUT_RING_REG(RADEON_REG_PP_TXABLEND_0,
 		    RADEON_ALPHA_ARG_A_ZERO |
 		    RADEON_ALPHA_ARG_B_ZERO |
 		    RADEON_ALPHA_ARG_C_T0_ALPHA |
 		    RADEON_BLEND_CTL_ADD |
 		    RADEON_CLAMP_TX);
 
+		OUT_RING(DMA_PACKET0(RADEON_REG_PP_TEX_SIZE_0, 2));
+		OUT_RING_REG(RADEON_REG_PP_TEX_SIZE_0,
+		    (pPixmap->drawable.width - 1) |
+		    ((pPixmap->drawable.height - 1) << RADEON_TEX_VSIZE_SHIFT));
+		OUT_RING_REG(RADEON_REG_PP_TEX_PITCH_0,
+		    pPortPriv->src_pitch - 32);
+
 		END_DMA();
 	}
 
