diff --git a/composite/compositeext.h b/composite/compositeext.h
index 600604d..0b148f0 100644
--- a/composite/compositeext.h
+++ b/composite/compositeext.h
@@ -35,4 +35,6 @@ extern _X_EXPORT Bool CompositeRegisterAlternateVisuals(ScreenPtr pScreen,
                                                         VisualID * vids,
                                                         int nVisuals);
 
+extern _X_EXPORT RESTYPE CompositeClientWindowType;
+
 #endif                          /* _COMPOSITEEXT_H_ */
