diff --git a/hw/dmx/dmxextension.c b/hw/dmx/dmxextension.c
index feadb5e..a047b6a 100644
--- a/hw/dmx/dmxextension.c
+++ b/hw/dmx/dmxextension.c
@@ -65,6 +65,9 @@
 #include <X11/extensions/dmxproto.h>  /* For DMX_BAD_* */
 #include "cursorstr.h"
 
+/* XXX nasty hack for MIT-SCREEN-SAVER.  Please fix me properly. */
+const Bool screenSaverSuspended = FALSE;
+
 /* The default font is declared in dix/globals.c, but is not included in
  * _any_ header files. */
 extern FontPtr  defaultFont;
