diff --git a/hw/xquartz/quartz.c b/hw/xquartz/quartz.c
index c4b889e..b17e561 100644
--- a/hw/xquartz/quartz.c
+++ b/hw/xquartz/quartz.c
@@ -71,7 +71,8 @@ int                     quartzUseAGL = 1;
 int                     quartzEnableKeyEquivalents = 1;
 int                     quartzServerVisible = TRUE;
 int                     quartzServerQuitting = FALSE;
-DevPrivateKey           quartzScreenKey = &quartzScreenKey;
+static int              quartzScreenKeyIndex;
+DevPrivateKey           quartzScreenKey = &quartzScreenKeyIndex;
 int                     aquaMenuBarHeight = 0;
 QuartzModeProcsPtr      quartzProcs = NULL;
 const char             *quartzOpenGLBundle = NULL;
