diff --git a/dix/devices.c b/dix/devices.c
index fe70e78..656707c 100644
--- a/dix/devices.c
+++ b/dix/devices.c
@@ -830,12 +830,12 @@ CloseDownDevices(void)
     for (dev = inputInfo.devices; dev; dev = next)
     {
 	next = dev->next;
-	CloseDevice(dev);
+        DeleteInputDeviceRequest(dev);
     }
     for (dev = inputInfo.off_devices; dev; dev = next)
     {
 	next = dev->next;
-	CloseDevice(dev);
+        DeleteInputDeviceRequest(dev);
     }
 
     inputInfo.devices = NULL;
