diff --git a/hw/xfree86/parser/xf86Optrec.h b/hw/xfree86/parser/xf86Optrec.h
index 61a8c5f..b8e9d61 100644
--- a/hw/xfree86/parser/xf86Optrec.h
+++ b/hw/xfree86/parser/xf86Optrec.h
@@ -77,7 +77,7 @@ extern _X_EXPORT char *xf86optionValue(XF86OptionPtr opt);
 extern _X_EXPORT XF86OptionPtr xf86newOption(char *name, char *value);
 extern _X_EXPORT XF86OptionPtr xf86nextOption(XF86OptionPtr list);
 extern _X_EXPORT XF86OptionPtr xf86findOption(XF86OptionPtr list, const char *name);
-extern _X_EXPORT char *xf86findOptionValue(XF86OptionPtr list, const char *name);
+extern _X_EXPORT const char *xf86findOptionValue(XF86OptionPtr list, const char *name);
 extern _X_EXPORT XF86OptionPtr xf86optionListCreate(const char **options, int count, int used);
 extern _X_EXPORT XF86OptionPtr xf86optionListMerge(XF86OptionPtr head, XF86OptionPtr tail);
 extern _X_EXPORT int xf86nameCompare (const char *s1, const char *s2);
