diff --git a/hw/xfree86/xf8_16bpp/cfb8_16.h b/hw/xfree86/xf8_16bpp/cfb8_16.h
index 3a3a68d..d847a60 100644
--- a/hw/xfree86/xf8_16bpp/cfb8_16.h
+++ b/hw/xfree86/xf8_16bpp/cfb8_16.h
@@ -28,7 +28,7 @@
 
 /* this has to stay misnamed for ABI reasons */
 
-_X_EXPORT Bool
+extern _X_EXPORT Bool
 cfb8_16ScreenInit(ScreenPtr pScreen, pointer pbits16, pointer pbits8,
                   int xsize, int ysize, int dpix, int dpiy,
                   int width16, int width8);
