diff --git a/damageext/damageextint.h b/damageext/damageextint.h
index 6f14e4e..1ed07de 100644
--- a/damageext/damageextint.h
+++ b/damageext/damageextint.h
@@ -44,13 +44,6 @@
 #include "damage.h" 
 #include "xfixes.h"
 
-extern unsigned char	DamageReqCode;
-extern int		DamageEventBase;
-extern int		DamageErrorBase;
-extern int		DamageClientPrivateIndex;
-extern RESTYPE		DamageExtType;
-extern RESTYPE		DamageExtWinType;
-
 typedef struct _DamageClient {
     CARD32	major_version;
     CARD32	minor_version;
@@ -67,9 +60,6 @@ typedef struct _DamageExt {
     XID			id;
 } DamageExtRec, *DamageExtPtr;
 
-extern int	(*ProcDamageVector[/*XDamageNumberRequests*/])(ClientPtr);
-extern int	(*SProcDamageVector[/*XDamageNumberRequests*/])(ClientPtr);
-
 #define VERIFY_DAMAGEEXT(pDamageExt, rid, client, mode) { \
     pDamageExt = SecurityLookupIDByType (client, rid, DamageExtType, mode); \
     if (!pDamageExt) { \
@@ -79,10 +69,6 @@ extern int	(*SProcDamageVector[/*XDamageNumberRequests*/])(ClientPtr);
 }
 
 void
-SDamageNotifyEvent (xDamageNotifyEvent *from,
-		    xDamageNotifyEvent *to);
-
-void
 DamageExtSetCritical (ClientPtr pClient, Bool critical);
 
 #endif /* _DAMAGEEXTINT_H_ */
