diff --git a/ilbm/ilbm.h b/ilbm/ilbm.h
index 7e015eb..eba35c2 100644
--- a/ilbm/ilbm.h
+++ b/ilbm/ilbm.h
@@ -55,6 +55,10 @@ SOFTWARE.
 /* Modified jun 95 by Geert Uytterhoeven (Geert.Uytterhoeven@cs.kuleuven.ac.be)
    to use interleaved bitplanes instead of normal bitplanes */
 
+#ifdef HAVE_DIX_CONFIG_H
+#include <dix-config.h>
+#endif
+
 #include "pixmap.h"
 #include "region.h"
 #include "gc.h"
