diff --git a/Xi/setbmap.h b/Xi/setbmap.h
index dab5fcb..20ad8e0 100644
--- a/Xi/setbmap.h
+++ b/Xi/setbmap.h
@@ -1,4 +1,3 @@
-/* $XFree86: xc/programs/Xserver/Xi/setbmap.h,v 3.1 1996/04/15 11:19:03 dawes Exp $ */
 /************************************************************
 
 Copyright 1996 by Thomas E. Dickey <dickey@clark.net>
@@ -31,19 +30,13 @@ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 #ifndef SETBMAP_H
 #define SETBMAP_H 1
 
-int
-SProcXSetDeviceButtonMapping(
-	ClientPtr              /* client */
+int SProcXSetDeviceButtonMapping(ClientPtr	/* client */
 	);
 
-int
-ProcXSetDeviceButtonMapping(
-	ClientPtr              /* client */
+int ProcXSetDeviceButtonMapping(ClientPtr	/* client */
 	);
 
-void
-SRepXSetDeviceButtonMapping(
-	ClientPtr              /* client */,
+void SRepXSetDeviceButtonMapping(ClientPtr /* client */ ,
 	int                    /* size */,
 	xSetDeviceButtonMappingReply * /* rep */
 	);
