diff --git a/dix/swaprep.c b/dix/swaprep.c
index 493def5..9eb6765 100644
--- a/dix/swaprep.c
+++ b/dix/swaprep.c
@@ -99,7 +99,7 @@ CopySwap32Write(ClientPtr pClient, int size, CARD32 *pbuf)
     CARD32 tmpbuf[1];
     
     /* Allocate as big a buffer as we can... */
-    while (!(pbufT = (CARD32 *) xalloc(bufsize)))
+    while (!(pbufT = xalloc(bufsize)))
     {
         bufsize >>= 1;
 	if (bufsize == 4)
@@ -131,7 +131,7 @@ CopySwap32Write(ClientPtr pClient, int size, CARD32 *pbuf)
 	}
 
     if (pbufT != tmpbuf)
-	xfree ((char *) pbufT);
+	xfree (pbufT);
 }
 
 /**
@@ -147,7 +147,7 @@ CopySwap16Write(ClientPtr pClient, int size, short *pbuf)
     short tmpbuf[2];
     
     /* Allocate as big a buffer as we can... */
-    while (!(pbufT = (short *) xalloc(bufsize)))
+    while (!(pbufT = xalloc(bufsize)))
     {
         bufsize >>= 1;
 	if (bufsize == 4)
@@ -179,7 +179,7 @@ CopySwap16Write(ClientPtr pClient, int size, short *pbuf)
 	}
 
     if (pbufT != tmpbuf)
-	xfree ((char *) pbufT);
+	xfree (pbufT);
 }
 
 
@@ -1262,7 +1262,7 @@ WriteSConnectionInfo(ClientPtr pClient, unsigned long size, char *pInfo)
 {
     char	*pInfoTBase;
 
-    pInfoTBase = (char *) xalloc(size);
+    pInfoTBase = xalloc(size);
     if (!pInfoTBase)
     {
 	pClient->noClientException = -1;
