diff --git a/dix/devices.c b/dix/devices.c
index df53613..0a68c0e 100644
--- a/dix/devices.c
+++ b/dix/devices.c
@@ -83,7 +83,6 @@ SOFTWARE.
 #include "exglobals.h"
 #include "exevents.h"
 #include "listdev.h" /* for CopySwapXXXClass */
-#include "xiproperty.h"
 
 /** @file
  * This file handles input device-related stuff.
@@ -174,12 +173,6 @@ AddInputDevice(ClientPtr client, DeviceProc deviceProc, Bool autoStart)
     memset(dev->last.valuators, 0, sizeof(dev->last.valuators));
     dev->last.numValuators = 0;
 
-    /* device properties */
-    dev->properties  = NULL;
-    dev->SetProperty = NULL;
-    dev->GetProperty = NULL;
-    dev->pendingProperties = FALSE;
-
     /*  security creation/labeling check
      */
     if (XaceHook(XACE_DEVICE_ACCESS, client, dev, DixCreateAccess)) {
@@ -777,8 +770,6 @@ CloseDevice(DeviceIntPtr dev)
     if (!dev)
         return;
 
-    XIDeleteAllDeviceProperties(dev);
-
     if (dev->inited)
 	(void)(*dev->deviceProc)(dev, DEVICE_CLOSE);
 
