diff --git a/dix/dixfonts.c b/dix/dixfonts.c
index cb0df3c..ef23fa8 100644
--- a/dix/dixfonts.c
+++ b/dix/dixfonts.c
@@ -1889,7 +1889,7 @@ InitFonts ()
 {
     patternCache = MakeFontPatternCache();
 
-#ifndef KDRIVESERVER
+#ifndef BUILTIN_FONTS
     if (screenInfo.numScreens > screenInfo.numVideoScreens) {
 	PrinterFontRegisterFpeFunctions();
 	FontFileCheckRegisterFpeFunctions();
@@ -1897,10 +1897,11 @@ InitFonts ()
     } else 
 #endif
     {
-#ifdef KDRIVESERVER
+#ifdef BUILTIN_FONTS
 	BuiltinRegisterFpeFunctions();
-#endif
+#else
 	FontFileRegisterFpeFunctions();
+#endif
 #ifndef NOFONTSERVERACCESS
 	fs_register_fpe_functions();
 #endif
