diff --git a/hw/xwin/winmultiwindowclass.h b/hw/xwin/winmultiwindowclass.h
index c635ab2..1c9a5e5 100644
--- a/hw/xwin/winmultiwindowclass.h
+++ b/hw/xwin/winmultiwindowclass.h
@@ -47,6 +47,16 @@ typedef struct {
   /* this structure may be extended in the future */
 } WinXWMHints;
 
+/* Window manager hints mask bits */
+#define	InputHint	(1L << 0)
+#define	StateHint	(1L << 1)
+#define	IconPixmapHint	(1L << 2)
+#define	IconWindowHint	(1L << 3)
+#define	IconPositionHint	(1L << 4)
+#define	IconMaskHint	(1L << 5)
+#define	WindowGroupHint	(1L << 6)
+#define	UrgencyHint	(1L << 8)
+#define	AllHints 	(InputHint|StateHint|IconPixmapHint|IconWindowHint|IconPositionHint|IconMaskHint|WindowGroupHint)
 
 /*
  * new version containing base_width, base_height, and win_gravity fields;
