diff --git a/hw/xprint/pcl/PclFonts.c b/hw/xprint/pcl/PclFonts.c
index 716c48d..9e993f7 100644
--- a/hw/xprint/pcl/PclFonts.c
+++ b/hw/xprint/pcl/PclFonts.c
@@ -12,7 +12,6 @@
 **    *********************************************************
 ** 
 ********************************************************************/
-/* $XFree86$ */
 /*
 (c) Copyright 1996 Hewlett-Packard Company
 (c) Copyright 1996 International Business Machines Corp.
