diff --git a/GL/glx/g_renderswap.c b/GL/glx/g_renderswap.c
index e22494e..f485a26 100644
--- a/GL/glx/g_renderswap.c
+++ b/GL/glx/g_renderswap.c
@@ -3339,7 +3339,7 @@ void __glXDispSwap_PointParameterfvARB(GLbyte *pc)
 
 	__GLX_SWAP_INT(pc + 0);
 	pname = *(GLenum *)(pc + 0);
-	compsize = __glPointParameterfvARB_size(pname);
+	compsize = __glPointParameterfvEXT_size(pname);
 	if (compsize < 0) compsize = 0;
 	__GLX_SWAP_FLOAT_ARRAY(pc + 4, compsize);
 
