diff --git a/hw/dmx/input/usb-other.c b/hw/dmx/input/usb-other.c
index 9ac9e16..9bb2842 100644
--- a/hw/dmx/input/usb-other.c
+++ b/hw/dmx/input/usb-other.c
@@ -142,7 +142,7 @@ void othUSBGetInfo(DevicePtr pDev, DMXLocalInitInfoPtr info)
                 } else
                     info->numAbsAxes     = priv->numAbs;
                 for (j = 0; j < info->numAbsAxes; j++) {
-                    ioctl(priv->fd, EVIOCGABS(j), abs);
+                    ioctl(priv->fd, EVIOCGABS(j), absolute);
                     info->minval[1+j]    = absolute[1];
                     info->maxval[1+j]    = absolute[2];
                     info->res[1+j]       = absolute[3];
