diff --git a/Xext/xace.c b/Xext/xace.c
index 14a5e79..7f7944a 100644
--- a/Xext/xace.c
+++ b/Xext/xace.c
@@ -406,12 +406,14 @@ XaceCensorImage(client, pVisibleRegion, widthBytesLine, pDraw, x, y, w, h,
     unsigned int format;
     char * pBuf;
 {
-    ScreenPtr pScreen = pDraw->pScreen;
+    ScreenPtr pScreen;
     RegionRec imageRegion;  /* region representing x,y,w,h */
     RegionRec censorRegion; /* region to obliterate */
     BoxRec imageBox;
     int nRects;
 
+    pScreen = pDraw->pScreen;
+
     imageBox.x1 = x;
     imageBox.y1 = y;
     imageBox.x2 = x + w;
