diff --git a/hw/xgl/xglget.c b/hw/xgl/xglget.c
index 70f29e2..a88e0fe 100644
--- a/hw/xgl/xglget.c
+++ b/hw/xgl/xglget.c
@@ -45,6 +45,9 @@ xglGetImage (DrawablePtr   pDrawable,
     /* Many apps use GetImage to sync with the visable frame buffer */
     if (pDrawable->type == DRAWABLE_WINDOW)
     {
+	if (!xglSyncSurface (&pScreenPriv->pScreenPixmap->drawable))
+	    FatalError (XGL_SW_FAILURE_STRING);
+
 	glitz_surface_flush (pScreenPriv->surface);
 	glitz_drawable_finish (pScreenPriv->drawable);
     }
