diff --git a/hw/xgl/xgl.h b/hw/xgl/xgl.h
index cd4cec1..5710bbf 100644
--- a/hw/xgl/xgl.h
+++ b/hw/xgl/xgl.h
@@ -293,8 +293,6 @@ typedef struct _xglScreen {
     RealizeGlyphProcPtr		  RealizeGlyph;
     UnrealizeGlyphProcPtr	  UnrealizeGlyph;
 #endif
-
-    BSFuncRec			  BackingStoreFuncs;
 } xglScreenRec, *xglScreenPtr;
 
 extern int xglScreenPrivateIndex;
@@ -1110,23 +1108,6 @@ xglSetWindowPixmap (WindowPtr pWin,
 		    PixmapPtr pPixmap);
 
 
-/* xglbstore.c */
-
-void
-xglSaveAreas (PixmapPtr	pPixmap,
-	      RegionPtr	prgnSave,
-	      int	xorg,
-	      int	yorg,
-	      WindowPtr	pWin);
-
-void
-xglRestoreAreas (PixmapPtr pPixmap,
-		 RegionPtr prgnRestore,
-		 int	   xorg,
-		 int	   yorg,
-		 WindowPtr pWin);
-
-
 /* xglget.c */
 
 void
