diff --git a/hw/xfree86/modes/xf86cvt.c b/hw/xfree86/modes/xf86cvt.c
index 244f3b9..d5feccc 100644
--- a/hw/xfree86/modes/xf86cvt.c
+++ b/hw/xfree86/modes/xf86cvt.c
@@ -152,7 +152,7 @@ xf86CVTMode(int HDisplay, int VDisplay, float VRefresh, Bool Reduced,
 
     if (!Reduced) { /* simplified GTF calculation */
 
-        /* 4) Minimum time of vertical sync + back porch interval (µs) 
+        /* 4) Minimum time of vertical sync + back porch interval (Âµs) 
          * default 550.0 */
 #define CVT_MIN_VSYNC_BP 550.0
 
@@ -223,7 +223,7 @@ xf86CVTMode(int HDisplay, int VDisplay, float VRefresh, Bool Reduced,
         Mode->VSyncEnd = Mode->VSyncStart + VSync;
 
     } else { /* Reduced blanking */
-        /* Minimum vertical blanking interval time (µs) - default 460 */
+        /* Minimum vertical blanking interval time (Âµs) - default 460 */
 #define CVT_RB_MIN_VBLANK 460.0
 
         /* Fixed number of clocks for horizontal sync */
