diff --git a/Xext/xevie.c b/Xext/xevie.c
index bbbf045..292f207 100644
--- a/Xext/xevie.c
+++ b/Xext/xevie.c
@@ -77,9 +77,11 @@ DeviceIntPtr		xeviemouse = NULL;
 Mask			xevieMask = 0;
 int       		xevieEventSent = 0;
 int			xevieKBEventSent = 0;
-static DevPrivateKey    xevieDevicePrivateKey = &xevieDevicePrivateKey;
 static Bool             xevieModifiersOn = FALSE;
 
+static int xevieDevicePrivateKeyIndex;
+static DevPrivateKey xevieDevicePrivateKey = &xevieDevicePrivateKeyIndex;
+
 #define XEVIEINFO(dev)  ((xevieDeviceInfoPtr) \
     dixLookupPrivate(&(dev)->devPrivates, xevieDevicePrivateKey))
 
