diff --git a/hw/dmx/glxProxy/unpack.h b/hw/dmx/glxProxy/unpack.h
index 880f9ae..f34b6fd 100644
--- a/hw/dmx/glxProxy/unpack.h
+++ b/hw/dmx/glxProxy/unpack.h
@@ -138,7 +138,9 @@ extern xGLXSingleReply __glXReply;
 ** conceivably be replaced with routines that do the job faster.
 */
 #define __GLX_DECLARE_SWAP_VARIABLES \
-	GLbyte sw; \
+	GLbyte sw
+
+#define __GLX_DECLARE_SWAP_ARRAY_VARIABLES \
   	GLbyte *swapPC;		\
   	GLbyte *swapEnd
 
