diff --git a/Xext/panoramiXsrv.h b/Xext/panoramiXsrv.h
index 6fc903b..39d4952 100644
--- a/Xext/panoramiXsrv.h
+++ b/Xext/panoramiXsrv.h
@@ -19,7 +19,7 @@ extern _X_EXPORT PanoramiXRes * PanoramiXFindIDByScrnum(RESTYPE, XID, int);
 extern _X_EXPORT Bool XineramaRegisterConnectionBlockCallback(void (*func)(void));
 extern _X_EXPORT int XineramaDeleteResource(pointer, XID);
 
-extern _X_EXPORT void XineramaReinitData(ScreenPtr);
+extern _X_EXPORT void XineramaReinitData(void);
 
 extern _X_EXPORT RESTYPE XRC_DRAWABLE;
 extern _X_EXPORT RESTYPE XRT_WINDOW;
