diff --git a/hw/kdrive/mach64/mach64.h b/hw/kdrive/mach64/mach64.h
index ca042c2..5757cf7 100644
--- a/hw/kdrive/mach64/mach64.h
+++ b/hw/kdrive/mach64/mach64.h
@@ -570,6 +570,7 @@ typedef struct _mach64ScreenInfo {
     Mach64Cursor		cursor;
     CARD32			colorKey;
     KdVideoAdaptorPtr		pAdaptor;
+    KaaScreenInfoRec		kaa;
 } Mach64ScreenInfo;
 
 #define getMach64ScreenInfo(kd) ((Mach64ScreenInfo *) ((kd)->screen->driver))
@@ -624,9 +625,6 @@ void
 mach64DrawEnable (ScreenPtr pScreen);
 
 void
-mach64DrawSync (ScreenPtr pScreen);
-
-void
 mach64DrawDisable (ScreenPtr pScreen);
 
 void
