diff --git a/hw/xfree86/int10/generic.c b/hw/xfree86/int10/generic.c
index 9d39e99..fe8bb69 100644
--- a/hw/xfree86/int10/generic.c
+++ b/hw/xfree86/int10/generic.c
@@ -137,7 +137,7 @@ xf86ExtendedInitInt10(int entityIndex, int Flags)
     options = xf86HandleInt10Options(xf86Screens[screen],entityIndex);
     
     if (int10skip(options)) {
-	xfree(options);
+	free(options);
 	return NULL;
     }
     
@@ -285,17 +285,17 @@ xf86ExtendedInitInt10(int entityIndex, int Flags)
     xf86ExecX86int10(pInt);
     UnlockLegacyVGA(pInt, &vga);
 #endif
-    xfree(options);
+    free(options);
     return pInt;
 
  error1:
-    xfree(base);
+    free(base);
     UnmapVRam(pInt);
-    xfree(INTPriv(pInt)->alloc);
-    xfree(pInt->private);
+    free(INTPriv(pInt)->alloc);
+    free(pInt->private);
  error0:
-    xfree(pInt);
-    xfree(options);
+    free(pInt);
+    free(options);
     
     return NULL;
 }
@@ -339,11 +339,11 @@ xf86FreeInt10(xf86Int10InfoPtr pInt)
 #endif
     if (Int10Current == pInt)
 	Int10Current = NULL;
-    xfree(INTPriv(pInt)->base);
+    free(INTPriv(pInt)->base);
     UnmapVRam(pInt);
-    xfree(INTPriv(pInt)->alloc);
-    xfree(pInt->private);
-    xfree(pInt);
+    free(INTPriv(pInt)->alloc);
+    free(pInt->private);
+    free(pInt);
 }
 
 void *
