diff --git a/record/record.c b/record/record.c
index e4d4dc5..d206172 100644
--- a/record/record.c
+++ b/record/record.c
@@ -1,4 +1,4 @@
-/* $XdotOrg$ */
+/* $XdotOrg: xc/programs/Xserver/record/record.c,v 1.2 2004/04/23 19:54:28 eich Exp $ */
 /* $Xorg: record.c,v 1.4 2001/02/09 02:05:27 xorgcvs Exp $ */
 
 /*
@@ -39,7 +39,7 @@ and Jim Haggerty of Metheus.
 #include "dixstruct.h"
 #include "extnsionst.h"
 #define _XRECORD_SERVER_
-#include "recordstr.h"
+#include <X11/extensions/recordstr.h>
 #include "set.h"
 
 #ifndef XFree86LOADER
