diff --git a/miext/rootless/rootlessScreen.c b/miext/rootless/rootlessScreen.c
index 6b76973..356fec7 100644
--- a/miext/rootless/rootlessScreen.c
+++ b/miext/rootless/rootlessScreen.c
@@ -28,7 +28,6 @@
  * holders shall not be used in advertising or otherwise to promote the sale,
  * use or other dealings in this Software without prior written authorization.
  */
-/* $XFree86: xc/programs/Xserver/miext/rootless/rootlessScreen.c,v 1.2 2003/04/30 23:15:35 torrey Exp $ */
 
 
 #ifdef HAVE_DIX_CONFIG_H
