diff --git a/afb/afbbresd.c b/afb/afbbresd.c
index 52cb62c..4495c8c 100644
--- a/afb/afbbresd.c
+++ b/afb/afbbresd.c
@@ -46,7 +46,6 @@ ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
 SOFTWARE.
 
 ******************************************************************/
-/* $XConsortium: afbbresd.c,v 1.10 94/04/17 20:28:18 dpw Exp $ */
 
 #ifdef HAVE_DIX_CONFIG_H
 #include <dix-config.h>
