diff --git a/hw/xfree86/xf8_32bpp/cfbimage.c b/hw/xfree86/xf8_32bpp/cfbimage.c
index 25736fa..8b0ea4a 100644
--- a/hw/xfree86/xf8_32bpp/cfbimage.c
+++ b/hw/xfree86/xf8_32bpp/cfbimage.c
@@ -1,5 +1,9 @@
 /* $XFree86: xc/programs/Xserver/hw/xfree86/xf8_32bpp/cfbimage.c,v 1.2 2000/02/25 00:21:40 mvojkovi Exp $ */
 
+#ifdef HAVE_XORG_CONFIG_H
+#include <xorg-config.h>
+#endif
+
 #include <X11/X.h>
 #include "windowstr.h"
 #include "pixmapstr.h"
