diff --git a/hw/xquartz/quartzAudio.c b/hw/xquartz/quartzAudio.c
index 1eb099b..86bb200 100644
--- a/hw/xquartz/quartzAudio.c
+++ b/hw/xquartz/quartzAudio.c
@@ -246,10 +246,10 @@ static void QuartzCoreAudioBell(
 
 
 /*
- * DarwinModeBell
+ * QuartzBell
  *  Ring the bell
  */
-void DarwinModeBell(
+void QuartzBell(
     int volume,             // volume in percent of max
     DeviceIntPtr pDevice,
     pointer ctrl,
