diff --git a/ilbm/ilbmfillarc.c b/ilbm/ilbmfillarc.c
index f61a016..a5ab6c4 100644
--- a/ilbm/ilbmfillarc.c
+++ b/ilbm/ilbmfillarc.c
@@ -31,6 +31,10 @@ in this Software without prior written authorization from the X Consortium.
 /* Modified jun 95 by Geert Uytterhoeven (Geert.Uytterhoeven@cs.kuleuven.ac.be)
    to use interleaved bitplanes instead of normal bitplanes */
 
+#ifdef HAVE_DIX_CONFIG_H
+#include <dix-config.h>
+#endif
+
 #include <X11/X.h>
 #include <X11/Xprotostr.h>
 #include "miscstruct.h"
