diff --git a/include/windowstr.h b/include/windowstr.h
index 0e45262..2d11f6e 100644
--- a/include/windowstr.h
+++ b/include/windowstr.h
@@ -115,7 +115,7 @@ typedef struct _WindowOpt {
 #endif
     struct _OtherInputMasks *inputMasks;   /* default: NULL */
     DevCursorList       deviceCursors;     /* default: NULL */
-    struct _GenericClientMasks *geMasks;/* default: NULL */
+    struct _GenericClientMasks *geMasks;   /* default: NULL */
     WindowAccessRec     access;
 } WindowOptRec, *WindowOptPtr;
 
