diff --git a/hw/xfree86/shadowfb/shadowfb.h b/hw/xfree86/shadowfb/shadowfb.h
index d177c5f..e428226 100644
--- a/hw/xfree86/shadowfb/shadowfb.h
+++ b/hw/xfree86/shadowfb/shadowfb.h
@@ -1,9 +1,5 @@
 /* $XFree86: xc/programs/Xserver/hw/xfree86/shadowfb/shadowfb.h,v 1.3 2002/10/16 22:12:54 alanh Exp $ */
 
-#ifdef HAVE_XORG_CONFIG_H
-#include <xorg-config.h>
-#endif
-
 #ifndef _SHADOWFB_H
 #define _SHADOWFB_H
 
