diff --git a/dix/devices.c b/dix/devices.c
index 4e78971..e5e3832 100644
--- a/dix/devices.c
+++ b/dix/devices.c
@@ -1465,7 +1465,7 @@ InitStringFeedbackClassDeviceStruct (
     for (i=0; i<num_symbols_supported; i++)
 	*(feedc->ctrl.symbols_supported+i) = *symbols++;
     for (i=0; i<max_symbols; i++)
-	*(feedc->ctrl.symbols_displayed+i) = (KeySym) NULL;
+	*(feedc->ctrl.symbols_displayed+i) = (KeySym) 0;
     feedc->ctrl.id = 0;
     if ( (feedc->next = dev->stringfeed) )
 	feedc->ctrl.id = dev->stringfeed->ctrl.id + 1;
