diff --git a/lbx/lbxswap.c b/lbx/lbxswap.c
index cfed6a4..848be7f 100644
--- a/lbx/lbxswap.c
+++ b/lbx/lbxswap.c
@@ -57,8 +57,8 @@ in this Software without prior written authorization from The Open Group.
 #include <X11/X.h>
 #include <X11/Xproto.h>
 #include <X11/Xos.h>
-#include <X11/misc.h>
-#include <X11/os.h>
+#include "misc.h"
+#include "os.h"
 #include "dixstruct.h"
 #include "pixmapstr.h"
 #include "scrnintstr.h"
