diff --git a/Xext/sampleEVI.c b/Xext/sampleEVI.c
index 4f23ca3..a3d1376 100644
--- a/Xext/sampleEVI.c
+++ b/Xext/sampleEVI.c
@@ -23,13 +23,13 @@ THE USE OR PERFORMANCE OF THIS SOFTWARE.
 ********************************************************/
 /* $XFree86$ */
 
-#include "X.h"
-#include "Xproto.h"
+#include <X11/X.h>
+#include <X11/Xproto.h>
 #include "dixstruct.h"
 #include "extnsionst.h"
 #include "dix.h"
 #define _XEVI_SERVER_
-#include "XEVIstr.h"
+#include <X11/extensions/XEVIstr.h>
 #include "EVIstruct.h"
 #include "scrnintstr.h"
 static int sampleGetVisualInfo(
