diff --git a/mi/mipointer.c b/mi/mipointer.c
index be18956..4358759 100644
--- a/mi/mipointer.c
+++ b/mi/mipointer.c
@@ -284,10 +284,11 @@ miPointerDeviceInitialize(DeviceIntPtr pDev, ScreenPtr pScreen)
 static void
 miPointerDeviceCleanup(DeviceIntPtr pDev, ScreenPtr pScreen)
 {
+    SetupScreen(pScreen);
+
     if (!pDev->isMaster && pDev->u.master)
         return;
 
-    SetupScreen(pScreen);
     (*pScreenPriv->spriteFuncs->DeviceCursorCleanup)(pDev, pScreen);
     xfree(dixLookupPrivate(&pDev->devPrivates, miPointerPrivKey));
     dixSetPrivate(&pDev->devPrivates, miPointerPrivKey, NULL);
@@ -302,8 +303,8 @@ miPointerWarpCursor (DeviceIntPtr pDev, ScreenPtr pScreen, int x, int y)
     miPointerPtr pPointer;
     BOOL changedScreen = FALSE;
 
-    pPointer = MIPOINTER(pDev);
     SetupScreen (pScreen);
+    pPointer = MIPOINTER(pDev);
 
     if (pPointer->pScreen != pScreen)
     {
