diff --git a/Xi/getfctl.h b/Xi/getfctl.h
index 0ad58aa..f0b240d 100644
--- a/Xi/getfctl.h
+++ b/Xi/getfctl.h
@@ -30,15 +30,15 @@ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 #ifndef GETFCTL_H
 #define GETFCTL_H 1
 
-int SProcXGetFeedbackControl(ClientPtr	/* client */
+int SProcXGetFeedbackControl(ClientPtr  /* client */
     );
 
-int ProcXGetFeedbackControl(ClientPtr	/* client */
+int ProcXGetFeedbackControl(ClientPtr   /* client */
     );
 
 void SRepXGetFeedbackControl(ClientPtr /* client */ ,
-			     int /* size */ ,
-			     xGetFeedbackControlReply *	/* rep */
+                             int /* size */ ,
+                             xGetFeedbackControlReply * /* rep */
     );
 
-#endif /* GETFCTL_H */
+#endif                          /* GETFCTL_H */
