diff --git a/hw/xfree86/dummylib/xf86allocscripi.c b/hw/xfree86/dummylib/xf86allocscripi.c
deleted file mode 100644
index d747bd5..0000000
--- a/hw/xfree86/dummylib/xf86allocscripi.c
+++ /dev/null
@@ -1,19 +0,0 @@
-#ifdef HAVE_XORG_CONFIG_H
-#include <xorg-config.h>
-#endif
-
-#include <X11/X.h>
-#include "os.h"
-#include "xf86.h"
-#include "xf86Priv.h"
-
-/*
- * Utility functions required by libxf86_os. 
- */
-
-int
-xf86AllocateScrnInfoPrivateIndex()
-{
-    return -1;
-}
-
