diff --git a/hw/xfree86/os-support/usl/usl_io.c b/hw/xfree86/os-support/usl/usl_io.c
index 45c511f..4cb23d4 100644
--- a/hw/xfree86/os-support/usl/usl_io.c
+++ b/hw/xfree86/os-support/usl/usl_io.c
@@ -1,4 +1,3 @@
-/* $XdotOrg: xserver/xorg/hw/xfree86/os-support/usl/usl_io.c,v 1.2 2005/11/08 06:33:30 jkj Exp $ */
 /*
  * Copyright 2001-2005 by Kean Johnston <jkj@sco.com>
  * Copyright 1990,91 by Thomas Roell, Dinkelscherben, Germany
