diff --git a/hw/xfree86/scanpci/xf86ScanPci.c b/hw/xfree86/scanpci/xf86ScanPci.c
index 7a67ee0..4a82e65 100644
--- a/hw/xfree86/scanpci/xf86ScanPci.c
+++ b/hw/xfree86/scanpci/xf86ScanPci.c
@@ -52,7 +52,7 @@
 /* XXX This is including a lot of stuff that modules should not include! */
 
 #include <X11/X.h>
-#include "os.h"
+#include <X11/os.h>
 #include "xf86.h"
 #include "xf86Priv.h"
 #include "xf86Pci.h"
