diff --git a/hw/xwin/winwin32rootless.c b/hw/xwin/winwin32rootless.c
index 26d5977..a17c219 100755
--- a/hw/xwin/winwin32rootless.c
+++ b/hw/xwin/winwin32rootless.c
@@ -74,9 +74,6 @@ winMWExtWMSetNativeProperty (RootlessWindowPtr pFrame);
 
 Bool			g_fNoConfigureWindow = FALSE;
 
-
-extern void winSelectIcons(WindowPtr pWin, HICON *pIcon, HICON *pSmallIcon);
-
 /*
  * Internal function to get the DIB format that is compatible with the screen
  * Fixme: Share code with winshadgdi.c
