diff --git a/exa/exa_migration.c b/exa/exa_migration.c
index 2bb2ad9..2f90b82 100644
--- a/exa/exa_migration.c
+++ b/exa/exa_migration.c
@@ -43,39 +43,6 @@
 #endif
 
 /**
- * Returns TRUE if the pixmap has damage.
- * EXA only migrates the parts of a destination 
- * that are affected by rendering.
- * It uses the current damage as indication.
- * So anything that does not need to be updated won't be.
- * For clarity this seperate function was made.
- * Note that some situations don't use this, 
- * because their calls are wrapped by the damage layer.
- */
-Bool
-exaDamageDestForMigration(DrawablePtr pDrawable, PixmapPtr pPix, RegionPtr region)
-{
-    ScreenPtr pScreen = pDrawable->pScreen;
-    (void) pScreen; /* the macros don't use pScreen currently */
-    ExaPixmapPriv (pPix);
-    int x_offset, y_offset;
-    RegionPtr pending_damage;
-
-    if (!pExaPixmap->pDamage)
-	return FALSE;
-
-    exaGetDrawableDeltas(pDrawable, pPix, &x_offset, &y_offset);
-
-    REGION_TRANSLATE(pScreen, region, x_offset, y_offset);
-    pending_damage = DamagePendingRegion(pExaPixmap->pDamage);
-    REGION_UNION(pScreen, pending_damage, pending_damage, region);
-    /* Restore region as we got it. */
-    REGION_TRANSLATE(pScreen, region, -x_offset, -y_offset);
-
-    return TRUE;
-}
-
-/**
  * Returns TRUE if the pixmap is not movable.  This is the case where it's a
  * fake pixmap for the frontbuffer (no pixmap private) or it's a scratch
  * pixmap created by some other X Server internals (the score says it's
@@ -332,9 +299,6 @@ exaDoMoveInPixmap (ExaMigrationPtr migrate)
     ExaScreenPriv (pScreen);
     ExaPixmapPriv (pPixmap);
 
-    if (migrate->as_dst)
-	pExaPixmap->pendingDamage = TRUE;
-
     /* If we're VT-switched away, no touching card memory allowed. */
     if (pExaScr->swappedOut)
 	return;
@@ -403,9 +367,6 @@ exaDoMoveOutPixmap (ExaMigrationPtr migrate)
     PixmapPtr pPixmap = migrate->pPix;
     ExaPixmapPriv (pPixmap);
 
-    if (migrate->as_dst)
-	pExaPixmap->pendingDamage = TRUE;
-
     if (!pExaPixmap->area || exaPixmapIsPinned(pPixmap))
 	return;
 
