diff --git a/hw/xfree86/common/xf86Io.c b/hw/xfree86/common/xf86Io.c
index 4e1dc4b..4dbbd7e 100644
--- a/hw/xfree86/common/xf86Io.c
+++ b/hw/xfree86/common/xf86Io.c
@@ -49,7 +49,7 @@
  */
 
 /* $XConsortium: xf86Io.c /main/27 1996/10/19 17:58:55 kaleb $ */
-/* $XdotOrg: xserver/xorg/hw/xfree86/common/xf86Io.c,v 1.5 2005/07/03 07:01:24 daniels Exp $ */
+/* $XdotOrg: xserver/xorg/hw/xfree86/common/xf86Io.c,v 1.6 2006/03/25 19:52:03 ajax Exp $ */
 
 #define NEED_EVENTS
 #ifdef HAVE_XORG_CONFIG_H
@@ -369,12 +369,6 @@ xf86KbdProc (pKeyboard, what)
 	XkbDescPtr		desc;
 	Bool			foundTerminate = FALSE;
 	int			keyc;
-	if (XkbInitialMap) {
-	    if ((xf86Info.xkbkeymap = strchr(XkbInitialMap, '/')) != NULL)
-		xf86Info.xkbkeymap++;
-	    else
-		xf86Info.xkbkeymap = XkbInitialMap;
-	}
 	if (xf86Info.xkbkeymap) {
 	    names.keymap = xf86Info.xkbkeymap;
 	    names.keycodes = NULL;
