diff --git a/hw/xfree86/ddc/xf86DDC.h b/hw/xfree86/ddc/xf86DDC.h
index 07411b8..2deadce 100644
--- a/hw/xfree86/ddc/xf86DDC.h
+++ b/hw/xfree86/ddc/xf86DDC.h
@@ -50,7 +50,7 @@ extern _X_EXPORT xf86MonPtr xf86InterpretEEDID(
 );
 
 extern _X_EXPORT void
-xf86DDCMonitorSet(int scrnIndex, MonPtr Monitor, xf86MonPtr DDC);
+xf86EdidMonitorSet(int scrnIndex, MonPtr Monitor, xf86MonPtr DDC);
 
 extern _X_EXPORT Bool xf86SetDDCproperties(
     ScrnInfoPtr pScreen,
