diff --git a/hw/xfree86/os-support/linux/lnxResource.c b/hw/xfree86/os-support/linux/lnxResource.c
index 04e2f07..3b29618 100644
--- a/hw/xfree86/os-support/linux/lnxResource.c
+++ b/hw/xfree86/os-support/linux/lnxResource.c
@@ -2,7 +2,7 @@
 
 /* Resource information code */
 
-#include "X.h"
+#include <X11/X.h>
 #include "xf86.h"
 #include "xf86Priv.h"
 #include "xf86Privstr.h"
@@ -190,7 +190,8 @@ xf86AccResFromOS(resPtr ret)
       defined(__s390__) || \
       defined(__hppa__)
 
- /* XXX this isn't exactly correct but it will get the server working 
+ /*
+  * XXX this isn't exactly correct but it will get the server working 
   * for now until we get something better.
   */
   
@@ -203,7 +204,7 @@ xf86BusAccWindowsFromOS(void)
     RANGE(range, 0x00000000, 0xffffffff, ResExcMemBlock);
     ret = xf86AddResToList(ret, &range, -1);
 
-#ifdef __sparc__
+#if defined(__sparc__) || defined(__powerpc__)
     RANGE(range, 0x00000000, 0x00ffffff, ResExcIoBlock);
 #else
     RANGE(range, 0x00000000, 0x0000ffff, ResExcIoBlock);
@@ -221,7 +222,7 @@ xf86PciBusAccWindowsFromOS(void)
     RANGE(range, 0x00000000, 0xffffffff, ResExcMemBlock);
     ret = xf86AddResToList(ret, &range, -1);
 
-#ifdef __sparc__
+#if defined(__sparc__) || defined(__powerpc__)
     RANGE(range, 0x00000000, 0x00ffffff, ResExcIoBlock);
 #else
     RANGE(range, 0x00000000, 0x0000ffff, ResExcIoBlock);
@@ -230,7 +231,7 @@ xf86PciBusAccWindowsFromOS(void)
     return ret;
 }
 
-#ifdef INCLUDE_UNUSED */
+#ifdef INCLUDE_UNUSED
 
 resPtr
 xf86IsaBusAccWindowsFromOS(void)
@@ -241,7 +242,7 @@ xf86IsaBusAccWindowsFromOS(void)
     RANGE(range, 0x00000000, 0xffffffff, ResExcMemBlock);
     ret = xf86AddResToList(ret, &range, -1);
 
-#ifdef __sparc__
+#if defined(__sparc__) || defined(__powerpc__)
     RANGE(range, 0x00000000, 0x00ffffff, ResExcIoBlock);
 #else
     RANGE(range, 0x00000000, 0x0000ffff, ResExcIoBlock);
@@ -267,7 +268,7 @@ xf86AccResFromOS(resPtr ret)
     ret = xf86AddResToList(ret, &range, -1);
     RANGE(range, 0x00000000, 0x00000000, ResExcIoBlock);
     ret = xf86AddResToList(ret, &range, -1);
-#ifdef __sparc__
+#if defined(__sparc__) || defined(__powerpc__)
     RANGE(range, 0x00ffffff, 0x00ffffff, ResExcIoBlock);
 #else
     RANGE(range, 0x0000ffff, 0x0000ffff, ResExcIoBlock);
