diff --git a/hw/dmx/input/dmxmotion.c b/hw/dmx/input/dmxmotion.c
index e7f5b59..73580a2 100644
--- a/hw/dmx/input/dmxmotion.c
+++ b/hw/dmx/input/dmxmotion.c
@@ -113,7 +113,7 @@ void dmxPointerPutMotionEvent(DeviceIntPtr pDevice,
                                      * DMX_MOTION_SIZE);
         dmxLocal->head      = 0;
         dmxLocal->tail      = 0;
-        dmxLocal->valuators = xcalloc(sizeof(*dmxLocal->valuators), numAxes);
+        dmxLocal->valuators = calloc(sizeof(*dmxLocal->valuators), numAxes);
     } else {
         if (++dmxLocal->tail >= DMX_MOTION_SIZE) dmxLocal->tail = 0;
         if (dmxLocal->head == dmxLocal->tail)
