diff --git a/afb/afbpntarea.c b/afb/afbpntarea.c
index ab3031b..648a2d1 100644
--- a/afb/afbpntarea.c
+++ b/afb/afbpntarea.c
@@ -49,6 +49,10 @@ SOFTWARE.
 ******************************************************************/
 /* $XConsortium: afbpntarea.c,v 5.7 94/04/17 20:28:29 dpw Exp $ */
 
+#ifdef HAVE_DIX_CONFIG_H
+#include <dix-config.h>
+#endif
+
 #include <X11/X.h>
 
 #include "windowstr.h"
