diff --git a/hw/xwin/windialogs.c b/hw/xwin/windialogs.c
index 582b865..0b590cf 100755
--- a/hw/xwin/windialogs.c
+++ b/hw/xwin/windialogs.c
@@ -116,8 +116,8 @@ winDrawURLWindow (LPARAM lParam)
     crText = RGB(0,0,128+64);
   SetTextColor (draw->hDC, crText);
   
-  /* Create underlined font 14 high, standard dialog font */
-  font = CreateFont (-14, 0, 0, 0, FW_NORMAL, FALSE, TRUE, FALSE,
+  /* Create font 8 high, standard dialog font */
+  font = CreateFont (-8, 0, 0, 0, FW_DONTCARE, FALSE, FALSE, FALSE,
 		     0, 0, 0, 0, 0, "MS Sans Serif");
   if (!font)
     {
