diff --git a/include/globals.h b/include/globals.h
index 7de262f..8b80a65 100644
--- a/include/globals.h
+++ b/include/globals.h
@@ -22,8 +22,6 @@ extern _X_EXPORT int defaultColorVisualClass;
 extern _X_EXPORT int GrabInProgress;
 extern _X_EXPORT Bool noTestExtensions;
 
-extern _X_EXPORT DDXPointRec dixScreenOrigins[MAXSCREENS];
-
 extern _X_EXPORT char *ConnectionInfo;
 
 #ifdef DPMSExtension
