diff --git a/miext/cw/cw_render.c b/miext/cw/cw_render.c
index dfe2681..165c44f 100644
--- a/miext/cw/cw_render.c
+++ b/miext/cw/cw_render.c
@@ -73,7 +73,7 @@ cwCreatePicturePrivate (PicturePtr pPicture)
     int		    error;
     cwPicturePtr    pPicturePrivate;
 
-    pPicturePrivate = xalloc (sizeof (cwPictureRec));
+    pPicturePrivate = malloc(sizeof (cwPictureRec));
     if (!pPicturePrivate)
 	return NULL;
     
@@ -83,7 +83,7 @@ cwCreatePicturePrivate (PicturePtr pPicture)
 						      &error);
     if (!pPicturePrivate->pBackingPicture)
     {
-	xfree (pPicturePrivate);
+	free(pPicturePrivate);
 	return NULL;
     }
 
@@ -107,7 +107,7 @@ cwDestroyPicturePrivate (PicturePtr pPicture)
     {
 	if (pPicturePrivate->pBackingPicture)
 	    FreePicture (pPicturePrivate->pBackingPicture, 0);
-	xfree (pPicturePrivate);
+	free(pPicturePrivate);
 	setCwPicture(pPicture, NULL);
     }
 }
