diff --git a/mi/mipolyutil.c b/mi/mipolyutil.c
index b596914..8bb736c 100644
--- a/mi/mipolyutil.c
+++ b/mi/mipolyutil.c
@@ -46,6 +46,10 @@ SOFTWARE.
 
 ******************************************************************/
 /* $Xorg: mipolyutil.c,v 1.4 2001/02/09 02:05:21 xorgcvs Exp $ */
+#ifdef HAVE_DIX_CONFIG_H
+#include <dix-config.h>
+#endif
+
 #include "regionstr.h"
 #include "gc.h"
 #include "miscanfill.h"
