diff --git a/include/misc.h b/include/misc.h
index e4bdee4..62d813e 100644
--- a/include/misc.h
+++ b/include/misc.h
@@ -100,18 +100,6 @@ typedef unsigned long ATOM;
 #define FALSE 0
 #endif
 
-#ifdef __GNUC__
-#define X_DEPRECATED __attribute__((deprecated))
-#else
-#define X_DEPRECATED
-#endif
-
-#if defined(__GNUC__) && (__GNUC__ > 2)
-#define X_NORETURN __attribute__((noreturn))
-#else
-#define X_NORETURN
-#endif
-
 #ifndef _XTYPEDEF_CALLBACKLISTPTR
 typedef struct _CallbackList *CallbackListPtr; /* also in dix.h */
 #define _XTYPEDEF_CALLBACKLISTPTR
