diff --git a/hw/xfree86/common/xf86xv.c b/hw/xfree86/common/xf86xv.c
index cc84525..85b13d3 100644
--- a/hw/xfree86/common/xf86xv.c
+++ b/hw/xfree86/common/xf86xv.c
@@ -113,12 +113,12 @@ static Bool xf86XVInitAdaptors(ScreenPtr, XF86VideoAdaptorPtr*, int);
 static int XF86XVWindowKeyIndex;
 static DevPrivateKey XF86XVWindowKey = &XF86XVWindowKeyIndex;
 static int XF86XvScreenKeyIndex;
-_X_EXPORT DevPrivateKey XF86XvScreenKey = &XF86XvScreenKeyIndex;
+DevPrivateKey XF86XvScreenKey = &XF86XvScreenKeyIndex;
 static unsigned long PortResource = 0;
 
-_X_EXPORT DevPrivateKey (*XvGetScreenKeyProc)(void) = NULL;
-_X_EXPORT unsigned long (*XvGetRTPortProc)(void) = NULL;
-_X_EXPORT int (*XvScreenInitProc)(ScreenPtr) = NULL;
+DevPrivateKey (*XvGetScreenKeyProc)(void) = NULL;
+unsigned long (*XvGetRTPortProc)(void) = NULL;
+int (*XvScreenInitProc)(ScreenPtr) = NULL;
 
 #define GET_XV_SCREEN(pScreen) \
     ((XvScreenPtr)dixLookupPrivate(&(pScreen)->devPrivates, XF86XvScreenKey))
@@ -132,7 +132,7 @@ _X_EXPORT int (*XvScreenInitProc)(ScreenPtr) = NULL;
 static xf86XVInitGenericAdaptorPtr *GenDrivers = NULL;
 static int NumGenDrivers = 0;
 
-_X_EXPORT int
+int
 xf86XVRegisterGenericAdaptorDriver(
     xf86XVInitGenericAdaptorPtr InitFunc
 ){
@@ -149,7 +149,7 @@ xf86XVRegisterGenericAdaptorDriver(
   return 1;
 }
 
-_X_EXPORT int
+int
 xf86XVListGenericAdaptors(
     ScrnInfoPtr pScrn,
     XF86VideoAdaptorPtr **adaptors
@@ -188,7 +188,7 @@ typedef struct {
 
 static OffscreenImageRec OffscreenImages[MAXSCREENS];
 
-_X_EXPORT Bool
+Bool
 xf86XVRegisterOffscreenImages(
     ScreenPtr pScreen,
     XF86OffscreenImagePtr images,
@@ -200,7 +200,7 @@ xf86XVRegisterOffscreenImages(
     return TRUE;
 }
 
-_X_EXPORT XF86OffscreenImagePtr
+XF86OffscreenImagePtr
 xf86XVQueryOffscreenImages(
    ScreenPtr pScreen,
    int *num
@@ -210,20 +210,20 @@ xf86XVQueryOffscreenImages(
 }
 
 
-_X_EXPORT XF86VideoAdaptorPtr
+XF86VideoAdaptorPtr
 xf86XVAllocateVideoAdaptorRec(ScrnInfoPtr pScrn)
 {
     return xcalloc(1, sizeof(XF86VideoAdaptorRec));
 }
 
-_X_EXPORT void
+void
 xf86XVFreeVideoAdaptorRec(XF86VideoAdaptorPtr ptr)
 {
     xfree(ptr);
 }
 
 
-_X_EXPORT Bool
+Bool
 xf86XVScreenInit(
    ScreenPtr pScreen,
    XF86VideoAdaptorPtr *adaptors,
@@ -1814,7 +1814,7 @@ xf86XVQueryImageAttributes(
 }
 
 
-_X_EXPORT void
+void
 xf86XVFillKeyHelperDrawable (DrawablePtr pDraw, CARD32 key, RegionPtr clipboxes)
 {
    ScreenPtr pScreen = pDraw->pScreen;
@@ -1864,7 +1864,7 @@ xf86XVFillKeyHelperDrawable (DrawablePtr pDraw, CARD32 key, RegionPtr clipboxes)
    xfree(rects);
 }
 
-_X_EXPORT void
+void
 xf86XVFillKeyHelper (ScreenPtr pScreen, CARD32 key, RegionPtr clipboxes)
 {
    DrawablePtr root = &WindowTable[pScreen->myNum]->drawable;
@@ -1909,7 +1909,7 @@ xf86XVFillKeyHelper (ScreenPtr pScreen, CARD32 key, RegionPtr clipboxes)
 
 #define DummyScreen screenInfo.screens[0]
 
-_X_EXPORT Bool
+Bool
 xf86XVClipVideoHelper(
     BoxPtr dst,
     INT32 *xa,
@@ -1991,7 +1991,7 @@ xf86XVClipVideoHelper(
     return TRUE;
 }
 
-_X_EXPORT void
+void
 xf86XVCopyYUV12ToPacked(
     const void *srcy,
     const void *srcv,
@@ -2049,7 +2049,7 @@ xf86XVCopyYUV12ToPacked(
     }
 }
 
-_X_EXPORT void
+void
 xf86XVCopyPacked(
     const void *src,
     void *dst,
