diff --git a/hw/xfree86/utils/kbd_mode/bsd-kbd_mode.c b/hw/xfree86/utils/kbd_mode/bsd-kbd_mode.c
index afa8510..5f8bdbd 100644
--- a/hw/xfree86/utils/kbd_mode/bsd-kbd_mode.c
+++ b/hw/xfree86/utils/kbd_mode/bsd-kbd_mode.c
@@ -1,10 +1,5 @@
-
-
 /* Keyboard mode control program for 386BSD */
 
-
-/* $XConsortium: kbd_mode.c /main/7 1996/03/11 10:46:12 kaleb $ */
-
 #include <sys/types.h>
 #include <fcntl.h>
 #include <errno.h>
