diff --git a/hw/xfree86/xaa/xaaInit.c b/hw/xfree86/xaa/xaaInit.c
index 35ee34c..1513256 100644
--- a/hw/xfree86/xaa/xaaInit.c
+++ b/hw/xfree86/xaa/xaaInit.c
@@ -503,6 +503,9 @@ XAACreatePixmap(ScreenPtr pScreen, int w, int h, int depth)
     PixmapPtr pPix = NULL;
     int size = w * h;
     
+    if (w > 32767 || h > 32767)
+	return NullPixmap;
+    
     if (!infoRec->offscreenDepthsInitialized)
 	XAAInitializeOffscreenDepths (pScreen);
 
