diff --git a/mi/mipolyrect.c b/mi/mipolyrect.c
index 7886248..e7504ef 100644
--- a/mi/mipolyrect.c
+++ b/mi/mipolyrect.c
@@ -56,11 +56,7 @@ SOFTWARE.
 #include "mi.h"
 
 _X_EXPORT void
-miPolyRectangle(pDraw, pGC, nrects, pRects)
-    DrawablePtr	pDraw;
-    GCPtr	pGC;
-    int		nrects;
-    xRectangle	*pRects;
+miPolyRectangle(DrawablePtr pDraw, GCPtr pGC, int nrects, xRectangle *pRects)
 {
     int i;
     xRectangle *pR = pRects;
