diff --git a/exa/exa.c b/exa/exa.c
index dfad0e7..3e8054d 100644
--- a/exa/exa.c
+++ b/exa/exa.c
@@ -755,9 +755,10 @@ exaDriverInit (ScreenPtr		pScreen,
 	return FALSE;
     }
 
+    if (!pScreenInfo->CreatePixmap) {
     if (!pScreenInfo->memoryBase) {
-	LogMessage(X_ERROR, "EXA(%d): ExaDriverRec::memoryBase must be "
-		   "non-zero\n", pScreen->myNum);
+	    LogMessage(X_ERROR, "EXA(%d): ExaDriverRec::memoryBase "
+		       "must be non-zero\n", pScreen->myNum);
 	return FALSE;
     }
 
@@ -768,10 +769,11 @@ exaDriverInit (ScreenPtr		pScreen,
     }
 
     if (pScreenInfo->offScreenBase > pScreenInfo->memorySize) {
-	LogMessage(X_ERROR, "EXA(%d): ExaDriverRec::offScreenBase must be <= "
-		   "ExaDriverRec::memorySize\n", pScreen->myNum);
+	    LogMessage(X_ERROR, "EXA(%d): ExaDriverRec::offScreenBase must "
+		       "be <= ExaDriverRec::memorySize\n", pScreen->myNum);
 	return FALSE;
     }
+    }
 
     if (!pScreenInfo->PrepareSolid) {
 	LogMessage(X_ERROR, "EXA(%d): ExaDriverRec::PrepareSolid must be "
@@ -881,8 +883,7 @@ exaDriverInit (ScreenPtr		pScreen,
     /*
      * Hookup offscreen pixmaps
      */
-    if ((pExaScr->info->flags & EXA_OFFSCREEN_PIXMAPS) &&
-	pExaScr->info->offScreenBase < pExaScr->info->memorySize)
+    if (pExaScr->info->flags & EXA_OFFSCREEN_PIXMAPS)
     {
 	if (!AllocatePixmapPrivate(pScreen, exaPixmapPrivateIndex,
 				   sizeof (ExaPixmapPrivRec))) {
@@ -899,10 +900,15 @@ exaDriverInit (ScreenPtr		pScreen,
 
 	pExaScr->SavedModifyPixmapHeader = pScreen->ModifyPixmapHeader;
 	pScreen->ModifyPixmapHeader = exaModifyPixmapHeader;
-
-	LogMessage(X_INFO, "EXA(%d): Offscreen pixmap area of %d bytes\n",
+	if (!pExaScr->info->CreatePixmap) {
+	    LogMessage(X_INFO, "EXA(%d): Offscreen pixmap area of %lu bytes\n",
 		   pScreen->myNum,
 		   pExaScr->info->memorySize - pExaScr->info->offScreenBase);
+	} else {
+	    LogMessage(X_INFO, "EXA(%d): Driver allocated offscreen pixmaps\n",
+		       pScreen->myNum);
+
+	}
     }
     else
     {
@@ -911,6 +917,7 @@ exaDriverInit (ScreenPtr		pScreen,
 	    return FALSE;
     }
 
+    if (!pExaScr->info->CreatePixmap) {
     DBG_PIXMAP(("============== %ld < %ld\n", pExaScr->info->offScreenBase,
                 pExaScr->info->memorySize));
     if (pExaScr->info->offScreenBase < pExaScr->info->memorySize) {
@@ -920,6 +927,7 @@ exaDriverInit (ScreenPtr		pScreen,
             return FALSE;
         }
     }
+    }
 
     LogMessage(X_INFO, "EXA(%d): Driver registered support for the following"
 	       " operations:\n", pScreen->myNum);
