diff --git a/Xi/extinit.c b/Xi/extinit.c
index 20d49e1..1376f4f 100644
--- a/Xi/extinit.c
+++ b/Xi/extinit.c
@@ -712,7 +712,7 @@ SDeviceChangedEvent(xXIDeviceChangedEvent* from, xXIDeviceChangedEvent* to)
             case ValuatorClass:
                 {
                     xXIValuatorInfo* ai = (xXIValuatorInfo*)any;
-                    swapl(&ai->name, n);
+                    swapl(&ai->label, n);
                     swapl(&ai->min.integral, n);
                     swapl(&ai->min.frac, n);
                     swapl(&ai->max.integral, n);
