diff --git a/xkb/xkbAccessX.c b/xkb/xkbAccessX.c
index 28107d0..d3b585a 100644
--- a/xkb/xkbAccessX.c
+++ b/xkb/xkbAccessX.c
@@ -309,15 +309,14 @@ AccessXRepeatKeyExpire(OsTimerPtr timer,CARD32 now,pointer arg)
 DeviceIntPtr    dev = (DeviceIntPtr) arg;
 XkbSrvInfoPtr	xkbi = dev->key->xkbInfo;
 KeyCode		key;
-BOOL            is_core;
 
     if (xkbi->repeatKey == 0)
 	return 0;
 
-    is_core = (dev == inputInfo.keyboard);
     key = xkbi->repeatKey;
     AccessXKeyboardEvent(dev, DeviceKeyRelease, key, True);
     AccessXKeyboardEvent(dev, DeviceKeyPress, key, True);
+
     return xkbi->desc->ctrls->repeat_interval;
 }
 
