diff --git a/hw/xfree86/modes/xf86RandR12.c b/hw/xfree86/modes/xf86RandR12.c
index 83a4f4f..ab39329 100644
--- a/hw/xfree86/modes/xf86RandR12.c
+++ b/hw/xfree86/modes/xf86RandR12.c
@@ -588,7 +588,6 @@ xf86RandR12CreateScreenResources (ScreenPtr pScreen)
 	else
 	{
 	    xf86OutputPtr   output = config->output[config->compat_output];
-	    xf86CrtcPtr	    crtc = output->crtc;
 
 	    if (output->conf_monitor &&
 		(output->conf_monitor->mon_width  > 0 &&
