diff --git a/hw/xfree86/common/xf86MiscExt.c b/hw/xfree86/common/xf86MiscExt.c
index 8817b79..9c81a6f 100644
--- a/hw/xfree86/common/xf86MiscExt.c
+++ b/hw/xfree86/common/xf86MiscExt.c
@@ -52,8 +52,6 @@
 #include <X11/extensions/XIproto.h>
 #include "xf86Xinput.h"
 
-#include "xf86OSmouse.h"
-
 #ifdef DEBUG
 # define DEBUG_P(x) ErrorF(x"\n");
 #else
@@ -404,17 +402,8 @@ MiscExtApply(pointer structure, MiscExtStructType mse_or_kbd)
 	    = LoaderSymbol("xf86MouseProtocolIDToName");
 	if (!xf86MouseProtocolIDToName)
 	    return MISC_RET_NOMODULE;
-	if (mse->type < MTYPE_MICROSOFT
-		|| (mse->type > MTYPE_EXPPS2
-		    && (mse->type != MTYPE_OSMOUSE)))
-	    return MISC_RET_BADMSEPROTO;
-#ifdef OSMOUSE_ONLY
-	if (mse->type != MTYPE_OSMOUSE)
-	    return MISC_RET_BADMSEPROTO;
-#else
-	if (mse->type == MTYPE_OSMOUSE)
+	if (mse->type < MTYPE_MICROSOFT || mse->type > MTYPE_EXPPS2)
 	    return MISC_RET_BADMSEPROTO;
-#endif /* OSMOUSE_ONLY */
 
 	if (mse->em3timeout < 0)
 	    return MISC_RET_BADVAL;
@@ -431,8 +420,7 @@ MiscExtApply(pointer structure, MiscExtStructType mse_or_kbd)
 	    reopen = TRUE;
 	    mse->flags &= ~MF_REOPEN;
 	}
-	if (mse->type != MTYPE_OSMOUSE
-		&& mse->type != MTYPE_PS_2
+	if (mse->type != MTYPE_PS_2
 		&& mse->type != MTYPE_BUSMOUSE
 		&& mse->type != MTYPE_IMPS2
 		&& mse->type != MTYPE_THINKINGPS2
@@ -451,12 +439,8 @@ MiscExtApply(pointer structure, MiscExtStructType mse_or_kbd)
 		&& (mse->type != MTYPE_MOUSESYS))
 	    return MISC_RET_BADFLAGS;
 
-	if (mse->type != MTYPE_OSMOUSE
-		&& mse->type != MTYPE_BUSMOUSE)
-	{
-	    if (mse->samplerate < 0)
+	if (mse->type != MTYPE_BUSMOUSE && mse->samplerate < 0)
 		return MISC_RET_BADVAL;
-	}
 
 	if (mse->resolution < 0)
 	    return MISC_RET_BADVAL;
