diff --git a/lbx/lbxprop.c b/lbx/lbxprop.c
index 09d818f..53c4ef4 100644
--- a/lbx/lbxprop.c
+++ b/lbx/lbxprop.c
@@ -60,8 +60,8 @@ in this Software without prior written authorization from The Open Group.
 #define NEED_EVENTS
 #include <X11/X.h>
 #include <X11/Xproto.h>
-#include <X11/misc.h>
-#include <X11/os.h>
+#include "misc.h"
+#include "os.h"
 #include "dixstruct.h"
 #include "resource.h"
 #include "servermd.h"
