diff --git a/dix/swapreq.c b/dix/swapreq.c
index d795a3d..5baadf7 100644
--- a/dix/swapreq.c
+++ b/dix/swapreq.c
@@ -62,9 +62,7 @@ SOFTWARE.
 /* Byte swap a list of longs */
 
 void
-SwapLongs (list, count)
-	register CARD32 *list;
-	register unsigned long count;
+SwapLongs (register CARD32 *list, register unsigned long count)
 {
 	register char n;
 
@@ -91,9 +89,7 @@ SwapLongs (list, count)
 /* Byte swap a list of shorts */
 
 void
-SwapShorts (list, count)
-	register short *list;
-	register unsigned long count;
+SwapShorts (register short *list, register unsigned long count)
 {
 	register char n;
 
@@ -128,8 +124,7 @@ SwapShorts (list, count)
 /* The following is used for all requests that have
    no fields to be swapped (except "length") */
 int
-SProcSimpleReq(client)
-	register ClientPtr client;
+SProcSimpleReq(register ClientPtr client)
 {
     register char n;
 
@@ -142,8 +137,7 @@ SProcSimpleReq(client)
    only a single 32-bit field to be swapped, coming
    right after the "length" field */
 int
-SProcResourceReq(client)
-	register ClientPtr client;
+SProcResourceReq(register ClientPtr client)
 {
     register char n;
 
@@ -155,8 +149,7 @@ SProcResourceReq(client)
 }
 
 int
-SProcCreateWindow(client)
-    register ClientPtr client;
+SProcCreateWindow(register ClientPtr client)
 {
     register char n;
 
@@ -178,8 +171,7 @@ SProcCreateWindow(client)
 }
 
 int
-SProcChangeWindowAttributes(client)
-    register ClientPtr client;
+SProcChangeWindowAttributes(register ClientPtr client)
 {
     register char n;
 
@@ -193,8 +185,7 @@ SProcChangeWindowAttributes(client)
 }
 
 int
-SProcReparentWindow(client)
-    register ClientPtr client;
+SProcReparentWindow(register ClientPtr client)
 {
     register char n;
     REQUEST(xReparentWindowReq);
@@ -208,8 +199,7 @@ SProcReparentWindow(client)
 }
 
 int
-SProcConfigureWindow(client)
-    register ClientPtr client;
+SProcConfigureWindow(register ClientPtr client)
 {
     register char n;
     REQUEST(xConfigureWindowReq);
@@ -224,8 +214,7 @@ SProcConfigureWindow(client)
 
 
 int
-SProcInternAtom(client)
-    register ClientPtr client;
+SProcInternAtom(register ClientPtr client)
 {
     register char n;
     REQUEST(xInternAtomReq);
@@ -236,8 +225,7 @@ SProcInternAtom(client)
 }
 
 int
-SProcChangeProperty(client)
-    register ClientPtr client;
+SProcChangeProperty(register ClientPtr client)
 {
     register char n;
     REQUEST(xChangePropertyReq);
@@ -261,8 +249,7 @@ SProcChangeProperty(client)
 }
 
 int 
-SProcDeleteProperty(client)
-    register ClientPtr client;
+SProcDeleteProperty(register ClientPtr client)
 {
     register char n;
     REQUEST(xDeletePropertyReq);
@@ -275,8 +262,7 @@ SProcDeleteProperty(client)
 }
 
 int 
-SProcGetProperty(client)
-    register ClientPtr client;
+SProcGetProperty(register ClientPtr client)
 {
     register char n;
     REQUEST(xGetPropertyReq);
@@ -291,8 +277,7 @@ SProcGetProperty(client)
 }
 
 int
-SProcSetSelectionOwner(client)
-    register ClientPtr client;
+SProcSetSelectionOwner(register ClientPtr client)
 {
     register char n;
     REQUEST(xSetSelectionOwnerReq);
@@ -305,8 +290,7 @@ SProcSetSelectionOwner(client)
 }
 
 int
-SProcConvertSelection(client)
-    register ClientPtr client;
+SProcConvertSelection(register ClientPtr client)
 {
     register char n;
     REQUEST(xConvertSelectionReq);
@@ -321,8 +305,7 @@ SProcConvertSelection(client)
 }
 
 int
-SProcSendEvent(client)
-    register ClientPtr client;
+SProcSendEvent(register ClientPtr client)
 {
     register char n;
     xEvent eventT;
@@ -344,8 +327,7 @@ SProcSendEvent(client)
 }
 
 int
-SProcGrabPointer(client)
-    register ClientPtr client;
+SProcGrabPointer(register ClientPtr client)
 {
     register char n;
     REQUEST(xGrabPointerReq);
@@ -360,8 +342,7 @@ SProcGrabPointer(client)
 }
 
 int
-SProcGrabButton(client)
-    register ClientPtr client;
+SProcGrabButton(register ClientPtr client)
 {
     register char n;
     REQUEST(xGrabButtonReq);
@@ -376,8 +357,7 @@ SProcGrabButton(client)
 }
 
 int
-SProcUngrabButton(client)
-    register ClientPtr client;
+SProcUngrabButton(register ClientPtr client)
 {
     register char n;
     REQUEST(xUngrabButtonReq);
@@ -389,8 +369,7 @@ SProcUngrabButton(client)
 }
 
 int
-SProcChangeActivePointerGrab(client)
-    register ClientPtr client;
+SProcChangeActivePointerGrab(register ClientPtr client)
 {
     register char n;
     REQUEST(xChangeActivePointerGrabReq);
@@ -403,8 +382,7 @@ SProcChangeActivePointerGrab(client)
 }
 
 int
-SProcGrabKeyboard(client)
-    register ClientPtr client;
+SProcGrabKeyboard(register ClientPtr client)
 {
     register char n;
     REQUEST(xGrabKeyboardReq);
@@ -416,8 +394,7 @@ SProcGrabKeyboard(client)
 }
 
 int
-SProcGrabKey(client)
-    register ClientPtr client;
+SProcGrabKey(register ClientPtr client)
 {
     register char n;
     REQUEST(xGrabKeyReq);
@@ -429,8 +406,7 @@ SProcGrabKey(client)
 }
 
 int
-SProcUngrabKey(client)
-    register ClientPtr client;
+SProcUngrabKey(register ClientPtr client)
 {
     register char n;
     REQUEST(xUngrabKeyReq);
@@ -442,8 +418,7 @@ SProcUngrabKey(client)
 }
 
 int
-SProcGetMotionEvents(client)
-    register ClientPtr client;
+SProcGetMotionEvents(register ClientPtr client)
 {
     register char n;
     REQUEST(xGetMotionEventsReq);
@@ -456,8 +431,7 @@ SProcGetMotionEvents(client)
 }
 
 int
-SProcTranslateCoords(client)
-    register ClientPtr client;
+SProcTranslateCoords(register ClientPtr client)
 {
     register char n;
     REQUEST(xTranslateCoordsReq);
@@ -471,8 +445,7 @@ SProcTranslateCoords(client)
 }
 
 int
-SProcWarpPointer(client)
-    register ClientPtr client;
+SProcWarpPointer(register ClientPtr client)
 {
     register char n;
     REQUEST(xWarpPointerReq);
@@ -490,8 +463,7 @@ SProcWarpPointer(client)
 }
 
 int
-SProcSetInputFocus(client)
-    register ClientPtr client;
+SProcSetInputFocus(register ClientPtr client)
 {
     register char n;
     REQUEST(xSetInputFocusReq);
@@ -503,8 +475,7 @@ SProcSetInputFocus(client)
 }
 
 int
-SProcOpenFont(client)
-    register ClientPtr client;
+SProcOpenFont(register ClientPtr client)
 {
     register char n;
     REQUEST(xOpenFontReq);
@@ -516,8 +487,7 @@ SProcOpenFont(client)
 }
 
 int
-SProcListFonts(client)
-    register ClientPtr client;
+SProcListFonts(register ClientPtr client)
 {
     register char n;
     REQUEST(xListFontsReq);
@@ -529,8 +499,7 @@ SProcListFonts(client)
 }
 
 int
-SProcListFontsWithInfo(client)
-    register ClientPtr client;
+SProcListFontsWithInfo(register ClientPtr client)
 {
     register char n;
     REQUEST(xListFontsWithInfoReq);
@@ -542,8 +511,7 @@ SProcListFontsWithInfo(client)
 }
 
 int
-SProcSetFontPath(client)
-    register ClientPtr client;
+SProcSetFontPath(register ClientPtr client)
 {
     register char n;
     REQUEST(xSetFontPathReq);
@@ -554,8 +522,7 @@ SProcSetFontPath(client)
 }
 
 int
-SProcCreatePixmap(client)
-    register ClientPtr client;
+SProcCreatePixmap(register ClientPtr client)
 {
     register char n;
     REQUEST(xCreatePixmapReq);
@@ -570,8 +537,7 @@ SProcCreatePixmap(client)
 }
 
 int
-SProcCreateGC(client)
-    register ClientPtr client;
+SProcCreateGC(register ClientPtr client)
 {
     register char n;
     REQUEST(xCreateGCReq);
@@ -585,8 +551,7 @@ SProcCreateGC(client)
 }
 
 int
-SProcChangeGC(client)
-    register ClientPtr client;
+SProcChangeGC(register ClientPtr client)
 {
     register char n;
     REQUEST(xChangeGCReq);
@@ -599,8 +564,7 @@ SProcChangeGC(client)
 }
 
 int
-SProcCopyGC(client)
-    register ClientPtr client;
+SProcCopyGC(register ClientPtr client)
 {
     register char n;
     REQUEST(xCopyGCReq);
@@ -613,8 +577,7 @@ SProcCopyGC(client)
 }
 
 int
-SProcSetDashes(client)
-    register ClientPtr client;
+SProcSetDashes(register ClientPtr client)
 {
     register char n;
     REQUEST(xSetDashesReq);
@@ -628,8 +591,7 @@ SProcSetDashes(client)
 }
 
 int
-SProcSetClipRectangles(client)
-    register ClientPtr client;
+SProcSetClipRectangles(register ClientPtr client)
 {
     register char n;
     REQUEST(xSetClipRectanglesReq);
@@ -643,8 +605,7 @@ SProcSetClipRectangles(client)
 }
 
 int
-SProcClearToBackground(client)
-    register ClientPtr client;
+SProcClearToBackground(register ClientPtr client)
 {
     register char n;
     REQUEST(xClearAreaReq);
@@ -659,8 +620,7 @@ SProcClearToBackground(client)
 }
 
 int
-SProcCopyArea(client)
-    register ClientPtr client;
+SProcCopyArea(register ClientPtr client)
 {
     register char n;
     REQUEST(xCopyAreaReq);
@@ -679,8 +639,7 @@ SProcCopyArea(client)
 }
 
 int
-SProcCopyPlane(client)
-    register ClientPtr client;
+SProcCopyPlane(register ClientPtr client)
 {
     register char n;
     REQUEST(xCopyPlaneReq);
@@ -702,8 +661,7 @@ SProcCopyPlane(client)
 /* The following routine is used for all Poly drawing requests
    (except FillPoly, which uses a different request format) */
 int
-SProcPoly(client)
-    register ClientPtr client;
+SProcPoly(register ClientPtr client)
 {
     register char n;
 
@@ -720,8 +678,7 @@ SProcPoly(client)
    is longer than xPolyPointReq, and we don't want to swap
    the difference as shorts! */
 int
-SProcFillPoly(client)
-    register ClientPtr client;
+SProcFillPoly(register ClientPtr client)
 {
     register char n;
 
@@ -735,8 +692,7 @@ SProcFillPoly(client)
 }
 
 int
-SProcPutImage(client)
-    register ClientPtr client;
+SProcPutImage(register ClientPtr client)
 {
     register char n;
     REQUEST(xPutImageReq);
@@ -754,8 +710,7 @@ SProcPutImage(client)
 }
 
 int
-SProcGetImage(client)
-    register ClientPtr	client;
+SProcGetImage(register ClientPtr client)
 {
     register char n;
     REQUEST(xGetImageReq);
@@ -773,8 +728,7 @@ SProcGetImage(client)
 /* ProcPolyText used for both PolyText8 and PolyText16 */
 
 int
-SProcPolyText(client)
-    register ClientPtr client;
+SProcPolyText(register ClientPtr client)
 {
     register char n;
     REQUEST(xPolyTextReq);
@@ -790,8 +744,7 @@ SProcPolyText(client)
 /* ProcImageText used for both ImageText8 and ImageText16 */
 
 int
-SProcImageText(client)
-    register ClientPtr client;
+SProcImageText(register ClientPtr client)
 {
     register char n;
     REQUEST(xImageTextReq);
@@ -805,8 +758,7 @@ SProcImageText(client)
 }
 
 int
-SProcCreateColormap(client)
-    register ClientPtr client;
+SProcCreateColormap(register ClientPtr client)
 {
     register char n;
     REQUEST(xCreateColormapReq);
@@ -820,8 +772,7 @@ SProcCreateColormap(client)
 
 
 int
-SProcCopyColormapAndFree(client)
-    register ClientPtr client;
+SProcCopyColormapAndFree(register ClientPtr client)
 {
     register char n;
     REQUEST(xCopyColormapAndFreeReq);
@@ -834,8 +785,7 @@ SProcCopyColormapAndFree(client)
 }
 
 int
-SProcAllocColor                (client)
-    register ClientPtr client;
+SProcAllocColor(register ClientPtr client)
 {
     register char n;
     REQUEST(xAllocColorReq);
@@ -849,8 +799,7 @@ SProcAllocColor                (client)
 }
 
 int
-SProcAllocNamedColor           (client)
-    register ClientPtr client;
+SProcAllocNamedColor(register ClientPtr client)
 {
     register char n;
 
@@ -863,8 +812,7 @@ SProcAllocNamedColor           (client)
 }
 
 int
-SProcAllocColorCells           (client)
-    register ClientPtr client;
+SProcAllocColorCells(register ClientPtr client)
 {
     register char n;
     REQUEST(xAllocColorCellsReq);
@@ -877,8 +825,7 @@ SProcAllocColorCells           (client)
 }
 
 int
-SProcAllocColorPlanes(client)
-    register ClientPtr client;
+SProcAllocColorPlanes(register ClientPtr client)
 {
     register char n;
     REQUEST(xAllocColorPlanesReq);
@@ -893,8 +840,7 @@ SProcAllocColorPlanes(client)
 }
 
 int
-SProcFreeColors          (client)
-    register ClientPtr client;
+SProcFreeColors(register ClientPtr client)
 {
     register char n;
     REQUEST(xFreeColorsReq);
@@ -908,8 +854,7 @@ SProcFreeColors          (client)
 }
 
 void
-SwapColorItem(pItem)
-    xColorItem	*pItem;
+SwapColorItem(xColorItem *pItem)
 {
     register char n;
 
@@ -920,8 +865,7 @@ SwapColorItem(pItem)
 }
 
 int
-SProcStoreColors               (client)
-    register ClientPtr client;
+SProcStoreColors(register ClientPtr client)
 {
     register char n;
     long count;
@@ -938,8 +882,7 @@ SProcStoreColors               (client)
 }
 
 int
-SProcStoreNamedColor           (client)
-    register ClientPtr client;
+SProcStoreNamedColor (register ClientPtr client)
 {
     register char n;
     REQUEST(xStoreNamedColorReq);
@@ -952,8 +895,7 @@ SProcStoreNamedColor           (client)
 }
 
 int
-SProcQueryColors(client)
-    register ClientPtr client;
+SProcQueryColors (register ClientPtr client)
 {
     register char n;
     REQUEST(xQueryColorsReq);
@@ -965,8 +907,7 @@ SProcQueryColors(client)
 } 
 
 int
-SProcLookupColor(client)
-    register ClientPtr client;
+SProcLookupColor (register ClientPtr client)
 {
     register char n;
     REQUEST(xLookupColorReq);
@@ -978,8 +919,7 @@ SProcLookupColor(client)
 }
 
 int
-SProcCreateCursor( client)
-    register ClientPtr client;
+SProcCreateCursor (register ClientPtr client)
 {
     register char n;
     REQUEST(xCreateCursorReq);
@@ -1000,8 +940,7 @@ SProcCreateCursor( client)
 }
 
 int
-SProcCreateGlyphCursor( client)
-    register ClientPtr client;
+SProcCreateGlyphCursor (register ClientPtr client)
 {
     register char n;
     REQUEST(xCreateGlyphCursorReq);
@@ -1023,8 +962,7 @@ SProcCreateGlyphCursor( client)
 
 
 int
-SProcRecolorCursor(client)
-    register ClientPtr client;
+SProcRecolorCursor (register ClientPtr client)
 {
     register char n;
     REQUEST(xRecolorCursorReq);
@@ -1041,8 +979,7 @@ SProcRecolorCursor(client)
 }
 
 int
-SProcQueryBestSize   (client)
-    register ClientPtr client;
+SProcQueryBestSize (register ClientPtr client)
 {
     register char n;
     REQUEST(xQueryBestSizeReq);
@@ -1056,8 +993,7 @@ SProcQueryBestSize   (client)
 }
 
 int
-SProcQueryExtension   (client)
-    register ClientPtr client;
+SProcQueryExtension (register ClientPtr client)
 {
     register char n;
     REQUEST(xQueryExtensionReq);
@@ -1068,8 +1004,7 @@ SProcQueryExtension   (client)
 }
 
 int
-SProcChangeKeyboardMapping   (client)
-    register ClientPtr client;
+SProcChangeKeyboardMapping (register ClientPtr client)
 {
     register char n;
     REQUEST(xChangeKeyboardMappingReq);
@@ -1081,8 +1016,7 @@ SProcChangeKeyboardMapping   (client)
 
 
 int
-SProcChangeKeyboardControl   (client)
-    register ClientPtr client;
+SProcChangeKeyboardControl (register ClientPtr client)
 {
     register char n;
     REQUEST(xChangeKeyboardControlReq);
@@ -1094,8 +1028,7 @@ SProcChangeKeyboardControl   (client)
 }
 
 int
-SProcChangePointerControl   (client)
-    register ClientPtr client;
+SProcChangePointerControl (register ClientPtr client)
 {
     register char n;
     REQUEST(xChangePointerControlReq);
@@ -1109,8 +1042,7 @@ SProcChangePointerControl   (client)
 
 
 int
-SProcSetScreenSaver            (client)
-    register ClientPtr client;
+SProcSetScreenSaver (register ClientPtr client)
 {
     register char n;
     REQUEST(xSetScreenSaverReq);
@@ -1122,8 +1054,7 @@ SProcSetScreenSaver            (client)
 }
 
 int
-SProcChangeHosts(client)
-    register ClientPtr client;
+SProcChangeHosts (register ClientPtr client)
 {
     register char n;
 
@@ -1135,8 +1066,7 @@ SProcChangeHosts(client)
 
 }
 
-int SProcRotateProperties(client)
-    register ClientPtr client;
+int SProcRotateProperties (register ClientPtr client)
 {
     register char n;
     REQUEST(xRotatePropertiesReq);
@@ -1149,10 +1079,8 @@ int SProcRotateProperties(client)
     return ((* ProcVector[X_RotateProperties])(client));
 }
 
-/*ARGSUSED*/
 int
-SProcNoOperation(client)
-    ClientPtr client;
+SProcNoOperation(register ClientPtr client)
 {
     register char n;
     REQUEST(xReq);
@@ -1161,8 +1089,7 @@ SProcNoOperation(client)
 }
 
 void
-SwapConnClientPrefix(pCCP)
-    xConnClientPrefix	*pCCP;
+SwapConnClientPrefix(xConnClientPrefix *pCCP)
 {
     register char n;
 
