diff --git a/dix/ptrveloc.c b/dix/ptrveloc.c
index a73bef8..304a1c9 100644
--- a/dix/ptrveloc.c
+++ b/dix/ptrveloc.c
@@ -694,16 +694,16 @@ ProcessVelocityData2D(
  * constant-velocity response
  */
 static inline float
-ApplySimpleSoftening(int od, int d)
+ApplySimpleSoftening(int prev_delta, int delta)
 {
-    float res = d;
-    if (d <= 1 && d >= -1)
-        return res;
-    if (d > od)
-        res -= 0.5;
-    else if (d < od)
-        res += 0.5;
-    return res;
+    float result = delta;
+    if (delta <= 1 && delta >= -1)
+        return result;
+    if (delta > prev_delta)
+        result -= 0.5;
+    else if (delta < prev_delta)
+        result += 0.5;
+    return result;
 }
 
 
