diff --git a/hw/dmx/dmxclient.h b/hw/dmx/dmxclient.h
index 8f018d1..91ba049 100644
--- a/hw/dmx/dmxclient.h
+++ b/hw/dmx/dmxclient.h
@@ -98,9 +98,7 @@ typedef XID           KeySym64;
 #include <X11/extensions/XKBstr.h>
 #endif
 
-#ifdef XINPUT
 #include <X11/extensions/XI.h>
-#endif
 
 /* Always include these, since we query them even if we don't export XINPUT. */
 #include <X11/extensions/XInput.h> /* For XDevice */
