diff --git a/hw/xfree86/shadowfb/shadowfb.h b/hw/xfree86/shadowfb/shadowfb.h
index 56e0e86..bcf59f9 100644
--- a/hw/xfree86/shadowfb/shadowfb.h
+++ b/hw/xfree86/shadowfb/shadowfb.h
@@ -20,10 +20,7 @@ typedef void (*RefreshAreaFuncPtr)(ScrnInfoPtr, int, BoxPtr);
  * Returns FALSE in the event of an error.
  */
 extern _X_EXPORT Bool
-ShadowFBInit (
-    ScreenPtr		pScreen,
-    RefreshAreaFuncPtr  refreshArea
-);
+ ShadowFBInit(ScreenPtr pScreen, RefreshAreaFuncPtr refreshArea);
 
 /*
  * ShadowFBInit2 is a more featureful refinement of the original shadowfb.
@@ -34,10 +31,9 @@ ShadowFBInit (
  * Returns FALSE in the event of an error
  */
 extern _X_EXPORT Bool
-ShadowFBInit2 (
-    ScreenPtr		pScreen,
+
+ShadowFBInit2(ScreenPtr pScreen,
     RefreshAreaFuncPtr  preRefreshArea,
-    RefreshAreaFuncPtr  postRefreshArea
-);
+              RefreshAreaFuncPtr postRefreshArea);
 
 #endif /* _SHADOWFB_H */
