diff --git a/hw/xfree86/os-support/bus/sparcPci.c b/hw/xfree86/os-support/bus/sparcPci.c
index 48e10ed..5e0ef5b 100644
--- a/hw/xfree86/os-support/bus/sparcPci.c
+++ b/hw/xfree86/os-support/bus/sparcPci.c
@@ -592,13 +592,13 @@ sparcPciInit(void)
 
 #ifndef INCLUDE_XF86_NO_DOMAIN
 
-int
+_X_EXPORT int
 xf86GetPciDomain(PCITAG Tag)
 {
     return PCI_DOM_FROM_TAG(Tag);
 }
 
-pointer
+_X_EXPORT pointer
 xf86MapDomainMemory(int ScreenNum, int Flags, PCITAG Tag,
 		    ADDRESS Base, unsigned long Size)
 {
@@ -622,7 +622,7 @@ xf86MapDomainMemory(int ScreenNum, int Flags, PCITAG Tag,
     return result;
 }
 
-IOADDRESS
+_X_EXPORT IOADDRESS
 xf86MapDomainIO(int ScreenNum, int Flags, PCITAG Tag,
 		IOADDRESS Base, unsigned long Size)
 {
@@ -649,7 +649,7 @@ xf86MapDomainIO(int ScreenNum, int Flags, PCITAG Tag,
     return (IOADDRESS)pDomain->io + Base;
 }
 
-int
+_X_EXPORT int
 xf86ReadDomainMemory(PCITAG Tag, ADDRESS Base, int Len, unsigned char *Buf)
 {
     unsigned char *ptr, *src;
