diff --git a/mi/mieq.c b/mi/mieq.c
index 8a4f5fc..41b632a 100644
--- a/mi/mieq.c
+++ b/mi/mieq.c
@@ -386,7 +386,7 @@ mieqProcessInputEvents(void)
         }
 
         /* Update the sprite now. Next event may be from different device. */
-        if (type == DeviceMotionNotify && dev->coreEvents)
+        if (type == DeviceMotionNotify && master)
             miPointerUpdateSprite(dev);
     }
 }
