diff --git a/hw/xfree86/os-support/bsd/memrange.h b/hw/xfree86/os-support/bsd/memrange.h
index 0f4b270..03c4791 100644
--- a/hw/xfree86/os-support/bsd/memrange.h
+++ b/hw/xfree86/os-support/bsd/memrange.h
@@ -1,9 +1,6 @@
 /*
  * Memory range attribute operations, peformed on /dev/mem
- *
- * $FreeBSD: src/sys/sys/memrange.h,v 1.4 1999/12/29 04:24:44 peter Exp $
  */
-/* $XFree86$ */
 
 #ifdef HAVE_XORG_CONFIG_H
 #include <xorg-config.h>
