diff --git a/mi/misprite.c b/mi/misprite.c
index b23adb7..a8731f1 100644
--- a/mi/misprite.c
+++ b/mi/misprite.c
@@ -111,8 +111,8 @@ miSpriteIsDown(miCursorInfoPtr pDevCursor)
 
 static int miSpriteScreenKeyIndex;
 static DevPrivateKey miSpriteScreenKey = &miSpriteScreenKeyIndex;
-static int mmiSpriteDevPrivatesKeyIndex;
-static DevPrivateKey miSpriteDevPrivatesKey = &mmiSpriteDevPrivatesKeyIndex;
+static int miSpriteDevPrivatesKeyIndex;
+static DevPrivateKey miSpriteDevPrivatesKey = &miSpriteDevPrivatesKeyIndex;
 
 static Bool	    miSpriteCloseScreen(int i, ScreenPtr pScreen);
 static void	    miSpriteGetImage(DrawablePtr pDrawable, int sx, int sy,
