diff --git a/glx/glxdri2.c b/glx/glxdri2.c
index 4df406b..77b5307 100644
--- a/glx/glxdri2.c
+++ b/glx/glxdri2.c
@@ -251,12 +251,15 @@ __glXDRIbindTexImage(__GLXcontext *baseContext,
     if (texBuffer == NULL)
         return Success;
 
+#if __DRI_TEX_BUFFER_VERSION >= 2
     if (texBuffer->base.version >= 2 && texBuffer->setTexBuffer2 != NULL) {
 	(*texBuffer->setTexBuffer2)(context->driContext,
 				    glxPixmap->target,
 				    glxPixmap->format,
 				    drawable->driDrawable);
-    } else {
+    } else
+#endif
+    {
 	texBuffer->setTexBuffer(context->driContext,
 				glxPixmap->target,
 				drawable->driDrawable);
