diff --git a/org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java b/org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java
index b2beaa9..3b68674 100644
--- a/org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java
+++ b/org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java
@@ -934,14 +934,6 @@ public final class JavaCore extends Plugin {
 	 * @since 3.2
 	 */
 	public static final String TIMEOUT_FOR_PARAMETER_NAME_FROM_ATTACHED_JAVADOC = PLUGIN_ID + ".timeoutForParameterNameFromAttachedJavadoc"; //$NON-NLS-1$
-
-	/**
-	 * Possible  configurable option ID.
-	 * @see #getDefaultOptions()
-	 * @since 3.2
-	 */
-	public static final String ENABLE_JAVADOC_ATTACHMENTS_ACCESSES = PLUGIN_ID + ".enable.external.javadoc.attachments.acccesses"; //$NON-NLS-1$
-	// *************** Possible values for configurable options. ********************
 	
 	/**
 	 * Possible  configurable option value.
@@ -2391,13 +2383,6 @@ public final class JavaCore extends Plugin {
 	 *     - possible values:	"&lt;n&gt;", where n is an integer greater than or equal to 0
 	 *     - default:           "50"
 	 * 
-	 * JAVACORE / Enabling the javadoc attachments accesses.
-	 *    When disabled, javadoc attachments are not used to retrieve
-	 *    the javadoc of java elements in {@link IJavaElement#getAttachedJavadoc(IProgressMonitor) IJavaElement.getAttachedJavadoc}.
-	 *     - option id:         "org.eclipse.jdt.core.enable.javadoc.attachments.accesses"
-	 *     - possible values:   { "enabled", "disabled" }
-	 *     - default:           "enabled"
-	 * 
 	 * FORMATTER / Inserting New Line Before Opening Brace
 	 *    When Insert, a new line is inserted before an opening brace, otherwise nothing
 	 *    is inserted
