diff --git a/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/parser/AbstractCommentParser.java b/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/parser/AbstractCommentParser.java
index b7da464..e0ac7cd 100644
--- a/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/parser/AbstractCommentParser.java
+++ b/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/parser/AbstractCommentParser.java
@@ -142,7 +142,7 @@ public abstract class AbstractCommentParser implements JavadocTagConstants {
 			int invalidInlineTagLineEnd = -1;
 			
 			// Loop on each comment character
-			characterLoop: while (!abort && this.index < this.javadocEnd) {
+			while (!abort && this.index < this.javadocEnd) {
 				previousPosition = this.index;
 				previousChar = nextCharacter;
 				
@@ -1492,7 +1492,7 @@ public abstract class AbstractCommentParser implements JavadocTagConstants {
 		int previousPosition = this.index;
 		this.starPosition = -1;
 		ch = readChar();
-		nextChar: while (this.index<this.source.length) {
+		while (this.index<this.source.length) {
 			switch (ch) {
 				case '*':
 					// valid whatever the number of star before last '/'
