diff --git a/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/parser/ParserBasicInformation.java b/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/parser/ParserBasicInformation.java
index 01bf9a2..6d5124a 100644
--- a/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/parser/ParserBasicInformation.java
+++ b/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/parser/ParserBasicInformation.java
@@ -16,20 +16,20 @@ public interface ParserBasicInformation {
 
 	int ERROR_SYMBOL = 110,
 		MAX_NAME_LENGTH = 41,
-		NUM_STATES = 955,
+		NUM_STATES = 954,
 
 		NT_OFFSET = 110,
 		SCOPE_UBOUND = 131,
 		SCOPE_SIZE = 132,
-		LA_STATE_OFFSET = 12619,
+		LA_STATE_OFFSET = 12754,
 		MAX_LA = 1,
-		NUM_RULES = 691,
+		NUM_RULES = 692,
 		NUM_TERMINALS = 110,
-		NUM_NON_TERMINALS = 306,
-		NUM_SYMBOLS = 416,
-		START_STATE = 1466,
+		NUM_NON_TERMINALS = 307,
+		NUM_SYMBOLS = 417,
+		START_STATE = 803,
 		EOFT_SYMBOL = 66,
 		EOLT_SYMBOL = 66,
-		ACCEPT_ACTION = 12618,
-		ERROR_ACTION = 12619;
+		ACCEPT_ACTION = 12753,
+		ERROR_ACTION = 12754;
 }
