diff --git a/org.eclipse.jdt.core.tests.model/workspace/Completion/src/CompletionFindConstructor.java b/org.eclipse.jdt.core.tests.model/workspace/Completion/src/CompletionFindConstructor.java
deleted file mode 100644
index ececce4..0000000
--- a/org.eclipse.jdt.core.tests.model/workspace/Completion/src/CompletionFindConstructor.java
+++ /dev/null
@@ -1,8 +0,0 @@
-public class CompletionFindConstructor {
-	public CompletionFindConstructor (int i) {
-	}
-	publuc void foo(){
-		int x = 45;
-		new CompletionFindConstructor(i);
-	}
-}
\ No newline at end of file
