diff --git a/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/builder/ReferenceCollection.java b/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/builder/ReferenceCollection.java
index 158f00c..893ec62 100644
--- a/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/builder/ReferenceCollection.java
+++ b/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/builder/ReferenceCollection.java
@@ -52,7 +52,7 @@ void addDependencies(String[] typeNameDependencies) {
 				if (temp == EmptyQualifiedNames)
 					continue next; // qualifiedTypeName is a well known name
 				qualifiedTypeName = temp[0];
-			}		
+			}
 		}
 	}
 }
