diff --git a/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/ast/ForeachStatement.java b/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/ast/ForeachStatement.java
index 0da87a0..e8ac3e2 100644
--- a/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/ast/ForeachStatement.java
+++ b/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/ast/ForeachStatement.java
@@ -64,11 +64,9 @@ public class ForeachStatement extends Statement {
 	
 	public ForeachStatement(
 		LocalDeclaration elementVariable,
-		Expression collection,
 		int start) {
 
 		this.elementVariable = elementVariable;
-		this.collection = collection;
 		this.sourceStart = start;
 		this.kind = -1;
 	}
