diff --git a/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/CharacterLiteral.java b/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/CharacterLiteral.java
index 7431400..14ebf27 100644
--- a/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/CharacterLiteral.java
+++ b/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/CharacterLiteral.java
@@ -265,22 +265,21 @@ public class CharacterLiteral extends Expression {
 								throw new IllegalArgumentException("illegal character literal");//$NON-NLS-1$
 							}
 							number = (number * 8) + Character.getNumericValue(nextChar);
-						}
-						nextChar = (char) scanner.getNextChar();
-						if (nextChar == -1) {
-							throw new IllegalArgumentException("illegal character literal");//$NON-NLS-1$
-						}
-						if (nextChar != '\'') {
-							if (!Character.isDigit(nextChar)) {
+							nextChar = (char) scanner.getNextChar();
+							if (nextChar == -1) {
 								throw new IllegalArgumentException("illegal character literal");//$NON-NLS-1$
 							}
-							number = (number * 8) + Character.getNumericValue(nextChar);
+							if (nextChar != '\'') {
+								if (!Character.isDigit(nextChar)) {
+									throw new IllegalArgumentException("illegal character literal");//$NON-NLS-1$
+								}
+								number = (number * 8) + Character.getNumericValue(nextChar);
+							}
 						}
-						value = (char) number;
+						return (char) number;			
 					} else {
 						throw new IllegalArgumentException("illegal character literal");//$NON-NLS-1$
 					}
-					break;
 			}
 			nextChar = (char) scanner.getNextChar();
 			if (nextChar == -1) {
