diff --git a/org.eclipse.jdt.core.tests.model/workspace/Formatter/test588/A_in.java b/org.eclipse.jdt.core.tests.model/workspace/Formatter/test588/A_in.java
new file mode 100644
index 0000000..de80b2e
--- /dev/null
+++ b/org.eclipse.jdt.core.tests.model/workspace/Formatter/test588/A_in.java
@@ -0,0 +1,6 @@
+public class X {
+  public static void main(String[] args) {
+    if(true) throw new RuntimeException("X");
+      loop: System.out.println("X");
+    }
+  }
