diff --git a/org.eclipse.jdt.apt.tests/src/org/eclipse/jdt/apt/tests/annotations/valueconversion/Annotation.java b/org.eclipse.jdt.apt.tests/src/org/eclipse/jdt/apt/tests/annotations/valueconversion/Annotation.java
new file mode 100644
index 0000000..ca6ee0a
--- /dev/null
+++ b/org.eclipse.jdt.apt.tests/src/org/eclipse/jdt/apt/tests/annotations/valueconversion/Annotation.java
@@ -0,0 +1,12 @@
+package org.eclipse.jdt.apt.tests.annotations.valueconversion;
+
+public @interface Annotation {
+	boolean z();
+	char c();
+	byte b();
+	short s();
+	int i(); 
+	long l();
+	float f();
+	double d();
+}
