diff --git a/org.eclipse.jdt.apt.core/src/org/eclipse/jdt/apt/core/internal/generatedfile/FileGenerationResult.java b/org.eclipse.jdt.apt.core/src/org/eclipse/jdt/apt/core/internal/generatedfile/FileGenerationResult.java
index a865a8d..46f028c 100644
--- a/org.eclipse.jdt.apt.core/src/org/eclipse/jdt/apt/core/internal/generatedfile/FileGenerationResult.java
+++ b/org.eclipse.jdt.apt.core/src/org/eclipse/jdt/apt/core/internal/generatedfile/FileGenerationResult.java
@@ -13,10 +13,12 @@ public class FileGenerationResult {
 
 	private final IFile file;
 	private final boolean modified;
+	private final boolean sourcepathChanged;
 	
-	public FileGenerationResult(final IFile file, final boolean modified) {
+	public FileGenerationResult(final IFile file, final boolean modified, final boolean sourcepathChanged) {
 		this.file = file;
 		this.modified = modified;
+		this.sourcepathChanged = sourcepathChanged;
 	}
 	
 	public IFile getFile() {
@@ -26,4 +28,8 @@ public class FileGenerationResult {
 	public boolean isModified() {
 		return modified;
 	}	
+	
+	public boolean hasSourcepathChanged() {
+		return sourcepathChanged;
+	}
 }
