diff --git a/org.eclipse.jdt.apt.core/src/org/eclipse/jdt/apt/core/util/AptUtil.java b/org.eclipse.jdt.apt.core/src/org/eclipse/jdt/apt/core/util/AptUtil.java
index e5073ef..86b9324 100644
--- a/org.eclipse.jdt.apt.core/src/org/eclipse/jdt/apt/core/util/AptUtil.java
+++ b/org.eclipse.jdt.apt.core/src/org/eclipse/jdt/apt/core/util/AptUtil.java
@@ -19,7 +19,7 @@ import org.eclipse.jdt.apt.core.internal.AnnotationProcessorFactoryLoader;
 
 import com.sun.mirror.apt.AnnotationProcessorFactory;
 
-public class AptUtil {
+public final class AptUtil {
 	
 	// Private c-tor to prevent construction
 	private AptUtil() {}
