diff --git a/org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/complete/CompletionScanner.java b/org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/complete/CompletionScanner.java
index 7523889..7f8f80f 100644
--- a/org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/complete/CompletionScanner.java
+++ b/org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/complete/CompletionScanner.java
@@ -118,11 +118,6 @@ public int getNextToken() throws InvalidInputException {
 						/* might be completing at eof (e.g. behind a dot) */
 						if (this.completionIdentifier == null && 
 							this.startPosition == this.cursorLocation + 1){
-							// compute end of empty identifier.
-							// if the empty identifier is at the start of a next token the end of
-							// empty identifier is the end of the next token (eg. "<empty token>next").
-						 	while(getNextCharAsJavaIdentifierPart()){/*empty*/}
-						 	this.endOfEmptyToken = this.currentPosition - 1;
 							this.currentPosition = this.startPosition; // for being detected as empty free identifier
 							return TokenNameIdentifier;
 						}
