diff --git a/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/AnnotationTest.java b/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/AnnotationTest.java
index 145d4f6..5e391b3 100644
--- a/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/AnnotationTest.java
+++ b/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/AnnotationTest.java
@@ -2983,10 +2983,10 @@ public class AnnotationTest extends AbstractComparableTest {
 		
 		String expectedOutput = 
 			"  Inner classes:\n" + 
-			"    [inner class info: #30 X$MyAnon, outer class info: #2 X\n" + 
+			"    [inner class info: #29 X$MyAnon, outer class info: #1 X\n" + 
 			"     inner name: #68 MyAnon, accessflags: 9737 public abstract static ],\n" + 
-			"    [inner class info: #70 X$I, outer class info: #2 X\n" + 
-			"     inner name: #71 I, accessflags: 1545 public abstract static ]"; 
+			"    [inner class info: #69 X$I, outer class info: #1 X\n" + 
+			"     inner name: #71 I, accessflags: 1545 public abstract static ]\n"; 
 			
 		int index = actualOutput.indexOf(expectedOutput);
 		if (index == -1 || expectedOutput.length() == 0) {
