diff --git a/org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/InternalCompletionContext.java b/org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/InternalCompletionContext.java
index 6cedbb9..18f2be8 100644
--- a/org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/InternalCompletionContext.java
+++ b/org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/InternalCompletionContext.java
@@ -48,6 +48,13 @@ public class InternalCompletionContext {
 	protected void setTokenRange(int start, int end, int endOfEmptyToken) {
 		this.tokenStart = start;
 		this.tokenEnd = endOfEmptyToken > end ? endOfEmptyToken : end;
+		
+		// Work around for bug 132558 (https://bugs.eclipse.org/bugs/show_bug.cgi?id=132558).
+		// completionLocation can be -1 if the completion occur at the start of a file or
+		// the start of a code snippet but this API isn't design to support negative position.
+		if(this.tokenEnd == -1) {
+			this.tokenEnd = 0;
+		}
 	}
 	
 	protected void setToken(char[] token) {
