diff --git a/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTConverter15Test.java b/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTConverter15Test.java
index 4d8e527..63bc7d9 100644
--- a/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTConverter15Test.java
+++ b/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTConverter15Test.java
@@ -50,7 +50,7 @@ public class ASTConverter15Test extends ConverterTestSetup {
 //		TESTS_NAMES = new String[] {"test0204"};
 	}
 	public static Test suite() {
-		return buildTestSuite(ASTConverter15Test.class);
+		return buildModelTestSuite(ASTConverter15Test.class);
 	}
 	
 	protected void tearDown() throws Exception {
