diff --git a/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavadocMethodCompletionModelTest.java b/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavadocMethodCompletionModelTest.java
index 235f9e4..50fc68b 100644
--- a/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavadocMethodCompletionModelTest.java
+++ b/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavadocMethodCompletionModelTest.java
@@ -32,7 +32,7 @@ static {
 //	TESTS_RANGE = new int[] { 58, 69 };
 }
 public static Test suite() {
-	return buildTestSuite(JavadocMethodCompletionModelTest.class);
+	return buildModelTestSuite(JavadocMethodCompletionModelTest.class);
 }
 
 /* (non-Javadoc)
