diff --git a/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/compiler/ISourceElementRequestor.java b/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/compiler/ISourceElementRequestor.java
index 253a680..4bc41f1 100644
--- a/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/compiler/ISourceElementRequestor.java
+++ b/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/compiler/ISourceElementRequestor.java
@@ -39,7 +39,6 @@ import org.eclipse.jdt.core.compiler.IProblem;
 public interface ISourceElementRequestor {
 	
 	public static class TypeInfo {
-		public int kind;
 		public int declarationStart;
 		public int modifiers;
 		public char[] name;
