diff --git a/org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/CompletionEngine.java b/org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/CompletionEngine.java
index d9a6487..b1bcb3f 100644
--- a/org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/CompletionEngine.java
+++ b/org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/CompletionEngine.java
@@ -2329,6 +2329,8 @@ public final class CompletionEngine
 			FieldBinding field = fields[f];
 
 			if (field.isSynthetic())	continue next;
+			
+			if (field.name == ClassScope.IncompleteHierarchy) continue next;
 
 			if (onlyStaticFields && !field.isStatic()) continue next;
