diff --git a/org.eclipse.jdt.core.tests.model/workspace/Encoding/src/testBug110576/Test.java b/org.eclipse.jdt.core.tests.model/workspace/Encoding/src/testBug110576/Test.java
new file mode 100644
index 0000000..78b4bff
--- /dev/null
+++ b/org.eclipse.jdt.core.tests.model/workspace/Encoding/src/testBug110576/Test.java
@@ -0,0 +1,10 @@
+﻿
+package pack;
+/**
+ * C'est un exemple de fichier contenant
+ * des caractères UTF-8: Noël même où çà?
+ * x
+ */
+class Test {
+}
+
