diff --git a/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/JavadocTest_1_3.java b/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/JavadocTest_1_3.java
index 0edd205..2b93c5c 100644
--- a/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/JavadocTest_1_3.java
+++ b/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/JavadocTest_1_3.java
@@ -3009,4 +3009,51 @@ public class JavadocTest_1_3 extends JavadocTest {
 			"----------\n"
 		);
 	}
+
+	/**
+	 * Bug 112346: [javadoc] {@inheritedDoc} should be inactive for non-overridden method
+	 * @see "http://bugs.eclipse.org/bugs/show_bug.cgi?id=112346"
+	 */
+	public void testBug112346() {
+		runNegativeTest(
+			new String[] {
+				"Test.java",
+				"/**\n" + 
+				" * Test references\n" + 
+				" * @see Test#field\n" + 
+				" * @see Test#foo()\n" + 
+				" */\n" + 
+				"public class Test<T> {\n" + 
+				"	T field;\n" + 
+				"	T foo() { return null; }\n" + 
+				"}\n"
+			},
+			"----------\n" + 
+			"1. ERROR in Test.java (at line 3)\n" + 
+			"	* @see Test#field\n" + 
+			"	            ^^^^^\n" + 
+			"Javadoc: field cannot be resolved or is not a field\n" + 
+			"----------\n" + 
+			"2. ERROR in Test.java (at line 4)\n" + 
+			"	* @see Test#foo()\n" + 
+			"	            ^^^\n" + 
+			"Javadoc: The method foo() is undefined for the type Test\n" + 
+			"----------\n" + 
+			"3. ERROR in Test.java (at line 6)\n" + 
+			"	public class Test<T> {\n" + 
+			"	                  ^\n" + 
+			"Syntax error, type parameters are only available if source level is 5.0\n" + 
+			"----------\n" + 
+			"4. ERROR in Test.java (at line 7)\n" + 
+			"	T field;\n" + 
+			"	^\n" + 
+			"T cannot be resolved to a type\n" + 
+			"----------\n" + 
+			"5. ERROR in Test.java (at line 8)\n" + 
+			"	T foo() { return null; }\n" + 
+			"	^\n" + 
+			"T cannot be resolved to a type\n" + 
+			"----------\n"
+		);
+	}
 }
