diff --git a/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/problem/AbortCompilationUnit.java b/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/problem/AbortCompilationUnit.java
index d9b937c..050bc41 100644
--- a/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/problem/AbortCompilationUnit.java
+++ b/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/problem/AbortCompilationUnit.java
@@ -10,7 +10,7 @@
  *******************************************************************************/
 package org.eclipse.jdt.internal.compiler.problem;
 
-import org.eclipse.jdt.core.compiler.IProblem;
+import org.eclipse.jdt.core.compiler.CategorizedProblem;
 import org.eclipse.jdt.internal.compiler.CompilationResult;
 
 /*
@@ -23,7 +23,7 @@ public class AbortCompilationUnit extends AbortCompilation {
 
 	private static final long serialVersionUID = -4253893529982226734L; // backward compatible
 	
-public AbortCompilationUnit(CompilationResult compilationResult, IProblem problem) {
+public AbortCompilationUnit(CompilationResult compilationResult, CategorizedProblem problem) {
 	super(compilationResult, problem);
 }
 }
