diff --git a/org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java b/org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java
index 5378648..eda1758 100644
--- a/org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java
+++ b/org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java
@@ -18,6 +18,7 @@
  *     IBM Corporation - added specific output location to source classpath entries
  *     IBM Corporation - added the following constants:
  *                                 CORE_JAVA_BUILD_CLEAN_OUTPUT_FOLDER
+ *                                 CORE_JAVA_BUILD_MAKE_OUTPUT_FOLDER_CONSISTENT
  *                                 CLEAN
  *     IBM Corporation - added getClasspathContainerInitializer(String)
  *     IBM Corporation - added the following constants:
@@ -696,6 +697,12 @@ public final class JavaCore extends Plugin {
 	/**
 	 * Possible  configurable option ID.
 	 * @see #getDefaultOptions()
+	 * @since 3.2
+	 */
+	public static final String CORE_JAVA_BUILD_MAKE_OUTPUT_FOLDER_CONSISTENT = PLUGIN_ID + ".builder.makeOutputFolderConsistent"; //$NON-NLS-1$	 	
+	/**
+	 * Possible  configurable option ID.
+	 * @see #getDefaultOptions()
 	 * @since 2.1
 	 */
 	public static final String CORE_INCOMPLETE_CLASSPATH = PLUGIN_ID + ".incompleteClasspath"; //$NON-NLS-1$
@@ -2305,6 +2312,14 @@ public final class JavaCore extends Plugin {
 	 *     - possible values:   { "clean", "ignore" }
 	 *     - default:           "clean"
 	 * 
+	 * BUILDER / Make Output Folder(s) Consistent
+	 *    Indicate whether the JavaBuilder should check that the output folders
+	 *    are consistent while performing incremental build operations. If changes
+	 *    are detected to managed .class files, then a full build is performed.
+	 *     - option id:         "org.eclipse.jdt.core.builder.makeOutputFolderConsistent"
+	 *     - possible values:   { "enabled", "ignore" }
+	 *     - default:           "ignore"
+	 * 
 	 * BUILDER / Reporting Duplicate Resources
 	 *    Indicate the severity of the problem reported when more than one occurrence
 	 *    of a resource is to be copied into the output location.
