diff --git a/org.eclipse.jdt.apt.core/src/org/eclipse/jdt/apt/core/internal/util/FactoryPathUtil.java b/org.eclipse.jdt.apt.core/src/org/eclipse/jdt/apt/core/internal/util/FactoryPathUtil.java
index d526dda..db03d47 100644
--- a/org.eclipse.jdt.apt.core/src/org/eclipse/jdt/apt/core/internal/util/FactoryPathUtil.java
+++ b/org.eclipse.jdt.apt.core/src/org/eclipse/jdt/apt/core/internal/util/FactoryPathUtil.java
@@ -225,7 +225,7 @@ public final class FactoryPathUtil {
 					break;
 				
 				case PLUGIN :
-					container = FactoryPath.newPluginFactoryContainer(idString);
+					container = FactoryPath.getPluginFactoryContainer(idString);
 					break;
 					
 				default :
