diff --git a/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/util/PublicScanner.java b/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/util/PublicScanner.java
index b8788d9..d6ddecc 100644
--- a/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/util/PublicScanner.java
+++ b/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/util/PublicScanner.java
@@ -1439,7 +1439,7 @@ public int getNextToken() throws InvalidInputException {
 						switch (ScannerHelper.ObviousIdentCharNatures[c]) {
 							case ScannerHelper.C_UPPER_LETTER :
 							case ScannerHelper.C_LOWER_LETTER :
-							case ScannerHelper.C_LETTER :
+							case ScannerHelper.C_IDENT_PART :
 								return scanIdentifierOrKeyword();
 							case ScannerHelper.C_DIGIT :
 								return scanNumber(false);
@@ -1882,7 +1882,7 @@ public final void jumpOverMethodBody() {
 							switch (ScannerHelper.ObviousIdentCharNatures[c]) {
 								case ScannerHelper.C_UPPER_LETTER :
 								case ScannerHelper.C_LOWER_LETTER :
-								case ScannerHelper.C_LETTER :
+								case ScannerHelper.C_IDENT_PART :
 									scanIdentifierOrKeyword();
 									break NextToken;
 								case ScannerHelper.C_DIGIT :
@@ -2459,7 +2459,7 @@ public int scanIdentifierOrKeyword() {
 				switch (ScannerHelper.ObviousIdentCharNatures[c]) {
 					case ScannerHelper.C_UPPER_LETTER :
 					case ScannerHelper.C_LOWER_LETTER :
-					case ScannerHelper.C_LETTER :
+					case ScannerHelper.C_IDENT_PART :
 					case ScannerHelper.C_DIGIT :
 		               if (this.withoutUnicodePtr != 0) {
 							this.currentCharacter = c;
