diff --git a/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/JavaProject.java b/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/JavaProject.java
index a982b89..e935858 100644
--- a/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/JavaProject.java
+++ b/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/JavaProject.java
@@ -3171,7 +3171,7 @@ public class JavaProject
 			    }
 			}
 			try {
-				// save immediately old preferences
+				// save immediately new preferences
 				preferences.flush();
 			} catch (BackingStoreException e) {
 				// fails silently
