diff --git a/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/BasicCompilationUnit.java b/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/BasicCompilationUnit.java
index 9d31c94..200f766 100644
--- a/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/BasicCompilationUnit.java
+++ b/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/BasicCompilationUnit.java
@@ -29,7 +29,13 @@ import org.eclipse.jdt.internal.compiler.util.Util;
  */
 public class BasicCompilationUnit implements ICompilationUnit {
 	protected char[] contents;
-	protected char[] fileName;
+	
+	// Note that if this compiler ICompilationUnit's content is known in advance, the fileName is not used to retrieve this content.
+	// Instead it is used to keep enough information to recreate the IJavaElement corresponding to this compiler ICompilationUnit.
+	// Thus the fileName can be a path to a .class file, or even a path in a .jar to a .class file.
+	// (e.g. /P/lib/mylib.jar|org/eclipse/test/X.class)
+	protected char[] fileName; 
+	
 	protected char[][] packageName;
 	protected char[] mainTypeName;
 	protected String encoding;
