diff --git a/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/ast/ArrayInitializer.java b/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/ast/ArrayInitializer.java
index 13358c0..0bfcf3f 100644
--- a/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/ast/ArrayInitializer.java
+++ b/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/ast/ArrayInitializer.java
@@ -168,6 +168,7 @@ public class ArrayInitializer extends Expression {
 					expression.computeConversion(scope, elementType, exprType);
 				} else if (scope.isBoxingCompatibleWith(exprType, elementType) 
 									|| (exprType.isBaseType()  // narrowing then boxing ?
+											&& scope.compilerOptions().sourceLevel >= JDK1_5 // autoboxing
 											&& !elementType.isBaseType()
 											&& expression.isConstantValueOfTypeAssignableToType(exprType, scope.environment().computeBoxingType(elementType)))) {
 					expression.computeConversion(scope, elementType, exprType);
