diff --git a/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/util/Messages.java b/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/util/Messages.java
index 08ce1d0..c7c47f0 100644
--- a/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/util/Messages.java
+++ b/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/util/Messages.java
@@ -212,6 +212,7 @@ public final class Messages extends NLS {
 	public static String dom_addAncestorAsSibling;
 	public static String dom_addNullInterface;
 	public static String dom_nullInterfaces;
+	public static String importRewrite_processDescription;
 	public static String correction_nullRequestor;
 	public static String correction_nullUnit;
 	public static String engine_searching;
