diff --git a/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/ast/InstanceOfExpression.java b/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/ast/InstanceOfExpression.java
index dd79e71..c8bd410 100644
--- a/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/ast/InstanceOfExpression.java
+++ b/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/ast/InstanceOfExpression.java
@@ -89,8 +89,7 @@ public FlowInfo analyseCode(
 
 		if (!checkedType.isReifiable()) {
 			scope.problemReporter().illegalInstanceOfGenericType(checkedType, this);
-		}
-		if ((expressionType != TypeBinding.NULL && expressionType.isBaseType()) // disallow autoboxing
+		} else if ((expressionType != TypeBinding.NULL && expressionType.isBaseType()) // disallow autoboxing
 				|| !checkCastTypesCompatibility(scope, checkedType, expressionType, null)) {
 			scope.problemReporter().notCompatibleTypesError(this, expressionType, checkedType);
 		}
