diff --git a/org.eclipse.jdt.core/antadapter/org/eclipse/jdt/core/JDTCompilerAdapter.java b/org.eclipse.jdt.core/antadapter/org/eclipse/jdt/core/JDTCompilerAdapter.java
index d447565..8c3c69f 100644
--- a/org.eclipse.jdt.core/antadapter/org/eclipse/jdt/core/JDTCompilerAdapter.java
+++ b/org.eclipse.jdt.core/antadapter/org/eclipse/jdt/core/JDTCompilerAdapter.java
@@ -280,19 +280,8 @@ public class JDTCompilerAdapter extends DefaultCompilerAdapter {
 		/*
 		 * verbose option
 		 */
-		if (this.verbose && this.destDir != null) {
-			/*
-			 * if destDir is null, we don't generate any log.
-			 * See https://bugs.eclipse.org/bugs/show_bug.cgi?id=97744
-			 */
-			// Fix for https://bugs.eclipse.org/bugs/show_bug.cgi?id=96605
-			// cmd.createArgument().setValue("-verbose"); //$NON-NLS-1$
-			/*
-			 * extra option allowed by the Eclipse compiler
-			 */
-			cmd.createArgument().setValue("-log"); //$NON-NLS-1$
-			this.logFileName = this.destDir.getAbsolutePath() + ".log"; //$NON-NLS-1$
-			cmd.createArgument().setValue(this.logFileName);			
+		if (this.verbose) {
+			cmd.createArgument().setValue("-verbose"); //$NON-NLS-1$
 		}
 
 		/*
