diff --git a/org.eclipse.jdt.apt.tests/srcext/org/eclipse/jdt/apt/tests/external/annotations/batch/BatchProcessor.java b/org.eclipse.jdt.apt.tests/srcext/org/eclipse/jdt/apt/tests/external/annotations/batch/BatchProcessor.java
index a02403d..f6463f5 100644
--- a/org.eclipse.jdt.apt.tests/srcext/org/eclipse/jdt/apt/tests/external/annotations/batch/BatchProcessor.java
+++ b/org.eclipse.jdt.apt.tests/srcext/org/eclipse/jdt/apt/tests/external/annotations/batch/BatchProcessor.java
@@ -21,8 +21,7 @@ import com.sun.mirror.declaration.Declaration;
 import com.sun.mirror.declaration.TypeDeclaration;
 
 public class BatchProcessor implements AnnotationProcessor {
-
-	private boolean _succeed = false;
+	
 	private final AnnotationProcessorEnvironment _env;
 	BatchProcessor(AnnotationProcessorEnvironment env){
 		_env = env;
