diff --git a/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchBugsTests.java b/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchBugsTests.java
index 5239a71..f42a93a 100644
--- a/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchBugsTests.java
+++ b/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchBugsTests.java
@@ -45,7 +45,7 @@ public class JavaSearchBugsTests extends AbstractJavaSearchTests implements IJav
 //		org.eclipse.jdt.internal.codeassist.SelectionEngine.DEBUG = true;
 //		TESTS_PREFIX =  "testBug75816";
 //		TESTS_NAMES = new String[] { "testBug82208_SearchAllTypeNames_CLASS" };
-//		TESTS_NUMBERS = new int[] { 97322 };
+//		TESTS_NUMBERS = new int[] { 97087 };
 //		TESTS_RANGE = new int[] { 83304, -1 };
 		}
 
@@ -3320,6 +3320,34 @@ public class JavaSearchBugsTests extends AbstractJavaSearchTests implements IJav
 	}
 
 	/**
+	 * Bug 97087: [1.5][search] Can't find reference of generic class's constructor.
+	 * @see "https://bugs.eclipse.org/bugs/show_bug.cgi?id=97087"
+	 */
+	public void testBug97087() throws CoreException {
+		workingCopies = new ICompilationUnit[1];
+		this.resultCollector.showRule = true;
+		workingCopies[0] = getWorkingCopy("/JavaSearchBugs/src/b97087/Bug.java",
+			"package b97087;\n" + 
+			"public class Bug<Type> {\n" + 
+			"    Bug(){}\n" + 
+			"}\n" + 
+			"class Foo extends Bug<String>{\n" + 
+			"    Foo(){}\n" + 
+			"}\n" +
+			"class Bar extends Bug<Exception>{\n" + 
+			"    Bar(){super();}\n" + 
+			"}"
+		);
+		IType type = workingCopies[0].getType("Bug");
+		IMethod method= type.getMethods()[0];
+		search(method, REFERENCES, SearchPattern.R_ERASURE_MATCH);
+		assertSearchResults(
+			"src/b97087/Bug.java b97087.Foo() [Foo] EXACT_MATCH\n" + 
+			"src/b97087/Bug.java b97087.Bar() [super()] ERASURE_MATCH"
+		);
+	}
+
+	/**
 	 * Bug 97322: [search] Search for method references sometimes reports potential match with differing argument count
 	 * @see "https://bugs.eclipse.org/bugs/show_bug.cgi?id=97322"
 	 */
