diff --git a/org.eclipse.jdt.apt.tests/src/org/eclipse/jdt/apt/tests/TestAll.java b/org.eclipse.jdt.apt.tests/src/org/eclipse/jdt/apt/tests/TestAll.java
index bd9d0a5..8b93dfd 100644
--- a/org.eclipse.jdt.apt.tests/src/org/eclipse/jdt/apt/tests/TestAll.java
+++ b/org.eclipse.jdt.apt.tests/src/org/eclipse/jdt/apt/tests/TestAll.java
@@ -46,6 +46,7 @@ public class TestAll extends TestCase {
 		suite.addTest(AnnotationValueConversionTests.suite());
 		suite.addTest(JavaVersionTests.suite());
 		suite.addTest(RegressionTests.suite());
+		suite.addTest(FileGenerationTests.suite());
 	
 		return suite;
