diff --git a/org.eclipse.jdt.core/compiler/org/eclipse/jdt/core/compiler/IProblem.java b/org.eclipse.jdt.core/compiler/org/eclipse/jdt/core/compiler/IProblem.java
index 524fd50..8e2129a 100644
--- a/org.eclipse.jdt.core/compiler/org/eclipse/jdt/core/compiler/IProblem.java
+++ b/org.eclipse.jdt.core/compiler/org/eclipse/jdt/core/compiler/IProblem.java
@@ -748,9 +748,9 @@ void setSourceStart(int sourceStart);
 	 * Javadoc comments
 	 */
 	/** @since 3.2 */
-	int JavadocNonStaticTypeFromStaticInvocation = Javadoc + Internal + 468;
-	/** @since 3.2 */
 	int JavadocMissingIdentifier = Javadoc + Internal + 467;
+	/** @since 3.2 */
+	int JavadocNonStaticTypeFromStaticInvocation = Javadoc + Internal + 468;
 	/** @since 3.1 */
 	int JavadocInvalidParamTagTypeParameter = Javadoc + Internal + 469;
 	/** @since 3.0 */
