diff --git a/org.eclipse.jdt.core.tests.model/workspace/Formatter/test588/A_out.java b/org.eclipse.jdt.core.tests.model/workspace/Formatter/test588/A_out.java
new file mode 100644
index 0000000..1a14137
--- /dev/null
+++ b/org.eclipse.jdt.core.tests.model/workspace/Formatter/test588/A_out.java
@@ -0,0 +1,7 @@
+public class X {
+	public static void main(String[] args) {
+		if (true)
+			throw new RuntimeException("X");
+		loop: System.out.println("X");
+	}
+}
