diff --git a/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/ast/ContinueStatement.java b/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/ast/ContinueStatement.java
index 13e392d..0e3025f 100644
--- a/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/ast/ContinueStatement.java
+++ b/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/ast/ContinueStatement.java
@@ -16,85 +16,79 @@ import org.eclipse.jdt.internal.compiler.lookup.*;
 
 public class ContinueStatement extends BranchStatement {
 
-	public ContinueStatement(char[] label, int sourceStart, int sourceEnd) {
-		
-		super(label, sourceStart, sourceEnd);
-	}
-	
-	public FlowInfo analyseCode(
-		BlockScope currentScope,
-		FlowContext flowContext,
-		FlowInfo flowInfo) {
+public ContinueStatement(char[] label, int sourceStart, int sourceEnd) {
+	super(label, sourceStart, sourceEnd);
+}
 
-		// here requires to generate a sequence of finally blocks invocations depending corresponding
-		// to each of the traversed try statements, so that execution will terminate properly.
+public FlowInfo analyseCode(BlockScope currentScope, FlowContext flowContext, FlowInfo flowInfo) {
 
-		// lookup the label, this should answer the returnContext
-		FlowContext targetContext = (label == null)
-				? flowContext.getTargetContextForDefaultContinue()
-				: flowContext.getTargetContextForContinueLabel(label);
+	// here requires to generate a sequence of finally blocks invocations depending corresponding
+	// to each of the traversed try statements, so that execution will terminate properly.
 
-		if (targetContext == null) {
-			if (label == null) {
-				currentScope.problemReporter().invalidContinue(this);
-			} else {
-				currentScope.problemReporter().undefinedLabel(this); 
-			}
-			return flowInfo; // pretend it did not continue since no actual target			
-		} 
+	// lookup the label, this should answer the returnContext
+	FlowContext targetContext = (label == null)
+			? flowContext.getTargetContextForDefaultContinue()
+			: flowContext.getTargetContextForContinueLabel(label);
 
-		if (targetContext == FlowContext.NotContinuableContext) {
+	if (targetContext == null) {
+		if (label == null) {
 			currentScope.problemReporter().invalidContinue(this);
-			return flowInfo; // pretend it did not continue since no actual target
+		} else {
+			currentScope.problemReporter().undefinedLabel(this); 
 		}
-		targetLabel = targetContext.continueLabel();
-		FlowContext traversedContext = flowContext;
-		int subIndex = 0, maxSub = 5;
-		subroutines = new SubRoutineStatement[maxSub];
+		return flowInfo; // pretend it did not continue since no actual target			
+	} 
 
-		do {
-			SubRoutineStatement sub;
-			if ((sub = traversedContext.subRoutine()) != null) {
-				if (subIndex == maxSub) {
-					System.arraycopy(subroutines, 0, (subroutines = new SubRoutineStatement[maxSub*=2]), 0, subIndex); // grow
-				}
-				subroutines[subIndex++] = sub;
-				if (sub.isSubRoutineEscaping()) {
-					break;
-				}
-			}
-			traversedContext.recordReturnFrom(flowInfo.unconditionalInits());
+	if (targetContext == FlowContext.NotContinuableContext) {
+		currentScope.problemReporter().invalidContinue(this);
+		return flowInfo; // pretend it did not continue since no actual target
+	}
+	targetLabel = targetContext.continueLabel();
+	FlowContext traversedContext = flowContext;
+	int subCount = 0;
+	subroutines = new SubRoutineStatement[5];
 
-			ASTNode node;
-			if ((node = traversedContext.associatedNode) instanceof TryStatement) {
-				TryStatement tryStatement = (TryStatement) node;
-				flowInfo.addInitializationsFrom(tryStatement.subRoutineInits); // collect inits			
-			} else if (traversedContext == targetContext) {
-				// only record continue info once accumulated through subroutines, and only against target context
-				targetContext.recordContinueFrom(flowContext, flowInfo);
+	do {
+		SubRoutineStatement sub;
+		if ((sub = traversedContext.subroutine()) != null) {
+			if (subCount == subroutines.length) {
+				System.arraycopy(subroutines, 0, subroutines = new SubRoutineStatement[subCount*2], 0, subCount); // grow
+			}
+			subroutines[subCount++] = sub;
+			if (sub.isSubRoutineEscaping()) {
 				break;
 			}
-		} while ((traversedContext = traversedContext.parent) != null);
-		
-		// resize subroutines
-		if (subIndex != maxSub) {
-			System.arraycopy(subroutines, 0, (subroutines = new SubRoutineStatement[subIndex]), 0, subIndex);
 		}
-		return FlowInfo.DEAD_END;
-	}
+		traversedContext.recordReturnFrom(flowInfo.unconditionalInits());
 
-	public StringBuffer printStatement(int tab, StringBuffer output) {
-
-		printIndent(tab, output).append("continue "); //$NON-NLS-1$
-		if (label != null) output.append(label);
-		return output.append(';');
+		if (traversedContext instanceof InsideSubRoutineFlowContext) {
+			ASTNode node = traversedContext.associatedNode;
+			if (node instanceof TryStatement) {
+				TryStatement tryStatement = (TryStatement) node;
+				flowInfo.addInitializationsFrom(tryStatement.subRoutineInits); // collect inits			
+			}		
+		} else if (traversedContext == targetContext) {
+			// only record continue info once accumulated through subroutines, and only against target context
+			targetContext.recordContinueFrom(flowContext, flowInfo);
+			break;
+		}
+	} while ((traversedContext = traversedContext.parent) != null);
+	
+	// resize subroutines
+	if (subCount != subroutines.length) {
+		System.arraycopy(subroutines, 0, subroutines = new SubRoutineStatement[subCount], 0, subCount);
 	}
+	return FlowInfo.DEAD_END;
+}
 
-	public void traverse(
-		ASTVisitor visitor,
-		BlockScope blockScope) {
+public StringBuffer printStatement(int tab, StringBuffer output) {
+	printIndent(tab, output).append("continue "); //$NON-NLS-1$
+	if (label != null) output.append(label);
+	return output.append(';');
+}
 
-		visitor.visit(this, blockScope);
-		visitor.endVisit(this, blockScope);
-	}
+public void traverse(ASTVisitor visitor, 	BlockScope blockScope) {
+	visitor.visit(this, blockScope);
+	visitor.endVisit(this, blockScope);
+}
 }
