diff --git a/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/lookup/SourceTypeBinding.java b/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/lookup/SourceTypeBinding.java
index 372fe4a..5b90335 100644
--- a/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/lookup/SourceTypeBinding.java
+++ b/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/lookup/SourceTypeBinding.java
@@ -130,7 +130,7 @@ private void addDefaultAbstractMethods() {
 			// re-sort methods
 			length = length+defaultAbstractsCount;
 			if (length > 1) {
-				ReferenceBinding.sortMethods(this.methods, 0, length - 1);
+				ReferenceBinding.sortMethods(this.methods, 0, length);
 			}
 			// this.tagBits |= TagBits.AreMethodsSorted; -- already set in #methods()
 		}
@@ -597,7 +597,7 @@ public FieldBinding[] fields() {
 		if ((this.tagBits & TagBits.AreFieldsSorted) == 0) {
 			int length = this.fields.length;
 			if (length > 1)
-				ReferenceBinding.sortFields(this.fields, 0, length - 1);
+				ReferenceBinding.sortFields(this.fields, 0, length);
 			this.tagBits |= TagBits.AreFieldsSorted;
 		}			
 		for (int i = 0, length = this.fields.length; i < length; i++) {
@@ -719,7 +719,7 @@ public MethodBinding getExactConstructor(TypeBinding[] argumentTypes) {
 		if ((this.tagBits & TagBits.AreMethodsSorted) == 0) {
 			int length = this.methods.length;
 			if (length > 1)
-				ReferenceBinding.sortMethods(this.methods, 0, length - 1);
+				ReferenceBinding.sortMethods(this.methods, 0, length);
 			this.tagBits |= TagBits.AreMethodsSorted;
 		}		
 		long range;
@@ -770,7 +770,7 @@ public MethodBinding getExactMethod(char[] selector, TypeBinding[] argumentTypes
 		if ((this.tagBits & TagBits.AreMethodsSorted) == 0) {
 			int length = this.methods.length;
 			if (length > 1)
-				ReferenceBinding.sortMethods(this.methods, 0, length - 1);
+				ReferenceBinding.sortMethods(this.methods, 0, length);
 			this.tagBits |= TagBits.AreMethodsSorted;
 		}
 		
@@ -839,7 +839,7 @@ public FieldBinding getField(char[] fieldName, boolean needResolve) {
 	if ((this.tagBits & TagBits.AreFieldsSorted) == 0) {
 		int length = this.fields.length;
 		if (length > 1)
-			ReferenceBinding.sortFields(this.fields, 0, length - 1);
+			ReferenceBinding.sortFields(this.fields, 0, length);
 		this.tagBits |= TagBits.AreFieldsSorted;
 	}		
 	// always resolve anyway on source types
@@ -889,7 +889,7 @@ public MethodBinding[] getMethods(char[] selector) {
 	if ((this.tagBits & TagBits.AreMethodsSorted) == 0) {
 		int length = this.methods.length;
 		if (length > 1)
-			ReferenceBinding.sortMethods(this.methods, 0, length - 1);
+			ReferenceBinding.sortMethods(this.methods, 0, length);
 		this.tagBits |= TagBits.AreMethodsSorted;
 	}
 	MethodBinding[] result;	
@@ -1084,7 +1084,7 @@ public MethodBinding[] methods() {
 	if ((this.tagBits & TagBits.AreMethodsSorted) == 0) {
 		int length = this.methods.length;
 		if (length > 1)
-			ReferenceBinding.sortMethods(this.methods, 0, length - 1);
+			ReferenceBinding.sortMethods(this.methods, 0, length);
 		this.tagBits |= TagBits.AreMethodsSorted;
 	}
