diff --git a/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/CompilerInvocationTests.java b/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/CompilerInvocationTests.java
index 566a433..69470e2 100644
--- a/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/CompilerInvocationTests.java
+++ b/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/CompilerInvocationTests.java
@@ -46,7 +46,7 @@ public CompilerInvocationTests(String name) {
   	}
 
 public static Test suite() {
-    return buildAllCompliancesRegressionTestSetupSuite(testClass());
+    return buildAllCompliancesTestSuite(testClass());
 }
   
 public static Class testClass() {
