diff --git a/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/DeltaProcessor.java b/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/DeltaProcessor.java
index 95ed1ab..b187cd7 100644
--- a/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/DeltaProcessor.java
+++ b/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/DeltaProcessor.java
@@ -324,11 +324,6 @@ public class DeltaProcessor {
 						for (int i = 0; i < length; i++) {
 							IProject project = projectsToTouch[i];
 							
-							// reset the corresponding project built state, since the builder would miss this change
-							if (JavaBuilder.DEBUG)
-								System.out.println("Clearing last state for project with external jar changed: " + project); //$NON-NLS-1$						
-							DeltaProcessor.this.manager.setLastBuiltState(project, null /*no state*/);
-							
 							// touch to force a build of this project
 							project.touch(progressMonitor);
 						}
