diff --git a/org.eclipse.jdt.core.tests.model/workspace/Formatter/test030/A_out.java b/org.eclipse.jdt.core.tests.model/workspace/Formatter/test030/A_out.java
index 1c76425..4857c72 100644
--- a/org.eclipse.jdt.core.tests.model/workspace/Formatter/test030/A_out.java
+++ b/org.eclipse.jdt.core.tests.model/workspace/Formatter/test030/A_out.java
@@ -5,8 +5,7 @@ public class Test {
 	public int i;
 	public void foo(int k) {
 		if (k > 0) {
-			i = 3;
-			;;
+			i = 3;;;
 		} else if (k == 0) {
 			i = 2;
 		}
