diff --git a/org.eclipse.jdt.core.tests.model/workspace/JavaSearchBugs/src/b108088/Test108088.java b/org.eclipse.jdt.core.tests.model/workspace/JavaSearchBugs/src/b108088/Test108088.java
new file mode 100644
index 0000000..7e8e877
--- /dev/null
+++ b/org.eclipse.jdt.core.tests.model/workspace/JavaSearchBugs/src/b108088/Test108088.java
@@ -0,0 +1,12 @@
+package b108088;
+class A108088 {
+    B108088 fB;
+    public void subroutine(float subArg) {
+        subsub();
+    }
+    public void subsub() {
+    }
+    
+}
+public class Test108088 {
+}
