diff --git a/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/compiler/SourceElementParser.java b/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/compiler/SourceElementParser.java
index 1a968b5..f1da087 100644
--- a/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/compiler/SourceElementParser.java
+++ b/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/compiler/SourceElementParser.java
@@ -157,12 +157,6 @@ public void checkComment() {
 			if (this.javadocParser.checkDeprecation(lastComment)) {
 				checkAndSetModifiers(ClassFileConstants.AccDeprecated);
 			}
-			if (this.javadocParser.notNull) {
-				checkAndSetModifiers(ExtraCompilerModifiers.AccNotNull);
-			}
-			if (this.javadocParser.nullable) { // no else on purpose
-				checkAndSetModifiers(ExtraCompilerModifiers.AccNullable);
-			}
 			this.javadoc = this.javadocParser.docComment;	// null if check javadoc is not activated
 			if (currentElement == null) this.lastJavadocEnd = commentEnd;
 		}
