diff --git a/org.eclipse.jdt.apt.tests/src/org/eclipse/jdt/apt/tests/TestAll.java b/org.eclipse.jdt.apt.tests/src/org/eclipse/jdt/apt/tests/TestAll.java
index 3bd94d9..437bf08 100644
--- a/org.eclipse.jdt.apt.tests/src/org/eclipse/jdt/apt/tests/TestAll.java
+++ b/org.eclipse.jdt.apt.tests/src/org/eclipse/jdt/apt/tests/TestAll.java
@@ -45,6 +45,7 @@ public class TestAll extends TestCase {
 		suite.addTest(MirrorUtilTests.suite());
 		suite.addTest(AnnotationValueConversionTests.suite());
 		suite.addTest(JavaVersionTests.suite());
+		suite.addTest(RegressionTests.suite());
 	
 		return suite;
