diff --git a/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavadocCompletionContextTests.java b/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavadocCompletionContextTests.java
index 051f1be..73b5ac2 100644
--- a/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavadocCompletionContextTests.java
+++ b/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavadocCompletionContextTests.java
@@ -10,8 +10,6 @@
  *******************************************************************************/
 package org.eclipse.jdt.core.tests.model;
 
-import java.lang.reflect.Method;
-
 import org.eclipse.jdt.core.*;
 
 import org.eclipse.jdt.internal.codeassist.RelevanceConstants;
@@ -36,20 +34,7 @@ public void tearDownSuite() throws Exception {
 	super.tearDownSuite();
 }
 public static Test suite() {
-	TestSuite suite = new Suite(JavadocCompletionContextTests.class.getName());		
-
-	if (true) {
-		Class c = JavadocCompletionContextTests.class;
-		Method[] methods = c.getMethods();
-		for (int i = 0, max = methods.length; i < max; i++) {
-			if (methods[i].getName().startsWith("test")) { //$NON-NLS-1$
-				suite.addTest(new JavadocCompletionContextTests(methods[i].getName()));
-			}
-		}
-		return suite;
-	}
-	suite.addTest(new JavadocCompletionContextTests("test0050"));			
-	return suite;
+	return buildModelTestSuite(JavadocCompletionContextTests.class);
 }
 public void test0001() throws JavaModelException {
 	this.workingCopies = new ICompilationUnit[1];
