diff --git a/org.eclipse.jdt.core.tests.model/workspace/Formatter/test612/A_out.java b/org.eclipse.jdt.core.tests.model/workspace/Formatter/test612/A_out.java
new file mode 100644
index 0000000..ad99c5c
--- /dev/null
+++ b/org.eclipse.jdt.core.tests.model/workspace/Formatter/test612/A_out.java
@@ -0,0 +1,15 @@
+package test612;
+
+@interface Annot {
+String value() default "";
+}
+
+@interface Annot2 {}
+
+public class A {
+	A() {
+	}
+}
+
+class B {
+}
