diff --git a/org.eclipse.jdt.apt.tests/src/org/eclipse/jdt/apt/tests/annotations/extradependency/ExtraDependencyAnnotationProcessor.java b/org.eclipse.jdt.apt.tests/src/org/eclipse/jdt/apt/tests/annotations/extradependency/ExtraDependencyAnnotationProcessor.java
index 7479ab0..15c8aff 100644
--- a/org.eclipse.jdt.apt.tests/src/org/eclipse/jdt/apt/tests/annotations/extradependency/ExtraDependencyAnnotationProcessor.java
+++ b/org.eclipse.jdt.apt.tests/src/org/eclipse/jdt/apt/tests/annotations/extradependency/ExtraDependencyAnnotationProcessor.java
@@ -12,21 +12,17 @@
 
 package org.eclipse.jdt.apt.tests.annotations.extradependency;
 
-import com.sun.mirror.apt.AnnotationProcessor;
+import org.eclipse.jdt.apt.tests.annotations.BaseProcessor;
+
 import com.sun.mirror.apt.AnnotationProcessorEnvironment;
 
-public class ExtraDependencyAnnotationProcessor implements AnnotationProcessor
-{
+public class ExtraDependencyAnnotationProcessor extends BaseProcessor {
 
-	public ExtraDependencyAnnotationProcessor(AnnotationProcessorEnvironment env)
-	{
-		_env = env;
+	public ExtraDependencyAnnotationProcessor(AnnotationProcessorEnvironment env) {
+		super(env);
 	}
 
-	public void process()
-	{
+	public void process() {
 		_env.getTypeDeclaration( "p1.p2.p3.p4.C" ); //$NON-NLS-1$
 	}
-
-	AnnotationProcessorEnvironment	_env;
 }
