diff --git a/org.eclipse.jdt.apt.tests/src/org/eclipse/jdt/apt/tests/annotations/generic/GenericAnnotation.java b/org.eclipse.jdt.apt.tests/src/org/eclipse/jdt/apt/tests/annotations/generic/GenericAnnotation.java
new file mode 100644
index 0000000..7824c9c
--- /dev/null
+++ b/org.eclipse.jdt.apt.tests/src/org/eclipse/jdt/apt/tests/annotations/generic/GenericAnnotation.java
@@ -0,0 +1,5 @@
+package org.eclipse.jdt.apt.tests.annotations.generic;
+
+public @interface GenericAnnotation {
+
+}
