diff --git a/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/MethodVerifyTest.java b/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/MethodVerifyTest.java
index 049aef2..3cfc334 100644
--- a/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/MethodVerifyTest.java
+++ b/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/MethodVerifyTest.java
@@ -5024,7 +5024,7 @@ public class MethodVerifyTest extends AbstractComparableTest {
 			""
 		);
 	}
-	public void testONLY_087() {
+	public void test087() {
 		this.runNegativeTest(
 			new String[] {
 				"X.java",
