diff --git a/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/GetSourceTests.java b/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/GetSourceTests.java
index a062499..72e55e5 100644
--- a/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/GetSourceTests.java
+++ b/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/GetSourceTests.java
@@ -78,7 +78,7 @@ public class GetSourceTests extends ModifyingResourceTests {
 //		TESTS_RANGE = new int[] { 16, -1 };
 	}
 	public static Test suite() {
-		return buildTestSuite(GetSourceTests.class);
+		return buildModelTestSuite(GetSourceTests.class);
 	}
 	
 	public void tearDownSuite() throws Exception {
