diff --git a/org.eclipse.jdt.core.tests.model/workspace/Formatter/test613/A_in.java b/org.eclipse.jdt.core.tests.model/workspace/Formatter/test613/A_in.java
new file mode 100644
index 0000000..215145d
--- /dev/null
+++ b/org.eclipse.jdt.core.tests.model/workspace/Formatter/test613/A_in.java
@@ -0,0 +1,13 @@
+package test612;
+
+@interface Annot {
+	String value() default "";
+}
+@interface Annot2 {
+}
+public class A {
+	A() {
+	}
+}
+class B {
+}
