diff --git a/org.eclipse.jdt.core/eval/org/eclipse/jdt/internal/eval/EvaluationContext.java b/org.eclipse.jdt.core/eval/org/eclipse/jdt/internal/eval/EvaluationContext.java
index 8119fd0..6dd5e77 100644
--- a/org.eclipse.jdt.core/eval/org/eclipse/jdt/internal/eval/EvaluationContext.java
+++ b/org.eclipse.jdt.core/eval/org/eclipse/jdt/internal/eval/EvaluationContext.java
@@ -335,6 +335,22 @@ public void evaluateVariables(INameEnvironment environment, Map options, IReques
 	ClassFile[] classes = evaluator.getClasses();
 	if (classes != null) {
 		if (classes.length > 0) {
+			// Sort classes so that enclosing types are cached before nested types
+			// otherwise an AbortCompilation is thrown in 1.5 mode since the enclosing type
+			// is needed to resolve a nested type
+			Util.sort(classes, new Util.Comparer() {
+				public int compare(Object a, Object b) {
+					if (a == b) return 0;
+					ClassFile enclosing = ((ClassFile) a).enclosingClassFile;
+					while (enclosing != null) {
+						if (enclosing == b)
+							return 1;
+						enclosing = enclosing.enclosingClassFile;
+					}
+					return -1;
+				}
+			});
+			
 			// Send classes
 			if (!requestor.acceptClassFiles(classes, null)) {
 				throw new InstallException();
