diff --git a/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/JavaProject.java b/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/JavaProject.java
index 01ed8d8..c24e2f1 100644
--- a/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/JavaProject.java
+++ b/org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/JavaProject.java
@@ -21,6 +21,7 @@ import javax.xml.parsers.DocumentBuilder;
 import javax.xml.parsers.DocumentBuilderFactory;
 import javax.xml.parsers.ParserConfigurationException;
 import org.eclipse.core.resources.ICommand;
+import org.eclipse.core.resources.IContainer;
 import org.eclipse.core.resources.IFile;
 import org.eclipse.core.resources.IFolder;
 import org.eclipse.core.resources.IMarker;
@@ -380,8 +381,10 @@ public class JavaProject
 		info.setChildren(
 			computePackageFragmentRoots(classpath, false, null /*no reverse map*/));		
 	}
-	
 
+	public IContainer[][] computeCleanedFolders() {
+		return org.eclipse.jdt.internal.core.builder.JavaBuilder.computeCleanedFolders(this);
+	}
 
 	/**
 	 * Internal computation of an expanded classpath. It will eliminate duplicates, and produce copies
