diff --git a/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/ast/Expression.java b/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/ast/Expression.java
index d516793..2a2916c 100644
--- a/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/ast/Expression.java
+++ b/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/ast/Expression.java
@@ -492,6 +492,9 @@ public void checkNPE(BlockScope scope, FlowContext flowContext,
 		}
 		flowInfo.markAsComparedEqualToNonNull(local); 
 			// from thereon it is set
+		if (flowContext.initsOnFinally != null) {
+			flowContext.initsOnFinally.markAsComparedEqualToNonNull(local); 
+		}
 	}
 }
