diff --git a/org.eclipse.jdt.core.tests.model/workspace/JavaSearchBugs/src/b124645/xy/Y_124645.java b/org.eclipse.jdt.core.tests.model/workspace/JavaSearchBugs/src/b124645/xy/Y_124645.java
new file mode 100644
index 0000000..e08e9fb
--- /dev/null
+++ b/org.eclipse.jdt.core.tests.model/workspace/JavaSearchBugs/src/b124645/xy/Y_124645.java
@@ -0,0 +1,4 @@
+package b124645.xy;
+import xy.BE_124645;
+public class Y_124645 extends BE_124645 { 
+}
