diff --git a/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTModelBridgeTests.java b/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTModelBridgeTests.java
index bf9e436..3f7fe3f 100644
--- a/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTModelBridgeTests.java
+++ b/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTModelBridgeTests.java
@@ -29,7 +29,7 @@ public class ASTModelBridgeTests extends AbstractASTTests {
 	}
 
 	public static Test suite() {
-		return buildTestSuite(ASTModelBridgeTests.class);
+		return buildModelTestSuite(ASTModelBridgeTests.class);
 	}
 	
 	// Use this static initializer to specify subset for tests
