diff --git a/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavadocCompletionContextTests_1_5.java b/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavadocCompletionContextTests_1_5.java
index a8993c0..7e121f7 100644
--- a/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavadocCompletionContextTests_1_5.java
+++ b/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavadocCompletionContextTests_1_5.java
@@ -10,8 +10,6 @@
  *******************************************************************************/
 package org.eclipse.jdt.core.tests.model;
 
-import java.lang.reflect.Method;
-
 import org.eclipse.jdt.core.*;
 import org.eclipse.jdt.internal.codeassist.RelevanceConstants;
 
@@ -35,20 +33,7 @@ public void tearDownSuite() throws Exception {
 	super.tearDownSuite();
 }
 public static Test suite() {
-	TestSuite suite = new Suite(JavadocCompletionContextTests_1_5.class.getName());		
-
-	if (true) {
-		Class c = JavadocCompletionContextTests_1_5.class;
-		Method[] methods = c.getMethods();
-		for (int i = 0, max = methods.length; i < max; i++) {
-			if (methods[i].getName().startsWith("test")) { //$NON-NLS-1$
-				suite.addTest(new JavadocCompletionContextTests_1_5(methods[i].getName()));
-			}
-		}
-		return suite;
-	}
-	suite.addTest(new JavadocCompletionContextTests_1_5("test0214"));			
-	return suite;
+	return buildModelTestSuite(JavadocCompletionContextTests_1_5.class);
 }
 public void test0001() throws JavaModelException {
 	this.workingCopies = new ICompilationUnit[1];
