diff --git a/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/parser/TestAll.java b/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/parser/TestAll.java
index 965e01b..a53e64a 100644
--- a/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/parser/TestAll.java
+++ b/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/parser/TestAll.java
@@ -61,6 +61,6 @@ public class TestAll extends TestCase {
 		testClasses.add(ParserTest.class);
 		testClasses.add(ComplianceDiagnoseTest.class);
 
-		return AbstractCompilerTest.buildAllCompliancesSetupSuite(TestAll.class, CompilerTestSetup.class, testClasses);
+		return AbstractCompilerTest.buildAllCompliancesTestSuite(TestAll.class, CompilerTestSetup.class, testClasses);
 	}
 }
