diff --git a/org.eclipse.jdt.apt.core/src/org/eclipse/jdt/apt/core/util/AptUtil.java b/org.eclipse.jdt.apt.core/src/org/eclipse/jdt/apt/core/util/AptUtil.java
index 88500b3..cc47f82 100644
--- a/org.eclipse.jdt.apt.core/src/org/eclipse/jdt/apt/core/util/AptUtil.java
+++ b/org.eclipse.jdt.apt.core/src/org/eclipse/jdt/apt/core/util/AptUtil.java
@@ -30,7 +30,6 @@ public final class AptUtil {
 	 * 
 	 * @param fullyQualifiedAnnotation the annotation for which a factory
 	 * is desired. This must be fully qualfied -- e.g. "org.eclipse.annotation.Foo"
-	 * @param project the project in which the annotation was found
 	 */
 	public static AnnotationProcessorFactory getFactoryForAnnotation(
 			final String fullyQualifiedAnnotation,
