diff --git a/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/BatchCompilerTest.java b/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/BatchCompilerTest.java
index b6127b5..09d50eb 100644
--- a/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/BatchCompilerTest.java
+++ b/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/BatchCompilerTest.java
@@ -52,9 +52,6 @@ public static Test suite() {
 	}
 	
 	private String getExtDirectory() {
-		if (Util.isMacOS()) {
-			return JRE_HOME_DIR + "/lib/ext";
-		}
 		return JRE_HOME_DIR + "/lib/ext";
 	}
 	/**
