diff --git a/org.eclipse.jdt.apt.core/src/org/eclipse/jdt/apt/core/internal/AptPlugin.java b/org.eclipse.jdt.apt.core/src/org/eclipse/jdt/apt/core/internal/AptPlugin.java
index 683e340..597ccde 100644
--- a/org.eclipse.jdt.apt.core/src/org/eclipse/jdt/apt/core/internal/AptPlugin.java
+++ b/org.eclipse.jdt.apt.core/src/org/eclipse/jdt/apt/core/internal/AptPlugin.java
@@ -58,8 +58,13 @@ public class AptPlugin extends Plugin {
 		AnnotationProcessorFactoryLoader.getLoader();
 		// register resource-changed listener
 		// TODO: can move this into AptProject.
-		int mask = IResourceChangeEvent.PRE_BUILD | IResourceChangeEvent.PRE_CLOSE | IResourceChangeEvent.PRE_DELETE;
+		int mask = 
+			IResourceChangeEvent.PRE_BUILD | 
+			IResourceChangeEvent.PRE_CLOSE | 
+			IResourceChangeEvent.PRE_DELETE | 
+			IResourceChangeEvent.POST_CHANGE;
 		JavaCore.addPreProcessingResourceChangedListener( new GeneratedResourceChangeListener(), mask );
+		
 		if( DEBUG )
 			trace("registered resource change listener"); //$NON-NLS-1$
 	}
