diff --git a/org.eclipse.jdt.core.tests.builder/src/org/eclipse/jdt/core/tests/builder/MultiProjectTests.java b/org.eclipse.jdt.core.tests.builder/src/org/eclipse/jdt/core/tests/builder/MultiProjectTests.java
index 9444a10..a5b5ac3 100644
--- a/org.eclipse.jdt.core.tests.builder/src/org/eclipse/jdt/core/tests/builder/MultiProjectTests.java
+++ b/org.eclipse.jdt.core.tests.builder/src/org/eclipse/jdt/core/tests/builder/MultiProjectTests.java
@@ -30,7 +30,7 @@ public class MultiProjectTests extends Tests {
 	}
 	
 	public static Test suite() {
-		if (true) {
+		if (false) {
 			TestSuite suite = new TestSuite(MultiProjectTests.class.getName());
 			suite.addTest(new MultiProjectTests("testIgnoreIfBetterNonAccessibleRule1"));
 			return suite;
