diff --git a/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/TypeHierarchyTests.java b/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/TypeHierarchyTests.java
index 8581000..a87d58a 100644
--- a/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/TypeHierarchyTests.java
+++ b/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/TypeHierarchyTests.java
@@ -37,7 +37,7 @@ static {
 //	TESTS_NAMES= new String[] { "testGeneric7" };
 }
 public static Test suite() {
-	return buildTestSuite(TypeHierarchyTests.class);
+	return buildModelTestSuite(TypeHierarchyTests.class);
 }
 
 /* (non-Javadoc)
