diff --git a/org.eclipse.jdt.core.tests.builder/src/org/eclipse/jdt/core/tests/builder/Tests.java b/org.eclipse.jdt.core.tests.builder/src/org/eclipse/jdt/core/tests/builder/Tests.java
index 79a4ade..0f90b70 100644
--- a/org.eclipse.jdt.core.tests.builder/src/org/eclipse/jdt/core/tests/builder/Tests.java
+++ b/org.eclipse.jdt.core.tests.builder/src/org/eclipse/jdt/core/tests/builder/Tests.java
@@ -407,7 +407,6 @@ public class Tests extends TestCase {
 		suite.addTest(AbstractMethodTests.suite());
 		suite.addTest(BasicBuildTests.suite());
 		suite.addTest(BuildpathTests.suite());
-		suite.addTest(CleanOutputFolderTests.suite());
 		suite.addTest(CopyResourceTests.suite());
 		suite.addTest(DependencyTests.suite());
 		suite.addTest(ErrorsTests.suite());
