diff --git a/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/parser/SelectionJavadocTest.java b/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/parser/SelectionJavadocTest.java
index 9fbbef5..479f4c1 100644
--- a/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/parser/SelectionJavadocTest.java
+++ b/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/parser/SelectionJavadocTest.java
@@ -55,7 +55,7 @@ public class SelectionJavadocTest extends AbstractSelectionTest {
 	}
 
 	public static Test suite() {
-		return buildTestSuite(SelectionJavadocTest.class);
+		return buildAllCompliancesRegressionTestSetupSuite(SelectionJavadocTest.class);
 	}
 
 	class SelectionVisitor extends ASTVisitor {
