diff --git a/lucene/contrib/misc/src/java/org/apache/lucene/index/IndexSplitter.java b/lucene/contrib/misc/src/java/org/apache/lucene/index/IndexSplitter.java
index 5641a74..e2f3ecf 100644
--- a/lucene/contrib/misc/src/java/org/apache/lucene/index/IndexSplitter.java
+++ b/lucene/contrib/misc/src/java/org/apache/lucene/index/IndexSplitter.java
@@ -26,6 +26,8 @@ import java.text.DecimalFormat;
 import java.util.ArrayList;
 import java.util.List;
 
+import org.apache.lucene.index.codecs.Codec;
+import org.apache.lucene.index.codecs.CodecProvider;
 import org.apache.lucene.store.FSDirectory;
 
 /**
@@ -47,6 +49,8 @@ import org.apache.lucene.store.FSDirectory;
  */
 public class IndexSplitter {
   public SegmentInfos infos;
+  
+  private final CodecProvider codecs;
 
   FSDirectory fsDir;
 
@@ -89,10 +93,15 @@ public class IndexSplitter {
   }
 
   public IndexSplitter(File dir) throws IOException {
+    this(dir, CodecProvider.getDefault());
+  }
+  
+  public IndexSplitter(File dir, CodecProvider codecs) throws IOException {
     this.dir = dir;
+    this.codecs = codecs;
     fsDir = FSDirectory.open(dir);
-    infos = new SegmentInfos();
-    infos.read(fsDir);
+    infos = new SegmentInfos(codecs);
+    infos.read(fsDir, codecs);
   }
 
   public void listSegments() throws IOException {
@@ -131,7 +140,7 @@ public class IndexSplitter {
   public void split(File destDir, String[] segs) throws IOException {
     destDir.mkdirs();
     FSDirectory destFSDir = FSDirectory.open(destDir);
-    SegmentInfos destInfos = new SegmentInfos();
+    SegmentInfos destInfos = new SegmentInfos(codecs);
     for (String n : segs) {
       SegmentInfo info = getInfo(n);
       destInfos.add(info);
