diff --git a/solr/src/java/org/apache/solr/response/TextResponseWriter.java b/solr/src/java/org/apache/solr/response/TextResponseWriter.java
index fae3e6b..9a7d9cf 100644
--- a/solr/src/java/org/apache/solr/response/TextResponseWriter.java
+++ b/solr/src/java/org/apache/solr/response/TextResponseWriter.java
@@ -37,7 +37,7 @@ import org.apache.solr.search.ReturnFields;
 
 /** Base class for text-oriented response writers.
  *
- * @version $Id$
+ *
  */
 public abstract class TextResponseWriter {
