diff --git a/solr/src/java/org/apache/solr/analysis/PatternTokenizerFactory.java b/solr/src/java/org/apache/solr/analysis/PatternTokenizerFactory.java
index 2e748f5..8996bb4 100644
--- a/solr/src/java/org/apache/solr/analysis/PatternTokenizerFactory.java
+++ b/solr/src/java/org/apache/solr/analysis/PatternTokenizerFactory.java
@@ -62,7 +62,7 @@ import org.apache.solr.common.SolrException;
  * 
  * @see PatternTokenizer
  * @since solr1.2
- * @version $Id$
+ *
  */
 public class PatternTokenizerFactory extends BaseTokenizerFactory 
 {
