diff --git a/lucene/contrib/queryparser/src/java/org/apache/lucene/queryParser/standard/parser/JavaCharStream.java b/lucene/contrib/queryparser/src/java/org/apache/lucene/queryParser/standard/parser/JavaCharStream.java
index 6c0bab9..94cd1c9 100644
--- a/lucene/contrib/queryparser/src/java/org/apache/lucene/queryParser/standard/parser/JavaCharStream.java
+++ b/lucene/contrib/queryparser/src/java/org/apache/lucene/queryParser/standard/parser/JavaCharStream.java
@@ -347,7 +347,7 @@ public class JavaCharStream
 
   @Deprecated
   /**
-   * @deprecated
+   * @deprecated (gen)
    * @see #getEndColumn
    */
   public int getColumn() {
@@ -356,7 +356,7 @@ public class JavaCharStream
 
   @Deprecated
   /**
-   * @deprecated
+   * @deprecated (gen)
    * @see #getEndLine
    */
   public int getLine() {
