diff --git a/solr/src/java/org/apache/solr/spelling/SpellingResult.java b/solr/src/java/org/apache/solr/spelling/SpellingResult.java
index aeeacbd..7eb1c66 100644
--- a/solr/src/java/org/apache/solr/spelling/SpellingResult.java
+++ b/solr/src/java/org/apache/solr/spelling/SpellingResult.java
@@ -66,7 +66,13 @@ public class SpellingResult {
     }
   }
 
-  public void add(Token token, int docFreq) {
+  /**
+   * Adds an original token with its document frequency
+   * 
+   * @param token original token
+   * @param docFreq original token's document frequency
+   */
+  public void addFrequency(Token token, int docFreq) {
     if (tokenFrequency == null) {
       tokenFrequency = new LinkedHashMap<Token, Integer>();
     }
