diff --git a/lucene/src/test/org/apache/lucene/util/LuceneTestCase.java b/lucene/src/test/org/apache/lucene/util/LuceneTestCase.java
index 861ca1e..986877b 100644
--- a/lucene/src/test/org/apache/lucene/util/LuceneTestCase.java
+++ b/lucene/src/test/org/apache/lucene/util/LuceneTestCase.java
@@ -659,7 +659,7 @@ public abstract class LuceneTestCase extends Assert {
   public static MockDirectoryWrapper newDirectory(Random r) throws IOException {
     StackTraceElement[] stack = new Exception().getStackTrace();
     Directory impl = newDirectoryImpl(r, TEST_DIRECTORY);
-    MockDirectoryWrapper dir = new MockDirectoryWrapper(impl);
+    MockDirectoryWrapper dir = new MockDirectoryWrapper(r, impl);
     stores.put(dir, stack);
     return dir;
   }
@@ -679,7 +679,7 @@ public abstract class LuceneTestCase extends Assert {
     for (String file : d.listAll()) {
      d.copy(impl, file, file);
     }
-    MockDirectoryWrapper dir = new MockDirectoryWrapper(impl);
+    MockDirectoryWrapper dir = new MockDirectoryWrapper(r, impl);
     stores.put(dir, stack);
     return dir;
   }
