diff --git a/solr/src/solrj/org/apache/solr/client/solrj/request/UpdateRequest.java b/solr/src/solrj/org/apache/solr/client/solrj/request/UpdateRequest.java
index eb6b332..1b5ae4a 100644
--- a/solr/src/solrj/org/apache/solr/client/solrj/request/UpdateRequest.java
+++ b/solr/src/solrj/org/apache/solr/client/solrj/request/UpdateRequest.java
@@ -47,6 +47,7 @@ public class UpdateRequest extends AbstractUpdateRequest {
    *
    * @deprecated Use {@link AbstractUpdateRequest.ACTION} instead
    */
+  @Deprecated
   public enum ACTION {
     COMMIT,
     OPTIMIZE
@@ -137,6 +138,7 @@ public class UpdateRequest extends AbstractUpdateRequest {
    *
    * @deprecated Use {@link org.apache.solr.client.solrj.request.AbstractUpdateRequest.ACTION} instead
    * */
+  @Deprecated
   public UpdateRequest setAction(ACTION action, boolean waitFlush, boolean waitSearcher ) {
     return setAction(action, waitFlush, waitSearcher, 1);
   }
@@ -145,6 +147,7 @@ public class UpdateRequest extends AbstractUpdateRequest {
    *
    * @deprecated Use {@link org.apache.solr.client.solrj.request.AbstractUpdateRequest.ACTION} instead
    */
+  @Deprecated
   public UpdateRequest setAction(ACTION action, boolean waitFlush, boolean waitSearcher, int maxSegments ) {
     if (params == null)
       params = new ModifiableSolrParams();
@@ -166,6 +169,7 @@ public class UpdateRequest extends AbstractUpdateRequest {
    *
    * @deprecated Use {@link org.apache.solr.client.solrj.request.AbstractUpdateRequest.ACTION} instead
    */
+  @Deprecated
   public UpdateRequest setAction(ACTION action, boolean waitFlush, boolean waitSearcher, int maxSegments , boolean expungeDeletes) {
     setAction(action, waitFlush, waitSearcher,maxSegments) ;
     params.set(UpdateParams.EXPUNGE_DELETES,""+expungeDeletes);
