diff --git a/solr/src/java/org/apache/solr/schema/LongField.java b/solr/src/java/org/apache/solr/schema/LongField.java
index 239dc1f..7c0d2ac 100644
--- a/solr/src/java/org/apache/solr/schema/LongField.java
+++ b/solr/src/java/org/apache/solr/schema/LongField.java
@@ -29,7 +29,7 @@ import org.apache.solr.search.function.LongFieldSource;
 import java.io.IOException;
 import java.util.Map;
 /**
- * @version $Id$
+ *
  */
 public class LongField extends FieldType {
   @Override
