diff --git a/solr/src/java/org/apache/solr/search/function/MaxDocValueSource.java b/solr/src/java/org/apache/solr/search/function/MaxDocValueSource.java
index 30d9452..908693f 100755
--- a/solr/src/java/org/apache/solr/search/function/MaxDocValueSource.java
+++ b/solr/src/java/org/apache/solr/search/function/MaxDocValueSource.java
@@ -17,6 +17,8 @@
 package org.apache.solr.search.function;
 
 import org.apache.lucene.index.IndexReader.AtomicReaderContext;
+import org.apache.lucene.queries.function.DocValues;
+import org.apache.lucene.queries.function.ValueSource;
 import org.apache.lucene.search.IndexSearcher;
 
 import java.io.IOException;
