diff --git a/solr/src/test/org/apache/solr/core/TestSolrDeletionPolicy2.java b/solr/src/test/org/apache/solr/core/TestSolrDeletionPolicy2.java
index 1235910..61ac85a 100644
--- a/solr/src/test/org/apache/solr/core/TestSolrDeletionPolicy2.java
+++ b/solr/src/test/org/apache/solr/core/TestSolrDeletionPolicy2.java
@@ -17,7 +17,6 @@
 package org.apache.solr.core;
 
 import org.apache.solr.SolrTestCaseJ4;
-import org.apache.solr.util.AbstractSolrTestCase;
 import org.junit.BeforeClass;
 import org.junit.Test;
