diff --git a/solr/src/test/org/apache/solr/schema/BadIndexSchemaTest.java b/solr/src/test/org/apache/solr/schema/BadIndexSchemaTest.java
index 96e5060..d8ae588 100644
--- a/solr/src/test/org/apache/solr/schema/BadIndexSchemaTest.java
+++ b/solr/src/test/org/apache/solr/schema/BadIndexSchemaTest.java
@@ -22,10 +22,6 @@ import org.apache.solr.common.SolrException;
 import org.apache.solr.common.SolrException.ErrorCode;
 import org.apache.solr.core.SolrConfig;
 
-import static org.junit.Assert.assertTrue;
-import static org.junit.Assert.fail;
-
-import org.junit.Before;
 import org.junit.Test;
 
 public class BadIndexSchemaTest extends SolrTestCaseJ4 {
