diff --git a/lucene/src/test-framework/org/apache/lucene/index/codecs/mockintblock/MockFixedIntBlockCodec.java b/lucene/src/test-framework/org/apache/lucene/index/codecs/mockintblock/MockFixedIntBlockCodec.java
index b653471..222d25a 100644
--- a/lucene/src/test-framework/org/apache/lucene/index/codecs/mockintblock/MockFixedIntBlockCodec.java
+++ b/lucene/src/test-framework/org/apache/lucene/index/codecs/mockintblock/MockFixedIntBlockCodec.java
@@ -186,10 +186,11 @@ public class MockFixedIntBlockCodec extends Codec {
   }
 
   @Override
-  public void files(Directory dir, SegmentInfo segmentInfo, String codecId, Set<String> files) {
-    SepPostingsReaderImpl.files(segmentInfo, codecId, files);
-    BlockTermsReader.files(dir, segmentInfo, codecId, files);
-    FixedGapTermsIndexReader.files(dir, segmentInfo, codecId, files);
+  public void files(Directory dir, SegmentInfo segmentInfo, int codecId, Set<String> files) {
+    final String codecIdAsString = "" + codecId;
+    SepPostingsReaderImpl.files(segmentInfo, codecIdAsString, files);
+    BlockTermsReader.files(dir, segmentInfo, codecIdAsString, files);
+    FixedGapTermsIndexReader.files(dir, segmentInfo, codecIdAsString, files);
   }
 
   @Override
