diff --git a/lucene/contrib/demo/src/java/org/apache/lucene/demo/SearchFiles.java b/lucene/contrib/demo/src/java/org/apache/lucene/demo/SearchFiles.java
index 65629fe..0d76e4c 100644
--- a/lucene/contrib/demo/src/java/org/apache/lucene/demo/SearchFiles.java
+++ b/lucene/contrib/demo/src/java/org/apache/lucene/demo/SearchFiles.java
@@ -19,6 +19,7 @@ package org.apache.lucene.demo;
 
 import java.io.BufferedReader;
 import java.io.File;
+import java.io.FileInputStream;
 import java.io.FileReader;
 import java.io.IOException;
 import java.io.InputStreamReader;
@@ -124,7 +125,7 @@ public class SearchFiles {
 
     BufferedReader in = null;
     if (queries != null) {
-      in = new BufferedReader(new FileReader(queries));
+      in = new BufferedReader(new InputStreamReader(new FileInputStream(queries), "UTF-8"));
     } else {
       in = new BufferedReader(new InputStreamReader(System.in, "UTF-8"));
     }
