diff --git a/solr/src/test/org/apache/solr/client/solrj/SolrQueryTest.java b/solr/src/test/org/apache/solr/client/solrj/SolrQueryTest.java
index 69d7d37..9c24c4d 100644
--- a/solr/src/test/org/apache/solr/client/solrj/SolrQueryTest.java
+++ b/solr/src/test/org/apache/solr/client/solrj/SolrQueryTest.java
@@ -24,7 +24,7 @@ import junit.framework.Assert;
 
 /**
  * 
- * @version $Id$
+ *
  * @since solr 1.3
  */
 public class SolrQueryTest extends LuceneTestCase {
