diff --git a/solr/src/java/org/apache/solr/response/JSONResponseWriter.java b/solr/src/java/org/apache/solr/response/JSONResponseWriter.java
index 7a51578..0e328fc 100644
--- a/solr/src/java/org/apache/solr/response/JSONResponseWriter.java
+++ b/solr/src/java/org/apache/solr/response/JSONResponseWriter.java
@@ -37,7 +37,7 @@ import org.apache.solr.schema.SchemaField;
 import org.apache.solr.search.ReturnFields;
 
 /**
- * @version $Id$
+ *
  */
 
 public class JSONResponseWriter implements QueryResponseWriter {
