diff --git a/solr/src/java/org/apache/solr/analysis/BaseCharFilterFactory.java b/solr/src/java/org/apache/solr/analysis/BaseCharFilterFactory.java
index 62ff65d..a7cef83 100644
--- a/solr/src/java/org/apache/solr/analysis/BaseCharFilterFactory.java
+++ b/solr/src/java/org/apache/solr/analysis/BaseCharFilterFactory.java
@@ -24,7 +24,6 @@ import org.slf4j.LoggerFactory;
 
 /**
 *
-* @version $Id$
 * @since Solr 1.4
 *
 */
