diff --git a/solr/src/java/org/apache/solr/analysis/TokenizerChain.java b/solr/src/java/org/apache/solr/analysis/TokenizerChain.java
index 3fd4884..cf04a82 100644
--- a/solr/src/java/org/apache/solr/analysis/TokenizerChain.java
+++ b/solr/src/java/org/apache/solr/analysis/TokenizerChain.java
@@ -25,7 +25,7 @@ import org.apache.lucene.analysis.Tokenizer;
 import java.io.Reader;
 
 /**
- * @version $Id$
+ *
  */
 
 //
