diff --git a/solr/src/java/org/apache/solr/search/SolrCacheBase.java b/solr/src/java/org/apache/solr/search/SolrCacheBase.java
index 742b779..23a7267 100644
--- a/solr/src/java/org/apache/solr/search/SolrCacheBase.java
+++ b/solr/src/java/org/apache/solr/search/SolrCacheBase.java
@@ -17,16 +17,6 @@
 
 package org.apache.solr.search;
 
-import org.apache.solr.common.SolrException;
-import org.apache.solr.common.util.NamedList;
-import org.apache.solr.common.util.SimpleOrderedMap;
-import org.apache.solr.core.SolrCore;
-
-import java.util.*;
-import java.util.concurrent.atomic.AtomicLong;
-import java.io.IOException;
-import java.net.URL;
-
 /**
  * Common base class of reusable functionality for SolrCaches
  */
