diff --git a/solr/src/java/org/apache/solr/response/CSVResponseWriter.java b/solr/src/java/org/apache/solr/response/CSVResponseWriter.java
index 3116467..d35ad3c 100755
--- a/solr/src/java/org/apache/solr/response/CSVResponseWriter.java
+++ b/solr/src/java/org/apache/solr/response/CSVResponseWriter.java
@@ -43,7 +43,7 @@ import java.io.Writer;
 import java.util.*;
 
 /**
- * @version $Id$
+ *
  */
 
 public class CSVResponseWriter implements QueryResponseWriter {
