diff --git a/solr/src/solrj/org/apache/solr/client/solrj/response/PivotField.java b/solr/src/solrj/org/apache/solr/client/solrj/response/PivotField.java
index 76621c6..a8ca4c8 100644
--- a/solr/src/solrj/org/apache/solr/client/solrj/response/PivotField.java
+++ b/solr/src/solrj/org/apache/solr/client/solrj/response/PivotField.java
@@ -18,7 +18,6 @@
 package org.apache.solr.client.solrj.response;
 
 import java.io.PrintStream;
-import java.io.PrintWriter;
 import java.io.Serializable;
 import java.util.List;
