diff --git a/solr/src/common/org/apache/solr/common/util/DOMUtil.java b/solr/src/common/org/apache/solr/common/util/DOMUtil.java
index d02be75..2d937aa 100644
--- a/solr/src/common/org/apache/solr/common/util/DOMUtil.java
+++ b/solr/src/common/org/apache/solr/common/util/DOMUtil.java
@@ -25,7 +25,7 @@ import org.w3c.dom.Node;
 import org.w3c.dom.NodeList;
 
 /**
- * @version $Id$
+ *
  */
 public class DOMUtil {
