diff --git a/solr/src/java/org/apache/solr/schema/DateField.java b/solr/src/java/org/apache/solr/schema/DateField.java
index d7bcef0..f7e16bd 100644
--- a/solr/src/java/org/apache/solr/schema/DateField.java
+++ b/solr/src/java/org/apache/solr/schema/DateField.java
@@ -93,7 +93,7 @@ import java.util.*;
  * acronym UTC was chosen as a compromise."
  * </blockquote>
  *
- * @version $Id$
+ *
  * @see <a href="http://www.w3.org/TR/xmlschema-2/#dateTime">XML schema part 2</a>
  *
  */
