diff --git a/lucene/contrib/queries/src/java/org/apache/lucene/search/ChainedFilter.java b/lucene/contrib/queries/src/java/org/apache/lucene/search/ChainedFilter.java
index f06d0e2..e95b506 100644
--- a/lucene/contrib/queries/src/java/org/apache/lucene/search/ChainedFilter.java
+++ b/lucene/contrib/queries/src/java/org/apache/lucene/search/ChainedFilter.java
@@ -149,18 +149,6 @@ public class ChainedFilter extends Filter
         return result;
     }
 
-    /** Provide a SortedVIntList when it is definitely
-     *  smaller than an OpenBitSet
-     *  @deprecated Either use CachingWrapperFilter, or
-     *  switch to a different DocIdSet implementation yourself.
-     *  This method will be removed in Lucene 4.0 
-     **/
-    @Deprecated
-    protected final DocIdSet finalResult(OpenBitSetDISI result, int maxDocs) {
-        return result;
-    }
-        
-
     /**
      * Delegates to each filter in the chain.
      * @param reader IndexReader
@@ -175,7 +163,7 @@ public class ChainedFilter extends Filter
         {
             doChain(result, logic, chain[index[0]].getDocIdSet(reader));
         }
-        return finalResult(result, reader.maxDoc());
+        return result;
     }
 
     /**
@@ -195,7 +183,7 @@ public class ChainedFilter extends Filter
         {
             doChain(result, logic[index[0]], chain[index[0]].getDocIdSet(reader));
         }
-        return finalResult(result, reader.maxDoc());
+        return result;
     }
 
     @Override
