diff --git a/solr/src/java/org/apache/solr/util/BoundedTreeSet.java b/solr/src/java/org/apache/solr/util/BoundedTreeSet.java
index dda5fda..4abe7b4 100644
--- a/solr/src/java/org/apache/solr/util/BoundedTreeSet.java
+++ b/solr/src/java/org/apache/solr/util/BoundedTreeSet.java
@@ -55,11 +55,13 @@ public class BoundedTreeSet<E> extends TreeSet<E> {
       remove(last());
     }
   }
+  @Override
   public boolean add(E item) {
     boolean out = super.add(item);
     adjust();
     return out;
   }
+  @Override
   public boolean addAll(Collection<? extends E> c) {
     boolean out = super.addAll(c);
     adjust();
