diff --git a/modules/analysis/stempel/src/java/org/apache/lucene/analysis/pl/PolishAnalyzer.java b/modules/analysis/stempel/src/java/org/apache/lucene/analysis/pl/PolishAnalyzer.java
index 80d9d6d..9593355 100644
--- a/modules/analysis/stempel/src/java/org/apache/lucene/analysis/pl/PolishAnalyzer.java
+++ b/modules/analysis/stempel/src/java/org/apache/lucene/analysis/pl/PolishAnalyzer.java
@@ -18,7 +18,6 @@ package org.apache.lucene.analysis.pl;
  */
 
 import java.io.IOException;
-import java.io.InputStream;
 import java.io.Reader;
 import java.util.Set;
