diff --git a/solr/contrib/clustering/src/test/java/org/apache/solr/handler/clustering/AbstractClusteringTestCase.java b/solr/contrib/clustering/src/test/java/org/apache/solr/handler/clustering/AbstractClusteringTestCase.java
index 19d7570..cb6bb1f 100644
--- a/solr/contrib/clustering/src/test/java/org/apache/solr/handler/clustering/AbstractClusteringTestCase.java
+++ b/solr/contrib/clustering/src/test/java/org/apache/solr/handler/clustering/AbstractClusteringTestCase.java
@@ -19,8 +19,6 @@ package org.apache.solr.handler.clustering;
 import org.apache.solr.SolrTestCaseJ4;
 import org.junit.BeforeClass;
 
-import static org.junit.Assert.*;
-
 
 /**
  *
