diff --git a/lucene/contrib/instantiated/src/java/org/apache/lucene/store/instantiated/InstantiatedTermsEnum.java b/lucene/contrib/instantiated/src/java/org/apache/lucene/store/instantiated/InstantiatedTermsEnum.java
index 4058f28..645e7e8 100644
--- a/lucene/contrib/instantiated/src/java/org/apache/lucene/store/instantiated/InstantiatedTermsEnum.java
+++ b/lucene/contrib/instantiated/src/java/org/apache/lucene/store/instantiated/InstantiatedTermsEnum.java
@@ -44,7 +44,20 @@ public class InstantiatedTermsEnum extends TermsEnum {
   }
 
   @Override
-  public SeekStatus seek(BytesRef text, boolean useCache) {
+  public boolean seekExact(BytesRef text, boolean useCache) {
+    final Term t = new Term(field, text);
+    int loc = Arrays.binarySearch(terms, t, InstantiatedTerm.termComparator);
+    if (loc < 0) {
+      return false;
+    } else {
+      upto = loc;
+      br.copy(text);
+      return true;
+    }
+  }
+
+  @Override
+  public SeekStatus seekCeil(BytesRef text, boolean useCache) {
     final Term t = new Term(field, text);
     int loc = Arrays.binarySearch(terms, t, InstantiatedTerm.termComparator);
     if (loc < 0) {
@@ -63,17 +76,10 @@ public class InstantiatedTermsEnum extends TermsEnum {
   }
 
   @Override
-  public SeekStatus seek(long ord) {
+  public void seekExact(long ord) {
+    assert (start + (int) ord) < terms.length;
     upto = start + (int) ord;
-    if (upto >= terms.length) {
-      return SeekStatus.END;
-    }
-
-    if (terms[upto].field().equals(field)) {
-      return SeekStatus.FOUND;
-    } else {
-      return SeekStatus.END;
-    }
+    assert field.equals(terms[upto].field());
   }
 
   @Override
@@ -140,9 +146,9 @@ public class InstantiatedTermsEnum extends TermsEnum {
   }
 
   @Override
-  public void seek(BytesRef term, TermState state) throws IOException {
+  public void seekExact(BytesRef term, TermState state) throws IOException {
     assert state != null && state instanceof OrdTermState;
-    seek(((OrdTermState)state).ord); // just use the ord for simplicity
+    seekExact(((OrdTermState)state).ord); // just use the ord for simplicity
   }
 }
