diff --git a/lucene/contrib/spatial/src/java/org/apache/lucene/spatial/tier/projections/SinusoidalProjector.java b/lucene/contrib/spatial/src/java/org/apache/lucene/spatial/tier/projections/SinusoidalProjector.java
index ea7694b..8cc151d 100644
--- a/lucene/contrib/spatial/src/java/org/apache/lucene/spatial/tier/projections/SinusoidalProjector.java
+++ b/lucene/contrib/spatial/src/java/org/apache/lucene/spatial/tier/projections/SinusoidalProjector.java
@@ -28,7 +28,7 @@ package org.apache.lucene.spatial.tier.projections;
  * flux and might change in incompatible ways in the next
  * release.</font>
  *
- * @deprecated Until we can put in place proper tests and a proper fix. 
+ * @deprecated (3.1) Until we can put in place proper tests and a proper fix. 
  */
 @Deprecated
 public class SinusoidalProjector implements IProjector {
