diff --git a/solr/src/java/org/apache/solr/request/SolrQueryRequest.java b/solr/src/java/org/apache/solr/request/SolrQueryRequest.java
index eeb3c60..931acbc 100644
--- a/solr/src/java/org/apache/solr/request/SolrQueryRequest.java
+++ b/solr/src/java/org/apache/solr/request/SolrQueryRequest.java
@@ -29,7 +29,7 @@ import java.util.Map;
  * <p>Container for a request to execute a query.</p>
  * <p><code>SolrQueryRequest</code> is not thread safe.</p>
  * 
- * @version $Id$
+ *
  */
 public interface SolrQueryRequest {
