diff --git a/solr/src/java/org/apache/solr/search/function/DivFloatFunction.java b/solr/src/java/org/apache/solr/search/function/DivFloatFunction.java
index a66c7c6..ad072b5 100755
--- a/solr/src/java/org/apache/solr/search/function/DivFloatFunction.java
+++ b/solr/src/java/org/apache/solr/search/function/DivFloatFunction.java
@@ -28,10 +28,12 @@ public class DivFloatFunction extends DualFloatFunction {
     super(a,b);
   }
 
+  @Override
   protected String name() {
     return "div";
   }
 
+  @Override
   protected float func(int doc, DocValues aVals, DocValues bVals) {
     return aVals.floatVal(doc) / bVals.floatVal(doc);
   }
