diff --git a/solr/src/test/org/apache/solr/util/TestUtils.java b/solr/src/test/org/apache/solr/util/TestUtils.java
index e1279e5..cf33955 100755
--- a/solr/src/test/org/apache/solr/util/TestUtils.java
+++ b/solr/src/test/org/apache/solr/util/TestUtils.java
@@ -28,7 +28,7 @@ import org.apache.solr.common.util.StrUtils;
 import org.junit.Assert;
 
 /**
- * @version $Id$
+ *
  */
 public class TestUtils extends LuceneTestCase {
   public void testSplitEscaping() {
