diff --git a/modules/analysis/common/src/java/org/apache/lucene/analysis/payloads/IdentityEncoder.java b/modules/analysis/common/src/java/org/apache/lucene/analysis/payloads/IdentityEncoder.java
index 0f30443..f143dda 100644
--- a/modules/analysis/common/src/java/org/apache/lucene/analysis/payloads/IdentityEncoder.java
+++ b/modules/analysis/common/src/java/org/apache/lucene/analysis/payloads/IdentityEncoder.java
@@ -28,23 +28,15 @@ import java.nio.charset.Charset;
  *
  **/
 public class IdentityEncoder extends AbstractEncoder implements PayloadEncoder{
-
   protected Charset charset = Charset.forName("UTF-8");
   
-  /** @deprecated This field is no longer used. Use {@link #charset} instead. */
-  @Deprecated
-  protected String charsetName = charset.name();
-
   public IdentityEncoder() {
   }
 
   public IdentityEncoder(Charset charset) {
     this.charset = charset;
-    // @deprecated, remove this in 4.0:
-    charsetName = charset.name();
   }
 
-
   public Payload encode(char[] buffer, int offset, int length) {
     final ByteBuffer bb = charset.encode(CharBuffer.wrap(buffer, offset, length));
     if (bb.hasArray()) {
