diff --git a/solr/src/java/org/apache/solr/search/QParserPlugin.java b/solr/src/java/org/apache/solr/search/QParserPlugin.java
index 55059e3..7873c5e 100755
--- a/solr/src/java/org/apache/solr/search/QParserPlugin.java
+++ b/solr/src/java/org/apache/solr/search/QParserPlugin.java
@@ -39,6 +39,7 @@ public abstract class QParserPlugin implements NamedListInitializedPlugin {
     NestedQParserPlugin.NAME, NestedQParserPlugin.class,
     FunctionRangeQParserPlugin.NAME, FunctionRangeQParserPlugin.class,
     SpatialFilterQParserPlugin.NAME, SpatialFilterQParserPlugin.class,
+    SpatialBoxQParserPlugin.NAME, SpatialBoxQParserPlugin.class,
   };
 
   /** return a {@link QParser} */
