diff --git a/solr/src/java/org/apache/solr/search/QueryUtils.java b/solr/src/java/org/apache/solr/search/QueryUtils.java
index 931776c..f8cd6e9 100755
--- a/solr/src/java/org/apache/solr/search/QueryUtils.java
+++ b/solr/src/java/org/apache/solr/search/QueryUtils.java
@@ -25,7 +25,7 @@ import org.apache.lucene.search.MatchAllDocsQuery;
 import java.util.List;
 
 /**
- * @version $Id$
+ *
  */
 public class QueryUtils {
