diff --git a/solr/src/test/org/apache/solr/cloud/ZkControllerTest.java b/solr/src/test/org/apache/solr/cloud/ZkControllerTest.java
index 0854a65..00b9568 100644
--- a/solr/src/test/org/apache/solr/cloud/ZkControllerTest.java
+++ b/solr/src/test/org/apache/solr/cloud/ZkControllerTest.java
@@ -33,7 +33,6 @@ import org.apache.zookeeper.KeeperException;
 
 import org.junit.BeforeClass;
 import org.junit.Test;
-import static org.junit.Assert.*;
 
 public class ZkControllerTest extends SolrTestCaseJ4 {
