diff --git a/solr/src/java/org/apache/solr/handler/admin/ThreadDumpHandler.java b/solr/src/java/org/apache/solr/handler/admin/ThreadDumpHandler.java
index 280b123..f885016 100644
--- a/solr/src/java/org/apache/solr/handler/admin/ThreadDumpHandler.java
+++ b/solr/src/java/org/apache/solr/handler/admin/ThreadDumpHandler.java
@@ -25,7 +25,6 @@ import java.lang.management.ThreadMXBean;
 import org.apache.solr.common.util.NamedList;
 import org.apache.solr.common.util.SimpleOrderedMap;
 import org.apache.solr.handler.RequestHandlerBase;
-import org.apache.solr.handler.RequestHandlerUtils;
 import org.apache.solr.request.SolrQueryRequest;
 import org.apache.solr.response.SolrQueryResponse;
