diff --git a/solr/src/test/org/apache/solr/update/DirectUpdateHandlerTest.java b/solr/src/test/org/apache/solr/update/DirectUpdateHandlerTest.java
index d58e780..3e4c1d2 100644
--- a/solr/src/test/org/apache/solr/update/DirectUpdateHandlerTest.java
+++ b/solr/src/test/org/apache/solr/update/DirectUpdateHandlerTest.java
@@ -354,8 +354,6 @@ public class DirectUpdateHandlerTest extends SolrTestCaseJ4 {
     // Delete the document
     DeleteUpdateCommand cmd = new DeleteUpdateCommand();
     cmd.id = id;
-    cmd.fromCommitted = true;
-    cmd.fromPending = true;
     
     updater.delete(cmd);
   }
