diff --git a/solr/src/test/org/apache/solr/handler/AnalysisRequestHandlerTestBase.java b/solr/src/test/org/apache/solr/handler/AnalysisRequestHandlerTestBase.java
index c46c275..feb827e 100644
--- a/solr/src/test/org/apache/solr/handler/AnalysisRequestHandlerTestBase.java
+++ b/solr/src/test/org/apache/solr/handler/AnalysisRequestHandlerTestBase.java
@@ -23,7 +23,7 @@ import org.apache.solr.common.util.NamedList;
 /**
  * A base class for all analysis request handler tests.
  *
- * @version $Id$
+ *
  * @since solr 1.4
  */
 public abstract class AnalysisRequestHandlerTestBase extends SolrTestCaseJ4 {
