diff --git a/lucene/contrib/db/bdb/src/java/org/apache/lucene/store/db/DbDirectory.java b/lucene/contrib/db/bdb/src/java/org/apache/lucene/store/db/DbDirectory.java
index 74478b1..829dbed 100644
--- a/lucene/contrib/db/bdb/src/java/org/apache/lucene/store/db/DbDirectory.java
+++ b/lucene/contrib/db/bdb/src/java/org/apache/lucene/store/db/DbDirectory.java
@@ -21,12 +21,7 @@ import java.io.FileNotFoundException;
 import java.io.IOException;
 import java.io.ByteArrayInputStream;
 import java.io.DataInputStream;
-import java.util.Set;
-import java.util.HashSet;
-import java.util.List;
-import java.util.ArrayList;
-import java.util.Iterator;
-import java.util.Collections;
+import java.util.*;
 
 import org.apache.lucene.store.Directory;
 import org.apache.lucene.store.Lock;
@@ -211,6 +206,10 @@ public class DbDirectory extends Directory {
     }
 
     @Override
+    public void sync(Collection<String> names) throws IOException {
+    }
+
+  @Override
     public IndexInput openInput(String name)
         throws IOException
     {
