diff --git a/lucene/contrib/queries/src/java/org/apache/lucene/search/TermsFilter.java b/lucene/contrib/queries/src/java/org/apache/lucene/search/TermsFilter.java
index f5c48c9..d9c4e51 100644
--- a/lucene/contrib/queries/src/java/org/apache/lucene/search/TermsFilter.java
+++ b/lucene/contrib/queries/src/java/org/apache/lucene/search/TermsFilter.java
@@ -71,7 +71,7 @@ public class TermsFilter extends Filter
       DocsEnum docs = null;
       for (Iterator<Term> iter = terms.iterator(); iter.hasNext();) {
         Term term = iter.next();
-        if (term.field() != lastField) {
+        if (!term.field().equals(lastField)) {
           termsC = fields.terms(term.field());
           termsEnum = termsC.iterator();
           lastField = term.field();
