diff --git a/solr/src/java/org/apache/solr/util/plugin/AbstractPluginLoader.java b/solr/src/java/org/apache/solr/util/plugin/AbstractPluginLoader.java
index 7860220..f3488b4 100644
--- a/solr/src/java/org/apache/solr/util/plugin/AbstractPluginLoader.java
+++ b/solr/src/java/org/apache/solr/util/plugin/AbstractPluginLoader.java
@@ -27,7 +27,6 @@ import org.apache.solr.common.SolrException;
 import org.apache.solr.common.SolrException.ErrorCode;
 import org.apache.solr.common.util.DOMUtil;
 import org.apache.solr.core.SolrConfig;
-import org.apache.solr.core.SolrResourceLoader;
 import org.w3c.dom.Node;
 import org.w3c.dom.NodeList;
