diff --git a/solr/src/java/org/apache/solr/response/XMLResponseWriter.java b/solr/src/java/org/apache/solr/response/XMLResponseWriter.java
index a883e9d..95d4654 100644
--- a/solr/src/java/org/apache/solr/response/XMLResponseWriter.java
+++ b/solr/src/java/org/apache/solr/response/XMLResponseWriter.java
@@ -24,7 +24,7 @@ import org.apache.solr.common.util.NamedList;
 import org.apache.solr.request.SolrQueryRequest;
 
 /**
- * @version $Id$
+ *
  */
 public class XMLResponseWriter implements QueryResponseWriter {
   public void init(NamedList n) {
