diff --git a/solr/src/test/org/apache/solr/BasicFunctionalityTest.java b/solr/src/test/org/apache/solr/BasicFunctionalityTest.java
index 1e76575..377bb30 100644
--- a/solr/src/test/org/apache/solr/BasicFunctionalityTest.java
+++ b/solr/src/test/org/apache/solr/BasicFunctionalityTest.java
@@ -162,7 +162,7 @@ public class BasicFunctionalityTest extends SolrTestCaseJ4 {
             ,"//*[@numFound='0']"
             );
 
-    // test allowDups default of false
+    // test overwrite default of true
 
     assertU(adoc("id", "42", "val_s", "AAA"));
     assertU(adoc("id", "42", "val_s", "BBB"));
@@ -181,12 +181,12 @@ public class BasicFunctionalityTest extends SolrTestCaseJ4 {
 
     // test deletes
     String [] adds = new String[] {
-      add( doc("id","101"), "allowDups", "false" ),
-      add( doc("id","101"), "allowDups", "false" ),
-      add( doc("id","105"), "allowDups", "true"  ),
-      add( doc("id","102"), "allowDups", "false" ),
-      add( doc("id","103"), "allowDups", "true"  ),
-      add( doc("id","101"), "allowDups", "false" ),
+      add( doc("id","101"), "overwrite", "true" ),
+      add( doc("id","101"), "overwrite", "true" ),
+      add( doc("id","105"), "overwrite", "false"  ),
+      add( doc("id","102"), "overwrite", "true" ),
+      add( doc("id","103"), "overwrite", "false"  ),
+      add( doc("id","101"), "overwrite", "true" ),
     };
     for (String a : adds) {
       assertU(a, a);
