diff --git a/solr/src/java/org/apache/solr/core/AbstractSolrEventListener.java b/solr/src/java/org/apache/solr/core/AbstractSolrEventListener.java
index bf593b3..0ff4521 100644
--- a/solr/src/java/org/apache/solr/core/AbstractSolrEventListener.java
+++ b/solr/src/java/org/apache/solr/core/AbstractSolrEventListener.java
@@ -23,7 +23,7 @@ import org.apache.solr.search.SolrIndexSearcher;
 
 /**
  */
-class AbstractSolrEventListener implements SolrEventListener {
+public class AbstractSolrEventListener implements SolrEventListener {
   protected final SolrCore core;
   public AbstractSolrEventListener(SolrCore core) {
     this.core = core;
