diff --git a/lucene/src/test/org/apache/lucene/index/TestRollingUpdates.java b/lucene/src/test/org/apache/lucene/index/TestRollingUpdates.java
index c38fd2d..a50dee8 100644
--- a/lucene/src/test/org/apache/lucene/index/TestRollingUpdates.java
+++ b/lucene/src/test/org/apache/lucene/index/TestRollingUpdates.java
@@ -19,6 +19,7 @@ package org.apache.lucene.index;
 
 import org.apache.lucene.analysis.MockAnalyzer;
 import org.apache.lucene.document.*;
+import org.apache.lucene.search.TermQuery;
 import org.apache.lucene.store.*;
 import org.apache.lucene.util.*;
 import org.junit.Test;
@@ -48,7 +49,21 @@ public class TestRollingUpdates extends LuceneTestCase {
         id++;
       }
       doc.getField("id").setValue(myID);
-      w.updateDocument(new Term("id", myID), doc);
+      int mode = docIter % 3;
+      switch (mode) {
+        case 0: {
+          w.deleteDocuments(new Term("id", myID));
+          w.addDocument(doc);
+          break;
+        }
+        case 1: {
+          w.deleteDocuments(new TermQuery(new Term("id", myID)));
+          w.addDocument(doc);
+          break;
+        }
+        default : w.updateDocument(new Term("id", myID), doc);
+      }
+
 
       if (docIter >= SIZE && random.nextInt(50) == 17) {
         if (r != null) {
