diff --git a/modules/analysis/common/src/test/org/apache/lucene/analysis/pattern/TestPatternReplaceFilter.java b/modules/analysis/common/src/test/org/apache/lucene/analysis/pattern/TestPatternReplaceFilter.java
index 12e0982..dd0e10c 100644
--- a/modules/analysis/common/src/test/org/apache/lucene/analysis/pattern/TestPatternReplaceFilter.java
+++ b/modules/analysis/common/src/test/org/apache/lucene/analysis/pattern/TestPatternReplaceFilter.java
@@ -25,7 +25,6 @@ import java.io.StringReader;
 import java.util.regex.Pattern;
 
 /**
- * @version $Id:$
  */
 public class TestPatternReplaceFilter extends BaseTokenStreamTestCase {
