diff --git a/solr/src/test/org/apache/solr/core/AlternateDirectoryTest.java b/solr/src/test/org/apache/solr/core/AlternateDirectoryTest.java
index d5a4616..93ca71c 100755
--- a/solr/src/test/org/apache/solr/core/AlternateDirectoryTest.java
+++ b/solr/src/test/org/apache/solr/core/AlternateDirectoryTest.java
@@ -18,6 +18,9 @@ package org.apache.solr.core;
 
 import java.io.File;
 import java.io.IOException;
+
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.store.Directory;
 import org.apache.lucene.store.FSDirectory;
 import org.apache.solr.SolrTestCaseJ4;
 import org.junit.BeforeClass;
@@ -38,6 +41,7 @@ public class AlternateDirectoryTest extends SolrTestCaseJ4 {
   public void testAltDirectoryUsed() throws Exception {
     assertQ(req("q","*:*","qt","standard"));
     assertTrue(TestFSDirectoryFactory.openCalled);
+    assertTrue(TestIndexReaderFactory.newReaderCalled);
   }
 
   static public class TestFSDirectoryFactory extends DirectoryFactory {
@@ -50,4 +54,15 @@ public class AlternateDirectoryTest extends SolrTestCaseJ4 {
 
   }
 
+
+  static public class TestIndexReaderFactory extends IndexReaderFactory {
+    static volatile boolean newReaderCalled = false;
+
+    public IndexReader newReader(Directory indexDir, boolean readOnly)
+        throws IOException {
+      TestIndexReaderFactory.newReaderCalled = true;
+      return IndexReader.open(indexDir, readOnly);
+    }
+  }
+
 }
