diff --git a/solr/src/test/org/apache/solr/core/TestJmxIntegration.java b/solr/src/test/org/apache/solr/core/TestJmxIntegration.java
index 6a91e59..46f42d1 100644
--- a/solr/src/test/org/apache/solr/core/TestJmxIntegration.java
+++ b/solr/src/test/org/apache/solr/core/TestJmxIntegration.java
@@ -95,7 +95,6 @@ public class TestJmxIntegration extends AbstractSolrTestCase {
     }
     if (bean==null) throw new RuntimeException("searcher was never registered");
     ObjectName searcher = getObjectName("searcher", bean);
-
     MBeanServer mbeanServer = servers.get(0);
     log.info("Mbeans in server: " + mbeanServer.queryNames(null, null));
 
@@ -114,7 +113,14 @@ public class TestJmxIntegration extends AbstractSolrTestCase {
     Hashtable<String, String> map = new Hashtable<String, String>();
     map.put("type", key);
     map.put("id", infoBean.getName());
-    return ObjectName.getInstance("solr", map);
+    String coreName = h.getCore().getName();
+    if (coreName.equals("")) {
+      String defaultCoreName = h.getCore().getCoreDescriptor().getCoreContainer().getDefaultCoreName();
+      if (!defaultCoreName.equals("")) {
+        coreName = defaultCoreName;
+      }
+    }
+    return ObjectName.getInstance(("solr" + (null != coreName ? "/" + coreName : "")), map);
   }
 }
