diff --git a/solr/src/test/org/apache/solr/TestTrie.java b/solr/src/test/org/apache/solr/TestTrie.java
index 861c349..4637e03 100644
--- a/solr/src/test/org/apache/solr/TestTrie.java
+++ b/solr/src/test/org/apache/solr/TestTrie.java
@@ -32,7 +32,7 @@ import java.util.TimeZone;
 /**
  * Tests for TrieField functionality
  *
- * @version $Id$
+ *
  * @since solr 1.4
  */
 public class TestTrie extends SolrTestCaseJ4 {
