diff --git a/solr/src/test/org/apache/solr/client/solrj/response/FieldAnalysisResponseTest.java b/solr/src/test/org/apache/solr/client/solrj/response/FieldAnalysisResponseTest.java
index f6f62aa..12f6100 100644
--- a/solr/src/test/org/apache/solr/client/solrj/response/FieldAnalysisResponseTest.java
+++ b/solr/src/test/org/apache/solr/client/solrj/response/FieldAnalysisResponseTest.java
@@ -28,7 +28,7 @@ import java.util.List;
 /**
  * A test case for the {@link FieldAnalysisResponse} class.
  *
- * @version $Id$
+ *
  * @since solr 1.4
  */
 @SuppressWarnings("unchecked")
