diff --git a/lucene/src/test/org/apache/lucene/index/TestIndexWriterOnJRECrash.java b/lucene/src/test/org/apache/lucene/index/TestIndexWriterOnJRECrash.java
index 4ad1a45..c0a8294 100644
--- a/lucene/src/test/org/apache/lucene/index/TestIndexWriterOnJRECrash.java
+++ b/lucene/src/test/org/apache/lucene/index/TestIndexWriterOnJRECrash.java
@@ -50,7 +50,7 @@ public class TestIndexWriterOnJRECrash extends TestNRTThreads {
   public void testNRTThreads() throws Exception {
     String vendor = Constants.JAVA_VENDOR;
     assumeTrue(vendor + " JRE not supported.",
-        vendor.startsWith("Sun") || vendor.startsWith("IBM") || vendor.startsWith("Apple"));
+        vendor.startsWith("Sun") || vendor.startsWith("Apple"));
 
     // if we are not the fork
     if (System.getProperty("tests.crashmode") == null) {
