diff --git a/solr/src/java/org/apache/solr/update/RollbackUpdateCommand.java b/solr/src/java/org/apache/solr/update/RollbackUpdateCommand.java
index 1dd12ae..1308c53 100644
--- a/solr/src/java/org/apache/solr/update/RollbackUpdateCommand.java
+++ b/solr/src/java/org/apache/solr/update/RollbackUpdateCommand.java
@@ -20,7 +20,7 @@ package org.apache.solr.update;
 import org.apache.solr.request.SolrQueryRequest;
 
 /**
- * @version $Id$
+ *
  * @since Solr 1.4
  */
 public class RollbackUpdateCommand extends UpdateCommand {
