diff --git a/lucene/src/test/org/apache/lucene/document/TestDateTools.java b/lucene/src/test/org/apache/lucene/document/TestDateTools.java
index 02c090f..b0f7c82 100644
--- a/lucene/src/test/org/apache/lucene/document/TestDateTools.java
+++ b/lucene/src/test/org/apache/lucene/document/TestDateTools.java
@@ -9,7 +9,6 @@ import java.util.TimeZone;
 import java.util.Locale;
 
 import org.apache.lucene.util.LuceneTestCase;
-import org.junit.runner.RunWith;
 
 /**
  * Licensed to the Apache Software Foundation (ASF) under one or more
@@ -27,7 +26,6 @@ import org.junit.runner.RunWith;
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-@RunWith(LuceneTestCase.LocalizedTestCaseRunner.class)
 public class TestDateTools extends LuceneTestCase {
 
   public void testStringToDate() throws ParseException {
