diff --git a/solr/src/test/org/apache/solr/search/TestSearchPerf.java b/solr/src/test/org/apache/solr/search/TestSearchPerf.java
index 57e0d86..36afad0 100755
--- a/solr/src/test/org/apache/solr/search/TestSearchPerf.java
+++ b/solr/src/test/org/apache/solr/search/TestSearchPerf.java
@@ -31,7 +31,7 @@ import java.util.*;
 import java.io.IOException;
 
 /**
- * @version $Id$
+ *
  */
 public class TestSearchPerf extends AbstractSolrTestCase {
