diff --git a/lucene/src/java/org/apache/lucene/search/BooleanClause.java b/lucene/src/java/org/apache/lucene/search/BooleanClause.java
index 6212a37..91af8e6 100644
--- a/lucene/src/java/org/apache/lucene/search/BooleanClause.java
+++ b/lucene/src/java/org/apache/lucene/search/BooleanClause.java
@@ -18,7 +18,7 @@ package org.apache.lucene.search;
  */
 
 /** A clause in a BooleanQuery. */
-public class BooleanClause implements java.io.Serializable {
+public class BooleanClause {
   
   /** Specifies how clauses are to occur in matching documents. */
   public static enum Occur {
