diff --git a/solr/src/java/org/apache/solr/search/DocSet.java b/solr/src/java/org/apache/solr/search/DocSet.java
index fcf8785..dd69c97 100644
--- a/solr/src/java/org/apache/solr/search/DocSet.java
+++ b/solr/src/java/org/apache/solr/search/DocSet.java
@@ -35,7 +35,7 @@ import java.io.IOException;
  * a cache and could be shared.
  * </p>
  *
- * @version $Id$
+ *
  * @since solr 0.9
  */
 public interface DocSet /* extends Collection<Integer> */ {
