diff --git a/solr/src/common/org/apache/solr/common/util/StrUtils.java b/solr/src/common/org/apache/solr/common/util/StrUtils.java
index fc2187a..931ef5f 100644
--- a/solr/src/common/org/apache/solr/common/util/StrUtils.java
+++ b/solr/src/common/org/apache/solr/common/util/StrUtils.java
@@ -26,7 +26,7 @@ import java.io.IOException;
 import org.apache.solr.common.SolrException;
 
 /**
- * @version $Id$
+ *
  */
 public class StrUtils {
   public static final char[] HEX_DIGITS = { '0', '1', '2', '3', '4', '5', '6',
