diff --git a/solr/src/solrj/org/apache/solr/client/solrj/impl/BinaryResponseParser.java b/solr/src/solrj/org/apache/solr/client/solrj/impl/BinaryResponseParser.java
index 1379460..7ac9294 100755
--- a/solr/src/solrj/org/apache/solr/client/solrj/impl/BinaryResponseParser.java
+++ b/solr/src/solrj/org/apache/solr/client/solrj/impl/BinaryResponseParser.java
@@ -26,7 +26,7 @@ import java.io.InputStream;
 import java.io.Reader;
 
 /**
- * @version $Id$
+ *
  * @since solr 1.3
  */
 public class BinaryResponseParser extends ResponseParser {
