diff --git a/lucene/src/java/org/apache/lucene/index/DocFieldProcessor.java b/lucene/src/java/org/apache/lucene/index/DocFieldProcessor.java
index 6416f51..3a85224 100644
--- a/lucene/src/java/org/apache/lucene/index/DocFieldProcessor.java
+++ b/lucene/src/java/org/apache/lucene/index/DocFieldProcessor.java
@@ -59,6 +59,13 @@ final class DocFieldProcessor extends DocConsumer {
     // FreqProxTermsWriter does this with
     // FieldInfo.storePayload.
     final String fileName = IndexFileNames.segmentFileName(state.segmentName, "", IndexFileNames.FIELD_INFOS_EXTENSION);
+
+    // If this segment only has docs that hit non-aborting exceptions,
+    // then no term vectors files will have been written; therefore we
+    // need to update the fieldInfos and clear the term vectors bits
+    if (!state.hasVectors) {
+      state.fieldInfos.clearVectors();
+    }
     state.fieldInfos.write(state.directory, fileName);
   }
