diff --git a/solr/src/java/org/apache/solr/schema/SchemaField.java b/solr/src/java/org/apache/solr/schema/SchemaField.java
index 8934c68..9d625bd 100644
--- a/solr/src/java/org/apache/solr/schema/SchemaField.java
+++ b/solr/src/java/org/apache/solr/schema/SchemaField.java
@@ -150,7 +150,6 @@ public final class SchemaField extends FieldProperties {
                               "can not sort on multivalued field: " 
                               + getName());
     }
-    
   }
 
   /**
