diff --git a/modules/analysis/common/src/java/org/tartarus/snowball/TestApp.java b/modules/analysis/common/src/java/org/tartarus/snowball/TestApp.java
index 855dfad..3da6f8c 100644
--- a/modules/analysis/common/src/java/org/tartarus/snowball/TestApp.java
+++ b/modules/analysis/common/src/java/org/tartarus/snowball/TestApp.java
@@ -54,9 +54,9 @@ public class TestApp {
             return;
         }
 
-	Class stemClass = Class.forName("org.tartarus.snowball.ext." +
-					args[0] + "Stemmer");
-        SnowballProgram stemmer = (SnowballProgram) stemClass.newInstance();
+	Class<? extends SnowballProgram> stemClass = Class.forName("org.tartarus.snowball.ext." +
+					args[0] + "Stemmer").asSubclass(SnowballProgram.class);
+        SnowballProgram stemmer = stemClass.newInstance();
 	Method stemMethod = stemClass.getMethod("stem", new Class[0]);
 
 	Reader reader;
