diff --git a/modules/analysis/common/src/test/org/apache/lucene/analysis/ru/TestRussianLetterTokenizer.java b/modules/analysis/common/src/test/org/apache/lucene/analysis/ru/TestRussianLetterTokenizer.java
index ce3ff5d..2fedf78 100644
--- a/modules/analysis/common/src/test/org/apache/lucene/analysis/ru/TestRussianLetterTokenizer.java
+++ b/modules/analysis/common/src/test/org/apache/lucene/analysis/ru/TestRussianLetterTokenizer.java
@@ -25,7 +25,7 @@ import org.apache.lucene.util.Version;
 
 /**
  * Testcase for {@link RussianLetterTokenizer}
- * @deprecated Remove this test class in Lucene 4.0
+ * @deprecated (3.1) Remove this test class in Lucene 5.0
  */
 @Deprecated
 public class TestRussianLetterTokenizer extends BaseTokenStreamTestCase {
