diff --git a/solr/src/java/org/apache/solr/analysis/GalicianStemFilterFactory.java b/solr/src/java/org/apache/solr/analysis/GalicianStemFilterFactory.java
index bb32d50..e6cf771 100644
--- a/solr/src/java/org/apache/solr/analysis/GalicianStemFilterFactory.java
+++ b/solr/src/java/org/apache/solr/analysis/GalicianStemFilterFactory.java
@@ -30,7 +30,7 @@ import org.apache.lucene.analysis.gl.GalicianStemFilter;
  *     &lt;filter class="solr.GalicianStemFilterFactory"/&gt;
  *   &lt;/analyzer&gt;
  * &lt;/fieldType&gt;</pre> 
- * @version $Id$
+ *
  */
 public class GalicianStemFilterFactory extends BaseTokenFilterFactory {
   public TokenStream create(TokenStream input) {
