diff --git a/lucene/src/java/org/apache/lucene/search/cache/EntryKey.java b/lucene/src/java/org/apache/lucene/search/cache/EntryKey.java
new file mode 100644
index 0000000..fb8d67e
--- /dev/null
+++ b/lucene/src/java/org/apache/lucene/search/cache/EntryKey.java
@@ -0,0 +1,9 @@
+package org.apache.lucene.search.cache;
+
+
+/**
+ * A Simple marker class -- Perhaps it could/should just be an Object
+ */
+public abstract class EntryKey {
+
+}
