diff --git a/solr/src/java/org/apache/solr/search/function/NumericFieldCacheSource.java b/solr/src/java/org/apache/solr/search/function/NumericFieldCacheSource.java
index 710a009..f7fde5c 100644
--- a/solr/src/java/org/apache/solr/search/function/NumericFieldCacheSource.java
+++ b/solr/src/java/org/apache/solr/search/function/NumericFieldCacheSource.java
@@ -22,7 +22,7 @@ import org.apache.lucene.search.cache.CachedArrayCreator;
 
 /**
  * 
- * @version $Id$
+ *
  */
 public abstract class NumericFieldCacheSource<T extends CachedArray> extends FieldCacheSource {
   protected final CachedArrayCreator<T> creator;
