diff --git a/lucene/contrib/queryparser/src/java/org/apache/lucene/queryParser/standard/nodes/RegexpQueryNode.java b/lucene/contrib/queryparser/src/java/org/apache/lucene/queryParser/standard/nodes/RegexpQueryNode.java
index 7e4f989..be8659d 100644
--- a/lucene/contrib/queryparser/src/java/org/apache/lucene/queryParser/standard/nodes/RegexpQueryNode.java
+++ b/lucene/contrib/queryparser/src/java/org/apache/lucene/queryParser/standard/nodes/RegexpQueryNode.java
@@ -29,7 +29,6 @@ import org.apache.lucene.util.BytesRef;
  */
 public class RegexpQueryNode extends QueryNodeImpl  implements TextableQueryNode,
 FieldableNode {
-  private static final long serialVersionUID = 0L;
   private CharSequence text;
   private CharSequence field;
   /**
