diff --git a/lucene/src/java/org/apache/lucene/index/Healthiness.java b/lucene/src/java/org/apache/lucene/index/Healthiness.java
index c302dbb..dcb9868 100644
--- a/lucene/src/java/org/apache/lucene/index/Healthiness.java
+++ b/lucene/src/java/org/apache/lucene/index/Healthiness.java
@@ -35,6 +35,7 @@ import org.apache.lucene.index.DocumentsWriterPerThreadPool.ThreadState;
  * than the number of active {@link ThreadState}s threads are released and can
  * continue indexing.
  */
+//TODO: rename this to DocumentsWriterStallControl (or something like that)?
 final class Healthiness {
 
   @SuppressWarnings("serial")
