diff --git a/solr/src/java/org/apache/solr/request/ServletSolrParams.java b/solr/src/java/org/apache/solr/request/ServletSolrParams.java
index 49922c8..18dc755 100644
--- a/solr/src/java/org/apache/solr/request/ServletSolrParams.java
+++ b/solr/src/java/org/apache/solr/request/ServletSolrParams.java
@@ -22,7 +22,7 @@ import javax.servlet.ServletRequest;
 import org.apache.solr.common.params.MultiMapSolrParams;
 
 /**
- * @version $Id$
+ *
  */
 public class ServletSolrParams extends MultiMapSolrParams {
   public ServletSolrParams(ServletRequest req) {
