diff --git a/modules/analysis/common/src/test/org/apache/lucene/analysis/core/TestUAX29Tokenizer.java b/modules/analysis/common/src/test/org/apache/lucene/analysis/core/TestUAX29Tokenizer.java
index f02f4e2..060c6dd 100644
--- a/modules/analysis/common/src/test/org/apache/lucene/analysis/core/TestUAX29Tokenizer.java
+++ b/modules/analysis/common/src/test/org/apache/lucene/analysis/core/TestUAX29Tokenizer.java
@@ -198,7 +198,7 @@ public class TestUAX29Tokenizer extends BaseTokenStreamTestCase {
   }
   
   public void testUnicodeWordBreaks() throws Exception {
-    WordBreakTestUnicode_5_2_0 wordBreakTest = new WordBreakTestUnicode_5_2_0();
+    WordBreakTestUnicode_6_0_0 wordBreakTest = new WordBreakTestUnicode_6_0_0();
     wordBreakTest.test(a);
   }
 }
