diff --git a/lucene/src/java/org/apache/lucene/search/SpanFilter.java b/lucene/src/java/org/apache/lucene/search/SpanFilter.java
index e2a5946..e46ff1e 100644
--- a/lucene/src/java/org/apache/lucene/search/SpanFilter.java
+++ b/lucene/src/java/org/apache/lucene/search/SpanFilter.java
@@ -15,7 +15,7 @@ package org.apache.lucene.search;
  * limitations under the License.
  */
 
-import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.index.IndexReader.AtomicReaderContext;
 
 import java.io.IOException;
 
@@ -30,9 +30,9 @@ import java.io.IOException;
 public abstract class SpanFilter extends Filter{
   /** Returns a SpanFilterResult with true for documents which should be permitted in
     search results, and false for those that should not and Spans for where the true docs match.
-   * @param reader The {@link org.apache.lucene.index.IndexReader} to load position and DocIdSet information from
+   * @param context The {@link AtomicReaderContext} to load position and DocIdSet information from
    * @return A {@link SpanFilterResult}
    * @throws java.io.IOException if there was an issue accessing the necessary information
    * */
-  public abstract SpanFilterResult bitSpans(IndexReader reader) throws IOException;
+  public abstract SpanFilterResult bitSpans(AtomicReaderContext context) throws IOException;
 }
