diff --git a/solr/src/common/org/apache/solr/common/params/MultiMapSolrParams.java b/solr/src/common/org/apache/solr/common/params/MultiMapSolrParams.java
index 781c1da..68202d5 100644
--- a/solr/src/common/org/apache/solr/common/params/MultiMapSolrParams.java
+++ b/solr/src/common/org/apache/solr/common/params/MultiMapSolrParams.java
@@ -24,7 +24,7 @@ import java.util.Map;
 import java.io.IOException;
 
 /**
- * @version $Id$
+ *
  */
 public class MultiMapSolrParams extends SolrParams {
   protected final Map<String,String[]> map;
