diff --git a/solr/src/test/org/apache/solr/client/solrj/response/FieldAnalysisResponseTest.java b/solr/src/test/org/apache/solr/client/solrj/response/FieldAnalysisResponseTest.java
index 65a42d0..73b506d 100644
--- a/solr/src/test/org/apache/solr/client/solrj/response/FieldAnalysisResponseTest.java
+++ b/solr/src/test/org/apache/solr/client/solrj/response/FieldAnalysisResponseTest.java
@@ -19,7 +19,6 @@ package org.apache.solr.client.solrj.response;
 
 import org.apache.lucene.util.LuceneTestCase;
 import org.apache.solr.common.util.NamedList;
-import static org.junit.Assert.*;
 import org.junit.Test;
 
 import java.util.ArrayList;
