diff --git a/solr/src/java/org/apache/solr/analysis/RussianLightStemFilterFactory.java b/solr/src/java/org/apache/solr/analysis/RussianLightStemFilterFactory.java
index e330759..6a11a3c 100644
--- a/solr/src/java/org/apache/solr/analysis/RussianLightStemFilterFactory.java
+++ b/solr/src/java/org/apache/solr/analysis/RussianLightStemFilterFactory.java
@@ -30,7 +30,7 @@ import org.apache.lucene.analysis.ru.RussianLightStemFilter;
  *     &lt;filter class="solr.RussianLightStemFilterFactory"/&gt;
  *   &lt;/analyzer&gt;
  * &lt;/fieldType&gt;</pre>
- * @version $Id$
+ *
  */
 public class RussianLightStemFilterFactory extends BaseTokenFilterFactory {
   public TokenStream create(TokenStream input) {
