diff --git a/solr/src/test/org/apache/solr/core/SolrCoreTest.java b/solr/src/test/org/apache/solr/core/SolrCoreTest.java
index 481d8c2..b979ede 100755
--- a/solr/src/test/org/apache/solr/core/SolrCoreTest.java
+++ b/solr/src/test/org/apache/solr/core/SolrCoreTest.java
@@ -25,7 +25,6 @@ import org.apache.solr.request.SolrQueryRequest;
 import org.apache.solr.request.SolrRequestHandler;
 import org.apache.solr.response.SolrQueryResponse;
 import org.apache.solr.util.plugin.SolrCoreAware;
-import org.junit.BeforeClass;
 import org.junit.Test;
 
 import java.util.concurrent.*;
