diff --git a/lucene/src/test-framework/org/apache/lucene/index/RandomIndexWriter.java b/lucene/src/test-framework/org/apache/lucene/index/RandomIndexWriter.java
index ea1f891..9362e15 100644
--- a/lucene/src/test-framework/org/apache/lucene/index/RandomIndexWriter.java
+++ b/lucene/src/test-framework/org/apache/lucene/index/RandomIndexWriter.java
@@ -65,6 +65,14 @@ public class RandomIndexWriter implements Closeable {
       // called from different threads; else test failures may
       // not be reproducible from the original seed
       this.r = new Random(r.nextInt());
+
+      if (r.nextBoolean()) {
+        if (LuceneTestCase.VERBOSE) {
+          System.out.println("NOTE: RIW ctor is setting new CodecProvider().copyFrom");
+        }
+        // Just to test CP.copyFrom:
+        conf.setCodecProvider(new CodecProvider().copyFrom(conf.getCodecProvider()));
+      }
     }
 
     @Override
