diff --git a/solr/src/test/org/apache/solr/update/DirectUpdateHandlerOptimizeTest.java b/solr/src/test/org/apache/solr/update/DirectUpdateHandlerOptimizeTest.java
index 647c100..768ca6c 100644
--- a/solr/src/test/org/apache/solr/update/DirectUpdateHandlerOptimizeTest.java
+++ b/solr/src/test/org/apache/solr/update/DirectUpdateHandlerOptimizeTest.java
@@ -36,7 +36,8 @@ public class DirectUpdateHandlerOptimizeTest extends AbstractSolrTestCase {
   }
 
   public String getSolrConfigFile() {
-    return "solrconfig-duh-optimize.xml";
+    // return "solrconfig-duh-optimize.xml";
+    return "solrconfig.xml";
   }
