diff --git a/lucene/src/java/org/apache/lucene/index/BufferedDeletes.java b/lucene/src/java/org/apache/lucene/index/BufferedDeletes.java
index 936d9fd..6b9ebcf 100644
--- a/lucene/src/java/org/apache/lucene/index/BufferedDeletes.java
+++ b/lucene/src/java/org/apache/lucene/index/BufferedDeletes.java
@@ -372,7 +372,7 @@ class BufferedDeletes {
           Query query = entry.getKey();
           int limit = entry.getValue().intValue();
           Weight weight = query.weight(searcher);
-          Scorer scorer = weight.scorer(readerContext, true, false);
+          Scorer scorer = weight.scorer(readerContext, Weight.ScorerContext.def());
           if (scorer != null) {
             while(true)  {
               int doc = scorer.nextDoc();
