diff --git a/solr/contrib/dataimporthandler/src/test/java/org/apache/solr/handler/dataimport/TestRegexTransformer.java b/solr/contrib/dataimporthandler/src/test/java/org/apache/solr/handler/dataimport/TestRegexTransformer.java
index ca3143b..50835d7 100644
--- a/solr/contrib/dataimporthandler/src/test/java/org/apache/solr/handler/dataimport/TestRegexTransformer.java
+++ b/solr/contrib/dataimporthandler/src/test/java/org/apache/solr/handler/dataimport/TestRegexTransformer.java
@@ -31,7 +31,7 @@ import java.util.Map;
 /**
  * <p> Test for RegexTransformer </p>
  *
- * @version $Id$
+ *
  * @since solr 1.3
  */
 public class TestRegexTransformer extends AbstractDataImportHandlerTestCase {
