diff --git a/solr/src/java/org/apache/solr/search/SolrConstantScoreQuery.java b/solr/src/java/org/apache/solr/search/SolrConstantScoreQuery.java
index 357ee66..69bf4b4 100755
--- a/solr/src/java/org/apache/solr/search/SolrConstantScoreQuery.java
+++ b/solr/src/java/org/apache/solr/search/SolrConstantScoreQuery.java
@@ -34,7 +34,9 @@ import java.util.Map;
  *
  * Experimental and subject to change.
  */
-public class SolrConstantScoreQuery extends ConstantScoreQuery {
+public class SolrConstantScoreQuery extends ConstantScoreQuery implements ExtendedQuery {
+  boolean cache = true;  // cache by default
+  int cost;
 
   public SolrConstantScoreQuery(Filter filter) {
     super(filter);
@@ -47,6 +49,36 @@ public class SolrConstantScoreQuery extends ConstantScoreQuery {
   }
 
   @Override
+  public void setCache(boolean cache) {
+    this.cache = cache;
+  }
+
+  @Override
+  public boolean getCache() {
+    return cache;
+  }
+
+  @Override
+  public void setCacheSep(boolean cacheSep) {
+  }
+
+  @Override
+  public boolean getCacheSep() {
+    return false;
+  }
+
+  @Override
+  public void setCost(int cost) {
+    this.cost = cost;
+  }
+
+  @Override
+  public int getCost() {
+    return cost;
+  }
+
+
+  @Override
   public Query rewrite(IndexReader reader) throws IOException {
     return this;
   }
