diff --git a/solr/src/java/org/apache/solr/handler/RequestHandlerUtils.java b/solr/src/java/org/apache/solr/handler/RequestHandlerUtils.java
index d7d8576..3d23322 100755
--- a/solr/src/java/org/apache/solr/handler/RequestHandlerUtils.java
+++ b/solr/src/java/org/apache/solr/handler/RequestHandlerUtils.java
@@ -32,7 +32,7 @@ import org.apache.solr.update.processor.UpdateRequestProcessor;
 /**
  * Common helper functions for RequestHandlers
  * 
- * @version $Id$
+ *
  * @since solr 1.2
  */
 public class RequestHandlerUtils
