diff --git a/solr/src/test/org/apache/solr/spelling/suggest/SuggesterTSTTest.java b/solr/src/test/org/apache/solr/spelling/suggest/SuggesterTSTTest.java
new file mode 100644
index 0000000..1fb336e
--- /dev/null
+++ b/solr/src/test/org/apache/solr/spelling/suggest/SuggesterTSTTest.java
@@ -0,0 +1,7 @@
+package org.apache.solr.spelling.suggest;
+
+public class SuggesterTSTTest extends SuggesterTest {
+  public SuggesterTSTTest() {
+    super.requestUri = "/suggest_tst";
+  }
+}
