diff --git a/solr/src/java/org/apache/solr/search/LuceneQParserPlugin.java b/solr/src/java/org/apache/solr/search/LuceneQParserPlugin.java
index 1c2e8da..0c8d1a1 100755
--- a/solr/src/java/org/apache/solr/search/LuceneQParserPlugin.java
+++ b/solr/src/java/org/apache/solr/search/LuceneQParserPlugin.java
@@ -117,7 +117,7 @@ class OldLuceneQParser extends LuceneQParser {
   public SortSpec getSort(boolean useGlobal) throws ParseException {
     SortSpec sort = super.getSort(useGlobal);
     if (sortStr != null && sortStr.length()>0 && sort.getSort()==null) {
-      Sort oldSort = QueryParsing.parseSort(sortStr, getReq().getSchema());
+      Sort oldSort = QueryParsing.parseSort(sortStr, getReq());
       if( oldSort != null ) {
         sort.sort = oldSort;
       }
