diff --git a/lucene/src/java/org/apache/lucene/util/fst/Builder.java b/lucene/src/java/org/apache/lucene/util/fst/Builder.java
index 99378cf..84245a6 100644
--- a/lucene/src/java/org/apache/lucene/util/fst/Builder.java
+++ b/lucene/src/java/org/apache/lucene/util/fst/Builder.java
@@ -21,7 +21,7 @@ import org.apache.lucene.util.ArrayUtil;
 import org.apache.lucene.util.RamUsageEstimator;
 import org.apache.lucene.util.BytesRef;
 import org.apache.lucene.util.IntsRef;
-import org.apache.lucene.util.fst.FST.INPUT_TYPE;
+import org.apache.lucene.util.fst.FST.INPUT_TYPE; // javadoc
 
 import java.io.IOException;
 
@@ -136,6 +136,13 @@ public class Builder<T> {
     return dedupHash == null ? 0 : fst.nodeCount;
   }
 
+  /** Pass false to disable the array arc optimization
+   *  while building the FST.  This is necessary if
+   *  encoding a single arc may take more than 255 bytes. */
+  public void setAllowArrayArcs(boolean b) {
+    fst.setAllowArrayArcs(b);
+  }
+
   private CompiledNode compileNode(UnCompiledNode<T> n) throws IOException {
 
     final int address;
