diff --git a/solr/src/java/org/apache/solr/analysis/EnglishMinimalStemFilterFactory.java b/solr/src/java/org/apache/solr/analysis/EnglishMinimalStemFilterFactory.java
index 5d2292e..98cdede 100644
--- a/solr/src/java/org/apache/solr/analysis/EnglishMinimalStemFilterFactory.java
+++ b/solr/src/java/org/apache/solr/analysis/EnglishMinimalStemFilterFactory.java
@@ -30,7 +30,7 @@ import org.apache.lucene.analysis.en.EnglishMinimalStemFilter;
  *     &lt;filter class="solr.EnglishMinimalStemFilterFactory"/&gt;
  *   &lt;/analyzer&gt;
  * &lt;/fieldType&gt;</pre>
- * @version $Id$
+ *
  */
 public class EnglishMinimalStemFilterFactory extends BaseTokenFilterFactory {
   public TokenStream create(TokenStream input) {
