diff --git a/solr/src/java/org/apache/solr/schema/SortableIntField.java b/solr/src/java/org/apache/solr/schema/SortableIntField.java
index b6db1cf..421e4bc 100644
--- a/solr/src/java/org/apache/solr/schema/SortableIntField.java
+++ b/solr/src/java/org/apache/solr/schema/SortableIntField.java
@@ -37,6 +37,8 @@ import java.util.Map;
 import java.io.IOException;
 /**
  * @version $Id$
+ * 
+ * @deprecated use {@link IntField} or {@link TrieIntField} - will be removed in 5.x
  */
 public class SortableIntField extends FieldType {
   protected void init(IndexSchema schema, Map<String,String> args) {
