diff --git a/hw/dmx/dmxstat.c b/hw/dmx/dmxstat.c
index ced4c8b..ae3f404 100644
--- a/hw/dmx/dmxstat.c
+++ b/hw/dmx/dmxstat.c
@@ -46,7 +46,7 @@
 #include "dmx.h"
 #include "dmxstat.h"
 #include "dmxlog.h"
-#include "Xos.h"                /* For sys/time.h */
+#include <X11/Xos.h>                /* For sys/time.h */
 
 /** Used to compute a running average of value. */
 typedef struct _DMXStatAvg {
