diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/state/ItemStateMap.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/state/ItemStateMap.java
index 051d342..8403437 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/state/ItemStateMap.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/state/ItemStateMap.java
@@ -16,7 +16,7 @@
  */
 package org.apache.jackrabbit.core.state;
 
-import org.apache.jackrabbit.core.ItemId;
+import org.apache.jackrabbit.core.id.ItemId;
 import org.apache.jackrabbit.core.util.Dumpable;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
