diff --git a/jackrabbit-jcr-server/src/main/java/org/apache/jackrabbit/webdav/simple/DavResourceImpl.java b/jackrabbit-jcr-server/src/main/java/org/apache/jackrabbit/webdav/simple/DavResourceImpl.java
index 1e9056c..7082e13 100644
--- a/jackrabbit-jcr-server/src/main/java/org/apache/jackrabbit/webdav/simple/DavResourceImpl.java
+++ b/jackrabbit-jcr-server/src/main/java/org/apache/jackrabbit/webdav/simple/DavResourceImpl.java
@@ -151,7 +151,7 @@ public class DavResourceImpl implements DavResource, JcrConstants {
         this(locator, factory, session, config, null);
         this.isCollection = isCollection;
     }
-    
+
     /**
      * Create a new {@link DavResource}.
      *
