diff --git a/jackrabbit-ocm-nodemanagement/src/main/java/org/apache/jackrabbit/ocm/nodemanagement/exception/OperationNotSupportedException.java b/jackrabbit-ocm-nodemanagement/src/main/java/org/apache/jackrabbit/ocm/nodemanagement/exception/OperationNotSupportedException.java
index 62f4614..ee5e4fd 100644
--- a/jackrabbit-ocm-nodemanagement/src/main/java/org/apache/jackrabbit/ocm/nodemanagement/exception/OperationNotSupportedException.java
+++ b/jackrabbit-ocm-nodemanagement/src/main/java/org/apache/jackrabbit/ocm/nodemanagement/exception/OperationNotSupportedException.java
@@ -23,13 +23,13 @@ package org.apache.jackrabbit.ocm.nodemanagement.exception;
  */
 public class OperationNotSupportedException extends BaseNodeManagementException
 {
-    
+
     /** Creates a new instance of NodeTypeRemovalException. */
     public OperationNotSupportedException()
     {
     }
-    
-    /** Creates a new instance of NodeTypeRemovalException. 
+
+    /** Creates a new instance of NodeTypeRemovalException.
      * @param wrappedException Root exception
      */
     public OperationNotSupportedException(Exception wrappedException)
