diff --git a/jackrabbit-spi/src/main/java/org/apache/jackrabbit/spi/ChildInfo.java b/jackrabbit-spi/src/main/java/org/apache/jackrabbit/spi/ChildInfo.java
index b8173ea..5e40f5d 100644
--- a/jackrabbit-spi/src/main/java/org/apache/jackrabbit/spi/ChildInfo.java
+++ b/jackrabbit-spi/src/main/java/org/apache/jackrabbit/spi/ChildInfo.java
@@ -16,8 +16,6 @@
  */
 package org.apache.jackrabbit.spi;
 
-import org.apache.jackrabbit.name.QName;
-
 /**
  * <code>ChildInfo</code>...
  */
@@ -28,7 +26,7 @@ public interface ChildInfo {
      *
      * @return The name of the child <code>Node</code>.
      */
-    public QName getName();
+    public Name getName();
 
     /**
      * Returns the uniqueID of the child <code>Node</code> or <code>null</code>
