diff --git a/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/cluster/ClusterRecordTest.java b/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/cluster/ClusterRecordTest.java
index 4e1f8e5..0695b3c 100644
--- a/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/cluster/ClusterRecordTest.java
+++ b/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/cluster/ClusterRecordTest.java
@@ -107,7 +107,7 @@ public class ClusterRecordTest extends JUnitTest {
         UpdateEventChannel channel = master.createUpdateChannel(DEFAULT_WORKSPACE);
         channel.updateCreated(update);
         channel.updatePrepared(update);
-        channel.updateCommitted(update);
+        channel.updateCommitted(update, null);
 
         SimpleEventListener listener = new SimpleEventListener();
         slave.createUpdateChannel(DEFAULT_WORKSPACE).setListener(listener);
