diff --git a/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/util/LogUtil.java b/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/util/LogUtil.java
index cbdf9e9..8a8a39f 100644
--- a/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/util/LogUtil.java
+++ b/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/util/LogUtil.java
@@ -65,9 +65,8 @@ public class LogUtil {
         try {
             return safeGetJCRPath(itemState.getHierarchyEntry().getPath(), pathResolver);
         } catch (RepositoryException e) {
-            ItemId id = itemState.getId();
-            log.error("failed to convert " + id + " to JCR path.");
-            return id.toString();
+            log.error("failed to convert " + itemState.toString() + " to JCR path.");
+            return itemState.toString();
         }
     }
