diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/sql/QueryBuilder.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/sql/QueryBuilder.java
index 55dabf5..4c44644 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/sql/QueryBuilder.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/sql/QueryBuilder.java
@@ -19,7 +19,7 @@ package org.apache.jackrabbit.core.query.sql;
 import org.apache.jackrabbit.core.query.QueryTreeBuilder;
 import org.apache.jackrabbit.core.query.QueryRootNode;
 import org.apache.jackrabbit.core.query.QueryNodeFactory;
-import org.apache.jackrabbit.name.NamespaceResolver;
+import org.apache.jackrabbit.conversion.NameResolver;
 
 import javax.jcr.query.InvalidQueryException;
 import javax.jcr.query.Query;
@@ -33,7 +33,7 @@ public class QueryBuilder implements QueryTreeBuilder {
      * @inheritDoc
      */
     public QueryRootNode createQueryTree(String statement,
-                                         NamespaceResolver resolver,
+                                         NameResolver resolver,
                                          QueryNodeFactory factory)
             throws InvalidQueryException {
         return JCRSQLQueryBuilder.createQuery(statement, resolver, factory);
@@ -57,7 +57,7 @@ public class QueryBuilder implements QueryTreeBuilder {
     /**
      * @inheritDoc
      */
-    public String toString(QueryRootNode root, NamespaceResolver resolver)
+    public String toString(QueryRootNode root, NameResolver resolver)
             throws InvalidQueryException {
         return JCRSQLQueryBuilder.toString(root, resolver);
     }
