diff --git a/jackrabbit-core/src/test/java/org/apache/jackrabbit/api/jsr283/TestAll.java b/jackrabbit-core/src/test/java/org/apache/jackrabbit/api/jsr283/TestAll.java
index 92a579b..31ab45f 100644
--- a/jackrabbit-core/src/test/java/org/apache/jackrabbit/api/jsr283/TestAll.java
+++ b/jackrabbit-core/src/test/java/org/apache/jackrabbit/api/jsr283/TestAll.java
@@ -26,8 +26,6 @@ public class TestAll extends TestCase {
     public static Test suite() {
         TestSuite suite = new TestSuite("javax.jcr tests");
 
-        suite.addTestSuite(SessionRemoveItemTest.class);
-
         return suite;
     }
 }
\ No newline at end of file
