diff --git a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/name/PathResolver.java b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/name/PathResolver.java
index bdb8a16..3f2a1b7 100644
--- a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/name/PathResolver.java
+++ b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/name/PathResolver.java
@@ -21,7 +21,7 @@ import javax.jcr.NamespaceException;
 /**
  * Resolver for JCR paths.
  *
- * @deprecated Use the PathResolver interface from 
+ * @deprecated Use the PathResolver interface from
  *             the org.apache.jackrabbit.spi.commons.conversion package of
  *             the jackrabbit-spi-commons component.
  */
