diff --git a/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/TestAll.java b/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/TestAll.java
index 07f9e93..f503e10 100644
--- a/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/TestAll.java
+++ b/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/TestAll.java
@@ -39,6 +39,7 @@ public class TestAll extends TestCase {
         suite.addTestSuite(RestoreAndCheckoutTest.class);
         suite.addTestSuite(NodeImplTest.class);
         suite.addTestSuite(RetentionRegistryImplTest.class);
+        suite.addTestSuite(InvalidDateTest.class);
         suite.addTestSuite(SessionGarbageCollectedTest.class);
 
         return suite;
