diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/cluster/UpdateEventChannel.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/cluster/UpdateEventChannel.java
index a798e60..5d83c44 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/cluster/UpdateEventChannel.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/cluster/UpdateEventChannel.java
@@ -39,8 +39,9 @@ public interface UpdateEventChannel {
      * Called when an a update operation has been committed.
      *
      * @param update update operation
+     * @param path the change path
      */
-    void updateCommitted(Update update);
+    void updateCommitted(Update update, String path);
 
     /**
      * Called when an a update operation has been cancelled.
