diff --git a/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/ConcurrentVersioningWithTransactionsTest.java b/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/ConcurrentVersioningWithTransactionsTest.java
index 8a36245..753798a 100644
--- a/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/ConcurrentVersioningWithTransactionsTest.java
+++ b/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/ConcurrentVersioningWithTransactionsTest.java
@@ -37,13 +37,13 @@ public class ConcurrentVersioningWithTransactionsTest extends AbstractConcurrenc
     /**
      * The number of threads.
      */
-    private static final int CONCURRENCY = 100;
+    private static final int CONCURRENCY = 10;
 
     /**
      * The total number of operations to execute. E.g. number of checkins
      * performed by the threads.
      */
-    private static final int NUM_OPERATIONS = 100;
+    private static final int NUM_OPERATIONS = 10 * getTestScale();
 
     public void testConcurrentAddVersionableInTransaction()
             throws RepositoryException {
