diff --git a/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/NodeImpl.java b/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/NodeImpl.java
index cba185d..ea2ae31 100644
--- a/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/NodeImpl.java
+++ b/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/NodeImpl.java
@@ -767,14 +767,14 @@ public class NodeImpl extends ItemImpl implements Node {
         checkIsVersionable();
         checkHasPendingChanges();
         checkIsLocked();
-        NodeEntry newVersion = null;
         if (isCheckedOut()) {
-            newVersion = session.getVersionManager().checkin(getNodeState());
+            NodeEntry newVersion = session.getVersionManager().checkin(getNodeState());
+            return (Version) itemMgr.getItem(newVersion);
         } else {
             // nothing to do
             log.debug("Node " + safeGetJCRPath() + " is already checked in.");
+            return getBaseVersion();
         }
-        return newVersion == null ? getBaseVersion() : (Version) itemMgr.getItem(newVersion);
     }
 
     /**
