diff --git a/jackrabbit-core/src/test/java/org/apache/jackrabbit/api/jsr283/security/AccessControlDiscoveryTest.java b/jackrabbit-core/src/test/java/org/apache/jackrabbit/api/jsr283/security/AccessControlDiscoveryTest.java
index 593e86b..d8cb03a 100644
--- a/jackrabbit-core/src/test/java/org/apache/jackrabbit/api/jsr283/security/AccessControlDiscoveryTest.java
+++ b/jackrabbit-core/src/test/java/org/apache/jackrabbit/api/jsr283/security/AccessControlDiscoveryTest.java
@@ -17,6 +17,7 @@
 package org.apache.jackrabbit.api.jsr283.security;
 
 import org.apache.jackrabbit.test.NotExecutableException;
+import org.apache.jackrabbit.spi.Name;
 
 import javax.jcr.PathNotFoundException;
 import javax.jcr.RepositoryException;
@@ -138,6 +139,40 @@ public class AccessControlDiscoveryTest extends AbstractAccessControlTest {
     }
 
     /**
+     * Tests if the privilege name is treated as JCR Name and consequently
+     * reflects changes made to the namespace prefix.
+     * 
+     * @throws RepositoryException
+     */
+    public void testPrivilegeName() throws RepositoryException {
+        Privilege allPriv = acMgr.privilegeFromName(Privilege.JCR_ALL);
+        try {
+            String remappedPrefix = "_jcr";
+            superuser.setNamespacePrefix(remappedPrefix, Name.NS_JCR_URI);
+
+            List l = new ArrayList();
+            l.add(acMgr.privilegeFromName(Privilege.JCR_ALL));
+            l.add(acMgr.privilegeFromName(remappedPrefix + ":all"));
+
+            for (Iterator it = l.iterator(); it.hasNext();) {
+                Privilege p = (Privilege) it.next();
+
+                assertEquals("The privilege name must reflect the modified namespace prefix.",remappedPrefix + ":all", p.getName());
+                assertEquals("jcr:all privileges must be equal.",allPriv, p);
+            }
+
+            try {
+                acMgr.privilegeFromName("jcr:all");
+                fail("Modified namespace prefix: 'jcr:all' privilege must not exist.");
+            } catch (RepositoryException e) {
+                // success
+            }
+        } finally {
+            superuser.setNamespacePrefix(Name.NS_JCR_PREFIX, Name.NS_JCR_URI);
+        }
+    }
+
+    /**
      *
      * @throws RepositoryException
      */
