diff --git a/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/query/MixinTest.java b/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/query/MixinTest.java
index d374d08..875c0dc 100644
--- a/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/query/MixinTest.java
+++ b/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/query/MixinTest.java
@@ -34,10 +34,10 @@ public class MixinTest extends AbstractQueryTest {
 
         JackrabbitNodeTypeManager manager = (JackrabbitNodeTypeManager)
             superuser.getWorkspace().getNodeTypeManager();
-        if (!manager.hasNodeType("test:referenceable")) {
+        if (!manager.hasNodeType("test:mimeType")) {
             String cnd =
                 "<test='http://www.apache.org/jackrabbit/test'>\n"
-                + "[test:referenceable] > mix:referenceable mixin";
+                + "[test:mimeType] > mix:mimeType mixin";
             manager.registerNodeTypes(
                     new ByteArrayInputStream(cnd.getBytes()),
                     JackrabbitNodeTypeManager.TEXT_X_JCR_CND);
@@ -45,7 +45,7 @@ public class MixinTest extends AbstractQueryTest {
     }
 
     public void testBuiltInMixin() throws RepositoryException {
-        // nt:resoure is referenceable by its node type definition
+        // nt:resource is mix:mimeType by its node type definition
         Node n1 = testRootNode.addNode("n1", "nt:resource");
         n1.setProperty("jcr:data", new ByteArrayInputStream("hello world".getBytes()));
         n1.setProperty("jcr:lastModified", Calendar.getInstance());
@@ -53,15 +53,15 @@ public class MixinTest extends AbstractQueryTest {
 
         // assign mix:referenceable to arbitrary node
         Node n2 = testRootNode.addNode("n2");
-        n2.addMixin("mix:referenceable");
+        n2.addMixin("mix:mimeType");
 
-        // make node referenceable using a mixin that extends from mix:referenceable
+        // make node referenceable using a mixin that extends from mix:mimeType
         Node n3 = testRootNode.addNode("n3");
-        n3.addMixin("test:referenceable");
+        n3.addMixin("test:mimeType");
 
         testRootNode.save();
 
-        String query = testPath + "//element(*, mix:referenceable)";
+        String query = testPath + "//element(*, mix:mimeType)";
         executeXPathQuery(query, new Node[]{n1, n2, n3});
     }
