diff --git a/jackrabbit-jcr-commons/src/test/java/org/apache/jackrabbit/commons/SerializingContentHandlerTest.java b/jackrabbit-jcr-commons/src/test/java/org/apache/jackrabbit/commons/SerializingContentHandlerTest.java
index d0f5a17..4b59f54 100644
--- a/jackrabbit-jcr-commons/src/test/java/org/apache/jackrabbit/commons/SerializingContentHandlerTest.java
+++ b/jackrabbit-jcr-commons/src/test/java/org/apache/jackrabbit/commons/SerializingContentHandlerTest.java
@@ -31,7 +31,7 @@ public class SerializingContentHandlerTest extends TestCase {
         StringWriter writer = new StringWriter();
 
         ContentHandler handler =
-            new SerializingContentHandler(new StreamResult(writer));
+            SerializingContentHandler.getSerializer(new StreamResult(writer));
         handler.startDocument();
         handler.startPrefixMapping("p", "uri");
         handler.startElement("uri", "a", "p:a", new AttributesImpl());
