diff --git a/jackrabbit-jcr-server/src/main/java/org/apache/jackrabbit/webdav/jcr/AbstractResource.java b/jackrabbit-jcr-server/src/main/java/org/apache/jackrabbit/webdav/jcr/AbstractResource.java
index 10f01b1..817a87c 100644
--- a/jackrabbit-jcr-server/src/main/java/org/apache/jackrabbit/webdav/jcr/AbstractResource.java
+++ b/jackrabbit-jcr-server/src/main/java/org/apache/jackrabbit/webdav/jcr/AbstractResource.java
@@ -93,6 +93,7 @@ abstract class AbstractResource implements DavResource, TransactionResource,
         DavCompliance.concatComplianceClasses(new String[] {
         DavCompliance._1_,
         DavCompliance._2_,
+        DavCompliance._3_,
         DavCompliance.VERSION_CONTROL,
         DavCompliance.VERSION_HISTORY,
         DavCompliance.CHECKOUT_IN_PLACE,
