diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/fs/mem/MemoryFile.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/fs/mem/MemoryFile.java
index 7197e43..7392673 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/fs/mem/MemoryFile.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/fs/mem/MemoryFile.java
@@ -16,6 +16,9 @@
  */
 package org.apache.jackrabbit.core.fs.mem;
 
+/**
+ * An in-memory file.
+ */
 public class MemoryFile extends MemoryFileSystemEntry {
 
     private byte[] data = new byte[0];
