diff --git a/jackrabbit-webdav/src/test/java/org/apache/jackrabbit/webdav/server/WebdavServerTests.java b/jackrabbit-webdav/src/test/java/org/apache/jackrabbit/webdav/server/WebdavServerTests.java
index e41dfde..ea3a60d 100644
--- a/jackrabbit-webdav/src/test/java/org/apache/jackrabbit/webdav/server/WebdavServerTests.java
+++ b/jackrabbit-webdav/src/test/java/org/apache/jackrabbit/webdav/server/WebdavServerTests.java
@@ -28,7 +28,8 @@ public class WebdavServerTests extends TestCase {
         suite.addTestSuite(BindTest.class);
         suite.addTestSuite(RFC4918DestinationHeaderTest.class);
         suite.addTestSuite(RFC4918IfHeaderTest.class);
+        suite.addTestSuite(RFC4918PropfindTest.class);
 
         return suite;
     }
-}
\ No newline at end of file
+}
