diff --git a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/value/DateValue.java b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/value/DateValue.java
index 8251e2d..1ac5241 100644
--- a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/value/DateValue.java
+++ b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/value/DateValue.java
@@ -80,7 +80,7 @@ public class DateValue extends BaseValue {
      * Note that the comparison still returns false when comparing the
      * same time in different time zones, but that seems to be the intent
      * of JSR 170. Compare the Value.getDate().getTime() values if you need
-     * an exact time comparison in UTC. 
+     * an exact time comparison in UTC.
      *
      * @param obj the reference object with which to compare.
      * @return <code>true</code> if this object is the same as the obj
