diff --git a/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/ConcurrencyTest.java b/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/ConcurrencyTest.java
index ce8686b..d0c4769 100644
--- a/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/ConcurrencyTest.java
+++ b/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/ConcurrencyTest.java
@@ -55,7 +55,7 @@ public class ConcurrencyTest extends AbstractJCRTest {
             Thread[] threads = new Thread[NUM_SESSIONS];
             for (int i = 0; i < threads.length; i++) {
                 // create new session
-                Session session = helper.getSuperuserSession();
+                Session session = getHelper().getSuperuserSession();
                 TestSession ts = new TestSession("s" + i, session);
                 Thread t = new Thread(ts);
                 t.setName((NUM_ITERATIONS - n) + "-s" + i);
