diff --git a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/util/Text.java b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/util/Text.java
index 3ff8a56..86fbaca 100644
--- a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/util/Text.java
+++ b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/util/Text.java
@@ -507,7 +507,7 @@ public class Text {
         buffer.append(name);
         return buffer.toString();
     }
-    
+
     /**
      * Returns the name part of the path
      *
