diff --git a/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/WorkspaceMoveVersionableTest.java b/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/WorkspaceMoveVersionableTest.java
index c4e297a..14fc504 100644
--- a/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/WorkspaceMoveVersionableTest.java
+++ b/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/WorkspaceMoveVersionableTest.java
@@ -41,7 +41,7 @@ public class WorkspaceMoveVersionableTest extends AbstractWorkspaceVersionableTe
     protected void initNodesW2() throws RepositoryException {
         // nothing to do.
     }
-    
+
     /**
      * A VersionException is thrown if the parent node of destAbsPath is
      * versionable and checked-in, or is non-versionable but its nearest
