diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/qom/DynamicOperandImpl.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/qom/DynamicOperandImpl.java
index 1c9c156..519c70c 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/qom/DynamicOperandImpl.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/qom/DynamicOperandImpl.java
@@ -16,7 +16,7 @@
  */
 package org.apache.jackrabbit.core.query.qom;
 
-import org.apache.jackrabbit.conversion.NamePathResolver;
+import org.apache.jackrabbit.spi.commons.conversion.NamePathResolver;
 
 import org.apache.jackrabbit.core.query.jsr283.qom.DynamicOperand;
