diff --git a/jackrabbit-ocm/src/main/java/org/apache/jackrabbit/ocm/manager/objectconverter/ObjectConverter.java b/jackrabbit-ocm/src/main/java/org/apache/jackrabbit/ocm/manager/objectconverter/ObjectConverter.java
index d4e16d1..fa44b87 100644
--- a/jackrabbit-ocm/src/main/java/org/apache/jackrabbit/ocm/manager/objectconverter/ObjectConverter.java
+++ b/jackrabbit-ocm/src/main/java/org/apache/jackrabbit/ocm/manager/objectconverter/ObjectConverter.java
@@ -20,6 +20,8 @@ import javax.jcr.Node;
 import javax.jcr.Session;
 
 import org.apache.jackrabbit.ocm.exception.ObjectContentManagerException;
+import org.apache.jackrabbit.ocm.manager.collectionconverter.CollectionConverter;
+import org.apache.jackrabbit.ocm.mapper.model.CollectionDescriptor;
 
 
 /**
@@ -144,5 +146,7 @@ public interface ObjectConverter
      * @throws ObjectContentManagerException when it is not possible to retrieve the object path
      */
     public String getPath(Session session , Object object)  throws ObjectContentManagerException;
+    
+    public CollectionConverter getCollectionConverter(Session session, CollectionDescriptor collectionDescriptor);
 
 }
