diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/cluster/PropertyDeletedOperation.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/cluster/PropertyDeletedOperation.java
index 79871e0..7b85575 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/cluster/PropertyDeletedOperation.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/cluster/PropertyDeletedOperation.java
@@ -51,4 +51,5 @@ public class PropertyDeletedOperation extends PropertyOperation {
         state.setStatus(PropertyState.STATUS_EXISTING_REMOVED);
         changeLog.deleted(state);
     }
-}
\ No newline at end of file
+
+}
