diff --git a/jackrabbit-ocm/src/test/java/org/apache/jackrabbit/ocm/manager/collectionconverter/DigesterArrayListTest.java b/jackrabbit-ocm/src/test/java/org/apache/jackrabbit/ocm/manager/collectionconverter/DigesterArrayListTest.java
index 89010a3..20506a1 100644
--- a/jackrabbit-ocm/src/test/java/org/apache/jackrabbit/ocm/manager/collectionconverter/DigesterArrayListTest.java
+++ b/jackrabbit-ocm/src/test/java/org/apache/jackrabbit/ocm/manager/collectionconverter/DigesterArrayListTest.java
@@ -51,7 +51,7 @@ public class DigesterArrayListTest extends DigesterTestBase
         // All methods starting with "test" will be executed in the test suite.
         return new RepositoryLifecycleTestSetup(new TestSuite(DigesterArrayListTest.class));
     }
-      
+
     public void testArrayList()
     {
         try
@@ -67,45 +67,45 @@ public class DigesterArrayListTest extends DigesterTestBase
             Main main = new Main();
             main.setPath("/test");
             main.setText("Main text");
-            
+
             ocm.insert(main);
             ocm.save();
-            
+
             // --------------------------------------------------------------------------------
             // Get the object
-            // --------------------------------------------------------------------------------           
+            // --------------------------------------------------------------------------------
             main = (Main) ocm.getObject( "/test");
             ArrayList arrayList = main.getList();
-            assertNull("main.getList is not null", arrayList ); 
-            
+            assertNull("main.getList is not null", arrayList );
+
             // --------------------------------------------------------------------------------
             // Update the object
             // --------------------------------------------------------------------------------
-            
+
             ArrayListElement arrayListElement = new ArrayListElement();
             Element e1 = new Element();
             e1.setId("e1");
             e1.setText("Element 1");
             arrayListElement.add(e1);
-            
+
             Element e2 = new Element();
             e2.setId("e2");
             e2.setText("Element 2");
             arrayListElement.add(e2);
-            
+
             main.setList(arrayListElement);
             ocm.update(main);
             ocm.save();
 
             // --------------------------------------------------------------------------------
             // Get the object
-            // --------------------------------------------------------------------------------           
+            // --------------------------------------------------------------------------------
             main = (Main) ocm.getObject( "/test");
             arrayList = main.getList();
-            assertNotNull("main.getList is null", arrayList ); 
+            assertNotNull("main.getList is null", arrayList );
             Element[] elements = (Element[]) arrayList.toArray(new Element[arrayList.size()]);
             assertTrue("Incorrect para element", elements[0].getText().equals("Element 1"));
-            
+
             // --------------------------------------------------------------------------------
             // Update the object
             // --------------------------------------------------------------------------------
@@ -114,40 +114,40 @@ public class DigesterArrayListTest extends DigesterTestBase
             e1.setId("e1");
             e1.setText("Element 1");
             arrayListElement.add(e1);
-            
+
             e2 = new Element();
             e2.setId("e3");
             e2.setText("Element 3");
             arrayListElement.add(e2);
-            
+
             Element e3 = new Element();
             e3.setId("e4");
             e3.setText("Element 4");
             arrayListElement.add(e3);
-            
-            main.setList(arrayListElement);            
-            
+
+            main.setList(arrayListElement);
+
             ocm.update(main);
             ocm.save();
 
             // --------------------------------------------------------------------------------
             // Get the object
-            // --------------------------------------------------------------------------------           
+            // --------------------------------------------------------------------------------
             main = (Main) ocm.getObject( "/test");
             arrayList = main.getList();
-            assertNotNull("main.getList() is null", arrayList ); 
+            assertNotNull("main.getList() is null", arrayList );
             elements = (Element[]) arrayList.toArray(new Element[arrayList.size()]);
             assertTrue("Incorrect element", elements[2].getText().equals("Element 4"));
-            
+
         }
         catch (Exception e)
         {
             e.printStackTrace();
             fail("Exception occurs during the unit test : " + e);
         }
-        
+
     }
 
 
-   
+
 }
\ No newline at end of file
