diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/value/BLOBInTempFile.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/value/BLOBInTempFile.java
index fd6bedb..91b644f 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/value/BLOBInTempFile.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/value/BLOBInTempFile.java
@@ -33,7 +33,7 @@ import javax.jcr.RepositoryException;
 /**
  * Represents binary data which is stored in a temporary file.
  */
-public class BLOBInTempFile extends BLOBFileValue {
+class BLOBInTempFile extends BLOBFileValue {
 
     /**
      * the prefix of the string representation of this value
