diff --git a/contrib/jackrabbit-jcr-mapping/jcr-mapping/src/test/java/org/apache/jackrabbit/ocm/testmodel/Default.java b/contrib/jackrabbit-jcr-mapping/jcr-mapping/src/test/java/org/apache/jackrabbit/ocm/testmodel/Default.java
index baa88ba..2471836 100644
--- a/contrib/jackrabbit-jcr-mapping/jcr-mapping/src/test/java/org/apache/jackrabbit/ocm/testmodel/Default.java
+++ b/contrib/jackrabbit-jcr-mapping/jcr-mapping/src/test/java/org/apache/jackrabbit/ocm/testmodel/Default.java
@@ -16,10 +16,14 @@
  */
 package org.apache.jackrabbit.ocm.testmodel;
 
+import org.apache.jackrabbit.ocm.mapper.impl.annotation.Field;
+import org.apache.jackrabbit.ocm.mapper.impl.annotation.Node;
+
 
 /**
  * Simple object used to test default value assignement
  */
+@Node(jcrType="ocm:DefTestPrimary", discriminator=false)
 public class Default {
     private String path;
 
@@ -33,6 +37,7 @@ public class Default {
 
     private String p5;
 
+    @Field(path=true)
     public String getPath() {
         return path;
     }
@@ -44,6 +49,7 @@ public class Default {
     /**
      * @return the p1
      */
+    @Field(jcrName="ocm:p1")
     public String getP1() {
         return p1;
     }
@@ -58,6 +64,7 @@ public class Default {
     /**
      * @return the p2
      */
+    @Field(jcrName="ocm:p2")
     public String getP2() {
         return p2;
     }
@@ -72,6 +79,7 @@ public class Default {
     /**
      * @return the p3
      */
+    @Field(jcrName="ocm:p3", jcrDefaultValue="p3DescriptorDefaultValue")
     public String getP3() {
         return p3;
     }
@@ -86,6 +94,7 @@ public class Default {
     /**
      * @return the p4
      */
+    @Field(jcrName="ocm:p4")
     public String getP4() {
         return p4;
     }
@@ -100,6 +109,7 @@ public class Default {
     /**
      * @return the p5
      */
+    @Field(jcrName="ocm:p5")
     public String getP5() {
         return p5;
     }
