diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/state/ISMLocking.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/state/ISMLocking.java
index fcec52d..f539d77 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/state/ISMLocking.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/state/ISMLocking.java
@@ -16,7 +16,7 @@
  */
 package org.apache.jackrabbit.core.state;
 
-import org.apache.jackrabbit.core.ItemId;
+import org.apache.jackrabbit.core.id.ItemId;
 
 /**
  * <code>ISMLocking</code> defines an interface for a locking strategy of an
