diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/security/authentication/AbstractLoginModule.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/security/authentication/AbstractLoginModule.java
index abc3c0c..b8043bc 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/security/authentication/AbstractLoginModule.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/security/authentication/AbstractLoginModule.java
@@ -656,9 +656,9 @@ public abstract class AbstractLoginModule implements LoginModule {
     }
 
     /**
-     * Sets the anonymous user id.
+     * Sets the administrator's user id.
      *
-     * @param anonymousId anonymous user id
+     * @param adminId the administrator's user id.
      */
     public void setAdminId(String adminId) {
         this.adminId = adminId;
