diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/qom/ChildNodeImpl.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/qom/ChildNodeImpl.java
index 5ebd6ca..206c0c4 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/qom/ChildNodeImpl.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/qom/ChildNodeImpl.java
@@ -16,9 +16,9 @@
  */
 package org.apache.jackrabbit.core.query.qom;
 
-import org.apache.jackrabbit.name.Path;
-import org.apache.jackrabbit.name.QName;
-import org.apache.jackrabbit.name.NamePathResolver;
+import org.apache.jackrabbit.spi.Path;
+import org.apache.jackrabbit.spi.Name;
+import org.apache.jackrabbit.conversion.NamePathResolver;
 
 import org.apache.jackrabbit.core.query.jsr283.qom.ChildNode;
 
@@ -30,14 +30,14 @@ public class ChildNodeImpl extends ConstraintImpl implements ChildNode {
     /**
      * The name of a selector.
      */
-    private final QName selectorName;
+    private final Name selectorName;
 
     /**
      * An absolute path.
      */
     private final Path path;
 
-    ChildNodeImpl(NamePathResolver resolver, QName selectorName, Path path) {
+    ChildNodeImpl(NamePathResolver resolver, Name selectorName, Path path) {
         super(resolver);
         this.selectorName = selectorName;
         this.path = path;
