diff --git a/jackrabbit-ocm/src/test/java/org/apache/jackrabbit/ocm/testmodel/C.java b/jackrabbit-ocm/src/test/java/org/apache/jackrabbit/ocm/testmodel/C.java
index 39f3eff..ebd7411 100644
--- a/jackrabbit-ocm/src/test/java/org/apache/jackrabbit/ocm/testmodel/C.java
+++ b/jackrabbit-ocm/src/test/java/org/apache/jackrabbit/ocm/testmodel/C.java
@@ -30,7 +30,7 @@ public class C
 	
 	@Field(jcrName="ocm:id", id=true) private String id;
     @Field(jcrName="ocm:name") private String name;
-     
+
     /**
      * @return Returns the id.
      */
