diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/journal/ReadRecord.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/journal/ReadRecord.java
index 02b8592..34eaaaf 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/journal/ReadRecord.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/journal/ReadRecord.java
@@ -26,7 +26,7 @@ import java.io.IOException;
 /**
  * Record used for reading.
  */
-class ReadRecord extends AbstractRecord {
+public class ReadRecord extends AbstractRecord {
 
     /**
      * This record's journal id.
