diff --git a/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/NodeMixinUtil.java b/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/NodeMixinUtil.java
index 5e1abef..aff07c9 100644
--- a/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/NodeMixinUtil.java
+++ b/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/NodeMixinUtil.java
@@ -21,6 +21,9 @@ import javax.jcr.Session;
 import javax.jcr.Node;
 import javax.jcr.nodetype.NodeTypeManager;
 import javax.jcr.nodetype.NodeTypeIterator;
+import javax.jcr.nodetype.NodeType;
+import java.util.List;
+import java.util.Arrays;
 
 /**
  * Utility class to locate mixins in the NodeTyeManager.
@@ -46,6 +49,20 @@ public class NodeMixinUtil {
         return null;
     }
 
+    public static String getNotAssignedMixinName(Session session, Node node) throws RepositoryException {
+        NodeTypeManager manager = session.getWorkspace().getNodeTypeManager();
+        NodeTypeIterator mixins = manager.getMixinNodeTypes();
+        List existingMixins = Arrays.asList(node.getMixinNodeTypes());
+
+        while (mixins.hasNext()) {
+            NodeType nt = mixins.nextNodeType();
+            if (!existingMixins.contains(nt)) {
+                return nt.getName();
+            }
+        }
+        return null;
+    }
+
     /**
      * @return a string that is not the name of a mixin type
      */
