diff --git a/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/UndefinedPropertyTest.java b/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/UndefinedPropertyTest.java
index 065f9b7..fdf2dd0 100644
--- a/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/UndefinedPropertyTest.java
+++ b/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/UndefinedPropertyTest.java
@@ -55,7 +55,7 @@ public class UndefinedPropertyTest extends AbstractJCRTest {
      * Tests that no actual property with type Undefined exists.
      */
     public void testUndefinedProperty() throws RepositoryException {
-        Session session = helper.getReadOnlySession();
+        Session session = getHelper().getReadOnlySession();
         try {
             Property prop = PropertyUtil.searchProp(session, session.getRootNode().getNode(testPath), PropertyType.UNDEFINED, null);
             assertNull("Property with type Undefined found.", prop);
