diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/virtual/VirtualPropertyState.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/virtual/VirtualPropertyState.java
index 571bae2..ca997f0 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/virtual/VirtualPropertyState.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/virtual/VirtualPropertyState.java
@@ -16,7 +16,7 @@
  */
 package org.apache.jackrabbit.core.virtual;
 
-import org.apache.jackrabbit.core.PropertyId;
+import org.apache.jackrabbit.core.id.PropertyId;
 import org.apache.jackrabbit.core.state.ItemState;
 import org.apache.jackrabbit.core.state.PropertyState;
 import org.apache.jackrabbit.core.value.InternalValue;
