diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/ItemManager.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/ItemManager.java
index 0d3d0ce..9786566 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/ItemManager.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/ItemManager.java
@@ -58,7 +58,7 @@ import java.util.Map;
  * instance. It is the factory for <code>Node</code> and <code>Property</code>
  * instances.
  * <p/>
- * The <code>ItemManager</code>'s responsabilities are:
+ * The <code>ItemManager</code>'s responsibilities are:
  * <ul>
  * <li>providing access to <code>Item</code> instances by <code>ItemId</code>
  * whereas <code>Node</code> and <code>Item</code> are only providing relative access.
