diff --git a/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/query/SkipDeletedNodesTest.java b/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/query/SkipDeletedNodesTest.java
index 4c436cb..b598662 100644
--- a/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/query/SkipDeletedNodesTest.java
+++ b/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/query/SkipDeletedNodesTest.java
@@ -37,7 +37,7 @@ public class SkipDeletedNodesTest extends AbstractQueryTest {
 
     protected void setUp() throws Exception {
         super.setUp();
-        s2 = helper.getSuperuserSession();
+        s2 = getHelper().getSuperuserSession();
         qm = s2.getWorkspace().getQueryManager();
     }
