diff --git a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/name/NamePathResolver.java b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/name/NamePathResolver.java
index 0a86ef6..df38170 100644
--- a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/name/NamePathResolver.java
+++ b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/name/NamePathResolver.java
@@ -21,7 +21,9 @@ package org.apache.jackrabbit.name;
  * name and path resolver interfaces when a single dependency for both
  * name and path resolution is needed.
  *
- * @deprecated Use {@link org.apache.jackrabbit.conversion.NamePathResolver} instead.
+ * @deprecated Use the NamePathResolver interface from 
+ *             the org.apache.jackrabbit.spi.commons.conversion package of
+ *             the jackrabbit-spi-commons component.
  */
 public interface NamePathResolver extends NameResolver, PathResolver {
 }
