diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/journal/FileRecordIterator.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/journal/FileRecordIterator.java
index e0221a0..12d3463 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/journal/FileRecordIterator.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/journal/FileRecordIterator.java
@@ -160,4 +160,5 @@ public class FileRecordIterator implements RecordIterator {
         String msg = "No log file found containing revision: " + revision;
         throw new IOException(msg);
     }
-}
\ No newline at end of file
+
+}
