diff --git a/jackrabbit-core/src/test/java/org/apache/jackrabbit/api/security/user/UserTest.java b/jackrabbit-core/src/test/java/org/apache/jackrabbit/api/security/user/UserTest.java
index 3467a2e..3832c26 100644
--- a/jackrabbit-core/src/test/java/org/apache/jackrabbit/api/security/user/UserTest.java
+++ b/jackrabbit-core/src/test/java/org/apache/jackrabbit/api/security/user/UserTest.java
@@ -44,7 +44,7 @@ public class UserTest extends AbstractUserTest {
     }
 
     public void testReadOnlyIsntAdmin() throws NotExecutableException, RepositoryException {
-        Session s = helper.getReadOnlySession();
+        Session s = getHelper().getReadOnlySession();
         try {
             User user = getTestUser(s);
             assertFalse(user.isAdmin());
@@ -60,7 +60,7 @@ public class UserTest extends AbstractUserTest {
     }
 
     public void testChangePassword() throws RepositoryException, NotExecutableException {
-        String oldPw = helper.getProperty("javax.jcr.tck.superuser.pwd");
+        String oldPw = getHelper().getProperty("javax.jcr.tck.superuser.pwd");
         if (oldPw == null) {
             // missing property
             throw new NotExecutableException();
@@ -70,7 +70,7 @@ public class UserTest extends AbstractUserTest {
         try {
             user.changePassword("pw");
             // make sure the user can login with the new pw
-            Session s = helper.getRepository().login(new SimpleCredentials(user.getID(), "pw".toCharArray()));
+            Session s = getHelper().getRepository().login(new SimpleCredentials(user.getID(), "pw".toCharArray()));
             s.logout();
         } finally {
             user.changePassword(oldPw);
@@ -78,7 +78,7 @@ public class UserTest extends AbstractUserTest {
     }
 
     public void testChangePassword2() throws RepositoryException, NotExecutableException {
-        String oldPw = helper.getProperty("javax.jcr.tck.superuser.pwd");
+        String oldPw = getHelper().getProperty("javax.jcr.tck.superuser.pwd");
         if (oldPw == null) {
             // missing property
             throw new NotExecutableException();
@@ -88,7 +88,7 @@ public class UserTest extends AbstractUserTest {
         try {
             user.changePassword("pw");
 
-            Session s = helper.getRepository().login(new SimpleCredentials(user.getID(), oldPw.toCharArray()));
+            Session s = getHelper().getRepository().login(new SimpleCredentials(user.getID(), oldPw.toCharArray()));
             s.logout();
             fail("superuser pw has changed. login must fail.");
         } catch (LoginException e) {
