diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/version/InternalFreeze.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/version/InternalFreeze.java
index 94dac58..a4a6bfb 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/version/InternalFreeze.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/version/InternalFreeze.java
@@ -16,7 +16,7 @@
  */
 package org.apache.jackrabbit.core.version;
 
-import org.apache.jackrabbit.name.QName;
+import org.apache.jackrabbit.spi.Name;
 
 /**
  * the base interface for nodes that were versioned and turned either into
@@ -29,6 +29,6 @@ public interface InternalFreeze extends InternalVersionItem {
      *
      * @return the name of the node.
      */
-    QName getName();
+    Name getName();
 
 }
