diff --git a/jackrabbit-spi/src/main/java/org/apache/jackrabbit/spi/ItemId.java b/jackrabbit-spi/src/main/java/org/apache/jackrabbit/spi/ItemId.java
index 148be0b..db141fa 100644
--- a/jackrabbit-spi/src/main/java/org/apache/jackrabbit/spi/ItemId.java
+++ b/jackrabbit-spi/src/main/java/org/apache/jackrabbit/spi/ItemId.java
@@ -43,6 +43,10 @@ package org.apache.jackrabbit.spi;
  *   <code>ItemId</code>.
  *   </td></tr>
  * </table>
+ * <p/>
+ * Two <code>ItemId</code>s should be considered equal if both the unique part
+ * and the path part are equal AND if they denote the same
+ * {@link #denotesNode() type} of <code>ItemId</code>.
  */
 public interface ItemId {
