diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/config/BeanConfig.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/config/BeanConfig.java
index dd2ef3a..c1f8c76 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/config/BeanConfig.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/config/BeanConfig.java
@@ -31,13 +31,13 @@ public class BeanConfig {
     /** The default class loader used by all instances of this class */
     private static ClassLoader DEFAULT_CLASS_LOADER =
         BeanConfig.class.getClassLoader();
-    
+
     /**
      * The current class loader used by this instance to create instances of
      * configured classes.
      */
     private ClassLoader classLoader = getDefaultClassLoader();
-    
+
     /**
      * The class name of the configured bean.
      */
@@ -126,11 +126,11 @@ public class BeanConfig {
     }
 
     //---------- Configurable class loader support ----------------------------
-    
+
     /**
      * Returns the current <code>ClassLoader</code> used to instantiate objects
      * in the {@link #newInstance()} method.
-     * 
+     *
      * @see #newInstance()
      * @see #setClassLoader(ClassLoader)
      * @see #getDefaultClassLoader()
@@ -139,15 +139,15 @@ public class BeanConfig {
     public ClassLoader getClassLoader() {
         return classLoader;
     }
-    
+
     /**
      * Sets the <code>ClassLoader</code> used to instantiate objects in the
      * {@link #newInstance()} method.
-     * 
+     *
      * @param classLoader The class loader to set on this instance. If this is
      *      <code>null</code> the system class loader will be used, which may
      *      lead to unexpected class loading failures.
-     *      
+     *
      * @see #newInstance()
      * @see #getClassLoader()
      * @see #getDefaultClassLoader()
@@ -156,12 +156,12 @@ public class BeanConfig {
     public void setClassLoader(ClassLoader classLoader) {
         this.classLoader = classLoader;
     }
-    
+
     /**
      * Returns the current <code>ClassLoader</code> used for new instances of
      * this class as the loader used to instantiate objects in the
      * {@link #newInstance()} method.
-     * 
+     *
      * @see #newInstance()
      * @see #getClassLoader()
      * @see #setClassLoader(ClassLoader)
@@ -170,15 +170,15 @@ public class BeanConfig {
     public static ClassLoader getDefaultClassLoader() {
         return DEFAULT_CLASS_LOADER;
     }
-    
+
     /**
      * Sets the <code>ClassLoader</code> used for new instances of this class as
      * the loader to instantiate objects in the {@link #newInstance()} method.
-     * 
+     *
      * @param classLoader The class loader to set as the default class loader.
      *      If this is <code>null</code> the system class loader will be used,
-     *      which may lead to unexpected class loading failures. 
-     *      
+     *      which may lead to unexpected class loading failures.
+     *
      * @see #newInstance()
      * @see #getClassLoader()
      * @see #setClassLoader(ClassLoader)
