diff --git a/jackrabbit-webdav/src/main/java/org/apache/jackrabbit/webdav/transaction/TransactionResource.java b/jackrabbit-webdav/src/main/java/org/apache/jackrabbit/webdav/transaction/TransactionResource.java
index e16465a..49c89d9 100644
--- a/jackrabbit-webdav/src/main/java/org/apache/jackrabbit/webdav/transaction/TransactionResource.java
+++ b/jackrabbit-webdav/src/main/java/org/apache/jackrabbit/webdav/transaction/TransactionResource.java
@@ -25,7 +25,6 @@ import org.apache.jackrabbit.webdav.DavResource;
  */
 public interface TransactionResource extends DavResource {
 
-    public static final String COMPLIANCE_CLASS = "";
     public static final String METHODS = "";
 
     /**
