diff --git a/jackrabbit-webdav/src/main/java/org/apache/jackrabbit/webdav/observation/ObservationResource.java b/jackrabbit-webdav/src/main/java/org/apache/jackrabbit/webdav/observation/ObservationResource.java
index f72418c..d27b7da 100644
--- a/jackrabbit-webdav/src/main/java/org/apache/jackrabbit/webdav/observation/ObservationResource.java
+++ b/jackrabbit-webdav/src/main/java/org/apache/jackrabbit/webdav/observation/ObservationResource.java
@@ -25,7 +25,6 @@ import org.apache.jackrabbit.webdav.DavResource;
  */
 public interface ObservationResource extends DavResource {
 
-    public String COMPLIANCE_CLASS = "observation";
     public String METHODS = "SUBSCRIBE, UNSUBSCRIBE, POLL";
 
     /**
