diff --git a/jackrabbit-spi/src/main/java/org/apache/jackrabbit/spi/QNodeTypeDefinition.java b/jackrabbit-spi/src/main/java/org/apache/jackrabbit/spi/QNodeTypeDefinition.java
index f6f16d7..aebfc2a 100644
--- a/jackrabbit-spi/src/main/java/org/apache/jackrabbit/spi/QNodeTypeDefinition.java
+++ b/jackrabbit-spi/src/main/java/org/apache/jackrabbit/spi/QNodeTypeDefinition.java
@@ -19,8 +19,8 @@ package org.apache.jackrabbit.spi;
 import java.util.Collection;
 
 /**
- * <code>QNodeTypeDefinition</code> is the qualified SPI representation of a
- * {@link javax.jcr.nodetype.NodeType node type}. It refers to qualified names
+ * <code>QNodeTypeDefinition</code> is the SPI representation of a
+ * {@link javax.jcr.nodetype.NodeType node type}. It refers to <code>Name</code>s
  * only and is therefore independant of session-specific namespace mappings.
  *
  * @see javax.jcr.nodetype.NodeType
