diff --git a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/util/ISO9075.java b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/util/ISO9075.java
index bd75154..fb831bc 100644
--- a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/util/ISO9075.java
+++ b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/util/ISO9075.java
@@ -16,8 +16,6 @@
  */
 package org.apache.jackrabbit.util;
 
-import org.apache.jackrabbit.name.QName;
-
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
@@ -25,11 +23,7 @@ import java.util.regex.Pattern;
  * Implements the encode and decode routines as specified for XML name to SQL
  * identifier conversion in ISO 9075-14:2003.<br/>
  * If a character <code>c</code> is not valid at a certain position in an XML 1.0
- * NCName it is encoded in the form: '_x' + hexValueOf(c) + '_'
- * <p/>
- * Note that only the local part of a {@link org.apache.jackrabbit.name.QName}
- * is encoded / decoded. A URI namespace will always be valid and does not
- * need encoding.
+ * NCName it is encoded in the form: '_x' + hexValueOf(c) + '_'.
  */
 public class ISO9075 {
 
@@ -46,21 +40,6 @@ public class ISO9075 {
     private static final String HEX_DIGITS = "0123456789abcdefABCDEF";
 
     /**
-     * Encodes the local part of <code>name</code> as specified in ISO 9075.
-     * @param name the <code>QName</code> to encode.
-     * @return the encoded <code>QName</code> or <code>name</code> if it does
-     *   not need encoding.
-     */
-    public static QName encode(QName name) {
-        String encoded = encode(name.getLocalName());
-        if (encoded == name.getLocalName()) {
-            return name;
-        } else {
-            return new QName(name.getNamespaceURI(), encoded);
-        }
-    }
-
-    /**
      * Encodes <code>name</code> as specified in ISO 9075.
      * @param name the <code>String</code> to encode.
      * @return the encoded <code>String</code> or <code>name</code> if it does
@@ -138,20 +117,6 @@ public class ISO9075 {
 
     /**
      * Decodes the <code>name</code>.
-     * @param name the <code>QName</code> to decode.
-     * @return the decoded <code>QName</code>.
-     */
-    public static QName decode(QName name) {
-        String decoded = decode(name.getLocalName());
-        if (decoded == name.getLocalName()) {
-            return name;
-        } else {
-            return new QName(name.getNamespaceURI(), decoded);
-        }
-    }
-
-    /**
-     * Decodes the <code>name</code>.
      * @param name the <code>String</code> to decode.
      * @return the decoded <code>String</code>.
      */
