diff --git a/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/lock/LockManager.java b/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/lock/LockManager.java
index 7a0b465..514c40f 100644
--- a/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/lock/LockManager.java
+++ b/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/lock/LockManager.java
@@ -27,7 +27,7 @@ import javax.jcr.lock.LockException;
  * Defines the functionality needed for locking and unlocking nodes.
  */
 public interface LockManager {
-    
+
     /**
      * Lock a node. Checks whether the node is not locked and then
      * returns a lock object for this node.
