diff --git a/jackrabbit-jcr2spi/src/test/java/org/apache/jackrabbit/jcr2spi/RevertMoveTest.java b/jackrabbit-jcr2spi/src/test/java/org/apache/jackrabbit/jcr2spi/RevertMoveTest.java
index f178a23..757a4bb 100644
--- a/jackrabbit-jcr2spi/src/test/java/org/apache/jackrabbit/jcr2spi/RevertMoveTest.java
+++ b/jackrabbit-jcr2spi/src/test/java/org/apache/jackrabbit/jcr2spi/RevertMoveTest.java
@@ -37,7 +37,7 @@ public class RevertMoveTest extends AbstractMoveTest {
      * Implementation specific test, that expects that the scope of a refresh(false)
      * must include all nodes affected by the operations that affected the
      * subtree to be refreshed.
-     * 
+     *
      * @throws RepositoryException
      */
     public void testRevertMovedNode() throws RepositoryException {
