diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/SingletonTokenStream.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/SingletonTokenStream.java
index 7d581a7..1e9fa40 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/SingletonTokenStream.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/SingletonTokenStream.java
@@ -46,6 +46,15 @@ public final class SingletonTokenStream extends TokenStream {
     }
 
     /**
+     * Creates a new SingleTokenStream with the given token.
+     *
+     * @param t the token.
+     */
+    public SingletonTokenStream(Token t) {
+        this.t = t;
+    }
+
+    /**
      * {@inheritDoc}
      */
     public Token next() {
