diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/data/DataStoreException.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/data/DataStoreException.java
index 9f2c50a..4b173ac 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/data/DataStoreException.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/data/DataStoreException.java
@@ -38,7 +38,7 @@ public class DataStoreException extends RepositoryException {
      * message and root cause.
      *
      * @param message the detailed message.
-     * @param rootCause root failure cause
+     * @param cause root failure cause
      */
     public DataStoreException(String message, Throwable cause) {
         super(message, cause);
