diff --git a/jackrabbit-webdav/src/main/java/org/apache/jackrabbit/webdav/bind/ParentSet.java b/jackrabbit-webdav/src/main/java/org/apache/jackrabbit/webdav/bind/ParentSet.java
index 28b0fad..fdac132 100644
--- a/jackrabbit-webdav/src/main/java/org/apache/jackrabbit/webdav/bind/ParentSet.java
+++ b/jackrabbit-webdav/src/main/java/org/apache/jackrabbit/webdav/bind/ParentSet.java
@@ -16,17 +16,9 @@
  */
 package org.apache.jackrabbit.webdav.bind;
 
-import org.apache.jackrabbit.webdav.xml.XmlSerializable;
-import org.apache.jackrabbit.webdav.xml.DomUtil;
 import org.apache.jackrabbit.webdav.property.AbstractDavProperty;
-import org.apache.jackrabbit.webdav.property.DavPropertyName;
-import org.w3c.dom.Element;
-import org.w3c.dom.Document;
-import org.w3c.dom.Node;
 
 import java.util.Collection;
-import java.util.Collections;
-import java.util.List;
 
 /**
  * <code>ParentSet</code> represents a DAV:parent-set property.
