diff --git a/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/state/TransientISFactory.java b/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/state/TransientISFactory.java
index 66e2b4b..2ff4d69 100644
--- a/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/state/TransientISFactory.java
+++ b/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/state/TransientISFactory.java
@@ -29,6 +29,7 @@ import org.apache.jackrabbit.jcr2spi.nodetype.ItemDefinitionProvider;
 
 import javax.jcr.RepositoryException;
 import javax.jcr.ItemNotFoundException;
+
 import java.util.Iterator;
 
 /**
@@ -141,9 +142,9 @@ public final class TransientISFactory extends AbstractItemStateFactory implement
      * @inheritDoc
      * @see ItemStateFactory#getNodeReferences(NodeState)
      */
-    public NodeReferences getNodeReferences(NodeState nodeState) {
+    public PropertyId[] getNodeReferences(NodeState nodeState) {
         if (nodeState.getStatus() == Status.NEW) {
-            return EmptyNodeReferences.getInstance();
+            return new PropertyId[0];
         }
         return workspaceStateFactory.getNodeReferences(nodeState);
     }
