diff --git a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/name/ParsingNameResolver.java b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/name/ParsingNameResolver.java
index 300ec46..4e00b4c 100644
--- a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/name/ParsingNameResolver.java
+++ b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/name/ParsingNameResolver.java
@@ -23,7 +23,9 @@ import javax.jcr.Session;
  * Name resolver that parsers and formats prefixed JCR names.
  * A {@link NamespaceResolver} is used for resolving the namespace prefixes.
  *
- * @deprecated Use {@link org.apache.jackrabbit.conversion.ParsingNameResolver}.
+ * @deprecated Use the ParsingNameResolver class from 
+ *             the org.apache.jackrabbit.spi.commons.conversion package of
+ *             the jackrabbit-spi-commons component.
  */
 public class ParsingNameResolver implements NameResolver {
