diff --git a/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/integration/MultiThreadingTest.java b/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/integration/MultiThreadingTest.java
index 1b8ab9c..22a4d59 100644
--- a/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/integration/MultiThreadingTest.java
+++ b/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/integration/MultiThreadingTest.java
@@ -41,7 +41,7 @@ public class MultiThreadingTest extends TestCase {
 
         suite.addTestSuite(ConcurrencyTest.class);
         suite.addTestSuite(ConcurrentLoginTest.class);
-        //suite.addTestSuite(ConcurrentNodeModificationTest.class);
+        suite.addTestSuite(ConcurrentNodeModificationTest.class);
         suite.addTestSuite(ConcurrentReadWriteTest.class);
         suite.addTestSuite(ConcurrentSaveTest.class);
         suite.addTestSuite(ConcurrentVersioningTest.class);
