diff --git a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/value/ReferenceValue.java b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/value/ReferenceValue.java
index 8d7f7a3..a00babf 100644
--- a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/value/ReferenceValue.java
+++ b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/value/ReferenceValue.java
@@ -24,6 +24,7 @@ import javax.jcr.RepositoryException;
 import javax.jcr.UnsupportedRepositoryOperationException;
 import javax.jcr.ValueFormatException;
 import java.util.Calendar;
+import java.math.BigDecimal;
 
 /**
  * A <code>ReferenceValue</code> provides an implementation
@@ -184,4 +185,15 @@ public class ReferenceValue extends BaseValue {
 
         throw new ValueFormatException("conversion to double failed: inconvertible types");
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    public BigDecimal getDecimal()
+            throws ValueFormatException, IllegalStateException,
+            RepositoryException {
+        setValueConsumed();
+
+        throw new ValueFormatException("conversion to Decimal failed: inconvertible types");
+    }
 }
