diff --git a/jackrabbit-ocm/src/test/java/org/apache/jackrabbit/ocm/manager/jcrnodetype/DigesterJcrPropertyTest.java b/jackrabbit-ocm/src/test/java/org/apache/jackrabbit/ocm/manager/jcrnodetype/DigesterJcrPropertyTest.java
index 58fae67..a21688a 100644
--- a/jackrabbit-ocm/src/test/java/org/apache/jackrabbit/ocm/manager/jcrnodetype/DigesterJcrPropertyTest.java
+++ b/jackrabbit-ocm/src/test/java/org/apache/jackrabbit/ocm/manager/jcrnodetype/DigesterJcrPropertyTest.java
@@ -48,10 +48,10 @@ public class DigesterJcrPropertyTest extends DigesterTestBase {
 		return new RepositoryLifecycleTestSetup(new TestSuite(DigesterJcrPropertyTest.class));
 	}
 
-	public void testRequiredProperty() 
+	public void testRequiredProperty()
 	{
 
-		try 
+		try
 		{
 			ObjectContentManager ocm = this.getObjectContentManager();
 			//---------------------------------------------------------------------------------------------------------
@@ -62,8 +62,8 @@ public class DigesterJcrPropertyTest extends DigesterTestBase {
             propertyTest.setPath("/test");
             propertyTest.setRequiredProp("requiredPropValue");
             propertyTest.setRequiredWithConstraintsProp("abc");
-            
-            try 
+
+            try
             {
                  ocm.insert(propertyTest);
                  fail("Incorrect insert operation - the mandatory fields have no value");
@@ -73,7 +73,7 @@ public class DigesterJcrPropertyTest extends DigesterTestBase {
                // Normal behaviour 	
             	ocm.refresh(false);
             }
-            
+
 			//---------------------------------------------------------------------------------------------------------
 			// Insert with the mandatory fields
 			//---------------------------------------------------------------------------------------------------------			
@@ -81,23 +81,23 @@ public class DigesterJcrPropertyTest extends DigesterTestBase {
             propertyTest.setMandatoryWithConstaintsProp("xx");
             ocm.insert(propertyTest);
             ocm.save();
-            
+
 			//---------------------------------------------------------------------------------------------------------
 			// Retrieve
 			//---------------------------------------------------------------------------------------------------------			
             propertyTest = (PropertyTest) ocm.getObject("/test");
             assertTrue("Invalid required property", propertyTest.getRequiredProp().equals("requiredPropValue"));
-            assertTrue("Invalid required property with constraints", propertyTest.getRequiredWithConstraintsProp().equals("abc"));            
-            assertTrue("Invalid autocreated property", propertyTest.getAutoCreatedProp().equals("aaa")); 
+            assertTrue("Invalid required property with constraints", propertyTest.getRequiredWithConstraintsProp().equals("abc"));
+            assertTrue("Invalid autocreated property", propertyTest.getAutoCreatedProp().equals("aaa"));
             assertTrue("Invalid autocreated property", propertyTest.getAutoCreatedWithConstraintsProp().equals("ccc"));
             assertTrue("Invalid protected property", propertyTest.getProtectedWithDefaultValueProp().equals("protectedValue"));
-            
+
             //---------------------------------------------------------------------------------------------------------
 			// update the property requiredWithConstraintsProp with bad value
 			//---------------------------------------------------------------------------------------------------------			
             propertyTest = (PropertyTest) ocm.getObject("/test");
             propertyTest.setRequiredWithConstraintsProp("invalid value");
-            try 
+            try
             {
             	ocm.update(propertyTest);
             	ocm.save();
@@ -107,13 +107,13 @@ public class DigesterJcrPropertyTest extends DigesterTestBase {
             {                	
                // Do nothing - normal behaviour, the value               	
             }
-            
+
             //---------------------------------------------------------------------------------------------------------
 			// update the property AutoCreatedWithConstraintsProp with bad value
 			//---------------------------------------------------------------------------------------------------------			
             propertyTest = (PropertyTest) ocm.getObject("/test");
             propertyTest.setAutoCreatedWithConstraintsProp("invalid value");
-            try 
+            try
             {
             	ocm.update(propertyTest);
             	ocm.save();
@@ -124,13 +124,13 @@ public class DigesterJcrPropertyTest extends DigesterTestBase {
                // Do nothing - normal behaviour, the value is not valid
                	
             }
-            
+
             //---------------------------------------------------------------------------------------------------------
 			// update the property mandatoryWithConstaintsProp with bad value
 			//---------------------------------------------------------------------------------------------------------			
             propertyTest = (PropertyTest) ocm.getObject("/test");
             propertyTest.setMandatoryWithConstaintsProp("yy");
-            try 
+            try
             {
             	ocm.update(propertyTest);
             	ocm.save();
@@ -139,7 +139,7 @@ public class DigesterJcrPropertyTest extends DigesterTestBase {
             catch(Exception e)
             {                	
                 e.printStackTrace();
-            }            
+            }
 			
 			
 		}
