diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/security/principal/EveryonePrincipal.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/security/principal/EveryonePrincipal.java
index 418161f..fae96af 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/security/principal/EveryonePrincipal.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/security/principal/EveryonePrincipal.java
@@ -27,12 +27,12 @@ public final class EveryonePrincipal implements java.security.acl.Group {
     private static final String NAME = "everyone";
     private static final EveryonePrincipal INSTANCE = new EveryonePrincipal();
 
-    private EveryonePrincipal() {}
+    private EveryonePrincipal() { }
 
     public static EveryonePrincipal getInstance() {
         return INSTANCE;
     }
-    
+
     //----------------------------------------------------------< Principal >---
     public String getName() {
         return NAME;
