diff --git a/jackrabbit-spi-commons/src/main/java/org/apache/jackrabbit/spi/commons/query/xpath/XPathQueryBuilder.java b/jackrabbit-spi-commons/src/main/java/org/apache/jackrabbit/spi/commons/query/xpath/XPathQueryBuilder.java
index c086324..478f8d1 100644
--- a/jackrabbit-spi-commons/src/main/java/org/apache/jackrabbit/spi/commons/query/xpath/XPathQueryBuilder.java
+++ b/jackrabbit-spi-commons/src/main/java/org/apache/jackrabbit/spi/commons/query/xpath/XPathQueryBuilder.java
@@ -296,10 +296,10 @@ public class XPathQueryBuilder implements XPathVisitor, XPathTreeConstants {
             }
             query.jjtAccept(this, root);
         } catch (ParseException e) {
-            throw new InvalidQueryException(e.getMessage(), e);
+            throw new InvalidQueryException(e.getMessage() + " for statement: " + statement, e);
         } catch (Throwable t) {
             // also catch any other exception
-            throw new InvalidQueryException(t.getMessage(), t);
+            throw new InvalidQueryException(t.getMessage() + " for statement: " + statement, t);
         }
         if (exceptions.size() > 0) {
             // simply report the first one
