diff --git a/jackrabbit-api/src/main/java/org/apache/jackrabbit/api/security/user/User.java b/jackrabbit-api/src/main/java/org/apache/jackrabbit/api/security/user/User.java
index c5036ec..5d29cd5 100644
--- a/jackrabbit-api/src/main/java/org/apache/jackrabbit/api/security/user/User.java
+++ b/jackrabbit-api/src/main/java/org/apache/jackrabbit/api/security/user/User.java
@@ -15,9 +15,6 @@
  */
 package org.apache.jackrabbit.api.security.user;
 
-import org.apache.jackrabbit.api.security.user.Authorizable;
-import org.apache.jackrabbit.api.security.user.Impersonation;
-
 import javax.jcr.RepositoryException;
 import java.util.Iterator;
