diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/AbstractNamespaceMappings.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/AbstractNamespaceMappings.java
index 00a0e79..248d791 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/AbstractNamespaceMappings.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/AbstractNamespaceMappings.java
@@ -32,7 +32,7 @@ public abstract class AbstractNamespaceMappings
         implements NamespaceMappings, NamespaceResolver {
 
     /**
-     * The name resolver used to translate the qualified name to JCR name
+     * The name resolver used to translate the <code>Name</code>s to JCR name strings.
      */
     private final NamePathResolver resolver;
