diff --git a/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/nodetype/NodeTypeManagerImpl.java b/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/nodetype/NodeTypeManagerImpl.java
index 00adb5f..fa99730 100644
--- a/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/nodetype/NodeTypeManagerImpl.java
+++ b/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/nodetype/NodeTypeManagerImpl.java
@@ -336,7 +336,7 @@ public class NodeTypeManagerImpl implements NodeTypeManager, NodeTypeRegistryLis
             throw new NoSuchNodeTypeException(nodeTypeName, e);
         }
     }
-    
+
     //-----------------------------------------------------------< Dumpable >---
     /**
      * {@inheritDoc}
