diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/AbstractExcerpt.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/AbstractExcerpt.java
index f13edea..1f13ccb 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/AbstractExcerpt.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/AbstractExcerpt.java
@@ -50,7 +50,7 @@ public abstract class AbstractExcerpt implements HighlightingExcerptProvider {
     /**
      * Logger instance for this class.
      */
-    private static final Logger log = LoggerFactory.getLogger(DefaultXMLExcerpt.class);
+    private static final Logger log = LoggerFactory.getLogger(AbstractExcerpt.class);
 
     /**
      * The search index.
