diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/QueryRootNode.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/QueryRootNode.java
index f09e331..2241b32 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/QueryRootNode.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/QueryRootNode.java
@@ -88,7 +88,7 @@ public class QueryRootNode extends QueryNode {
     }
 
     /**
-     * Returns the order node.
+     * Returns the order node or <code>null</code> if no order is specified.
      *
      * @return the order node.
      */
