diff --git a/jackrabbit-jca/src/main/java/org/apache/jackrabbit/jca/JCAManagedConnection.java b/jackrabbit-jca/src/main/java/org/apache/jackrabbit/jca/JCAManagedConnection.java
index 5b459fc..66a80a6 100644
--- a/jackrabbit-jca/src/main/java/org/apache/jackrabbit/jca/JCAManagedConnection.java
+++ b/jackrabbit-jca/src/main/java/org/apache/jackrabbit/jca/JCAManagedConnection.java
@@ -16,8 +16,8 @@
  */
 package org.apache.jackrabbit.jca;
 
+import org.apache.jackrabbit.api.XASession;
 import org.apache.jackrabbit.core.RepositoryImpl;
-import org.apache.jackrabbit.core.XASession;
 
 import javax.jcr.Repository;
 import javax.jcr.RepositoryException;
