diff --git a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/util/name/NamespaceAdder.java b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/util/name/NamespaceAdder.java
index 1efddbd..524ff60 100644
--- a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/util/name/NamespaceAdder.java
+++ b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/util/name/NamespaceAdder.java
@@ -24,7 +24,7 @@ import java.util.Map;
 import java.util.Iterator;
 
 /**
- * @deprecated Use the NamespaceAdder class from 
+ * @deprecated Use the NamespaceAdder class from
  *             the org.apache.jackrabbit.spi.commons.namespace package of
  *             the jackrabbit-spi-commons component.
  */
