diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/Transformable.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/Transformable.java
index f88ac29..5e73cea 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/Transformable.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/Transformable.java
@@ -31,5 +31,6 @@ public interface Transformable extends TransformConstants {
      * </ul>
      * @param transformation a transform constant.
      */
-    public void setTransformation(int transformation);
+    void setTransformation(int transformation);
+
 }
