diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/observation/ObservationDispatcher.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/observation/ObservationDispatcher.java
index 073ff51..b4742c7 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/observation/ObservationDispatcher.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/observation/ObservationDispatcher.java
@@ -83,7 +83,7 @@ public final class ObservationDispatcher extends EventDispatcher
 
     /**
      * Creates a new <code>ObservationDispatcher</code> instance
-     * and starts the notification thread deamon.
+     * and starts the notification thread daemon.
      */
     public ObservationDispatcher() {
         notificationThread = new Thread(this, "ObservationManager");
