diff --git a/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/WorkspaceManager.java b/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/WorkspaceManager.java
index 4f072a5..d620610 100644
--- a/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/WorkspaceManager.java
+++ b/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/WorkspaceManager.java
@@ -911,7 +911,8 @@ public class WorkspaceManager implements UpdatableItemStateManager, NamespaceSto
          * @see OperationVisitor#visit(Checkin)
          */
         public void visit(Checkin operation) throws UnsupportedRepositoryOperationException, LockException, InvalidItemStateException, RepositoryException {
-            service.checkin(sessionInfo, operation.getNodeId());
+            NodeId newId = service.checkin(sessionInfo, operation.getNodeId());
+            operation.setNewVersionId(newId);
         }
 
         /**
