diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/qom/JoinConditionImpl.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/qom/JoinConditionImpl.java
index 6ab8e16..96baf17 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/qom/JoinConditionImpl.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/qom/JoinConditionImpl.java
@@ -16,7 +16,7 @@
  */
 package org.apache.jackrabbit.core.query.qom;
 
-import org.apache.jackrabbit.name.NamePathResolver;
+import org.apache.jackrabbit.conversion.NamePathResolver;
 
 import org.apache.jackrabbit.core.query.jsr283.qom.JoinCondition;
