diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/CachingIndexReader.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/CachingIndexReader.java
index 28def43..bee930d 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/CachingIndexReader.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/CachingIndexReader.java
@@ -112,7 +112,7 @@ class CachingIndexReader extends FilterIndexReader {
         }
 
         if (parent == null) {
-            Document doc = document(n);
+            Document doc = document(n, FieldSelectors.UUID_AND_PARENT);
             String parentUUID = doc.get(FieldNames.PARENT);
             if (parentUUID == null || parentUUID.length() == 0) {
                 parent = DocId.NULL;
