diff --git a/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/query/qom/TestAll.java b/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/query/qom/TestAll.java
index 7a1eaa3..d4abfb3 100644
--- a/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/query/qom/TestAll.java
+++ b/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/query/qom/TestAll.java
@@ -32,6 +32,7 @@ public class TestAll extends TestCase {
         suite.addTestSuite(BindVariableValueTest.class);
         suite.addTestSuite(ChildNodeTest.class);
         suite.addTestSuite(DescendantNodeTest.class);
+        suite.addTestSuite(LengthTest.class);
         suite.addTestSuite(NodeNameTest.class);
         suite.addTestSuite(PropertyExistenceTest.class);
         suite.addTestSuite(QueryObjectModelFactoryTest.class);
