diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/ChildAxisQuery.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/ChildAxisQuery.java
index 5d92741..24b08ec 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/ChildAxisQuery.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/ChildAxisQuery.java
@@ -264,7 +264,7 @@ class ChildAxisQuery extends Query implements JackrabbitQuery {
             throws IOException {
         if (sort.getSort().length == 0 && matchesAnyChildNode()) {
             Query context = getContextQuery();
-            return new ChildNodesQueryHits(searcher.evaluate(context, sort), session);
+            return new ChildNodesQueryHits(searcher.evaluate(context), session);
         } else {
             return null;
         }
