diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/QueryHandler.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/QueryHandler.java
index 957cb68..eb573ba 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/QueryHandler.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/QueryHandler.java
@@ -17,9 +17,8 @@
 package org.apache.jackrabbit.core.query;
 
 import org.apache.jackrabbit.core.ItemManager;
-import org.apache.jackrabbit.core.NodeIdIterator;
 import org.apache.jackrabbit.core.SessionImpl;
-import org.apache.jackrabbit.core.NodeId;
+import org.apache.jackrabbit.core.id.NodeId;
 import org.apache.jackrabbit.core.state.NodeState;
 import org.apache.jackrabbit.core.state.NodeStateIterator;
 import org.apache.jackrabbit.spi.commons.query.qom.QueryObjectModelTree;
@@ -28,6 +27,7 @@ import javax.jcr.RepositoryException;
 import javax.jcr.query.InvalidQueryException;
 
 import java.io.IOException;
+import java.util.Iterator;
 
 /**
  * Defines an interface for the actual node indexing and query execution.
@@ -80,7 +80,7 @@ public interface QueryHandler {
      * @throws RepositoryException if an error occurs while indexing a node.
      * @throws IOException if an error occurs while updating the index.
      */
-    void updateNodes(NodeIdIterator remove, NodeStateIterator add)
+    void updateNodes(Iterator<NodeId> remove, NodeStateIterator add)
             throws RepositoryException, IOException;
 
     /**
