diff --git a/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/version/ActivitiesTest.java b/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/version/ActivitiesTest.java
index f0f141f..891dea8 100644
--- a/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/version/ActivitiesTest.java
+++ b/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/version/ActivitiesTest.java
@@ -92,7 +92,7 @@ public class ActivitiesTest extends AbstractVersionTest {
         }
     }
 
-    public void ignore_testActivitiesPath() throws Exception {
+    public void testActivitiesPath() throws Exception {
         
         Node an = null;
