diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/RowIteratorImpl.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/RowIteratorImpl.java
index 2fcaa3b..ac38c0a 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/RowIteratorImpl.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/RowIteratorImpl.java
@@ -370,8 +370,8 @@ class RowIteratorImpl implements RowIterator {
          *         function, <code>false</code> otherwise.
          */
         private boolean isExcerptFunction(Name name) {
-            return name.getNamespaceURI().equals(Name.NS_REP_URI) &&
-                    name.getLocalName().startsWith(EXCERPT_FUNC_LPAR);
+            return name.getNamespaceURI().equals(Name.NS_REP_URI)
+                && name.getLocalName().startsWith(EXCERPT_FUNC_LPAR);
         }
 
         /**
@@ -487,8 +487,8 @@ class RowIteratorImpl implements RowIterator {
          *         function, <code>false</code> otherwise.
          */
         private boolean isSpellCheckFunction(Name name) {
-            return name.getNamespaceURI().equals(Name.NS_REP_URI) &&
-                    name.getLocalName().startsWith(SPELLCHECK_FUNC_LPAR);
+            return name.getNamespaceURI().equals(Name.NS_REP_URI)
+                && name.getLocalName().startsWith(SPELLCHECK_FUNC_LPAR);
         }
 
         /**
