diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/jsr283/qom/DescendantNode.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/jsr283/qom/DescendantNode.java
index 3362c39..a2fa399 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/jsr283/qom/DescendantNode.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/jsr283/qom/DescendantNode.java
@@ -40,19 +40,20 @@ package org.apache.jackrabbit.core.query.jsr283.qom;
  *
  * @since JCR 2.0
  */
-public interface DescendantNode
-        extends Constraint {
+public interface DescendantNode extends Constraint {
+
     /**
      * Gets the name of the selector against which to apply this constraint.
      *
      * @return the selector name; non-null
      */
-    public String getSelectorName();
+    String getSelectorName();
 
     /**
      * Gets the absolute path.
      *
      * @return the path; non-null
      */
-    public String getPath();
+    String getPath();
+
 }
