diff --git a/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/query/TestAll.java b/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/query/TestAll.java
index fa0f767..e552009 100644
--- a/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/query/TestAll.java
+++ b/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/query/TestAll.java
@@ -55,6 +55,7 @@ public class TestAll extends TestCase {
         suite.addTestSuite(SynonymProviderTest.class);
         suite.addTestSuite(ArrayHitsTest.class);
         suite.addTestSuite(ExcerptTest.class);
+        suite.addTestSuite(IndexingAggregateTest.class);
 
         // exclude long running tests per default
         //suite.addTestSuite(MassiveRangeTest.class);
