diff --git a/jackrabbit-spi-commons/src/main/java/org/apache/jackrabbit/spi/commons/query/qom/JoinType.java b/jackrabbit-spi-commons/src/main/java/org/apache/jackrabbit/spi/commons/query/qom/JoinType.java
index 125d1bd..40c6448 100644
--- a/jackrabbit-spi-commons/src/main/java/org/apache/jackrabbit/spi/commons/query/qom/JoinType.java
+++ b/jackrabbit-spi-commons/src/main/java/org/apache/jackrabbit/spi/commons/query/qom/JoinType.java
@@ -32,11 +32,9 @@ public enum JoinType {
 
     INNER(QueryObjectModelConstants.JCR_JOIN_TYPE_INNER),
 
-    LEFT("{http://www.jcp.org/jcr/1.0}joinTypeLeftOuter"),
-    // LEFT(QueryObjectModelConstants.JCR_JOIN_TYPE_LEFT_OUTER),
+    LEFT(QueryObjectModelConstants.JCR_JOIN_TYPE_LEFT_OUTER),
 
-    RIGHT("{http://www.jcp.org/jcr/1.0}joinTypeRightOuter");
-    // RIGHT(QueryObjectModelConstants.JCR_JOIN_TYPE_RIGHT_OUTER);
+    RIGHT(QueryObjectModelConstants.JCR_JOIN_TYPE_RIGHT_OUTER);
 
     /**
      * JCR name of this join type.
