diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/hits/HitsIntersection.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/hits/HitsIntersection.java
index 2be884e..e2e4f79 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/hits/HitsIntersection.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/hits/HitsIntersection.java
@@ -40,7 +40,7 @@ public class HitsIntersection implements Hits {
     public void set(int doc) {
         throw new UnsupportedOperationException();
     }
-    
+
     /**
      * {@inheritDoc}
      */
