diff --git a/jackrabbit-ocm/src/test/java/org/apache/jackrabbit/ocm/manager/jcrnodetype/DigesterJcrNodeTypeTest.java b/jackrabbit-ocm/src/test/java/org/apache/jackrabbit/ocm/manager/jcrnodetype/DigesterJcrNodeTypeTest.java
index 58de43f..3a354bc 100644
--- a/jackrabbit-ocm/src/test/java/org/apache/jackrabbit/ocm/manager/jcrnodetype/DigesterJcrNodeTypeTest.java
+++ b/jackrabbit-ocm/src/test/java/org/apache/jackrabbit/ocm/manager/jcrnodetype/DigesterJcrNodeTypeTest.java
@@ -56,10 +56,10 @@ public class DigesterJcrNodeTypeTest extends DigesterTestBase {
 	}
 
 
-	public void testRetrieveSingleton() 
+	public void testRetrieveSingleton()
 	{
 
-		try 
+		try
 		{
 			ObjectContentManager ocm = this.getObjectContentManager();
 
@@ -67,18 +67,18 @@ public class DigesterJcrNodeTypeTest extends DigesterTestBase {
 			// Insert a  folder (class mapped to jcr:folder) with one file (class mapped to jcr:file)
 			//---------------------------------------------------------------------------------------------------------			
             Resource resource = new Resource();
-            resource.setData(new ByteArrayInputStream("this is the content".getBytes()));            
+            resource.setData(new ByteArrayInputStream("this is the content".getBytes()));
             resource.setLastModified(Calendar.getInstance());
             resource.setMimeType("plain/text");
-            File file = new File();    
+            File file = new File();
             file.setResource(resource);
-            
-            
+
+
             Folder folder = new Folder();
             folder.setPath("/folder1");
             folder.addChild(file);
-            
-            ocm.insert(folder);            
+
+            ocm.insert(folder);
 			ocm.save();
