diff --git a/jackrabbit-spi/src/main/java/org/apache/jackrabbit/spi/Batch.java b/jackrabbit-spi/src/main/java/org/apache/jackrabbit/spi/Batch.java
index d3cf6cf..f5b4d45 100644
--- a/jackrabbit-spi/src/main/java/org/apache/jackrabbit/spi/Batch.java
+++ b/jackrabbit-spi/src/main/java/org/apache/jackrabbit/spi/Batch.java
@@ -108,7 +108,7 @@ public interface Batch {
      * 
      * @param parentId NodeId identifying the parent node.
      * @param propertyName Name of the property to be created.
-     * @param value The qualified value of the property to be created.
+     * @param value The value of the property to be created.
      * @throws ValueFormatException
      * @throws VersionException
      * @throws LockException
@@ -139,7 +139,7 @@ public interface Batch {
      *
      * @param parentId NodeId identifying the parent node.
      * @param propertyName Name of the property to be created.
-     * @param values The qualified values of the property to be created.
+     * @param values The values of the property to be created.
      * @throws javax.jcr.ValueFormatException
      * @throws javax.jcr.version.VersionException
      * @throws javax.jcr.lock.LockException
