diff --git a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/value/LongValue.java b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/value/LongValue.java
index 5a94f0e..b50df06 100644
--- a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/value/LongValue.java
+++ b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/value/LongValue.java
@@ -66,7 +66,7 @@ public class LongValue extends BaseValue {
         try {
             return new LongValue(Long.parseLong(s));
         } catch (NumberFormatException e) {
-            throw new ValueFormatException("invalid format", e);
+            throw new ValueFormatException("not a valid long format: " + s, e);
         }
     }
