diff --git a/jackrabbit-spi/src/main/java/org/apache/jackrabbit/spi/EventBundle.java b/jackrabbit-spi/src/main/java/org/apache/jackrabbit/spi/EventBundle.java
index 3eea7f8..0d46294 100644
--- a/jackrabbit-spi/src/main/java/org/apache/jackrabbit/spi/EventBundle.java
+++ b/jackrabbit-spi/src/main/java/org/apache/jackrabbit/spi/EventBundle.java
@@ -24,7 +24,7 @@ import java.util.Iterator;
  * <code>EventIterator</code> an <code>EventBundle</code> allows to retrieve
  * the events multiple times using the {@link #getEvents} method.
  */
-public interface EventBundle {
+public interface EventBundle extends Iterable<Event> {
 
     /**
      * Returns the events of this bundle.
