diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/xml/TargetImportHandler.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/xml/TargetImportHandler.java
index d9c97a2..bbe7103 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/xml/TargetImportHandler.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/xml/TargetImportHandler.java
@@ -18,8 +18,8 @@ package org.apache.jackrabbit.core.xml;
 
 import org.xml.sax.SAXException;
 import org.xml.sax.helpers.DefaultHandler;
-import org.apache.jackrabbit.conversion.NamePathResolver;
-import org.apache.jackrabbit.conversion.DefaultNamePathResolver;
+import org.apache.jackrabbit.spi.commons.conversion.NamePathResolver;
+import org.apache.jackrabbit.spi.commons.conversion.DefaultNamePathResolver;
 
 import java.util.Map;
