diff --git a/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/query/TextNodeTest.java b/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/query/TextNodeTest.java
index e3a1b82..d04e93d 100644
--- a/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/query/TextNodeTest.java
+++ b/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/query/TextNodeTest.java
@@ -59,8 +59,9 @@ public class TextNodeTest extends AbstractQueryTest {
 
     /**
      * Tests if text() node test is equivalent with jcr:xmltext.
+     * @throws NotExecutableException 
      */
-    public void testTextNodeTest() throws RepositoryException {
+    public void testTextNodeTest() throws RepositoryException, NotExecutableException {
         Node text1 = testRootNode.addNode(jcrXMLText);
         text1.setProperty(jcrXMLCharacters, "foo");
         testRootNode.save();
@@ -71,8 +72,9 @@ public class TextNodeTest extends AbstractQueryTest {
     /**
      * Tests if text() node test is equivalent with jcr:xmltext and will select
      * multiple nodes with name jcr:xmltext.
+     * @throws NotExecutableException 
      */
-    public void testTextNodeTestMultiNodes() throws RepositoryException {
+    public void testTextNodeTestMultiNodes() throws RepositoryException, NotExecutableException {
         Node text1 = testRootNode.addNode(jcrXMLText);
         text1.setProperty(jcrXMLCharacters, "foo");
         Node text2 = testRootNode.addNode(nodeName1, testNodeType).addNode(jcrXMLText);
@@ -85,8 +87,9 @@ public class TextNodeTest extends AbstractQueryTest {
     /**
      * Tests if text() node test is equivalent with jcr:xmltext and jcr:contains
      * matches content in jcr:xmlcharacters property.
+     * @throws NotExecutableException 
      */
-    public void testTextNodeTestContains() throws RepositoryException {
+    public void testTextNodeTestContains() throws RepositoryException, NotExecutableException {
         Node text1 = testRootNode.addNode(jcrXMLText);
         text1.setProperty(jcrXMLCharacters, "the quick brown fox jumps over the lazy dog.");
         Node text2 = testRootNode.addNode(nodeName1, testNodeType).addNode(jcrXMLText);
