diff --git a/jackrabbit-spi-commons/src/main/java/org/apache/jackrabbit/spi/commons/query/qom/NotImpl.java b/jackrabbit-spi-commons/src/main/java/org/apache/jackrabbit/spi/commons/query/qom/NotImpl.java
index 9d89078..446c8cb 100644
--- a/jackrabbit-spi-commons/src/main/java/org/apache/jackrabbit/spi/commons/query/qom/NotImpl.java
+++ b/jackrabbit-spi-commons/src/main/java/org/apache/jackrabbit/spi/commons/query/qom/NotImpl.java
@@ -17,11 +17,10 @@
 package org.apache.jackrabbit.spi.commons.query.qom;
 
 import javax.jcr.query.qom.Constraint;
+import javax.jcr.query.qom.Not;
 
 import org.apache.jackrabbit.spi.commons.conversion.NamePathResolver;
 
-import org.apache.jackrabbit.spi.commons.query.jsr283.qom.Not;
-
 /**
  * <code>NotImpl</code>...
  */
