diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/persistence/bundle/util/BundleBinding.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/persistence/bundle/util/BundleBinding.java
index 32382af..4b69789 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/persistence/bundle/util/BundleBinding.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/persistence/bundle/util/BundleBinding.java
@@ -120,6 +120,7 @@ public class BundleBinding extends ItemStateBinding {
             if (name.equals(NameConstants.JCR_PRIMARYTYPE)
                 || name.equals(NameConstants.JCR_MIXINTYPES)
                 || name.equals(NameConstants.JCR_UUID)) {
+                name = readIndexedQName(in);
                 continue;
             }
             PropertyId pId = new PropertyId(bundle.getId(), name);
