diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/jsr283/qom/PropertyValue.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/jsr283/qom/PropertyValue.java
index 612d2e3..d79bb7e 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/jsr283/qom/PropertyValue.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/jsr283/qom/PropertyValue.java
@@ -33,19 +33,20 @@ package org.apache.jackrabbit.core.query.jsr283.qom;
  *
  * @since JCR 2.0
  */
-public interface PropertyValue
-        extends DynamicOperand {
+public interface PropertyValue extends DynamicOperand {
+
     /**
      * Gets the name of the selector against which to evaluate this operand.
      *
      * @return the selector name; non-null
      */
-    public String getSelectorName();
+    String getSelectorName();
 
     /**
      * Gets the name of the property.
      *
      * @return the property name; non-null
      */
-    public String getPropertyName();
+    String getPropertyName();
+
 }
