diff --git a/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/AddNodeTest.java b/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/AddNodeTest.java
index 1eed05c..db9d95c 100644
--- a/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/AddNodeTest.java
+++ b/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/AddNodeTest.java
@@ -211,7 +211,7 @@ public class AddNodeTest extends AbstractJCRTest {
         defaultRootNode.save();
 
         // use a different session to verify if the node is there
-        Session session = helper.getReadOnlySession();
+        Session session = getHelper().getReadOnlySession();
         try {
             session.getItem(testNode.getPath());
         } finally {
@@ -235,7 +235,7 @@ public class AddNodeTest extends AbstractJCRTest {
         superuser.save();
 
         // use a different session to verify if the node is there
-        Session session = helper.getReadOnlySession();
+        Session session = getHelper().getReadOnlySession();
         try {
             session.getItem(testNode.getPath());
         } finally {
