diff --git a/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/ItemManagerImpl.java b/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/ItemManagerImpl.java
index c164fa3..33bded3 100644
--- a/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/ItemManagerImpl.java
+++ b/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/ItemManagerImpl.java
@@ -200,7 +200,7 @@ public class ItemManagerImpl implements Dumpable, ItemManager, ItemStateCreation
         session.checkIsAlive();
         ItemState state = hierarchyEntry.getItemState();
         if (!state.isValid()) {
-            throw new ItemNotFoundException();
+            throw new ItemNotFoundException(LogUtil.safeGetJCRPath(state, session.getPathResolver()));
         }
 
         // first try to access item from cache
