diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/fs/mem/MemoryFileSystemEntry.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/fs/mem/MemoryFileSystemEntry.java
index d52811e..5053093 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/fs/mem/MemoryFileSystemEntry.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/fs/mem/MemoryFileSystemEntry.java
@@ -30,8 +30,4 @@ public abstract class MemoryFileSystemEntry {
         return lastModified;
     }
 
-    public void touch() {
-        this.lastModified = System.currentTimeMillis();
-    }
-
 }
