diff --git a/contrib/spi/client/src/test/java/org/apache/jackrabbit/jcr2spi/WorkspaceMoveTest.java b/contrib/spi/client/src/test/java/org/apache/jackrabbit/jcr2spi/WorkspaceMoveTest.java
index cbe1c7d..515ad37 100644
--- a/contrib/spi/client/src/test/java/org/apache/jackrabbit/jcr2spi/WorkspaceMoveTest.java
+++ b/contrib/spi/client/src/test/java/org/apache/jackrabbit/jcr2spi/WorkspaceMoveTest.java
@@ -66,7 +66,7 @@ public class WorkspaceMoveTest extends MoveTest {
         }
     }
 
-    public void testMoveTransientPropertyExistsException() throws RepositoryException, NotExecutableException {
+    public void testMoveTransientPropertyExists() throws RepositoryException, NotExecutableException {
         // try to create a property with the name of the node to be moved
         // to the destination parent
         Property destProperty;
@@ -76,12 +76,14 @@ public class WorkspaceMoveTest extends MoveTest {
             throw new NotExecutableException("Cannot create property with name '" +nodeName2+ "' and value 'anyString' at move destination.");
         }
 
+        // workspace-move the node (must succeed)
+        doMove(moveNode.getPath(), destProperty.getPath());
         try {
-            // move the node
-            doMove(moveNode.getPath(), destProperty.getPath());
-            fail("Moving a node to a location where a property exists must throw ItemExistsException");
-        } catch (ItemExistsException e) {
-            // ok, works as expected
+            // saving transient new property must fail
+            destParentNode.save();
+            fail("Saving new transient property must fail");
+        } catch (RepositoryException e) {
+            // ok.
         }
     }
 }
\ No newline at end of file
