diff --git a/contrib/jackrabbit-jcr-mapping/jcr-mapping/src/test/java/org/apache/jackrabbit/ocm/testmodel/Residual.java b/contrib/jackrabbit-jcr-mapping/jcr-mapping/src/test/java/org/apache/jackrabbit/ocm/testmodel/Residual.java
index 03ff8ab..17513d3 100644
--- a/contrib/jackrabbit-jcr-mapping/jcr-mapping/src/test/java/org/apache/jackrabbit/ocm/testmodel/Residual.java
+++ b/contrib/jackrabbit-jcr-mapping/jcr-mapping/src/test/java/org/apache/jackrabbit/ocm/testmodel/Residual.java
@@ -17,29 +17,28 @@
 package org.apache.jackrabbit.ocm.testmodel;
 
 import org.apache.jackrabbit.ocm.manager.collectionconverter.impl.ManagedHashMap;
+import org.apache.jackrabbit.ocm.manager.collectionconverter.impl.ResidualNodesCollectionConverterImpl;
+import org.apache.jackrabbit.ocm.manager.collectionconverter.impl.ResidualPropertiesCollectionConverterImpl;
+import org.apache.jackrabbit.ocm.mapper.impl.annotation.Collection;
+import org.apache.jackrabbit.ocm.mapper.impl.annotation.Field;
+import org.apache.jackrabbit.ocm.mapper.impl.annotation.Node;
 
 /**
  *
  * @author <a href="mailto:fmeschbe[at]apache[dot]com">Felix Meschberger</a>
  * 
+ * Note : 
+ * If the annotation are defined on the field declaration, it is mandatory to register the ancestor classes.
+ * Otherwise, the annotation mapper will not map the fields defined in the ancester classes. 
+ * 
  */
+@Node
 public class Residual
 {
-	private String path;
-    private ManagedHashMap elements;
-    
-    public static class ResidualProperties extends Residual {}
-    public static class ResidualNodes extends Residual {}
-    
-    protected Residual() {}
+	@Field(path=true) private String path;
     
+    private ManagedHashMap elements;
     
-    public String getPath() {
-		return path;
-	}
-	public void setPath(String path) {
-		this.path = path;
-	}
 	/**
      * @return Returns the elements.
      */
@@ -54,4 +53,34 @@ public class Residual
     {
         this.elements = elements;
     }
+
+    
+    @Node(extend=Residual.class) public static class ResidualProperties extends Residual 
+    {
+        @Collection( jcrName="value*",elementClassName=String.class,collectionConverter=ResidualPropertiesCollectionConverterImpl.class, 
+                collectionClassName=ManagedHashMap.class)
+        private ManagedHashMap elements;        
+        
+        
+    }
+    
+    @Node(extend=Residual.class) public static class ResidualNodes extends Residual 
+    {
+    	
+        @Collection( jcrName="value*",elementClassName=Object.class,collectionConverter=ResidualNodesCollectionConverterImpl.class, 
+                collectionClassName=ManagedHashMap.class)
+        private ManagedHashMap elements;
+            
+    	
+    }
+    
+    protected Residual() {}
+    
+    
+    public String getPath() {
+		return path;
+	}
+	public void setPath(String path) {
+		this.path = path;
+	}
 }
