diff --git a/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/observation/ObservationManagerImpl.java b/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/observation/ObservationManagerImpl.java
index 040cd13..e990c41 100644
--- a/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/observation/ObservationManagerImpl.java
+++ b/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/observation/ObservationManagerImpl.java
@@ -185,6 +185,13 @@ public class ObservationManagerImpl implements ObservationManager, InternalEvent
         throw new UnsupportedRepositoryOperationException("JCR-2108");
     }
 
+    public EventJournal getEventJournal(
+            int eventTypes, String absPath, boolean isDeep,
+            String[] uuid, String[] nodeTypeName, boolean noLocal)
+            throws RepositoryException {
+        throw new UnsupportedRepositoryOperationException("JCR-2108");
+    }
+
     /**
      * @see javax.jcr.observation.ObservationManager#setUserData(String) 
      */
