diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/PropertyImpl.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/PropertyImpl.java
index 96dbb9e..e2f42ac 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/PropertyImpl.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/PropertyImpl.java
@@ -754,10 +754,16 @@ public class PropertyImpl extends ItemImpl implements Property {
         internalSetValue(internalValues, reqType);
     }
 
+    /**
+     * {@inheritDoc}
+     */
     public long getLength() throws RepositoryException {
         return getLength(internalGetValue());
     }
 
+    /**
+     * {@inheritDoc}
+     */
     public long[] getLengths() throws RepositoryException {
         InternalValue[] values = internalGetValues();
         long[] lengths = new long[values.length];
@@ -777,10 +783,23 @@ public class PropertyImpl extends ItemImpl implements Property {
         return data.getPropertyDefinition();
     }
 
+    /**
+     * {@inheritDoc}
+     */
     public int getType() throws RepositoryException {
         return getPropertyState().getType();
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public boolean isMultiple() throws RepositoryException {
+        // check state of this instance
+        sanityCheck();
+
+        return data.getPropertyDefinition().isMultiple();
+    }
+
     //-----------------------------------------------------------------< Item >
     /**
      * {@inheritDoc}
