diff --git a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/value/DoubleValue.java b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/value/DoubleValue.java
index 37369ea..e178cb3 100644
--- a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/value/DoubleValue.java
+++ b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/value/DoubleValue.java
@@ -66,7 +66,7 @@ public class DoubleValue extends BaseValue {
         try {
             return new DoubleValue(Double.parseDouble(s));
         } catch (NumberFormatException e) {
-            throw new ValueFormatException("invalid format", e);
+            throw new ValueFormatException("not a valid double format: " + s, e);
         }
     }
