diff --git a/jackrabbit-jcr-server/src/main/java/org/apache/jackrabbit/server/jcr/JCRWebdavServer.java b/jackrabbit-jcr-server/src/main/java/org/apache/jackrabbit/server/jcr/JCRWebdavServer.java
index 6a86873..87967f0 100644
--- a/jackrabbit-jcr-server/src/main/java/org/apache/jackrabbit/server/jcr/JCRWebdavServer.java
+++ b/jackrabbit-jcr-server/src/main/java/org/apache/jackrabbit/server/jcr/JCRWebdavServer.java
@@ -126,7 +126,7 @@ public class JCRWebdavServer implements DavSessionProvider {
          * Removes the reference from this <code>DavSession</code>. If no
          * more references are present, this <code>DavSession</code> is removed
          * from the internal cache and the underlying session is released by
-         * calling {@link SessionProvider#releaseSession(javax.jcr.Session)} 
+         * calling {@link SessionProvider#releaseSession(javax.jcr.Session)}
          *
          * @see DavSession#removeReference(Object)
          */
