diff --git a/contrib/jackrabbit-jcr-mapping/jcr-mapping/src/test/java/org/apache/jackrabbit/ocm/AnnotationTestBase.java b/contrib/jackrabbit-jcr-mapping/jcr-mapping/src/test/java/org/apache/jackrabbit/ocm/AnnotationTestBase.java
index 9cbf74c..0931bda 100644
--- a/contrib/jackrabbit-jcr-mapping/jcr-mapping/src/test/java/org/apache/jackrabbit/ocm/AnnotationTestBase.java
+++ b/contrib/jackrabbit-jcr-mapping/jcr-mapping/src/test/java/org/apache/jackrabbit/ocm/AnnotationTestBase.java
@@ -26,8 +26,20 @@ import org.apache.jackrabbit.ocm.manager.impl.ObjectContentManagerImpl;
 import org.apache.jackrabbit.ocm.mapper.Mapper;
 import org.apache.jackrabbit.ocm.mapper.impl.annotation.AnnotationMapperImpl;
 import org.apache.jackrabbit.ocm.repository.RepositoryUtil;
+import org.apache.jackrabbit.ocm.testmodel.A;
 import org.apache.jackrabbit.ocm.testmodel.Atomic;
+import org.apache.jackrabbit.ocm.testmodel.B;
+import org.apache.jackrabbit.ocm.testmodel.C;
+import org.apache.jackrabbit.ocm.testmodel.D;
+import org.apache.jackrabbit.ocm.testmodel.DFull;
 import org.apache.jackrabbit.ocm.testmodel.Default;
+import org.apache.jackrabbit.ocm.testmodel.E;
+import org.apache.jackrabbit.ocm.testmodel.MultiValue;
+import org.apache.jackrabbit.ocm.testmodel.Page;
+import org.apache.jackrabbit.ocm.testmodel.Paragraph;
+import org.apache.jackrabbit.ocm.testmodel.Residual;
+import org.apache.jackrabbit.ocm.testmodel.Residual.ResidualNodes;
+import org.apache.jackrabbit.ocm.testmodel.Residual.ResidualProperties;
 import org.apache.jackrabbit.ocm.testmodel.auto.CmsObject;
 import org.apache.jackrabbit.ocm.testmodel.auto.Content;
 import org.apache.jackrabbit.ocm.testmodel.auto.Document;
@@ -36,6 +48,8 @@ import org.apache.jackrabbit.ocm.testmodel.auto.impl.CmsObjectImpl;
 import org.apache.jackrabbit.ocm.testmodel.auto.impl.ContentImpl;
 import org.apache.jackrabbit.ocm.testmodel.auto.impl.DocumentImpl;
 import org.apache.jackrabbit.ocm.testmodel.auto.impl.FolderImpl;
+import org.apache.jackrabbit.ocm.testmodel.collection.Element;
+import org.apache.jackrabbit.ocm.testmodel.collection.Main;
 
 /**
  * Base class for testcases. Provides priviledged access to the jcr test
@@ -68,9 +82,26 @@ public abstract class AnnotationTestBase extends AbstractTestBase
 		Repository repository = RepositoryUtil.getRepository("repositoryTest");	
 		session = RepositoryUtil.login(repository, "superuser", "superuser");
 		List<Class> classes = new ArrayList<Class>();
+		
+		// Register content classes used by the unit tests
 		classes.add(Atomic.class);
 		classes.add(Default.class);
+		classes.add(A.class);
+		classes.add(B.class);
+		classes.add(C.class);
+		classes.add(D.class);
+		classes.add(DFull.class);
+		classes.add(E.class);
+		classes.add(Page.class);
+		classes.add(Paragraph.class);
+		classes.add(Main.class);
+		classes.add(Element.class);
+		classes.add(MultiValue.class);
 		
+
+		classes.add(Residual.class); 
+		classes.add(ResidualProperties.class);
+		classes.add(ResidualNodes.class);
 		Mapper mapper = new AnnotationMapperImpl(classes);
 		ocm = new ObjectContentManagerImpl(session, mapper);
