diff --git a/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/data/TestAll.java b/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/data/TestAll.java
index 7e18065..1899782 100644
--- a/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/data/TestAll.java
+++ b/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/data/TestAll.java
@@ -40,6 +40,7 @@ public class TestAll extends TestCase {
         suite.addTestSuite(PersistenceManagerIteratorTest.class);
         suite.addTestSuite(CopyValueTest.class);
         suite.addTestSuite(TestTwoGetStreams.class);
+        suite.addTestSuite(TempFileInputStreamTest.class);
         return suite;
     }
