diff --git a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/commons/AbstractSession.java b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/commons/AbstractSession.java
index a724a4f..f36c91a 100644
--- a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/commons/AbstractSession.java
+++ b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/commons/AbstractSession.java
@@ -21,6 +21,7 @@ import java.io.InputStream;
 import java.io.OutputStream;
 
 import javax.jcr.Credentials;
+import javax.jcr.InvalidSerializedDataException;
 import javax.jcr.Item;
 import javax.jcr.Node;
 import javax.jcr.PathNotFoundException;
@@ -116,11 +117,13 @@ public abstract class AbstractSession implements Session {
      * @param in input stream to be parsed as XML and imported
      * @param uuidBehavior passed through
      * @throws IOException if an I/O error occurs
-     * @throws RepositoryException if another error occurs
+     * @throws InvalidSerializedDataException if an XML parsing error occurs
+     * @throws RepositoryException if a repository error occurs
      */
     public void importXML(
             String parentAbsPath, InputStream in, int uuidBehavior)
-            throws IOException, RepositoryException {
+            throws IOException, InvalidSerializedDataException,
+            RepositoryException {
         try {
             ContentHandler handler =
                 getImportContentHandler(parentAbsPath, uuidBehavior);
@@ -132,7 +135,7 @@ public abstract class AbstractSession implements Session {
             } else if (exception instanceof IOException) {
                 throw (IOException) exception;
             } else {
-                throw new RepositoryException("XML import failed", e);
+                throw new InvalidSerializedDataException("XML parse error", e);
             }
         }
     }
