diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/journal/FileRevision.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/journal/FileRevision.java
index b0b1f62..2b3305f 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/journal/FileRevision.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/journal/FileRevision.java
@@ -16,8 +16,8 @@
  */
 package org.apache.jackrabbit.core.journal;
 
-import org.slf4j.LoggerFactory;
 import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.io.File;
 import java.io.IOException;
