diff --git a/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/observation/FilteredEventIterator.java b/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/observation/FilteredEventIterator.java
index d261f65..6035d40 100644
--- a/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/observation/FilteredEventIterator.java
+++ b/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/observation/FilteredEventIterator.java
@@ -113,14 +113,6 @@ class FilteredEventIterator implements EventIterator {
     }
 
     /**
-     * @see javax.jcr.observation.EventIterator#getDate()
-     */
-    public long getDate() {
-        // TODO
-        throw new UnsupportedOperationException("JCR-2108. Implementation Missing");
-    }
-
-    /**
      * {@inheritDoc}
      */
     public void skip(long skipNum) {
