diff --git a/jackrabbit-ocm/src/test/java/org/apache/jackrabbit/ocm/testmodel/inheritance/impl/DocumentStream.java b/jackrabbit-ocm/src/test/java/org/apache/jackrabbit/ocm/testmodel/inheritance/impl/DocumentStream.java
index cbdd3ca..40a1867 100644
--- a/jackrabbit-ocm/src/test/java/org/apache/jackrabbit/ocm/testmodel/inheritance/impl/DocumentStream.java
+++ b/jackrabbit-ocm/src/test/java/org/apache/jackrabbit/ocm/testmodel/inheritance/impl/DocumentStream.java
@@ -31,17 +31,17 @@ import org.apache.jackrabbit.ocm.mapper.impl.annotation.Node;
  * Document Steam
  *
  * @author <a href="mailto:christophe.lombart@sword-technologies.com">Christophe Lombart</a>
- * 
+ *
  */
 @Node(jcrType="ocm:documentstream", discriminator=false)
-public class DocumentStream 
+public class DocumentStream
 {
     protected final static Log log = LogFactory.getLog(DocumentStream.class);
-      
+
     @Field(jcrName="ocm:binarycontent") protected byte[] content;
-    
+
     @Field(jcrName="ocm:encoding") protected String encoding;
-    
+
     @Field(path=true) protected String path;
 
 
@@ -50,7 +50,7 @@ public class DocumentStream
      */
     public InputStream getContentStream()
     {
-        return new ByteArrayInputStream(content);        
+        return new ByteArrayInputStream(content);
     }
 
     /**
@@ -58,7 +58,7 @@ public class DocumentStream
      */
     public byte[] getContent()
     {
-        
+
         return content;
     }
 
@@ -68,11 +68,11 @@ public class DocumentStream
      */
     public void setContent(byte[] stream)
     {
-        
+
         content = stream;
 
     }
-        
+
     /**
      * @return Returns the encoding.
      */
@@ -80,7 +80,7 @@ public class DocumentStream
     {
         return encoding;
     }
-    
+
     /**
      * @param encoding The encoding to set.
      */
@@ -96,8 +96,8 @@ public class DocumentStream
 	public void setPath(String path) {
 		this.path = path;
 	}
-    
-    
+
+
 }
