diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/nodetype/ItemDef.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/nodetype/ItemDef.java
index fa42e39..994df70 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/nodetype/ItemDef.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/nodetype/ItemDef.java
@@ -17,7 +17,7 @@
 package org.apache.jackrabbit.core.nodetype;
 
 import org.apache.jackrabbit.spi.Name;
-import org.apache.jackrabbit.name.NameConstants;
+import org.apache.jackrabbit.spi.commons.name.NameConstants;
 
 /**
  * <code>ItemDef</code> is the internal representation of
