diff --git a/jackrabbit-ocm/src/test/java/org/apache/jackrabbit/ocm/testmodel/version/Author.java b/jackrabbit-ocm/src/test/java/org/apache/jackrabbit/ocm/testmodel/version/Author.java
new file mode 100644
index 0000000..81f2a28
--- /dev/null
+++ b/jackrabbit-ocm/src/test/java/org/apache/jackrabbit/ocm/testmodel/version/Author.java
@@ -0,0 +1,28 @@
+package org.apache.jackrabbit.ocm.testmodel.version;
+
+import org.apache.jackrabbit.ocm.mapper.impl.annotation.Field;
+import org.apache.jackrabbit.ocm.mapper.impl.annotation.Node;
+
+@Node(jcrMixinTypes = "mix:versionable")
+public class Author {
+
+	//@Field(path = true) String path;
+	@Field String name;
+
+	public String getName() {
+		return name;
+	}
+
+	public void setName(String name) {
+		this.name = name;
+	}
+
+//	public String getPath() {
+//		return path;
+//	}
+//
+//	public void setPath(String path) {
+//		this.path = path;
+//	}
+
+}
\ No newline at end of file
