diff --git a/contrib/jackrabbit-jcr-mapping/jcr-mapping/src/main/java/org/apache/jackrabbit/ocm/mapper/impl/digester/DigesterDescriptorReader.java b/contrib/jackrabbit-jcr-mapping/jcr-mapping/src/main/java/org/apache/jackrabbit/ocm/mapper/impl/digester/DigesterDescriptorReader.java
index ff5ab48..b749ca1 100644
--- a/contrib/jackrabbit-jcr-mapping/jcr-mapping/src/main/java/org/apache/jackrabbit/ocm/mapper/impl/digester/DigesterDescriptorReader.java
+++ b/contrib/jackrabbit-jcr-mapping/jcr-mapping/src/main/java/org/apache/jackrabbit/ocm/mapper/impl/digester/DigesterDescriptorReader.java
@@ -117,6 +117,7 @@ public class DigesterDescriptorReader implements DescriptorReader
 			for (Iterator iter = configFileStreams.iterator(); iter.hasNext();) 
 			{
 				InputStream xmlMappingDescriptorFile = (InputStream) iter.next();
+				
 				Digester digester = new Digester();
 				digester.setValidating(this.validating);
 				if (null != this.dtdResolver)
