diff --git a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/commons/predicate/NtFilePredicate.java b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/commons/predicate/NtFilePredicate.java
index f484177..d065e14 100644
--- a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/commons/predicate/NtFilePredicate.java
+++ b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/commons/predicate/NtFilePredicate.java
@@ -75,7 +75,7 @@ public class NtFilePredicate implements Predicate {
 
     /**
      * @return <code>true</code> if the item is a nt:file or nt:resource property
-     * @see org.apache.commons.collections.Predicate#evaluate(java.lang.Object)
+     * @see org.apache.jackrabbit.commons.predicate.Predicate#evaluate(java.lang.Object)
      */
     public boolean evaluate(Object item) {
         if ( item instanceof Item ) {
