diff --git a/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/AbstractJCRTest.java b/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/AbstractJCRTest.java
index 8ee040a..ae2be83 100644
--- a/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/AbstractJCRTest.java
+++ b/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/AbstractJCRTest.java
@@ -667,4 +667,15 @@ public abstract class AbstractJCRTest extends JUnitTest {
         s.save();
         return testRootNode;
     }
+
+    /**
+     * @return the test scale value.
+     */
+    protected static int getTestScale() {
+        try {
+            return Integer.parseInt(System.getProperty("jackrabbit.test.scale", "1"));
+        } catch (NumberFormatException e) {
+            return 1;
+        }
+    }
 }
