diff --git a/jackrabbit-spi-commons/src/main/java/org/apache/jackrabbit/spi/commons/value/QValueValue.java b/jackrabbit-spi-commons/src/main/java/org/apache/jackrabbit/spi/commons/value/QValueValue.java
index 870aab0..917d912 100644
--- a/jackrabbit-spi-commons/src/main/java/org/apache/jackrabbit/spi/commons/value/QValueValue.java
+++ b/jackrabbit-spi-commons/src/main/java/org/apache/jackrabbit/spi/commons/value/QValueValue.java
@@ -164,11 +164,11 @@ public final class QValueValue implements Value {
     }
 
     private static final short STATE_UNDEFINED = 0;
-  
+
     private static final short STATE_VALUE_CONSUMED = 1;
-  
+
     private static final short STATE_STREAM_CONSUMED = 2;
-  
+
     private short state = STATE_UNDEFINED;
 
     /**
