diff --git a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/commons/xml/SystemViewExporter.java b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/commons/xml/SystemViewExporter.java
index 4eb293f..ecefc26 100644
--- a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/commons/xml/SystemViewExporter.java
+++ b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/commons/xml/SystemViewExporter.java
@@ -161,7 +161,7 @@ public class SystemViewExporter extends Exporter {
             String string = value.getString();
             for (int i = 0; i < string.length(); i++) {
                 char c = string.charAt(i);
-                if (c >= 0 && c < 32 && c != '\r' && c != '\n' && c != '\t') {
+                if (c >= 0 && c < 32 && c != '\n' && c != '\t') {
                     return true;
                 }
             }
