diff --git a/jackrabbit-jcr-client/src/test/java/org/apache/jackrabbit/client/RepositoryFactoryImplTest.java b/jackrabbit-jcr-client/src/test/java/org/apache/jackrabbit/client/RepositoryFactoryImplTest.java
index 9c7e7ef..ea67ee5 100644
--- a/jackrabbit-jcr-client/src/test/java/org/apache/jackrabbit/client/RepositoryFactoryImplTest.java
+++ b/jackrabbit-jcr-client/src/test/java/org/apache/jackrabbit/client/RepositoryFactoryImplTest.java
@@ -162,7 +162,7 @@ public class RepositoryFactoryImplTest extends TestCase {
             return null;
         }
 
-        public Map getRepositoryDescriptors() throws RepositoryException {
+        public Map<String, QValue[]> getRepositoryDescriptors() throws RepositoryException {
             return null;
         }
