diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/xml/NamespaceContext.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/xml/NamespaceContext.java
index c42fbae..31e324d 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/xml/NamespaceContext.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/xml/NamespaceContext.java
@@ -20,7 +20,6 @@ import org.apache.jackrabbit.spi.commons.namespace.NamespaceResolver;
 
 import javax.jcr.NamespaceException;
 import java.util.HashMap;
-import java.util.Iterator;
 import java.util.Map;
 
 /**
@@ -44,12 +43,12 @@ class NamespaceContext implements NamespaceResolver {
     /**
      * The namespace prefix to namespace URI mapping.
      */
-    private final Map prefixToURI;
+    private final Map<String, String> prefixToURI;
 
     /**
      * The namespace URI to namespace prefix mapping.
      */
-    private final Map uriToPrefix;
+    private final Map<String, String> uriToPrefix;
 
     /**
      * Creates a NamespaceContext instance with the given parent context
@@ -58,14 +57,12 @@ class NamespaceContext implements NamespaceResolver {
      * @param parent parent context
      * @param mappings local namespace mappings (prefix -> URI)
      */
-    public NamespaceContext(NamespaceContext parent, Map mappings) {
+    public NamespaceContext(NamespaceContext parent, Map<String, String> mappings) {
         this.parent = parent;
-        this.prefixToURI = new HashMap();
-        this.uriToPrefix = new HashMap();
+        this.prefixToURI = new HashMap<String, String>();
+        this.uriToPrefix = new HashMap<String, String>();
 
-        Iterator iterator = mappings.entrySet().iterator();
-        while (iterator.hasNext()) {
-            Map.Entry mapping = (Map.Entry) iterator.next();
+        for (Map.Entry<String, String> mapping : mappings.entrySet()) {
             prefixToURI.put(mapping.getKey(), mapping.getValue());
             uriToPrefix.put(mapping.getValue(), mapping.getKey());
         }
