diff --git a/jackrabbit-webdav/src/main/java/org/apache/jackrabbit/webdav/transaction/TransactionInfo.java b/jackrabbit-webdav/src/main/java/org/apache/jackrabbit/webdav/transaction/TransactionInfo.java
index 4f1569e..862c2df 100644
--- a/jackrabbit-webdav/src/main/java/org/apache/jackrabbit/webdav/transaction/TransactionInfo.java
+++ b/jackrabbit-webdav/src/main/java/org/apache/jackrabbit/webdav/transaction/TransactionInfo.java
@@ -74,7 +74,7 @@ public class TransactionInfo implements TransactionConstants, XmlSerializable {
         } else {
             log.warn("transactionInfo must contain a single 'transactionstatus' element.");
             throw new DavException(DavServletResponse.SC_BAD_REQUEST);
-        }     
+        }
     }
 
     /**
