diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/fs/db/DatabaseFileSystem.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/fs/db/DatabaseFileSystem.java
index a93cc23..a04920c 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/fs/db/DatabaseFileSystem.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/fs/db/DatabaseFileSystem.java
@@ -740,8 +740,14 @@ public class DatabaseFileSystem implements FileSystem {
 
             return new FilterOutputStream(new FileOutputStream(tmpFile)) {
 
+                public void write(byte[] bytes, int off, int len) throws IOException {
+                    out.write(bytes, off, len);
+                }
+
                 public void close() throws IOException {
-                    super.close();
+                    out.flush();
+                    ((FileOutputStream) out).getFD().sync();
+                    out.close();
 
                     InputStream in = null;
                     try {
