diff --git a/jackrabbit-core/src/test/java/org/apache/jackrabbit/api/jsr283/retention/TestAll.java b/jackrabbit-core/src/test/java/org/apache/jackrabbit/api/jsr283/retention/TestAll.java
index 1fa01b5..725408b 100644
--- a/jackrabbit-core/src/test/java/org/apache/jackrabbit/api/jsr283/retention/TestAll.java
+++ b/jackrabbit-core/src/test/java/org/apache/jackrabbit/api/jsr283/retention/TestAll.java
@@ -14,7 +14,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package javax.jcr.retention;
+package org.apache.jackrabbit.api.jsr283.retention;
 
 import junit.framework.Test;
 import junit.framework.TestCase;
