diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/SpellChecker.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/SpellChecker.java
index 507ddf4..d6ff178 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/SpellChecker.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/SpellChecker.java
@@ -17,7 +17,7 @@
 package org.apache.jackrabbit.core.query.lucene;
 
 import org.apache.jackrabbit.core.query.QueryHandler;
-import org.apache.jackrabbit.core.query.QueryRootNode;
+import org.apache.jackrabbit.spi.commons.query.QueryRootNode;
 
 import java.io.IOException;
