diff --git a/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/data/GarbageCollectorTest.java b/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/data/GarbageCollectorTest.java
index 5c1cc83..515f308 100644
--- a/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/data/GarbageCollectorTest.java
+++ b/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/data/GarbageCollectorTest.java
@@ -151,7 +151,7 @@ public class GarbageCollectorTest extends AbstractJCRTest implements ScanEventLi
     }
 
     public void afterScanning(Node n) throws RepositoryException {
-        if (n.getPath().startsWith("/testroot/node")) {
+        if (n != null && n.getPath().startsWith("/testroot/node")) {
             String path = n.getPath();
             LOG.debug("scanned: " + path);
         }
@@ -166,7 +166,7 @@ public class GarbageCollectorTest extends AbstractJCRTest implements ScanEventLi
     }
 
     public void beforeScanning(Node n) throws RepositoryException {
-        if (n.getPath().equals("/testroot/node2")) {
+        if (n != null && n.getPath().equals("/testroot/node2")) {
             Session session = n.getSession();
             list(session.getRootNode());
             session.move("/testroot/node2/nodeWithBlob", "/testroot/node1/nodeWithBlob");
