diff --git a/jackrabbit-jcr-server/src/main/java/org/apache/jackrabbit/webdav/jcr/version/report/RepositoryDescriptorsReport.java b/jackrabbit-jcr-server/src/main/java/org/apache/jackrabbit/webdav/jcr/version/report/RepositoryDescriptorsReport.java
index f79a076..78a2e74 100644
--- a/jackrabbit-jcr-server/src/main/java/org/apache/jackrabbit/webdav/jcr/version/report/RepositoryDescriptorsReport.java
+++ b/jackrabbit-jcr-server/src/main/java/org/apache/jackrabbit/webdav/jcr/version/report/RepositoryDescriptorsReport.java
@@ -78,7 +78,7 @@ public class RepositoryDescriptorsReport extends AbstractJcrReport implements It
     }
 
     /**
-     * @see Report#init(DavResource, ReportInfo) 
+     * @see Report#init(DavResource, ReportInfo)
      */
     public void init(DavResource resource, ReportInfo info) throws DavException {
         // delegate validation to abstract super classe
