diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/RelationQueryNode.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/RelationQueryNode.java
index c1cdf9a..d1b218e 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/RelationQueryNode.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/RelationQueryNode.java
@@ -331,7 +331,7 @@ public class RelationQueryNode extends NAryQueryNode implements QueryConstants {
                     && valueLong == other.valueLong
                     && valuePosition == other.valuePosition
                     && (valueString == null ? other.valueString == null : valueString.equals(other.valueString))
-                    && (relPath == null ? other.relPath== null : relPath.equals(other.relPath));
+                    && (relPath == null ? other.relPath == null : relPath.equals(other.relPath));
         }
         return false;
     }
