diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/NodeDataRef.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/NodeDataRef.java
index 72b1f45..37f84a2 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/NodeDataRef.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/NodeDataRef.java
@@ -17,6 +17,8 @@
 package org.apache.jackrabbit.core;
 
 import javax.jcr.nodetype.ItemDefinition;
+
+import org.apache.jackrabbit.core.id.NodeId;
 import org.apache.jackrabbit.core.state.ItemState;
 
 /**
