diff --git a/jackrabbit-spi-commons/src/main/java/org/apache/jackrabbit/spi/commons/query/qom/LowerCaseImpl.java b/jackrabbit-spi-commons/src/main/java/org/apache/jackrabbit/spi/commons/query/qom/LowerCaseImpl.java
index 04d1712..8f45da6 100644
--- a/jackrabbit-spi-commons/src/main/java/org/apache/jackrabbit/spi/commons/query/qom/LowerCaseImpl.java
+++ b/jackrabbit-spi-commons/src/main/java/org/apache/jackrabbit/spi/commons/query/qom/LowerCaseImpl.java
@@ -32,7 +32,7 @@ public class LowerCaseImpl extends DynamicOperandImpl implements LowerCase {
     private final DynamicOperandImpl operand;
 
     LowerCaseImpl(NamePathResolver resolver, DynamicOperandImpl operand) {
-        super(resolver);
+        super(resolver, operand.getSelectorQName());
         this.operand = operand;
     }
