diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/NamePathResolverImpl.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/NamePathResolverImpl.java
index 9d3717b..31359b1 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/NamePathResolverImpl.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/lucene/NamePathResolverImpl.java
@@ -110,4 +110,5 @@ public class NamePathResolverImpl extends DefaultNamePathResolver {
             }
         }
     }
-}
\ No newline at end of file
+
+}
