diff --git a/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/query/qom/RowTest.java b/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/query/qom/RowTest.java
index e213089..9e9aae1 100644
--- a/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/query/qom/RowTest.java
+++ b/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/query/qom/RowTest.java
@@ -89,11 +89,11 @@ public class RowTest extends AbstractQOMTest {
     }
 
     private Row getRow() throws RepositoryException {
-        QueryObjectModel qom = qomFactory.createQuery(
-                qomFactory.selector(testNodeType, SELECTOR_NAME),
-                qomFactory.descendantNode(SELECTOR_NAME, testRoot),
+        QueryObjectModel qom = qf.createQuery(
+                qf.selector(testNodeType, SELECTOR_NAME),
+                qf.descendantNode(SELECTOR_NAME, testRoot),
                 null,
-                new Column[]{qomFactory.column(SELECTOR_NAME, propertyName1, propertyName1)});
+                new Column[]{qf.column(SELECTOR_NAME, propertyName1, propertyName1)});
         RowIterator rows = qom.execute().getRows();
         assertTrue("empty result", rows.hasNext());
         Row r = rows.nextRow();
