diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/cluster/AbstractClusterOperation.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/cluster/AbstractClusterOperation.java
index 8043da5..0309f74 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/cluster/AbstractClusterOperation.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/cluster/AbstractClusterOperation.java
@@ -78,7 +78,7 @@ abstract class AbstractClusterOperation implements ClusterOperation {
      * @throws JournalException if an error occurs
      */
     protected abstract void write() throws JournalException;
-    
+
     /**
      * {@inheritDoc}
      */
