diff --git a/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/ItemManagerImpl.java b/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/ItemManagerImpl.java
index 4b3ad52..137b9f2 100644
--- a/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/ItemManagerImpl.java
+++ b/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/ItemManagerImpl.java
@@ -58,7 +58,7 @@ public class ItemManagerImpl implements Dumpable, ItemManager {
 
     /**
      * A cache for item instances created by this <code>ItemManagerImpl</code>.
-     * 
+     *
      * The <code>ItemState</code>s act as keys for the map. In contrast to
      * o.a.j.core the item state are copied to transient space for reading and
      * will therefor not change upon transient modifications.
