diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/security/authentication/AuthContextProvider.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/security/authentication/AuthContextProvider.java
index 17f6ab3..600281e 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/security/authentication/AuthContextProvider.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/security/authentication/AuthContextProvider.java
@@ -128,7 +128,7 @@ public class AuthContextProvider {
             AppConfigurationEntry[] entries = getJAASConfig();
             if(entries != null) {
                 List tmp = new ArrayList(entries.length);
-                for(int i=0;i<entries.length;i++) {
+                for(int i=0; i < entries.length; i++) {
                     Map opt = entries[i].getOptions();
                     if (opt != null) {
                         Properties prop = new Properties();
