diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/OnWorkspaceInconsistency.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/OnWorkspaceInconsistency.java
index bd139d3..2037bab 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/OnWorkspaceInconsistency.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/query/OnWorkspaceInconsistency.java
@@ -77,9 +77,9 @@ public abstract class OnWorkspaceInconsistency {
     private final String name;
 
     /**
-     * Private constructor.
+     * Protected constructor.
      */
-    private OnWorkspaceInconsistency(String name) {
+    protected OnWorkspaceInconsistency(String name) {
         this.name = name;
     }
