diff --git a/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/observation/GetDateTest.java b/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/observation/GetDateTest.java
index 9330b29..313accc 100644
--- a/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/observation/GetDateTest.java
+++ b/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/observation/GetDateTest.java
@@ -25,8 +25,6 @@ import java.util.Collections;
 import javax.jcr.RepositoryException;
 import javax.jcr.observation.Event;
 
-import org.apache.jackrabbit.test.api.observation.AbstractObservationTest;
-
 /**
  * <code>GetDateTest</code> checks if the dates returned by events are
  * monotonically increasing.
