diff --git a/jackrabbit-jcr2spi/src/test/java/org/apache/jackrabbit/jcr2spi/query/QueryTest.java b/jackrabbit-jcr2spi/src/test/java/org/apache/jackrabbit/jcr2spi/query/QueryTest.java
index c7a740a..a7b0725 100644
--- a/jackrabbit-jcr2spi/src/test/java/org/apache/jackrabbit/jcr2spi/query/QueryTest.java
+++ b/jackrabbit-jcr2spi/src/test/java/org/apache/jackrabbit/jcr2spi/query/QueryTest.java
@@ -15,6 +15,7 @@
  */
 package org.apache.jackrabbit.jcr2spi.query;
 
+import org.apache.jackrabbit.test.NotExecutableException;
 import org.apache.jackrabbit.test.api.query.AbstractQueryTest;
 
 import javax.jcr.RepositoryException;
@@ -31,8 +32,9 @@ public class QueryTest extends AbstractQueryTest {
 
     /**
      * Queries the child nodes of the root node.
+     * @throws NotExecutableException 
      */
-    public void testQueryChildNodesOfRoot() throws RepositoryException {
+    public void testQueryChildNodesOfRoot() throws RepositoryException, NotExecutableException {
         List nodes = new ArrayList();
         for (NodeIterator it = superuser.getRootNode().getNodes(); it.hasNext(); ) {
             nodes.add(it.nextNode());
@@ -41,7 +43,7 @@ public class QueryTest extends AbstractQueryTest {
         executeXPathQuery(superuser, "/jcr:root/*", children);
     }
 
-    public void testRemappedNamespace() throws RepositoryException {
+    public void testRemappedNamespace() throws RepositoryException, NotExecutableException {
         String namespaceURI = "http://jackrabbit.apache.org/spi/test";
         String defaultPrefix = "spiTest";
