diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/ItemManager.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/ItemManager.java
index 4c79860..df049a2 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/ItemManager.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/ItemManager.java
@@ -894,18 +894,18 @@ public class ItemManager implements Dumpable, ItemStateListener {
             Iterator iter = itemCache.keySet().iterator();
             while (iter.hasNext()) {
                 ItemId id = (ItemId) iter.next();
-                ItemImpl item = (ItemImpl) itemCache.get(id);
+                ItemData item = (ItemData) itemCache.get(id);
                 if (item.isNode()) {
                     ps.print("Node: ");
                 } else {
                     ps.print("Property: ");
                 }
-                if (item.isTransient()) {
+                if (item.getState().isTransient()) {
                     ps.print("transient ");
                 } else {
                     ps.print("          ");
                 }
-                ps.println(id + "\t" + item.safeGetJCRPath() + " (" + item + ")");
+                ps.println(id + "\t" + safeGetJCRPath(id) + " (" + item + ")");
             }
         }
     }
