diff --git a/jackrabbit-spi-commons/src/main/java/org/apache/jackrabbit/spi/commons/query/jsr283/qom/ChildNodeJoinCondition.java b/jackrabbit-spi-commons/src/main/java/org/apache/jackrabbit/spi/commons/query/jsr283/qom/ChildNodeJoinCondition.java
index 6de92e8..0accc5a 100644
--- a/jackrabbit-spi-commons/src/main/java/org/apache/jackrabbit/spi/commons/query/jsr283/qom/ChildNodeJoinCondition.java
+++ b/jackrabbit-spi-commons/src/main/java/org/apache/jackrabbit/spi/commons/query/jsr283/qom/ChildNodeJoinCondition.java
@@ -16,6 +16,8 @@
  */
 package org.apache.jackrabbit.spi.commons.query.jsr283.qom;
 
+import javax.jcr.query.qom.JoinCondition;
+
 /**
  * Tests whether the {@link #getChildSelectorName childSelector} node is a child
  * of the {@link #getParentSelectorName parentSelector} node.  A node-tuple
