diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/config/LoginModuleConfig.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/config/LoginModuleConfig.java
index 5dc4d28..d51818d 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/config/LoginModuleConfig.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/config/LoginModuleConfig.java
@@ -29,22 +29,24 @@ import javax.security.auth.spi.LoginModule;
  */
 public class LoginModuleConfig extends BeanConfig {
 
+    /**
+     * UserId of the anonymous user. Optional parameter in the LoginModule
+     * configuration.
+     */
     public static final String PARAM_ANONYMOUS_ID = "anonymousId";
-    public static final String PARAM_ADMIN_ID = "adminId";
-
+    
     /**
-     * Name of the default user id to be used upon Repository.login with
-     * <code>null</code> credentials. Optional parameter in the LoginModule
+     * UserId of the administrator. Optional parameter in the LoginModule
      * configuration.
      */
-    public static final String PARAM_DEFAULT_USERID = "defaultUserId";
+    public static final String PARAM_ADMIN_ID = "adminId";
 
     /**
      * Property-Key for the fully qualified class name of the implementation of
      * {@link org.apache.jackrabbit.core.security.principal.PrincipalProvider}
      * to be used with the LoginModule.
      */
-    public static final String PARAM_PRINCIPAL_PROVIDER_CLASS = "principalprovider";
+    public static final String PARAM_PRINCIPAL_PROVIDER_CLASS = "principalProvider";
 
     /**
      * Creates an access manager configuration object from the
