diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/security/principal/AdminPrincipal.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/security/principal/AdminPrincipal.java
index 395aa95..3d6976c 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/security/principal/AdminPrincipal.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/security/principal/AdminPrincipal.java
@@ -21,7 +21,7 @@ package org.apache.jackrabbit.core.security.principal;
  * the access rights and is being authenticated (in contrast to the
  * {@link org.apache.jackrabbit.core.security.SystemPrincipal SystemPrincipal}).
  */
-public final class AdminPrincipal implements JackrabbitPrincipal {
+public class AdminPrincipal implements JackrabbitPrincipal {
 
     private final String adminId;
