diff --git a/jackrabbit-webdav/src/main/java/org/apache/jackrabbit/webdav/security/Principal.java b/jackrabbit-webdav/src/main/java/org/apache/jackrabbit/webdav/security/Principal.java
index 087a146..890f851 100644
--- a/jackrabbit-webdav/src/main/java/org/apache/jackrabbit/webdav/security/Principal.java
+++ b/jackrabbit-webdav/src/main/java/org/apache/jackrabbit/webdav/security/Principal.java
@@ -39,7 +39,7 @@ import java.util.Map;
 public class Principal implements XmlSerializable, SecurityConstants {
 
     public static final String XML_PRINCIPAL = "principal";
-    
+
     private static final String XML_ALL = "all";
     private static final String XML_AUTHENTICATED  = "authenticated";
     private static final String XML_UNAUTHENTICATED = "unauthenticated";
