diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/version/AbstractVersionManager.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/version/AbstractVersionManager.java
index 62d81fa..9195fe5 100755
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/version/AbstractVersionManager.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/version/AbstractVersionManager.java
@@ -115,14 +115,6 @@ abstract class AbstractVersionManager implements VersionManager {
         return hasItem(id);
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    public boolean hasVersion(NodeId id) {
-        // lock handling via hasItem()
-        return hasItem(id);
-    }
-
     //-------------------------------------------------------< implementation >
 
     /**
