diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/state/obj/ObjectPersistenceManager.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/state/obj/ObjectPersistenceManager.java
index 7e4079b..ac0e8b8 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/state/obj/ObjectPersistenceManager.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/state/obj/ObjectPersistenceManager.java
@@ -26,4 +26,4 @@ import org.apache.jackrabbit.core.state.PersistenceManager;
 public class ObjectPersistenceManager
         extends org.apache.jackrabbit.core.persistence.obj.ObjectPersistenceManager
         implements PersistenceManager {
-}
\ No newline at end of file
+}
