diff --git a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/name/NoPrefixDeclaredException.java b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/name/NoPrefixDeclaredException.java
index 613eb86..248f265 100644
--- a/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/name/NoPrefixDeclaredException.java
+++ b/jackrabbit-jcr-commons/src/main/java/org/apache/jackrabbit/name/NoPrefixDeclaredException.java
@@ -22,7 +22,7 @@ package org.apache.jackrabbit.name;
  * prefix is not found into a JCR name string. The JCR name string can not be
  * created without the namespace prefix.
  *
- * @deprecated 
+ * @deprecated
  */
 public class NoPrefixDeclaredException extends NameException {
