diff --git a/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/security/authorization/JackrabbitAccessControlListTest.java b/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/security/authorization/JackrabbitAccessControlListTest.java
index 9ea5314..f1a99ea 100644
--- a/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/security/authorization/JackrabbitAccessControlListTest.java
+++ b/jackrabbit-core/src/test/java/org/apache/jackrabbit/core/security/authorization/JackrabbitAccessControlListTest.java
@@ -28,6 +28,7 @@ import org.apache.jackrabbit.test.NotExecutableException;
 
 import javax.jcr.Node;
 import javax.jcr.RepositoryException;
+import javax.jcr.PropertyType;
 import java.security.Principal;
 import java.util.Arrays;
 import java.util.Collections;
@@ -79,6 +80,18 @@ public class JackrabbitAccessControlListTest extends AbstractAccessControlTest {
         }
     }
 
+    public void testGetRestrictionNames() {
+        assertNotNull(templ.getRestrictionNames());
+    }
+
+    public void testGetRestrictionType() {
+        String[] names = templ.getRestrictionNames();
+        for (int i = 0; i < names.length; i++) {
+            int type = templ.getRestrictionType(names[i]);
+            assertTrue(type > PropertyType.UNDEFINED);
+        }
+    }
+
     public void testIsEmpty() throws RepositoryException {
         if (templ.isEmpty()) {
             assertEquals(0, templ.getAccessControlEntries().length);
