diff --git a/jackrabbit-api/src/main/java/org/apache/jackrabbit/api/jsr283/security/AccessControlEntry.java b/jackrabbit-api/src/main/java/org/apache/jackrabbit/api/jsr283/security/AccessControlEntry.java
index a596aa9..3a70b7c 100644
--- a/jackrabbit-api/src/main/java/org/apache/jackrabbit/api/jsr283/security/AccessControlEntry.java
+++ b/jackrabbit-api/src/main/java/org/apache/jackrabbit/api/jsr283/security/AccessControlEntry.java
@@ -25,6 +25,7 @@ import java.security.Principal;
  * @since JCR 2.0
  */
 public interface AccessControlEntry {
+
     /**
      * Returns the principal associated with this access control entry.
      * @return a <code>Principal</code>.
