diff --git a/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/observation/ObservationManagerImpl.java b/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/observation/ObservationManagerImpl.java
index 5ba9f9e..5df9799 100644
--- a/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/observation/ObservationManagerImpl.java
+++ b/jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/observation/ObservationManagerImpl.java
@@ -25,7 +25,6 @@ import java.util.Map;
 
 import javax.jcr.RepositoryException;
 import javax.jcr.UnsupportedRepositoryOperationException;
-import javax.jcr.ValueFactory;
 import javax.jcr.observation.EventJournal;
 import javax.jcr.observation.EventListener;
 import javax.jcr.observation.EventListenerIterator;
@@ -65,12 +64,6 @@ public class ObservationManagerImpl implements ObservationManager, InternalEvent
     private final NamePathResolver resolver;
 
     /**
-     * The ValueFactory associated with the session this observation
-     * manager belongs to.
-     */
-    private final ValueFactory valueFactory;
-
-    /**
      * The <code>NodeTypeRegistry</code> of the session.
      */
     private final NodeTypeRegistry ntRegistry;
@@ -93,11 +86,10 @@ public class ObservationManagerImpl implements ObservationManager, InternalEvent
      * @param valueFactory
      */
     public ObservationManagerImpl(WorkspaceManager wspManager, NamePathResolver resolver,
-                                  NodeTypeRegistry ntRegistry, ValueFactory valueFactory) {
+                                  NodeTypeRegistry ntRegistry) {
         this.wspManager = wspManager;
         this.resolver = resolver;
         this.ntRegistry = ntRegistry;
-        this.valueFactory = valueFactory;
     }
 
     /**
@@ -221,7 +213,7 @@ public class ObservationManagerImpl implements ObservationManager, InternalEvent
             Map.Entry entry = (Map.Entry) it.next();
             EventListener listener = (EventListener) entry.getKey();
             EventFilter filter = (EventFilter) entry.getValue();
-            FilteredEventIterator eventIter = new FilteredEventIterator(eventBundle, filter, resolver, valueFactory, wspManager.getIdFactory());
+            FilteredEventIterator eventIter = new FilteredEventIterator(eventBundle, filter, resolver, wspManager.getIdFactory());
             if (eventIter.hasNext()) {
                 try {
                     listener.onEvent(eventIter);
