diff --git a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/security/principal/PrincipalProvider.java b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/security/principal/PrincipalProvider.java
index bd49187..d0a80d4 100644
--- a/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/security/principal/PrincipalProvider.java
+++ b/jackrabbit-core/src/main/java/org/apache/jackrabbit/core/security/principal/PrincipalProvider.java
@@ -28,7 +28,7 @@ import java.util.Properties;
  * sources for authorization and authentication, as may be used by a custom
  * {@link javax.security.auth.spi.LoginModule} for example.
  *
- * @see PrincipalManager for more details about principals, users and groups.
+ * @see org.apache.jackrabbit.api.security.principal.PrincipalManager for more details about principals, users and groups.
  */
 public interface PrincipalProvider {
 
@@ -56,9 +56,9 @@ public interface PrincipalProvider {
      * @param simpleFilter
      * @param searchType searchType Any of the following constants:
      * <ul>
-     * <li>{@link PrincipalManager#SEARCH_TYPE_ALL}</li>
-     * <li>{@link PrincipalManager#SEARCH_TYPE_GROUP}</li>
-     * <li>{@link PrincipalManager#SEARCH_TYPE_NOT_GROUP}</li>
+     * <li>{@link org.apache.jackrabbit.api.security.principal.PrincipalManager#SEARCH_TYPE_ALL}</li>
+     * <li>{@link org.apache.jackrabbit.api.security.principal.PrincipalManager#SEARCH_TYPE_GROUP}</li>
+     * <li>{@link org.apache.jackrabbit.api.security.principal.PrincipalManager#SEARCH_TYPE_NOT_GROUP}</li>
      * </ul>
      * @return
      * @see #findPrincipals(String)
@@ -71,9 +71,9 @@ public interface PrincipalProvider {
      * @return an iterator over all principals that match the given search type.
      * @param searchType searchType Any of the following constants:
      * <ul>
-     * <li>{@link PrincipalManager#SEARCH_TYPE_ALL}</li>
-     * <li>{@link PrincipalManager#SEARCH_TYPE_GROUP}</li>
-     * <li>{@link PrincipalManager#SEARCH_TYPE_NOT_GROUP}</li>
+     * <li>{@link org.apache.jackrabbit.api.security.principal.PrincipalManager#SEARCH_TYPE_ALL}</li>
+     * <li>{@link org.apache.jackrabbit.api.security.principal.PrincipalManager#SEARCH_TYPE_GROUP}</li>
+     * <li>{@link org.apache.jackrabbit.api.security.principal.PrincipalManager#SEARCH_TYPE_NOT_GROUP}</li>
      * </ul>
      */
     PrincipalIterator getPrincipals(int searchType);
@@ -81,7 +81,7 @@ public interface PrincipalProvider {
     /**
      * Returns an iterator over all group principals for which the given
      * principal is either direct or indirect member of. If a principal is
-     * a direct member of a group, then <code>{@link Group#isMember(Principal)}</code>
+     * a direct member of a group, then <code>{@link java.security.acl.Group#isMember(Principal)}</code>
      * evaluates to <code>true</code>. A principal is an indirect member of a
      * group if any of its groups (to any degree of separation) is direct memeber
      * of the group.
