diff --git a/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/WorkspaceCopyTest.java b/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/WorkspaceCopyTest.java
index d37eaeb..2ed659a 100644
--- a/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/WorkspaceCopyTest.java
+++ b/jackrabbit-jcr-tests/src/main/java/org/apache/jackrabbit/test/api/WorkspaceCopyTest.java
@@ -24,6 +24,8 @@ import javax.jcr.Session;
 import javax.jcr.lock.LockException;
 import javax.jcr.nodetype.ConstraintViolationException;
 
+import org.apache.jackrabbit.test.NotExecutableException;
+
 /**
  * <code>WorkspaceCopyTest</code> contains tests for copying nodes in one
  * workspace.
@@ -136,7 +138,8 @@ public class WorkspaceCopyTest extends AbstractWorkspaceCopyTest {
     /**
      * A LockException is thrown if a lock prevents the copy.
      */
-    public void testCopyNodesLocked() throws RepositoryException {
+    public void testCopyNodesLocked() throws RepositoryException,
+            NotExecutableException {
         // we assume repository supports locking
         String dstAbsPath = node2.getPath() + "/" + node1.getName();
 
@@ -148,10 +151,8 @@ public class WorkspaceCopyTest extends AbstractWorkspaceCopyTest {
             Node lockTarget = (Node) otherSession.getItem(node2.getPath());
 
             // add mixin "lockable" to be able to lock the node
-            if (!lockTarget.getPrimaryNodeType().isNodeType(mixLockable)) {
-                lockTarget.addMixin(mixLockable);
-                lockTarget.getParent().save();
-            }
+            ensureMixinType(lockTarget, mixLockable);
+            lockTarget.getParent().save();
 
             // lock dst parent node using other session
             lockTarget.lock(true, true);
